{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "e435836c-450c-417f-9d5d-01c66e6dd564",
   "metadata": {},
   "source": [
    "# TMEM175 - Single gene analysis in GP2 NeuroBooster genotyping data (all ancestries)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "faf6f591-e3f0-4862-85b0-a94db6a53a2b",
   "metadata": {},
   "source": [
    "`GP2 ‚ù§Ô∏è Open Science üòç`\n",
    "\n",
    "**Last updated:** June 2025\n",
    "\n",
    "\n",
    "**Project:** GP2 TMEM175\n",
    "\n",
    "**Version:** Python/3.10.15, R/4.3.3\n",
    "\n",
    "\n",
    "## Description\n",
    "\n",
    "This notebook contains the code and workflow used in the study: **‚ÄúTMEM175, SCARB2 and CTSB associations with Parkinson's disease risk across populations‚Äù**.\n",
    "\n",
    "In this notebook performed a single variant based and gene based analysis of TMEM175 with GP2 data."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a787b17-a756-41ca-99a3-30efd0019c98",
   "metadata": {},
   "source": [
    "## Notebook Overview\n",
    "\n",
    "1. Loading Python libraries\n",
    "* Set paths\n",
    "* Make working directory\n",
    "2. Installing packages\n",
    "\n",
    "3. Create a covariate file with GP2 data\n",
    "\n",
    "4. Annotation of the gene TMEM175\n",
    "\n",
    "5. Association analysis to compare allele frequencies between cases and controls\n",
    "\n",
    "6. GLM analysis adjusting for gender, age, PC1-5\n",
    "\n",
    "7. Burden test (SkatO, Skat, cmc,zeggini,mb,fp,cmcWald)\n",
    "\n",
    "8. Conditional analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db70b75b-56b6-4611-9e5d-63f35668d365",
   "metadata": {},
   "source": [
    "## 1. Loading Python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6ead6872-5637-4e31-a6e3-543238a8e5e2",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Use pathlib for file path manipulation\n",
    "import pathlib\n",
    "\n",
    "# Install numpy\n",
    "import numpy as np\n",
    "\n",
    "# Install Pandas for tabular data\n",
    "import pandas as pd\n",
    "\n",
    "# Install plotnine: a ggplot2-compatible Python plotting package\n",
    "from plotnine import *\n",
    "\n",
    "# Always show all columns in a Pandas DataFrame\n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ef9c9cce-81a5-4c44-8dbf-cc91cc67ef9b",
   "metadata": {},
   "source": [
    "### Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "854afae5-dc23-4c88-96fb-89ff5175aeee",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "REL7_PATH = pathlib.Path(pathlib.Path.home(), '~/workspace/path_to_release7')\n",
    "!ls -hal {REL7_PATH}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "351766ac-da37-42a1-aed8-38e0d044c429",
   "metadata": {},
   "source": [
    "### Make working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d2df0a09-9eed-421f-815e-47fe9ea36a6a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/TMEM175/\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fcece3b3",
   "metadata": {},
   "source": [
    "## 2. Installing packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28d5c734-cb5f-498c-bb37-b9e6d2872fdb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LICENSE\t\t       plink2\t\t\t\trvtests\n",
      "annovar\t\t       plink2_linux_x86_64_latest.zip\ttoy.map\n",
      "annovar.latest.tar.gz  plink_linux_x86_64_20190304.zip\ttoy.ped\n",
      "plink\t\t       prettify\n"
     ]
    }
   ],
   "source": [
    "# Make sure all tools are installed\n",
    "! ls /home/jupyter/tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9150c8dc-16b1-44ad-9d51-7c3c9708990a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Give permission\n",
    "\n",
    "# chmod to make sure you have permission to run the program\n",
    "! chmod u+x /home/jupyter/tools/plink\n",
    "! chmod u+x /home/jupyter/tools/plink2\n",
    "! chmod 777 /home/jupyter/tools/rvtests/executable/rvtest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20f842ba-c03c-4604-972f-3e9b5b37d530",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "# making working directory\n",
    "# Loop over all the ancestries\n",
    "for ancestry in {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'} ;\n",
    "do\n",
    "\n",
    "# Make a folder for each ancestry\n",
    "mkdir ~/workspace/ws_files/TMEM175/TMEM175_\"$ancestry\"\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a4b8934f-368c-4876-b461-0b20e7814b15",
   "metadata": {},
   "source": [
    "## 3. Create a covariate file with GP2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "805da8a5-55d5-45b0-a15e-a97bb8b60049",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "CLINICAL_DATA_PATH = pathlib.Path(REL7_PATH, 'clinical_data/master_key_release7_final_vwb.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6f7224e2-9a40-4b68-a648-a78ce85ecbc3",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Let's load the master key\n",
    "key = pd.read_csv(CLINICAL_DATA_PATH, low_memory=False)\n",
    "print(key.shape)\n",
    "key.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e81204c4-0636-4cb9-b2d4-417d8c040263",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Subsetting to keep only a few columns \n",
    "key = key[['GP2sampleID', 'baseline_GP2_phenotype_for_qc', 'biological_sex_for_qc', 'age_at_sample_collection', 'age_of_onset', 'label']]\n",
    "# Renaming the columns\n",
    "key.rename(columns = {'GP2sampleID':'IID',\n",
    "                                     'baseline_GP2_phenotype_for_qc':'phenotype',\n",
    "                                     'biological_sex_for_qc':'SEX', \n",
    "                                     'age_at_sample_collection':'AGE', \n",
    "                                     'age_of_onset':'AAO'}, inplace = True)\n",
    "key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d9fc4b4b-3517-43a9-b806-2193a5d2b082",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AFR\n",
      "Related individuals: (228, 9)\n",
      "Unrelated individuals: (2752, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          1553\n",
      "Female                        1197\n",
      "Other/Unknown/Not Reported       2\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       1553\n",
      "2       1197\n",
      "<NA>       2\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 835\n",
      "Number of controls missing age: 809\n",
      "WORKING ON: AAC\n",
      "Related individuals: (28, 9)\n",
      "Unrelated individuals: (1181, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Female                        690\n",
      "Male                          489\n",
      "Other/Unknown/Not Reported      2\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "2       690\n",
      "1       489\n",
      "<NA>      2\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 91\n",
      "Number of controls missing age: 96\n",
      "WORKING ON: FIN\n",
      "Related individuals: (7, 9)\n",
      "Unrelated individuals: (117, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Female    63\n",
      "Male      54\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "2    63\n",
      "1    54\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 62\n",
      "Number of controls missing age: 3\n",
      "WORKING ON: AMR\n",
      "Related individuals: (51, 9)\n",
      "Unrelated individuals: (672, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          374\n",
      "Female                        296\n",
      "Other/Unknown/Not Reported      2\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       374\n",
      "2       296\n",
      "<NA>      2\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 106\n",
      "Number of controls missing age: 55\n",
      "WORKING ON: EAS\n",
      "Related individuals: (350, 9)\n",
      "Unrelated individuals: (5303, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          3398\n",
      "Female                        1899\n",
      "Other/Unknown/Not Reported       6\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       3398\n",
      "2       1899\n",
      "<NA>       6\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 1349\n",
      "Number of controls missing age: 1268\n",
      "WORKING ON: CAH\n",
      "Related individuals: (65, 9)\n",
      "Unrelated individuals: (1108, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          585\n",
      "Female                        498\n",
      "Other/Unknown/Not Reported     25\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       585\n",
      "2       498\n",
      "<NA>     25\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 136\n",
      "Number of controls missing age: 52\n",
      "WORKING ON: CAS\n",
      "Related individuals: (137, 9)\n",
      "Unrelated individuals: (933, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Female    483\n",
      "Male      450\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "2    483\n",
      "1    450\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 111\n",
      "Number of controls missing age: 88\n",
      "WORKING ON: SAS\n",
      "Related individuals: (55, 9)\n",
      "Unrelated individuals: (624, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          397\n",
      "Female                        226\n",
      "Other/Unknown/Not Reported      1\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       397\n",
      "2       226\n",
      "<NA>      1\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 203\n",
      "Number of controls missing age: 83\n",
      "WORKING ON: EUR\n",
      "Related individuals: (3662, 9)\n",
      "Unrelated individuals: (39710, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          23733\n",
      "Female                        15784\n",
      "Other/Unknown/Not Reported      193\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       23733\n",
      "2       15784\n",
      "<NA>      193\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 5947\n",
      "Number of controls missing age: 3347\n",
      "WORKING ON: AJ\n",
      "Related individuals: (245, 9)\n",
      "Unrelated individuals: (2675, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          1655\n",
      "Female                        1009\n",
      "Other/Unknown/Not Reported      11\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       1655\n",
      "2       1009\n",
      "<NA>      11\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 95\n",
      "Number of controls missing age: 195\n",
      "WORKING ON: MDE\n",
      "Related individuals: (19, 9)\n",
      "Unrelated individuals: (589, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          368\n",
      "Female                        219\n",
      "Other/Unknown/Not Reported      2\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       368\n",
      "2       219\n",
      "<NA>      2\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 140\n",
      "Number of controls missing age: 12\n"
     ]
    }
   ],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    # Subset to keep ancestry of interest \n",
    "    ancestry_key = key[key['label']==ancestry].copy()\n",
    "    ancestry_key.reset_index(drop=True)\n",
    "    \n",
    "     # Load information about related individuals in the ancestry analyzed\n",
    "    related_df = pd.read_csv(f'{REL7_PATH}/meta_data/related_samples/{ancestry}_release7_vwb.related')\n",
    "    print(f'Related individuals: {related_df.shape}')\n",
    "    \n",
    "    # Make a list of just one set of related people\n",
    "    related_list = list(related_df['IID1'])\n",
    "    \n",
    "    # Check value counts of related and remove only one related individual\n",
    "    ancestry_key = ancestry_key[~ancestry_key[\"IID\"].isin(related_list)]\n",
    "    \n",
    "    # Check size\n",
    "    print(f'Unrelated individuals: {ancestry_key.shape}')\n",
    "    \n",
    "    # Convert phenotype to binary (1/2)\n",
    "    # Assign conditions so case=2 and controls=1, and -9 otherwise (matching PLINK convention)\n",
    "    # PD = 2; control = 1\n",
    "    pheno_mapping = {\"PD\": 2, \"Control\": 1}\n",
    "    ancestry_key['PHENO'] = ancestry_key['phenotype'].map(pheno_mapping).astype('Int64')\n",
    "    \n",
    "    # Check value counts of pheno\n",
    "    ancestry_key['PHENO'].value_counts(dropna=False)\n",
    "    \n",
    "    # Get the PCs\n",
    "    pcs = pd.read_csv(f'{REL7_PATH}/meta_data/qc_metrics/projected_pcs_vwb.csv')\n",
    "    \n",
    "     # Select just first 5 PCs\n",
    "    selected_columns = ['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5']\n",
    "    pcs = pd.DataFrame(data=pcs.iloc[:, 1:7].values, columns=selected_columns)\n",
    "    \n",
    "     # Drop the first row (since it's now the column names)\n",
    "    pcs = pcs.drop(0)\n",
    "    \n",
    "    # Reset the index to remove any potential issues\n",
    "    pcs = pcs.reset_index(drop=True)\n",
    "    \n",
    "    # Check size\n",
    "    print(f'PCs: {pcs.shape}')\n",
    "    \n",
    "    # Check value counts of SEX\n",
    "    sex_og_values = ancestry_key['SEX'].value_counts(dropna=False)\n",
    "    print(f'Sex value counts - original:\\n {sex_og_values.to_string()}')\n",
    "    \n",
    "    # Convert sex to binary (1/2)\n",
    "    # Assign conditions so female=2 and men=1, and -9 otherwise (matching PLINK convention)\n",
    "    # Female = 2; Male = 1\n",
    "    sex_mapping = {\"Female\": 2, \"Male\": 1}\n",
    "    ancestry_key['SEX'] = ancestry_key['SEX'].map(sex_mapping).astype('Int64')\n",
    "    \n",
    "    # Check value counts of SEX after recoding\n",
    "    sex_recode_values = ancestry_key['SEX'].value_counts(dropna=False)\n",
    "    print(f'Sex value counts - recoded:\\n{sex_recode_values.to_string()}')\n",
    "    \n",
    "    # Make covariate file\n",
    "    df = pd.merge(pcs, ancestry_key, on='IID', how='left')\n",
    "    print(f'Check columns for covariate file: {df.columns}')\n",
    "    \n",
    "    # Make additional columns - FID, fatid and matid - these are needed for RVtests!!\n",
    "    # RVtests needs the first 5 columns to be fid, iid, fatid, matid and sex otherwise it does not run correctly\n",
    "    # Uppercase column name is ok\n",
    "    # See https://zhanxw.github.io/rvtests/# phenotype-file\n",
    "    df['FID'] = 0\n",
    "    df['FATID'] = 0\n",
    "    df['MATID'] = 0\n",
    "    \n",
    "    # Clean up and keep columns we need \n",
    "    final_df = df[['FID','IID', 'FATID', 'MATID', 'SEX', 'AGE', 'PHENO','PC1', 'PC2', 'PC3', 'PC4', 'PC5']].copy()\n",
    "    \n",
    "    ## DO NOT replace missing values with -9 as this is misinterpreted by RVtests - needs to be nonnumeric\n",
    "    # Leave missing values as NA\n",
    "    \n",
    "    # Check number of PD cases missing age\n",
    "    pd_missAge = final_df[(final_df['PHENO']==2)&(final_df['AGE'].isna())]\n",
    "    print(f'Number of PD cases missing age: {pd_missAge.shape[0]}')\n",
    "    \n",
    "    # Check number of controls missing age\n",
    "    control_missAge = final_df[(final_df['PHENO']==1)&(final_df['AGE'].isna())]\n",
    "    print(f'Number of controls missing age: {control_missAge.shape[0]}')\n",
    "    \n",
    "    # Make file of sample IDs to keep \n",
    "    samples_toKeep = final_df[['FID', 'IID']].copy()\n",
    "    samples_toKeep.to_csv(f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}/{ancestry}.samplestoKeep', sep = '\\t', index=False, header=None)\n",
    "    \n",
    "    # Make your covariate file\n",
    "    # Included na_rep to write out missing/NA values explicitly as string/text, not as blank otherwise they are misread in RVtests\n",
    "    final_df.to_csv(f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}/{ancestry}_covariate_file.txt', sep = '\\t', na_rep='NA', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e23131f5-8bd6-482d-85c9-008c2aff08a7",
   "metadata": {},
   "source": [
    "## 4. Annotation of the gene TMEM175\n",
    "\n",
    "* Extract the region using PLINK\n",
    "\n",
    "* Extract TMEM175 gene\n",
    "\n",
    "TMEM175 coordinates: \n",
    "* Chromosome 4:932387-958656 (GRCh38/hg38)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "45bdfb86-1535-40b2-8d8d-2255c06a3d10",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Extract region using plink\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --pfile {REL7_PATH}/imputed_genotypes/{ancestry}/chr4_{ancestry}_release7_vwb \\\n",
    "    --chr 4 \\\n",
    "    --from-bp 882387 \\\n",
    "    --to-bp 1008656 \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/{ancestry}_TMEM175"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65d2818f-be1b-4fbd-993f-1741cfe3a5b9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4\tchr4:882390:C:A\t0\t882390\tA\tC\n",
      "4\tchr4:882397:C:T\t0\t882397\tT\tC\n",
      "4\tchr4:882422:C:T\t0\t882422\tT\tC\n",
      "4\tchr4:882432:G:A\t0\t882432\tA\tG\n",
      "4\tchr4:882454:C:T\t0\t882454\tT\tC\n",
      "4\tchr4:882471:G:A\t0\t882471\tA\tG\n",
      "4\tchr4:882483:C:G\t0\t882483\tG\tC\n",
      "4\tchr4:882498:C:G\t0\t882498\tG\tC\n",
      "4\tchr4:882507:G:A\t0\t882507\tA\tG\n",
      "4\tchr4:882517:C:A\t0\t882517\tA\tC\n"
     ]
    }
   ],
   "source": [
    "# Visualize bim file\n",
    "WORK_DIR = '/home/jupyter/workspace/ws_files/TMEM175/'\n",
    "! head {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a75b27d6-b85d-4678-aa3b-336aad9b9f88",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Visualize bim file\n",
    "! head {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "c5e0bdd8-d997-4932-8c48-e65a34338186",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}/'\n",
    "    \n",
    "    ! head -n 1 {WORK_DIR}/{ancestry}_TMEM175.fam > {WORK_DIR}/{ancestry}_s1.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "02fda682-a3db-41ee-a76b-e7eb853ce4b5",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "! head /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_s1.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "305bd2da-933b-4b45-862c-8997ee82ffb2",
   "metadata": {},
   "source": [
    "### Turn binary files into VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03654fe1-3d77-4a0e-8ddf-f8012d0f664c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:13:40 2024\n",
      "52223 MiB RAM detected, ~50482 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam.\n",
      "3853 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim.\n",
      "1 binary phenotype loaded (942 cases, 1679 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.bed\n",
      "... done.\n",
      "End time: Wed Jul 31 10:13:40 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:13:40 2024\n",
      "52223 MiB RAM detected, ~50465 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1111 samples (656 females, 455 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam.\n",
      "3449 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim.\n",
      "1 binary phenotype loaded (285 cases, 801 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.bed\n",
      "... done.\n",
      "End time: Wed Jul 31 10:13:41 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:13:41 2024\n",
      "52223 MiB RAM detected, ~50489 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "114 samples (61 females, 53 males; 114 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam.\n",
      "605 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim.\n",
      "1 binary phenotype loaded (98 cases, 8 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.bed\n",
      "... done.\n",
      "End time: Wed Jul 31 10:13:42 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:13:42 2024\n",
      "52223 MiB RAM detected, ~50502 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "646 samples (283 females, 363 males; 646 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam.\n",
      "1987 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim.\n",
      "1 binary phenotype loaded (458 cases, 155 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.bed\n",
      "... done.\n",
      "End time: Wed Jul 31 10:13:42 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:13:43 2024\n",
      "52223 MiB RAM detected, ~50501 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam.\n",
      "3171 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim.\n",
      "1 binary phenotype loaded (2662 cases, 2461 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.bed\n",
      "... done.\n",
      "End time: Wed Jul 31 10:13:43 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:13:43 2024\n",
      "52223 MiB RAM detected, ~50501 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "851 samples (405 females, 446 males; 851 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam.\n",
      "3090 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim.\n",
      "1 binary phenotype loaded (525 cases, 302 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.bed\n",
      "... done.\n",
      "End time: Wed Jul 31 10:13:44 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:13:44 2024\n",
      "52223 MiB RAM detected, ~50501 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "903 samples (464 females, 439 males; 903 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam.\n",
      "1955 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim.\n",
      "1 binary phenotype loaded (552 cases, 343 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.bed\n",
      "... done.\n",
      "End time: Wed Jul 31 10:13:45 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:13:45 2024\n",
      "52223 MiB RAM detected, ~50496 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "635 samples (227 females, 408 males; 635 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam.\n",
      "1728 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim.\n",
      "1 binary phenotype loaded (387 cases, 222 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.bed\n",
      "... done.\n",
      "End time: Wed Jul 31 10:13:45 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:13:45 2024\n",
      "52223 MiB RAM detected, ~50502 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam.\n",
      "9504 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.bed\n",
      "... done.\n",
      "End time: Wed Jul 31 10:13:46 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:13:46 2024\n",
      "52223 MiB RAM detected, ~50502 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam.\n",
      "1285 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim.\n",
      "1 binary phenotype loaded (1292 cases, 411 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.bed\n",
      "... done.\n",
      "End time: Wed Jul 31 10:13:47 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:13:47 2024\n",
      "52223 MiB RAM detected, ~50497 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "581 samples (212 females, 369 males; 581 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam.\n",
      "1945 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim.\n",
      "1 binary phenotype loaded (311 cases, 225 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.bed\n",
      "... done.\n",
      "End time: Wed Jul 31 10:13:48 2024\n"
     ]
    }
   ],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    \n",
    "    # Turn binary files into VCF\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_TMEM175 \\\n",
    "    --keep {WORK_DIR}/{ancestry}_s1.txt \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/{ancestry}_TMEM175_v1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a988e523-8aad-44dd-b516-1803ba26c42b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:15:08 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50568 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.fam.\n",
      "3853 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.vcf ...\n",
      "1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Wed Jul 31 10:15:09 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:15:09 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50561 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.fam.\n",
      "3449 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.vcf ...\n",
      "1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Wed Jul 31 10:15:09 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:15:10 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50553 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.fam.\n",
      "605 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.vcf ...\n",
      "10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Wed Jul 31 10:15:10 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:15:10 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50571 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.fam.\n",
      "1987 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.vcf ...\n",
      "1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Wed Jul 31 10:15:10 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:15:11 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50572 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.fam.\n",
      "3171 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.vcf ...\n",
      "1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Wed Jul 31 10:15:11 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:15:11 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50555 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.fam.\n",
      "3090 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.vcf ...\n",
      "101011111212131314141515161617171818192020212122222323242425252626272728282930303131323233333434353536363737383839404041414242434344444545464647474848495050515152525353545455555656575758585960606161626263636464656566666767686869707071717272737374747575767677777878798080818182828383848485858686878788888990909191929293939494959596969797989899done.\n",
      "End time: Wed Jul 31 10:15:12 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:15:12 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50555 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.fam.\n",
      "1955 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.vcf ...\n",
      "10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Wed Jul 31 10:15:12 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:15:13 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50555 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.fam.\n",
      "1728 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.vcf ...\n",
      "1010111112121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373738383939404041414242434344444545464647474848495050515152525353545455555656575758585959606061616262636364646565666667676868696970707171727273737475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Wed Jul 31 10:15:13 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:15:13 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50556 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.fam.\n",
      "9504 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.vcf ...\n",
      "1010111112121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373738383939404041414242434344444545464647474848495050515152525353545455555656575758585959606061616262636364646565666667676868696970707171727273737475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Wed Jul 31 10:15:13 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:15:14 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50557 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.fam.\n",
      "1285 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.vcf ... 10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Wed Jul 31 10:15:14 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1\n",
      "\n",
      "Start time: Wed Jul 31 10:15:14 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50557 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.fam.\n",
      "1945 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.vcf ...\n",
      "10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Wed Jul 31 10:15:14 2024\n"
     ]
    }
   ],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    \n",
    "    # Turn binary files into VCF\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_TMEM175_v1 \\\n",
    "    --recode vcf-fid \\\n",
    "    --out {WORK_DIR}/{ancestry}_TMEM175_v1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d0f3130-fefc-40d0-b3ea-ca92fccd12a9",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "## Bgzip and Tabix (zip and index the file)\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    ! bgzip -f {WORK_DIR}/{ancestry}_TMEM175_v1.vcf\n",
    "    ! tabix -f -p vcf {WORK_DIR}/{ancestry}_TMEM175_v1.vcf.gz "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b70ddab-a941-4024-a640-37d6bc978640",
   "metadata": {},
   "source": [
    "### Annotate using ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "95cc7254-f11d-4798-aab0-e1c72cc8ce35",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 3860 lines from VCF file\n",
      "NOTICE: A total of 3853 locus in VCF file passed QC threshold, representing 3628 SNPs (2760 transitions and 868 transversions) and 225 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3628 SNP genotypes (2760 transitions and 868 transversions) and 225 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3853 unique variants in 3853 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3853 unique variants in 3853 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 3456 lines from VCF file\n",
      "NOTICE: A total of 3449 locus in VCF file passed QC threshold, representing 3247 SNPs (2444 transitions and 803 transversions) and 202 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3247 SNP genotypes (2444 transitions and 803 transversions) and 202 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3449 unique variants in 3449 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3449 unique variants in 3449 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 612 lines from VCF file\n",
      "NOTICE: A total of 605 locus in VCF file passed QC threshold, representing 573 SNPs (427 transitions and 146 transversions) and 32 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 573 SNP genotypes (427 transitions and 146 transversions) and 32 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 605 unique variants in 605 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 605 unique variants in 605 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 1994 lines from VCF file\n",
      "NOTICE: A total of 1987 locus in VCF file passed QC threshold, representing 1870 SNPs (1403 transitions and 467 transversions) and 117 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 1870 SNP genotypes (1403 transitions and 467 transversions) and 117 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 1987 unique variants in 1987 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 1987 unique variants in 1987 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 3178 lines from VCF file\n",
      "NOTICE: A total of 3171 locus in VCF file passed QC threshold, representing 3001 SNPs (2330 transitions and 671 transversions) and 170 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3001 SNP genotypes (2330 transitions and 671 transversions) and 170 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3171 unique variants in 3171 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3171 unique variants in 3171 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 3097 lines from VCF file\n",
      "NOTICE: A total of 3090 locus in VCF file passed QC threshold, representing 2917 SNPs (2209 transitions and 708 transversions) and 173 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2917 SNP genotypes (2209 transitions and 708 transversions) and 173 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3090 unique variants in 3090 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3090 unique variants in 3090 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 1962 lines from VCF file\n",
      "NOTICE: A total of 1955 locus in VCF file passed QC threshold, representing 1848 SNPs (1430 transitions and 418 transversions) and 107 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 1848 SNP genotypes (1430 transitions and 418 transversions) and 107 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 1955 unique variants in 1955 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 1955 unique variants in 1955 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 1735 lines from VCF file\n",
      "NOTICE: A total of 1728 locus in VCF file passed QC threshold, representing 1628 SNPs (1231 transitions and 397 transversions) and 100 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 1628 SNP genotypes (1231 transitions and 397 transversions) and 100 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 1728 unique variants in 1728 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 1728 unique variants in 1728 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 9511 lines from VCF file\n",
      "NOTICE: A total of 9504 locus in VCF file passed QC threshold, representing 9001 SNPs (6707 transitions and 2294 transversions) and 503 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 9001 SNP genotypes (6707 transitions and 2294 transversions) and 503 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 9504 unique variants in 9504 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 9504 unique variants in 9504 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 1292 lines from VCF file\n",
      "NOTICE: A total of 1285 locus in VCF file passed QC threshold, representing 1212 SNPs (912 transitions and 300 transversions) and 73 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 1212 SNP genotypes (912 transitions and 300 transversions) and 73 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 1285 unique variants in 1285 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 1285 unique variants in 1285 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 1952 lines from VCF file\n",
      "NOTICE: A total of 1945 locus in VCF file passed QC threshold, representing 1842 SNPs (1404 transitions and 438 transversions) and 103 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 1842 SNP genotypes (1404 transitions and 438 transversions) and 103 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 1945 unique variants in 1945 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 1945 unique variants in 1945 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "# annotate using ANNOVAR\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    \n",
    "    ! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/{ancestry}_TMEM175_v1.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "    -out {WORK_DIR}/{ancestry}_TMEM175.annovar \\\n",
    "    -remove -protocol refGene,clinvar_20140902 \\\n",
    "    -operation g,f \\\n",
    "    --nopolish \\\n",
    "    -nastring . \\\n",
    "    -vcfinput"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ddc28160",
   "metadata": {},
   "source": [
    "### AAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b20d62a-e603-4be0-9a46-a3b5091a1aa0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882400</td>\n",
       "      <td>882400</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882400</td>\n",
       "      <td>chr4:882400:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>882432</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>chr4:882432:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>882455</td>\n",
       "      <td>882455</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882455</td>\n",
       "      <td>chr4:882455:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>882471</td>\n",
       "      <td>882471</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882471</td>\n",
       "      <td>chr4:882471:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>882498</td>\n",
       "      <td>882498</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882498</td>\n",
       "      <td>chr4:882498:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3444</th>\n",
       "      <td>4</td>\n",
       "      <td>1008499</td>\n",
       "      <td>1008499</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3935;dist=3127</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008499</td>\n",
       "      <td>chr4:1008499:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3445</th>\n",
       "      <td>4</td>\n",
       "      <td>1008515</td>\n",
       "      <td>1008515</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3951;dist=3111</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008515</td>\n",
       "      <td>chr4:1008515:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3446</th>\n",
       "      <td>4</td>\n",
       "      <td>1008535</td>\n",
       "      <td>1008535</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3971;dist=3091</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008535</td>\n",
       "      <td>chr4:1008535:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3447</th>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>1008649</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4085;dist=2977</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>chr4:1008649:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3448</th>\n",
       "      <td>4</td>\n",
       "      <td>1008654</td>\n",
       "      <td>1008654</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4090;dist=2972</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008654</td>\n",
       "      <td>chr4:1008654:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3449 rows √ó 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr    Start      End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       4   882400   882400   G   A     intronic          GAK   \n",
       "1       4   882432   882432   G   A     intronic          GAK   \n",
       "2       4   882455   882455   G   A     intronic          GAK   \n",
       "3       4   882471   882471   G   A     intronic          GAK   \n",
       "4       4   882498   882498   C   G     intronic          GAK   \n",
       "...   ...      ...      ...  ..  ..          ...          ...   \n",
       "3444    4  1008499  1008499   C   T   intergenic  IDUA;FGFRL1   \n",
       "3445    4  1008515  1008515   C   T   intergenic  IDUA;FGFRL1   \n",
       "3446    4  1008535  1008535   G   C   intergenic  IDUA;FGFRL1   \n",
       "3447    4  1008649  1008649   C   T   intergenic  IDUA;FGFRL1   \n",
       "3448    4  1008654  1008654   G   C   intergenic  IDUA;FGFRL1   \n",
       "\n",
       "       GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0                       .                  .                .   \n",
       "1                       .                  .                .   \n",
       "2                       .                  .                .   \n",
       "3                       .                  .                .   \n",
       "4                       .                  .                .   \n",
       "...                   ...                ...              ...   \n",
       "3444  dist=3935;dist=3127                  .                .   \n",
       "3445  dist=3951;dist=3111                  .                .   \n",
       "3446  dist=3971;dist=3091                  .                .   \n",
       "3447  dist=4085;dist=2977                  .                .   \n",
       "3448  dist=4090;dist=2972                  .                .   \n",
       "\n",
       "     clinvar_20140902  Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .         0.0          .          .           4   \n",
       "1                   .         0.0          .          .           4   \n",
       "2                   .         0.0          .          .           4   \n",
       "3                   .         0.0          .          .           4   \n",
       "4                   .         0.0          .          .           4   \n",
       "...               ...         ...        ...        ...         ...   \n",
       "3444                .         0.0          .          .           4   \n",
       "3445                .         0.0          .          .           4   \n",
       "3446                .         0.0          .          .           4   \n",
       "3447                .         0.0          .          .           4   \n",
       "3448                .         0.0          .          .           4   \n",
       "\n",
       "      Otherinfo5        Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0         882400   chr4:882400:G:A          G          A          .   \n",
       "1         882432   chr4:882432:G:A          G          A          .   \n",
       "2         882455   chr4:882455:G:A          G          A          .   \n",
       "3         882471   chr4:882471:G:A          G          A          .   \n",
       "4         882498   chr4:882498:C:G          C          G          .   \n",
       "...          ...               ...        ...        ...        ...   \n",
       "3444     1008499  chr4:1008499:C:T          C          T          .   \n",
       "3445     1008515  chr4:1008515:C:T          C          T          .   \n",
       "3446     1008535  chr4:1008535:G:C          G          C          .   \n",
       "3447     1008649  chr4:1008649:C:T          C          T          .   \n",
       "3448     1008654  chr4:1008654:G:C          G          C          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "3444           .          PR          GT         0/0  \n",
       "3445           .          PR          GT         0/0  \n",
       "3446           .          PR          GT         0/0  \n",
       "3447           .          PR          GT         0/0  \n",
       "3448           .          PR          GT         0/0  \n",
       "\n",
       "[3449 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/TMEM175_AAC/AAC_TMEM175.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "183f61f5-40f9-48fd-ad39-f655d33915c3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic           2746\n",
       "exonic              287\n",
       "intergenic          175\n",
       "UTR3                102\n",
       "downstream           58\n",
       "UTR5                 42\n",
       "upstream             36\n",
       "splicing              2\n",
       "exonic;splicing       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "022b5a71-dc71-4585-a9df-f198923ad6c3",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "ff40e234-de59-492d-9323-c6a91f7b5d0e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV         155\n",
       "synonymous SNV            123\n",
       "stopgain                    4\n",
       "nonframeshift deletion      3\n",
       "frameshift deletion         2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ee7bed0b",
   "metadata": {},
   "source": [
    "### AFR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5bfb8ae0-3f37-430f-8db4-b8f092555733",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882400</td>\n",
       "      <td>882400</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882400</td>\n",
       "      <td>chr4:882400:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>882432</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>chr4:882432:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>882454</td>\n",
       "      <td>882454</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882454</td>\n",
       "      <td>chr4:882454:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>882471</td>\n",
       "      <td>882471</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882471</td>\n",
       "      <td>chr4:882471:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>882498</td>\n",
       "      <td>882498</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882498</td>\n",
       "      <td>chr4:882498:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3848</th>\n",
       "      <td>4</td>\n",
       "      <td>1008499</td>\n",
       "      <td>1008499</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3935;dist=3127</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008499</td>\n",
       "      <td>chr4:1008499:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3849</th>\n",
       "      <td>4</td>\n",
       "      <td>1008524</td>\n",
       "      <td>1008524</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3960;dist=3102</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008524</td>\n",
       "      <td>chr4:1008524:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3850</th>\n",
       "      <td>4</td>\n",
       "      <td>1008620</td>\n",
       "      <td>1008620</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4056;dist=3006</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008620</td>\n",
       "      <td>chr4:1008620:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3851</th>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>1008649</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4085;dist=2977</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>chr4:1008649:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3852</th>\n",
       "      <td>4</td>\n",
       "      <td>1008654</td>\n",
       "      <td>1008654</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4090;dist=2972</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008654</td>\n",
       "      <td>chr4:1008654:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3853 rows √ó 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr    Start      End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       4   882400   882400   G   A     intronic          GAK   \n",
       "1       4   882432   882432   G   A     intronic          GAK   \n",
       "2       4   882454   882454   C   T     intronic          GAK   \n",
       "3       4   882471   882471   G   A     intronic          GAK   \n",
       "4       4   882498   882498   C   G     intronic          GAK   \n",
       "...   ...      ...      ...  ..  ..          ...          ...   \n",
       "3848    4  1008499  1008499   C   T   intergenic  IDUA;FGFRL1   \n",
       "3849    4  1008524  1008524   G   T   intergenic  IDUA;FGFRL1   \n",
       "3850    4  1008620  1008620   G   T   intergenic  IDUA;FGFRL1   \n",
       "3851    4  1008649  1008649   C   T   intergenic  IDUA;FGFRL1   \n",
       "3852    4  1008654  1008654   G   C   intergenic  IDUA;FGFRL1   \n",
       "\n",
       "       GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0                       .                  .                .   \n",
       "1                       .                  .                .   \n",
       "2                       .                  .                .   \n",
       "3                       .                  .                .   \n",
       "4                       .                  .                .   \n",
       "...                   ...                ...              ...   \n",
       "3848  dist=3935;dist=3127                  .                .   \n",
       "3849  dist=3960;dist=3102                  .                .   \n",
       "3850  dist=4056;dist=3006                  .                .   \n",
       "3851  dist=4085;dist=2977                  .                .   \n",
       "3852  dist=4090;dist=2972                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           4   \n",
       "1                   .          0          .          .           4   \n",
       "2                   .          0          .          .           4   \n",
       "3                   .          0          .          .           4   \n",
       "4                   .          0          .          .           4   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "3848                .          0          .          .           4   \n",
       "3849                .          0          .          .           4   \n",
       "3850                .          0          .          .           4   \n",
       "3851                .          0          .          .           4   \n",
       "3852                .          0          .          .           4   \n",
       "\n",
       "      Otherinfo5        Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0         882400   chr4:882400:G:A          G          A          .   \n",
       "1         882432   chr4:882432:G:A          G          A          .   \n",
       "2         882454   chr4:882454:C:T          C          T          .   \n",
       "3         882471   chr4:882471:G:A          G          A          .   \n",
       "4         882498   chr4:882498:C:G          C          G          .   \n",
       "...          ...               ...        ...        ...        ...   \n",
       "3848     1008499  chr4:1008499:C:T          C          T          .   \n",
       "3849     1008524  chr4:1008524:G:T          G          T          .   \n",
       "3850     1008620  chr4:1008620:G:T          G          T          .   \n",
       "3851     1008649  chr4:1008649:C:T          C          T          .   \n",
       "3852     1008654  chr4:1008654:G:C          G          C          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "3848           .          PR          GT         0/0  \n",
       "3849           .          PR          GT         0/0  \n",
       "3850           .          PR          GT         0/0  \n",
       "3851           .          PR          GT         0/0  \n",
       "3852           .          PR          GT         0/0  \n",
       "\n",
       "[3853 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/TMEM175_AFR/AFR_TMEM175.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "e60f2eb3-68c6-42c9-bc59-8d472fcd9653",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic           3069\n",
       "exonic              314\n",
       "intergenic          203\n",
       "UTR3                118\n",
       "downstream           63\n",
       "UTR5                 45\n",
       "upstream             40\n",
       "exonic;splicing       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "32cdb2bc-755b-4c3c-9ec3-3d4e22bc00bf",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "ad2afccb-7b7a-4e27-98a4-5610366fa5f6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV          161\n",
       "synonymous SNV             143\n",
       "frameshift deletion          4\n",
       "nonframeshift deletion       3\n",
       "stopgain                     2\n",
       "nonframeshift insertion      1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7da87ea3",
   "metadata": {},
   "source": [
    "### AJ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c96f016a-dcbe-4666-af1c-b9d1fb2c4b9c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882498</td>\n",
       "      <td>882498</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882498</td>\n",
       "      <td>chr4:882498:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>882530</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>882611</td>\n",
       "      <td>882611</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882611</td>\n",
       "      <td>chr4:882611:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>882745</td>\n",
       "      <td>882745</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>GAK:NM_001318134:exon11:c.C1242T:p.H414H,GAK:N...</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882745</td>\n",
       "      <td>chr4:882745:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>882801</td>\n",
       "      <td>882801</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GAK:NM_001318134:exon11:c.G1186A:p.A396T,GAK:N...</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882801</td>\n",
       "      <td>chr4:882801:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1280</th>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>1008337</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3773;dist=3289</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1281</th>\n",
       "      <td>4</td>\n",
       "      <td>1008371</td>\n",
       "      <td>1008371</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3807;dist=3255</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008370</td>\n",
       "      <td>chr4:1008370:TC:T</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1282</th>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>1008480</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3916;dist=3146</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>chr4:1008480:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1283</th>\n",
       "      <td>4</td>\n",
       "      <td>1008538</td>\n",
       "      <td>1008538</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3974;dist=3088</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008538</td>\n",
       "      <td>chr4:1008538:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1284</th>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>1008649</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4085;dist=2977</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>chr4:1008649:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1285 rows √ó 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr    Start      End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       4   882498   882498   C   G     intronic          GAK   \n",
       "1       4   882530   882530   C   T     intronic          GAK   \n",
       "2       4   882611   882611   T   C     intronic          GAK   \n",
       "3       4   882745   882745   G   A       exonic          GAK   \n",
       "4       4   882801   882801   C   T       exonic          GAK   \n",
       "...   ...      ...      ...  ..  ..          ...          ...   \n",
       "1280    4  1008337  1008337   C   T   intergenic  IDUA;FGFRL1   \n",
       "1281    4  1008371  1008371   C   -   intergenic  IDUA;FGFRL1   \n",
       "1282    4  1008480  1008480   A   G   intergenic  IDUA;FGFRL1   \n",
       "1283    4  1008538  1008538   A   G   intergenic  IDUA;FGFRL1   \n",
       "1284    4  1008649  1008649   C   T   intergenic  IDUA;FGFRL1   \n",
       "\n",
       "       GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                       .                  .   \n",
       "1                       .                  .   \n",
       "2                       .                  .   \n",
       "3                       .     synonymous SNV   \n",
       "4                       .  nonsynonymous SNV   \n",
       "...                   ...                ...   \n",
       "1280  dist=3773;dist=3289                  .   \n",
       "1281  dist=3807;dist=3255                  .   \n",
       "1282  dist=3916;dist=3146                  .   \n",
       "1283  dist=3974;dist=3088                  .   \n",
       "1284  dist=4085;dist=2977                  .   \n",
       "\n",
       "                                       AAChange.refGene clinvar_20140902  \\\n",
       "0                                                     .                .   \n",
       "1                                                     .                .   \n",
       "2                                                     .                .   \n",
       "3     GAK:NM_001318134:exon11:c.C1242T:p.H414H,GAK:N...                .   \n",
       "4     GAK:NM_001318134:exon11:c.G1186A:p.A396T,GAK:N...                .   \n",
       "...                                                 ...              ...   \n",
       "1280                                                  .                .   \n",
       "1281                                                  .                .   \n",
       "1282                                                  .                .   \n",
       "1283                                                  .                .   \n",
       "1284                                                  .                .   \n",
       "\n",
       "      Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  Otherinfo5  \\\n",
       "0            0.0          .          .           4      882498   \n",
       "1            0.0          .          .           4      882530   \n",
       "2            0.0          .          .           4      882611   \n",
       "3            0.0          .          .           4      882745   \n",
       "4            0.0          .          .           4      882801   \n",
       "...          ...        ...        ...         ...         ...   \n",
       "1280         1.0          .          .           4     1008337   \n",
       "1281         0.0          .          .           4     1008370   \n",
       "1282         0.0          .          .           4     1008480   \n",
       "1283         0.0          .          .           4     1008538   \n",
       "1284         0.0          .          .           4     1008649   \n",
       "\n",
       "             Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9 Otherinfo10  \\\n",
       "0       chr4:882498:C:G          C          G          .           .   \n",
       "1       chr4:882530:C:T          C          T          .           .   \n",
       "2       chr4:882611:T:C          T          C          .           .   \n",
       "3       chr4:882745:G:A          G          A          .           .   \n",
       "4       chr4:882801:C:T          C          T          .           .   \n",
       "...                 ...        ...        ...        ...         ...   \n",
       "1280   chr4:1008337:C:T          C          T          .           .   \n",
       "1281  chr4:1008370:TC:T         TC          T          .           .   \n",
       "1282   chr4:1008480:A:G          A          G          .           .   \n",
       "1283   chr4:1008538:A:G          A          G          .           .   \n",
       "1284   chr4:1008649:C:T          C          T          .           .   \n",
       "\n",
       "     Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0             PR          GT         0/0  \n",
       "1             PR          GT         0/0  \n",
       "2             PR          GT         0/0  \n",
       "3             PR          GT         0/0  \n",
       "4             PR          GT         0/0  \n",
       "...          ...         ...         ...  \n",
       "1280          PR          GT         1/1  \n",
       "1281          PR          GT         0/0  \n",
       "1282          PR          GT         0/0  \n",
       "1283          PR          GT         0/0  \n",
       "1284          PR          GT         0/0  \n",
       "\n",
       "[1285 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/TMEM175_AJ/AJ_TMEM175.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "2b731629-eeb1-4241-b58e-17a6e4e57702",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic           1024\n",
       "exonic              125\n",
       "intergenic           60\n",
       "UTR3                 43\n",
       "downstream           20\n",
       "upstream              6\n",
       "UTR5                  6\n",
       "exonic;splicing       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8a69325-86ae-49ef-8bfe-f6d786b23a19",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "1e87e13a-ef0d-491d-a277-85c7ea9d53ef",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV         66\n",
       "synonymous SNV            55\n",
       "frameshift deletion        2\n",
       "stopgain                   1\n",
       "nonframeshift deletion     1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b7ada0e",
   "metadata": {},
   "source": [
    "### AMR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c178e42f-9588-4327-8319-5db30d88fea0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882397</td>\n",
       "      <td>882397</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882397</td>\n",
       "      <td>chr4:882397:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882400</td>\n",
       "      <td>882400</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882400</td>\n",
       "      <td>chr4:882400:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>882432</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>chr4:882432:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>882454</td>\n",
       "      <td>882454</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882454</td>\n",
       "      <td>chr4:882454:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>882498</td>\n",
       "      <td>882498</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882498</td>\n",
       "      <td>chr4:882498:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1982</th>\n",
       "      <td>4</td>\n",
       "      <td>1008395</td>\n",
       "      <td>1008395</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3831;dist=3231</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008395</td>\n",
       "      <td>chr4:1008395:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1983</th>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>1008480</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3916;dist=3146</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>chr4:1008480:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1984</th>\n",
       "      <td>4</td>\n",
       "      <td>1008515</td>\n",
       "      <td>1008515</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3951;dist=3111</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008515</td>\n",
       "      <td>chr4:1008515:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1985</th>\n",
       "      <td>4</td>\n",
       "      <td>1008571</td>\n",
       "      <td>1008571</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4007;dist=3055</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008571</td>\n",
       "      <td>chr4:1008571:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1986</th>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>1008649</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4085;dist=2977</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>chr4:1008649:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1987 rows √ó 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr    Start      End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       4   882397   882397   C   T     intronic          GAK   \n",
       "1       4   882400   882400   G   A     intronic          GAK   \n",
       "2       4   882432   882432   G   A     intronic          GAK   \n",
       "3       4   882454   882454   C   T     intronic          GAK   \n",
       "4       4   882498   882498   C   G     intronic          GAK   \n",
       "...   ...      ...      ...  ..  ..          ...          ...   \n",
       "1982    4  1008395  1008395   C   T   intergenic  IDUA;FGFRL1   \n",
       "1983    4  1008480  1008480   A   G   intergenic  IDUA;FGFRL1   \n",
       "1984    4  1008515  1008515   C   T   intergenic  IDUA;FGFRL1   \n",
       "1985    4  1008571  1008571   G   A   intergenic  IDUA;FGFRL1   \n",
       "1986    4  1008649  1008649   C   T   intergenic  IDUA;FGFRL1   \n",
       "\n",
       "       GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0                       .                  .                .   \n",
       "1                       .                  .                .   \n",
       "2                       .                  .                .   \n",
       "3                       .                  .                .   \n",
       "4                       .                  .                .   \n",
       "...                   ...                ...              ...   \n",
       "1982  dist=3831;dist=3231                  .                .   \n",
       "1983  dist=3916;dist=3146                  .                .   \n",
       "1984  dist=3951;dist=3111                  .                .   \n",
       "1985  dist=4007;dist=3055                  .                .   \n",
       "1986  dist=4085;dist=2977                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           4   \n",
       "1                   .          0          .          .           4   \n",
       "2                   .          0          .          .           4   \n",
       "3                   .          0          .          .           4   \n",
       "4                   .          0          .          .           4   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "1982                .          0          .          .           4   \n",
       "1983                .          0          .          .           4   \n",
       "1984                .          0          .          .           4   \n",
       "1985                .          0          .          .           4   \n",
       "1986                .          0          .          .           4   \n",
       "\n",
       "      Otherinfo5        Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0         882397   chr4:882397:C:T          C          T          .   \n",
       "1         882400   chr4:882400:G:A          G          A          .   \n",
       "2         882432   chr4:882432:G:A          G          A          .   \n",
       "3         882454   chr4:882454:C:T          C          T          .   \n",
       "4         882498   chr4:882498:C:G          C          G          .   \n",
       "...          ...               ...        ...        ...        ...   \n",
       "1982     1008395  chr4:1008395:C:T          C          T          .   \n",
       "1983     1008480  chr4:1008480:A:G          A          G          .   \n",
       "1984     1008515  chr4:1008515:C:T          C          T          .   \n",
       "1985     1008571  chr4:1008571:G:A          G          A          .   \n",
       "1986     1008649  chr4:1008649:C:T          C          T          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "1982           .          PR          GT         0/0  \n",
       "1983           .          PR          GT         0/0  \n",
       "1984           .          PR          GT         0/0  \n",
       "1985           .          PR          GT         0/0  \n",
       "1986           .          PR          GT         0/0  \n",
       "\n",
       "[1987 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/TMEM175_AMR/AMR_TMEM175.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "9ae7b44d-44b3-4373-a79c-8ea15c484de3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic           1596\n",
       "exonic              178\n",
       "intergenic          104\n",
       "UTR3                 56\n",
       "downstream           23\n",
       "UTR5                 17\n",
       "upstream             12\n",
       "exonic;splicing       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b76abdc-89ed-453d-a65b-e243be9c0a02",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "62bb1733-90be-48db-9383-33868e742cc3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV          89\n",
       "synonymous SNV             85\n",
       "frameshift deletion         2\n",
       "stopgain                    1\n",
       "nonframeshift insertion     1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af7c3683",
   "metadata": {},
   "source": [
    "### CAH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6f2f6d87-5e50-49b8-beb5-c4fcea98b4be",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882400</td>\n",
       "      <td>882400</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882400</td>\n",
       "      <td>chr4:882400:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>882432</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>chr4:882432:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>882454</td>\n",
       "      <td>882454</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882454</td>\n",
       "      <td>chr4:882454:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>882498</td>\n",
       "      <td>882498</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882498</td>\n",
       "      <td>chr4:882498:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>882524</td>\n",
       "      <td>882524</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882524</td>\n",
       "      <td>chr4:882524:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3085</th>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>1008480</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3916;dist=3146</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>chr4:1008480:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3086</th>\n",
       "      <td>4</td>\n",
       "      <td>1008515</td>\n",
       "      <td>1008515</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3951;dist=3111</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008515</td>\n",
       "      <td>chr4:1008515:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3087</th>\n",
       "      <td>4</td>\n",
       "      <td>1008570</td>\n",
       "      <td>1008570</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4006;dist=3056</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008570</td>\n",
       "      <td>chr4:1008570:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3088</th>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>1008649</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4085;dist=2977</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>chr4:1008649:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3089</th>\n",
       "      <td>4</td>\n",
       "      <td>1008654</td>\n",
       "      <td>1008654</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4090;dist=2972</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008654</td>\n",
       "      <td>chr4:1008654:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3090 rows √ó 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr    Start      End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       4   882400   882400   G   A     intronic          GAK   \n",
       "1       4   882432   882432   G   A     intronic          GAK   \n",
       "2       4   882454   882454   C   T     intronic          GAK   \n",
       "3       4   882498   882498   C   G     intronic          GAK   \n",
       "4       4   882524   882524   A   G     intronic          GAK   \n",
       "...   ...      ...      ...  ..  ..          ...          ...   \n",
       "3085    4  1008480  1008480   A   G   intergenic  IDUA;FGFRL1   \n",
       "3086    4  1008515  1008515   C   T   intergenic  IDUA;FGFRL1   \n",
       "3087    4  1008570  1008570   C   T   intergenic  IDUA;FGFRL1   \n",
       "3088    4  1008649  1008649   C   T   intergenic  IDUA;FGFRL1   \n",
       "3089    4  1008654  1008654   G   C   intergenic  IDUA;FGFRL1   \n",
       "\n",
       "       GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0                       .                  .                .   \n",
       "1                       .                  .                .   \n",
       "2                       .                  .                .   \n",
       "3                       .                  .                .   \n",
       "4                       .                  .                .   \n",
       "...                   ...                ...              ...   \n",
       "3085  dist=3916;dist=3146                  .                .   \n",
       "3086  dist=3951;dist=3111                  .                .   \n",
       "3087  dist=4006;dist=3056                  .                .   \n",
       "3088  dist=4085;dist=2977                  .                .   \n",
       "3089  dist=4090;dist=2972                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           4   \n",
       "1                   .          0          .          .           4   \n",
       "2                   .          0          .          .           4   \n",
       "3                   .          0          .          .           4   \n",
       "4                   .          0          .          .           4   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "3085                .          0          .          .           4   \n",
       "3086                .          0          .          .           4   \n",
       "3087                .          0          .          .           4   \n",
       "3088                .          0          .          .           4   \n",
       "3089                .          0          .          .           4   \n",
       "\n",
       "      Otherinfo5        Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0         882400   chr4:882400:G:A          G          A          .   \n",
       "1         882432   chr4:882432:G:A          G          A          .   \n",
       "2         882454   chr4:882454:C:T          C          T          .   \n",
       "3         882498   chr4:882498:C:G          C          G          .   \n",
       "4         882524   chr4:882524:A:G          A          G          .   \n",
       "...          ...               ...        ...        ...        ...   \n",
       "3085     1008480  chr4:1008480:A:G          A          G          .   \n",
       "3086     1008515  chr4:1008515:C:T          C          T          .   \n",
       "3087     1008570  chr4:1008570:C:T          C          T          .   \n",
       "3088     1008649  chr4:1008649:C:T          C          T          .   \n",
       "3089     1008654  chr4:1008654:G:C          G          C          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "3085           .          PR          GT         0/0  \n",
       "3086           .          PR          GT         0/0  \n",
       "3087           .          PR          GT         0/0  \n",
       "3088           .          PR          GT         0/0  \n",
       "3089           .          PR          GT         0/0  \n",
       "\n",
       "[3090 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/TMEM175_CAH/CAH_TMEM175.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "99ef454c-3694-4aca-826c-6d1554b4b0d9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic           2467\n",
       "exonic              260\n",
       "intergenic          168\n",
       "UTR3                 87\n",
       "downstream           46\n",
       "UTR5                 31\n",
       "upstream             30\n",
       "exonic;splicing       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "202ab877-11f8-4d19-9939-30e0ca0ac0d1",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "bf5af1bd-c7a5-4e09-b7c7-b77ba7154a05",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV         139\n",
       "synonymous SNV            113\n",
       "stopgain                    4\n",
       "frameshift deletion         2\n",
       "nonframeshift deletion      1\n",
       "frameshift insertion        1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0352f26",
   "metadata": {},
   "source": [
    "### CAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ebcfd55c-1755-474d-920d-60c3b00de695",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882429</td>\n",
       "      <td>882429</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882429</td>\n",
       "      <td>chr4:882429:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>882432</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>chr4:882432:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>882468</td>\n",
       "      <td>882468</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882468</td>\n",
       "      <td>chr4:882468:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>882498</td>\n",
       "      <td>882498</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882498</td>\n",
       "      <td>chr4:882498:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>882501</td>\n",
       "      <td>882501</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882501</td>\n",
       "      <td>chr4:882501:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1950</th>\n",
       "      <td>4</td>\n",
       "      <td>1008371</td>\n",
       "      <td>1008371</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3807;dist=3255</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008370</td>\n",
       "      <td>chr4:1008370:TC:T</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1951</th>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>1008480</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3916;dist=3146</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>chr4:1008480:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1952</th>\n",
       "      <td>4</td>\n",
       "      <td>1008538</td>\n",
       "      <td>1008538</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3974;dist=3088</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008538</td>\n",
       "      <td>chr4:1008538:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1953</th>\n",
       "      <td>4</td>\n",
       "      <td>1008570</td>\n",
       "      <td>1008570</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4006;dist=3056</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008570</td>\n",
       "      <td>chr4:1008570:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1954</th>\n",
       "      <td>4</td>\n",
       "      <td>1008600</td>\n",
       "      <td>1008600</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4036;dist=3026</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008600</td>\n",
       "      <td>chr4:1008600:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1955 rows √ó 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr    Start      End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       4   882429   882429   G   C     intronic          GAK   \n",
       "1       4   882432   882432   G   A     intronic          GAK   \n",
       "2       4   882468   882468   C   G     intronic          GAK   \n",
       "3       4   882498   882498   C   G     intronic          GAK   \n",
       "4       4   882501   882501   C   A     intronic          GAK   \n",
       "...   ...      ...      ...  ..  ..          ...          ...   \n",
       "1950    4  1008371  1008371   C   -   intergenic  IDUA;FGFRL1   \n",
       "1951    4  1008480  1008480   A   G   intergenic  IDUA;FGFRL1   \n",
       "1952    4  1008538  1008538   A   G   intergenic  IDUA;FGFRL1   \n",
       "1953    4  1008570  1008570   C   T   intergenic  IDUA;FGFRL1   \n",
       "1954    4  1008600  1008600   C   T   intergenic  IDUA;FGFRL1   \n",
       "\n",
       "       GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0                       .                  .                .   \n",
       "1                       .                  .                .   \n",
       "2                       .                  .                .   \n",
       "3                       .                  .                .   \n",
       "4                       .                  .                .   \n",
       "...                   ...                ...              ...   \n",
       "1950  dist=3807;dist=3255                  .                .   \n",
       "1951  dist=3916;dist=3146                  .                .   \n",
       "1952  dist=3974;dist=3088                  .                .   \n",
       "1953  dist=4006;dist=3056                  .                .   \n",
       "1954  dist=4036;dist=3026                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           4   \n",
       "1                   .          0          .          .           4   \n",
       "2                   .          0          .          .           4   \n",
       "3                   .          0          .          .           4   \n",
       "4                   .          0          .          .           4   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "1950                .          0          .          .           4   \n",
       "1951                .          0          .          .           4   \n",
       "1952                .          0          .          .           4   \n",
       "1953                .          0          .          .           4   \n",
       "1954                .          0          .          .           4   \n",
       "\n",
       "      Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0         882429    chr4:882429:G:C          G          C          .   \n",
       "1         882432    chr4:882432:G:A          G          A          .   \n",
       "2         882468    chr4:882468:C:G          C          G          .   \n",
       "3         882498    chr4:882498:C:G          C          G          .   \n",
       "4         882501    chr4:882501:C:A          C          A          .   \n",
       "...          ...                ...        ...        ...        ...   \n",
       "1950     1008370  chr4:1008370:TC:T         TC          T          .   \n",
       "1951     1008480   chr4:1008480:A:G          A          G          .   \n",
       "1952     1008538   chr4:1008538:A:G          A          G          .   \n",
       "1953     1008570   chr4:1008570:C:T          C          T          .   \n",
       "1954     1008600   chr4:1008600:C:T          C          T          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "1950           .          PR          GT         0/0  \n",
       "1951           .          PR          GT         0/0  \n",
       "1952           .          PR          GT         0/0  \n",
       "1953           .          PR          GT         0/0  \n",
       "1954           .          PR          GT         0/0  \n",
       "\n",
       "[1955 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/TMEM175_CAS/CAS_TMEM175.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "2e7db3c5-aa82-41c0-9dce-969fbf842974",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic           1568\n",
       "exonic              167\n",
       "intergenic           96\n",
       "UTR3                 62\n",
       "UTR5                 26\n",
       "downstream           22\n",
       "upstream             13\n",
       "exonic;splicing       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "742ef450-0bf8-4c77-8bcd-1a4bd7810dab",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "a6b69f47-a29c-49a0-836e-503d9fe25949",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV      89\n",
       "synonymous SNV         73\n",
       "stopgain                4\n",
       "frameshift deletion     1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f7ab8fcd",
   "metadata": {},
   "source": [
    "### EAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cfdcbdd5-b1a2-4960-a114-a83ea9006fa1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882390</td>\n",
       "      <td>882390</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882390</td>\n",
       "      <td>chr4:882390:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>882432</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>chr4:882432:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>882501</td>\n",
       "      <td>882501</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882501</td>\n",
       "      <td>chr4:882501:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>882507</td>\n",
       "      <td>882507</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882507</td>\n",
       "      <td>chr4:882507:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>882530</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3166</th>\n",
       "      <td>4</td>\n",
       "      <td>1008570</td>\n",
       "      <td>1008570</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4006;dist=3056</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008570</td>\n",
       "      <td>chr4:1008570:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3167</th>\n",
       "      <td>4</td>\n",
       "      <td>1008634</td>\n",
       "      <td>1008634</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4070;dist=2992</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008634</td>\n",
       "      <td>chr4:1008634:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3168</th>\n",
       "      <td>4</td>\n",
       "      <td>1008636</td>\n",
       "      <td>1008636</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4072;dist=2990</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008636</td>\n",
       "      <td>chr4:1008636:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3169</th>\n",
       "      <td>4</td>\n",
       "      <td>1008644</td>\n",
       "      <td>1008644</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4080;dist=2982</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008644</td>\n",
       "      <td>chr4:1008644:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3170</th>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>1008649</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4085;dist=2977</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>chr4:1008649:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3171 rows √ó 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr    Start      End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       4   882390   882390   C   G     intronic          GAK   \n",
       "1       4   882432   882432   G   A     intronic          GAK   \n",
       "2       4   882501   882501   C   A     intronic          GAK   \n",
       "3       4   882507   882507   G   A     intronic          GAK   \n",
       "4       4   882530   882530   C   T     intronic          GAK   \n",
       "...   ...      ...      ...  ..  ..          ...          ...   \n",
       "3166    4  1008570  1008570   C   T   intergenic  IDUA;FGFRL1   \n",
       "3167    4  1008634  1008634   A   G   intergenic  IDUA;FGFRL1   \n",
       "3168    4  1008636  1008636   C   A   intergenic  IDUA;FGFRL1   \n",
       "3169    4  1008644  1008644   T   C   intergenic  IDUA;FGFRL1   \n",
       "3170    4  1008649  1008649   C   T   intergenic  IDUA;FGFRL1   \n",
       "\n",
       "       GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0                       .                  .                .   \n",
       "1                       .                  .                .   \n",
       "2                       .                  .                .   \n",
       "3                       .                  .                .   \n",
       "4                       .                  .                .   \n",
       "...                   ...                ...              ...   \n",
       "3166  dist=4006;dist=3056                  .                .   \n",
       "3167  dist=4070;dist=2992                  .                .   \n",
       "3168  dist=4072;dist=2990                  .                .   \n",
       "3169  dist=4080;dist=2982                  .                .   \n",
       "3170  dist=4085;dist=2977                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           4   \n",
       "1                   .          0          .          .           4   \n",
       "2                   .          0          .          .           4   \n",
       "3                   .          0          .          .           4   \n",
       "4                   .          1          .          .           4   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "3166                .          0          .          .           4   \n",
       "3167                .          0          .          .           4   \n",
       "3168                .          0          .          .           4   \n",
       "3169                .          0          .          .           4   \n",
       "3170                .          0          .          .           4   \n",
       "\n",
       "      Otherinfo5        Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0         882390   chr4:882390:C:G          C          G          .   \n",
       "1         882432   chr4:882432:G:A          G          A          .   \n",
       "2         882501   chr4:882501:C:A          C          A          .   \n",
       "3         882507   chr4:882507:G:A          G          A          .   \n",
       "4         882530   chr4:882530:C:T          C          T          .   \n",
       "...          ...               ...        ...        ...        ...   \n",
       "3166     1008570  chr4:1008570:C:T          C          T          .   \n",
       "3167     1008634  chr4:1008634:A:G          A          G          .   \n",
       "3168     1008636  chr4:1008636:C:A          C          A          .   \n",
       "3169     1008644  chr4:1008644:T:C          T          C          .   \n",
       "3170     1008649  chr4:1008649:C:T          C          T          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         1/1  \n",
       "...          ...         ...         ...         ...  \n",
       "3166           .          PR          GT         0/0  \n",
       "3167           .          PR          GT         0/0  \n",
       "3168           .          PR          GT         0/0  \n",
       "3169           .          PR          GT         0/0  \n",
       "3170           .          PR          GT         0/0  \n",
       "\n",
       "[3171 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/TMEM175_EAS/EAS_TMEM175.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "152d60cd-05b5-48b6-8204-743f58f429cb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic           2518\n",
       "exonic              285\n",
       "intergenic          174\n",
       "UTR3                 99\n",
       "UTR5                 43\n",
       "downstream           33\n",
       "upstream             17\n",
       "exonic;splicing       1\n",
       "splicing              1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "467c7db6-6f1e-46c1-a29b-f8cb99319b20",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "e1a6f2c1-b2a5-4869-9df1-56cd26d3e73e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV         164\n",
       "synonymous SNV            109\n",
       "stopgain                    6\n",
       "frameshift deletion         4\n",
       "nonframeshift deletion      2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "86a0aa27",
   "metadata": {},
   "source": [
    "### MDE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd0eec6d-9c07-404f-a661-51410ef13813",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>882432</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>chr4:882432:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882471</td>\n",
       "      <td>882471</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882471</td>\n",
       "      <td>chr4:882471:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>882498</td>\n",
       "      <td>882498</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882498</td>\n",
       "      <td>chr4:882498:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>882517</td>\n",
       "      <td>882517</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882517</td>\n",
       "      <td>chr4:882517:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>882530</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1940</th>\n",
       "      <td>4</td>\n",
       "      <td>1008371</td>\n",
       "      <td>1008371</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3807;dist=3255</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008370</td>\n",
       "      <td>chr4:1008370:TC:T</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1941</th>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>1008480</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3916;dist=3146</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>chr4:1008480:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1942</th>\n",
       "      <td>4</td>\n",
       "      <td>1008515</td>\n",
       "      <td>1008515</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3951;dist=3111</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008515</td>\n",
       "      <td>chr4:1008515:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1943</th>\n",
       "      <td>4</td>\n",
       "      <td>1008521</td>\n",
       "      <td>1008521</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3957;dist=3105</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008521</td>\n",
       "      <td>chr4:1008521:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1944</th>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>1008649</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4085;dist=2977</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>chr4:1008649:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1945 rows √ó 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr    Start      End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       4   882432   882432   G   A     intronic          GAK   \n",
       "1       4   882471   882471   G   A     intronic          GAK   \n",
       "2       4   882498   882498   C   G     intronic          GAK   \n",
       "3       4   882517   882517   C   A     intronic          GAK   \n",
       "4       4   882530   882530   C   T     intronic          GAK   \n",
       "...   ...      ...      ...  ..  ..          ...          ...   \n",
       "1940    4  1008371  1008371   C   -   intergenic  IDUA;FGFRL1   \n",
       "1941    4  1008480  1008480   A   G   intergenic  IDUA;FGFRL1   \n",
       "1942    4  1008515  1008515   C   T   intergenic  IDUA;FGFRL1   \n",
       "1943    4  1008521  1008521   T   C   intergenic  IDUA;FGFRL1   \n",
       "1944    4  1008649  1008649   C   T   intergenic  IDUA;FGFRL1   \n",
       "\n",
       "       GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0                       .                  .                .   \n",
       "1                       .                  .                .   \n",
       "2                       .                  .                .   \n",
       "3                       .                  .                .   \n",
       "4                       .                  .                .   \n",
       "...                   ...                ...              ...   \n",
       "1940  dist=3807;dist=3255                  .                .   \n",
       "1941  dist=3916;dist=3146                  .                .   \n",
       "1942  dist=3951;dist=3111                  .                .   \n",
       "1943  dist=3957;dist=3105                  .                .   \n",
       "1944  dist=4085;dist=2977                  .                .   \n",
       "\n",
       "     clinvar_20140902  Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .         0.0          .          .           4   \n",
       "1                   .         0.0          .          .           4   \n",
       "2                   .         0.0          .          .           4   \n",
       "3                   .         0.0          .          .           4   \n",
       "4                   .         0.5          .          .           4   \n",
       "...               ...         ...        ...        ...         ...   \n",
       "1940                .         0.0          .          .           4   \n",
       "1941                .         0.5          .          .           4   \n",
       "1942                .         0.0          .          .           4   \n",
       "1943                .         0.0          .          .           4   \n",
       "1944                .         0.0          .          .           4   \n",
       "\n",
       "      Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0         882432    chr4:882432:G:A          G          A          .   \n",
       "1         882471    chr4:882471:G:A          G          A          .   \n",
       "2         882498    chr4:882498:C:G          C          G          .   \n",
       "3         882517    chr4:882517:C:A          C          A          .   \n",
       "4         882530    chr4:882530:C:T          C          T          .   \n",
       "...          ...                ...        ...        ...        ...   \n",
       "1940     1008370  chr4:1008370:TC:T         TC          T          .   \n",
       "1941     1008480   chr4:1008480:A:G          A          G          .   \n",
       "1942     1008515   chr4:1008515:C:T          C          T          .   \n",
       "1943     1008521   chr4:1008521:T:C          T          C          .   \n",
       "1944     1008649   chr4:1008649:C:T          C          T          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/1  \n",
       "...          ...         ...         ...         ...  \n",
       "1940           .          PR          GT         0/0  \n",
       "1941           .          PR          GT         0/1  \n",
       "1942           .          PR          GT         0/0  \n",
       "1943           .          PR          GT         0/0  \n",
       "1944           .          PR          GT         0/0  \n",
       "\n",
       "[1945 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/TMEM175_MDE/MDE_TMEM175.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "dba5c22f-2ef2-43e0-8c28-3a69f75d4e16",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic           1547\n",
       "exonic              177\n",
       "intergenic           94\n",
       "UTR3                 59\n",
       "downstream           34\n",
       "UTR5                 22\n",
       "upstream             11\n",
       "exonic;splicing       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3038cd0-f63b-46b7-ae85-042f100efc39",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "5681f86b-7d10-4321-ad82-6af8fd5d8db1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "synonymous SNV         91\n",
       "nonsynonymous SNV      84\n",
       "frameshift deletion     1\n",
       "stopgain                1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "daa46743",
   "metadata": {},
   "source": [
    "### SAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed235bcd-9dab-483a-86eb-c1aa46becf06",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882397</td>\n",
       "      <td>882397</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882397</td>\n",
       "      <td>chr4:882397:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>882530</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>882566</td>\n",
       "      <td>882566</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882565</td>\n",
       "      <td>chr4:882565:AG:A</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>882611</td>\n",
       "      <td>882611</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882611</td>\n",
       "      <td>chr4:882611:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>882678</td>\n",
       "      <td>882678</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882678</td>\n",
       "      <td>chr4:882678:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1723</th>\n",
       "      <td>4</td>\n",
       "      <td>1008246</td>\n",
       "      <td>1008246</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3682;dist=3380</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008246</td>\n",
       "      <td>chr4:1008246:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1724</th>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>1008337</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3773;dist=3289</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1725</th>\n",
       "      <td>4</td>\n",
       "      <td>1008371</td>\n",
       "      <td>1008371</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3807;dist=3255</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008370</td>\n",
       "      <td>chr4:1008370:TC:T</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1726</th>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>1008480</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3916;dist=3146</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>chr4:1008480:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1727</th>\n",
       "      <td>4</td>\n",
       "      <td>1008538</td>\n",
       "      <td>1008538</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3974;dist=3088</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008538</td>\n",
       "      <td>chr4:1008538:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1728 rows √ó 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr    Start      End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       4   882397   882397   C   T     intronic          GAK   \n",
       "1       4   882530   882530   C   T     intronic          GAK   \n",
       "2       4   882566   882566   G   -     intronic          GAK   \n",
       "3       4   882611   882611   T   C     intronic          GAK   \n",
       "4       4   882678   882678   C   G     intronic          GAK   \n",
       "...   ...      ...      ...  ..  ..          ...          ...   \n",
       "1723    4  1008246  1008246   C   T   intergenic  IDUA;FGFRL1   \n",
       "1724    4  1008337  1008337   C   T   intergenic  IDUA;FGFRL1   \n",
       "1725    4  1008371  1008371   C   -   intergenic  IDUA;FGFRL1   \n",
       "1726    4  1008480  1008480   A   G   intergenic  IDUA;FGFRL1   \n",
       "1727    4  1008538  1008538   A   G   intergenic  IDUA;FGFRL1   \n",
       "\n",
       "       GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0                       .                  .                .   \n",
       "1                       .                  .                .   \n",
       "2                       .                  .                .   \n",
       "3                       .                  .                .   \n",
       "4                       .                  .                .   \n",
       "...                   ...                ...              ...   \n",
       "1723  dist=3682;dist=3380                  .                .   \n",
       "1724  dist=3773;dist=3289                  .                .   \n",
       "1725  dist=3807;dist=3255                  .                .   \n",
       "1726  dist=3916;dist=3146                  .                .   \n",
       "1727  dist=3974;dist=3088                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           4   \n",
       "1                   .          0          .          .           4   \n",
       "2                   .          0          .          .           4   \n",
       "3                   .          0          .          .           4   \n",
       "4                   .          0          .          .           4   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "1723                .          0          .          .           4   \n",
       "1724                .        0.5          .          .           4   \n",
       "1725                .          0          .          .           4   \n",
       "1726                .          0          .          .           4   \n",
       "1727                .          0          .          .           4   \n",
       "\n",
       "      Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0         882397    chr4:882397:C:T          C          T          .   \n",
       "1         882530    chr4:882530:C:T          C          T          .   \n",
       "2         882565   chr4:882565:AG:A         AG          A          .   \n",
       "3         882611    chr4:882611:T:C          T          C          .   \n",
       "4         882678    chr4:882678:C:G          C          G          .   \n",
       "...          ...                ...        ...        ...        ...   \n",
       "1723     1008246   chr4:1008246:C:T          C          T          .   \n",
       "1724     1008337   chr4:1008337:C:T          C          T          .   \n",
       "1725     1008370  chr4:1008370:TC:T         TC          T          .   \n",
       "1726     1008480   chr4:1008480:A:G          A          G          .   \n",
       "1727     1008538   chr4:1008538:A:G          A          G          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "1723           .          PR          GT         0/0  \n",
       "1724           .          PR          GT         0/1  \n",
       "1725           .          PR          GT         0/0  \n",
       "1726           .          PR          GT         0/0  \n",
       "1727           .          PR          GT         0/0  \n",
       "\n",
       "[1728 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/TMEM175_SAS/SAS_TMEM175.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "e258c45e-a6ef-4a0a-bca1-6c1671be0e7b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic           1412\n",
       "exonic              143\n",
       "intergenic           72\n",
       "UTR3                 50\n",
       "UTR5                 24\n",
       "downstream           18\n",
       "upstream              8\n",
       "exonic;splicing       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14f514fe-fccd-416d-9ff7-94546b7e1546",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "bd7cfd6a-f766-4eee-8bb9-d7093d5bc574",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV      75\n",
       "synonymous SNV         64\n",
       "stopgain                3\n",
       "frameshift deletion     1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9c3c13c5",
   "metadata": {},
   "source": [
    "### FIN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d24c787-a2f9-4dcc-9283-35fd498f0279",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>882530</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882611</td>\n",
       "      <td>882611</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882611</td>\n",
       "      <td>chr4:882611:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>882849</td>\n",
       "      <td>882849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882849</td>\n",
       "      <td>chr4:882849:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>883018</td>\n",
       "      <td>883018</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>883018</td>\n",
       "      <td>chr4:883018:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>883133</td>\n",
       "      <td>883133</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>883133</td>\n",
       "      <td>chr4:883133:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>600</th>\n",
       "      <td>4</td>\n",
       "      <td>1008009</td>\n",
       "      <td>1008009</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3445;dist=3617</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008009</td>\n",
       "      <td>chr4:1008009:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>601</th>\n",
       "      <td>4</td>\n",
       "      <td>1008209</td>\n",
       "      <td>1008209</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3645;dist=3417</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008209</td>\n",
       "      <td>chr4:1008209:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>1008337</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3773;dist=3289</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>603</th>\n",
       "      <td>4</td>\n",
       "      <td>1008371</td>\n",
       "      <td>1008371</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3807;dist=3255</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008370</td>\n",
       "      <td>chr4:1008370:TC:T</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>604</th>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>1008480</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=3916;dist=3146</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>chr4:1008480:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>605 rows √ó 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr    Start      End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0      4   882530   882530   C   T     intronic          GAK   \n",
       "1      4   882611   882611   T   C     intronic          GAK   \n",
       "2      4   882849   882849   G   A     intronic          GAK   \n",
       "3      4   883018   883018   G   A     intronic          GAK   \n",
       "4      4   883133   883133   G   A     intronic          GAK   \n",
       "..   ...      ...      ...  ..  ..          ...          ...   \n",
       "600    4  1008009  1008009   C   T   intergenic  IDUA;FGFRL1   \n",
       "601    4  1008209  1008209   C   T   intergenic  IDUA;FGFRL1   \n",
       "602    4  1008337  1008337   C   T   intergenic  IDUA;FGFRL1   \n",
       "603    4  1008371  1008371   C   -   intergenic  IDUA;FGFRL1   \n",
       "604    4  1008480  1008480   A   G   intergenic  IDUA;FGFRL1   \n",
       "\n",
       "      GeneDetail.refGene ExonicFunc.refGene AAChange.refGene clinvar_20140902  \\\n",
       "0                      .                  .                .                .   \n",
       "1                      .                  .                .                .   \n",
       "2                      .                  .                .                .   \n",
       "3                      .                  .                .                .   \n",
       "4                      .                  .                .                .   \n",
       "..                   ...                ...              ...              ...   \n",
       "600  dist=3445;dist=3617                  .                .                .   \n",
       "601  dist=3645;dist=3417                  .                .                .   \n",
       "602  dist=3773;dist=3289                  .                .                .   \n",
       "603  dist=3807;dist=3255                  .                .                .   \n",
       "604  dist=3916;dist=3146                  .                .                .   \n",
       "\n",
       "    Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  Otherinfo5  \\\n",
       "0          0.5          .          .           4      882530   \n",
       "1            0          .          .           4      882611   \n",
       "2            0          .          .           4      882849   \n",
       "3            0          .          .           4      883018   \n",
       "4            0          .          .           4      883133   \n",
       "..         ...        ...        ...         ...         ...   \n",
       "600        0.5          .          .           4     1008009   \n",
       "601          0          .          .           4     1008209   \n",
       "602          1          .          .           4     1008337   \n",
       "603          0          .          .           4     1008370   \n",
       "604          0          .          .           4     1008480   \n",
       "\n",
       "            Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9 Otherinfo10  \\\n",
       "0      chr4:882530:C:T          C          T          .           .   \n",
       "1      chr4:882611:T:C          T          C          .           .   \n",
       "2      chr4:882849:G:A          G          A          .           .   \n",
       "3      chr4:883018:G:A          G          A          .           .   \n",
       "4      chr4:883133:G:A          G          A          .           .   \n",
       "..                 ...        ...        ...        ...         ...   \n",
       "600   chr4:1008009:C:T          C          T          .           .   \n",
       "601   chr4:1008209:C:T          C          T          .           .   \n",
       "602   chr4:1008337:C:T          C          T          .           .   \n",
       "603  chr4:1008370:TC:T         TC          T          .           .   \n",
       "604   chr4:1008480:A:G          A          G          .           .   \n",
       "\n",
       "    Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0            PR          GT         0/1  \n",
       "1            PR          GT         0/0  \n",
       "2            PR          GT         0/0  \n",
       "3            PR          GT         0/0  \n",
       "4            PR          GT         0/0  \n",
       "..          ...         ...         ...  \n",
       "600          PR          GT         0/1  \n",
       "601          PR          GT         0/0  \n",
       "602          PR          GT         1/1  \n",
       "603          PR          GT         0/0  \n",
       "604          PR          GT         0/0  \n",
       "\n",
       "[605 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/TMEM175_FIN/FIN_TMEM175.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "0950d33b-6128-42a7-acfb-787928219a5c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic           491\n",
       "exonic              50\n",
       "intergenic          29\n",
       "UTR3                20\n",
       "downstream           6\n",
       "upstream             5\n",
       "UTR5                 3\n",
       "exonic;splicing      1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "367cb26b-7ac4-4c74-8b4e-3e778db2aa6a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "2ef51c7e-e50e-4179-a8ea-fed89c7002f5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "synonymous SNV       27\n",
       "nonsynonymous SNV    23\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1ec415f6",
   "metadata": {},
   "source": [
    "### EUR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "636c9cb5-45be-4134-b557-00c2756945ea",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882390</td>\n",
       "      <td>882390</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882390</td>\n",
       "      <td>chr4:882390:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882397</td>\n",
       "      <td>882397</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882397</td>\n",
       "      <td>chr4:882397:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>882422</td>\n",
       "      <td>882422</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882422</td>\n",
       "      <td>chr4:882422:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>882432</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>chr4:882432:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>882454</td>\n",
       "      <td>882454</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882454</td>\n",
       "      <td>chr4:882454:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9499</th>\n",
       "      <td>4</td>\n",
       "      <td>1008570</td>\n",
       "      <td>1008570</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4006;dist=3056</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008570</td>\n",
       "      <td>chr4:1008570:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9500</th>\n",
       "      <td>4</td>\n",
       "      <td>1008571</td>\n",
       "      <td>1008571</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4007;dist=3055</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008571</td>\n",
       "      <td>chr4:1008571:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9501</th>\n",
       "      <td>4</td>\n",
       "      <td>1008596</td>\n",
       "      <td>1008596</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4032;dist=3030</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008596</td>\n",
       "      <td>chr4:1008596:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>./.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9502</th>\n",
       "      <td>4</td>\n",
       "      <td>1008600</td>\n",
       "      <td>1008600</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4036;dist=3026</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008600</td>\n",
       "      <td>chr4:1008600:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9503</th>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>1008649</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA;FGFRL1</td>\n",
       "      <td>dist=4085;dist=2977</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008649</td>\n",
       "      <td>chr4:1008649:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9504 rows √ó 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr    Start      End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       4   882390   882390   C   A     intronic          GAK   \n",
       "1       4   882397   882397   C   T     intronic          GAK   \n",
       "2       4   882422   882422   C   T     intronic          GAK   \n",
       "3       4   882432   882432   G   A     intronic          GAK   \n",
       "4       4   882454   882454   C   T     intronic          GAK   \n",
       "...   ...      ...      ...  ..  ..          ...          ...   \n",
       "9499    4  1008570  1008570   C   T   intergenic  IDUA;FGFRL1   \n",
       "9500    4  1008571  1008571   G   A   intergenic  IDUA;FGFRL1   \n",
       "9501    4  1008596  1008596   C   T   intergenic  IDUA;FGFRL1   \n",
       "9502    4  1008600  1008600   C   T   intergenic  IDUA;FGFRL1   \n",
       "9503    4  1008649  1008649   C   T   intergenic  IDUA;FGFRL1   \n",
       "\n",
       "       GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0                       .                  .                .   \n",
       "1                       .                  .                .   \n",
       "2                       .                  .                .   \n",
       "3                       .                  .                .   \n",
       "4                       .                  .                .   \n",
       "...                   ...                ...              ...   \n",
       "9499  dist=4006;dist=3056                  .                .   \n",
       "9500  dist=4007;dist=3055                  .                .   \n",
       "9501  dist=4032;dist=3030                  .                .   \n",
       "9502  dist=4036;dist=3026                  .                .   \n",
       "9503  dist=4085;dist=2977                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           4   \n",
       "1                   .          0          .          .           4   \n",
       "2                   .          0          .          .           4   \n",
       "3                   .          0          .          .           4   \n",
       "4                   .          0          .          .           4   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "9499                .          0          .          .           4   \n",
       "9500                .          0          .          .           4   \n",
       "9501                .          .          .          .           4   \n",
       "9502                .          0          .          .           4   \n",
       "9503                .          0          .          .           4   \n",
       "\n",
       "      Otherinfo5        Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0         882390   chr4:882390:C:A          C          A          .   \n",
       "1         882397   chr4:882397:C:T          C          T          .   \n",
       "2         882422   chr4:882422:C:T          C          T          .   \n",
       "3         882432   chr4:882432:G:A          G          A          .   \n",
       "4         882454   chr4:882454:C:T          C          T          .   \n",
       "...          ...               ...        ...        ...        ...   \n",
       "9499     1008570  chr4:1008570:C:T          C          T          .   \n",
       "9500     1008571  chr4:1008571:G:A          G          A          .   \n",
       "9501     1008596  chr4:1008596:C:T          C          T          .   \n",
       "9502     1008600  chr4:1008600:C:T          C          T          .   \n",
       "9503     1008649  chr4:1008649:C:T          C          T          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "9499           .          PR          GT         0/0  \n",
       "9500           .          PR          GT         0/0  \n",
       "9501           .          PR          GT         ./.  \n",
       "9502           .          PR          GT         0/0  \n",
       "9503           .          PR          GT         0/0  \n",
       "\n",
       "[9504 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/TMEM175_EUR/EUR_TMEM175.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "ba43863e-9f25-45a1-be06-f9f18d7cf601",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic           7451\n",
       "exonic              951\n",
       "intergenic          474\n",
       "UTR3                266\n",
       "downstream          152\n",
       "UTR5                114\n",
       "upstream             90\n",
       "splicing              5\n",
       "exonic;splicing       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "05abffe5-765f-4334-9595-283cf048038b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "85f7ace5-2ae5-4147-b33a-d6f43df01af6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV          548\n",
       "synonymous SNV             367\n",
       "nonframeshift deletion      12\n",
       "stopgain                    10\n",
       "frameshift deletion          9\n",
       "frameshift insertion         4\n",
       "nonframeshift insertion      1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "80cd9e67",
   "metadata": {},
   "source": [
    "### Make lists of variants to keep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "47b09ef1-37c1-48f7-9ab1-4ae876af99c8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AFR\n",
      "{'AFR'}\n",
      "Total variants:  3853\n",
      "Intronic:  3069\n",
      "UTR3:  118\n",
      "UTR5:  45\n",
      "Total exonic:  314\n",
      "  Synonymous:  153\n",
      "  Nonsynonymous:  161\n",
      "nonsynonymous_lof:  167\n",
      "\n",
      "\n",
      "WORKING ON: AAC\n",
      "{'AAC'}\n",
      "Total variants:  3449\n",
      "Intronic:  2746\n",
      "UTR3:  102\n",
      "UTR5:  42\n",
      "Total exonic:  287\n",
      "  Synonymous:  132\n",
      "  Nonsynonymous:  155\n",
      "nonsynonymous_lof:  161\n",
      "\n",
      "\n",
      "WORKING ON: FIN\n",
      "{'FIN'}\n",
      "Total variants:  605\n",
      "Intronic:  491\n",
      "UTR3:  20\n",
      "UTR5:  3\n",
      "Total exonic:  50\n",
      "  Synonymous:  27\n",
      "  Nonsynonymous:  23\n",
      "nonsynonymous_lof:  23\n",
      "\n",
      "\n",
      "WORKING ON: AMR\n",
      "{'AMR'}\n",
      "Total variants:  1987\n",
      "Intronic:  1596\n",
      "UTR3:  56\n",
      "UTR5:  17\n",
      "Total exonic:  178\n",
      "  Synonymous:  89\n",
      "  Nonsynonymous:  89\n",
      "nonsynonymous_lof:  92\n",
      "\n",
      "\n",
      "WORKING ON: EAS\n",
      "{'EAS'}\n",
      "Total variants:  3171\n",
      "Intronic:  2518\n",
      "UTR3:  99\n",
      "UTR5:  43\n",
      "Total exonic:  285\n",
      "  Synonymous:  121\n",
      "  Nonsynonymous:  164\n",
      "nonsynonymous_lof:  174\n",
      "\n",
      "\n",
      "WORKING ON: CAH\n",
      "{'CAH'}\n",
      "Total variants:  3090\n",
      "Intronic:  2467\n",
      "UTR3:  87\n",
      "UTR5:  31\n",
      "Total exonic:  260\n",
      "  Synonymous:  121\n",
      "  Nonsynonymous:  139\n",
      "nonsynonymous_lof:  146\n",
      "\n",
      "\n",
      "WORKING ON: CAS\n",
      "{'CAS'}\n",
      "Total variants:  1955\n",
      "Intronic:  1568\n",
      "UTR3:  62\n",
      "UTR5:  26\n",
      "Total exonic:  167\n",
      "  Synonymous:  78\n",
      "  Nonsynonymous:  89\n",
      "nonsynonymous_lof:  94\n",
      "\n",
      "\n",
      "WORKING ON: SAS\n",
      "{'SAS'}\n",
      "Total variants:  1728\n",
      "Intronic:  1412\n",
      "UTR3:  50\n",
      "UTR5:  24\n",
      "Total exonic:  143\n",
      "  Synonymous:  68\n",
      "  Nonsynonymous:  75\n",
      "nonsynonymous_lof:  79\n",
      "\n",
      "\n",
      "WORKING ON: EUR\n",
      "{'EUR'}\n",
      "Total variants:  9504\n",
      "Intronic:  7451\n",
      "UTR3:  266\n",
      "UTR5:  114\n",
      "Total exonic:  951\n",
      "  Synonymous:  403\n",
      "  Nonsynonymous:  548\n",
      "nonsynonymous_lof:  571\n",
      "\n",
      "\n",
      "WORKING ON: AJ\n",
      "{'AJ'}\n",
      "Total variants:  1285\n",
      "Intronic:  1024\n",
      "UTR3:  43\n",
      "UTR5:  6\n",
      "Total exonic:  125\n",
      "  Synonymous:  59\n",
      "  Nonsynonymous:  66\n",
      "nonsynonymous_lof:  69\n",
      "\n",
      "\n",
      "WORKING ON: MDE\n",
      "{'MDE'}\n",
      "Total variants:  1945\n",
      "Intronic:  1547\n",
      "UTR3:  59\n",
      "UTR5:  22\n",
      "Total exonic:  177\n",
      "  Synonymous:  93\n",
      "  Nonsynonymous:  84\n",
      "nonsynonymous_lof:  86\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Make lists of variants to keep - all coding, coding nonsynonymous (missense - as they are coded in ANNOVAR), deleterious (CADD_phred > 20)\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    # Read in ANNOVAR multianno file\n",
    "    gene = pd.read_csv(f'{WORK_DIR}/{ancestry}_TMEM175.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "    \n",
    "    # Print number of variants in the different categories\n",
    "    results = [] \n",
    "    \n",
    "    utr5 = gene[gene['Func.refGene']== 'UTR5']\n",
    "    intronic = gene[gene['Func.refGene']== 'intronic']\n",
    "    exonic = gene[gene['Func.refGene']== 'exonic']\n",
    "    utr3 = gene[gene['Func.refGene']== 'UTR3']\n",
    "    coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "    coding_synonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] != 'nonsynonymous SNV')]\n",
    "    lof = exonic[(exonic['ExonicFunc.refGene'] == 'stopgain') | (exonic['ExonicFunc.refGene'] == 'stoploss') | (exonic['ExonicFunc.refGene'] == 'frameshift deletion') | (exonic['ExonicFunc.refGene'] == 'frameshift insertion')]\n",
    "    nonsynonymous_lof = pd.concat([coding_nonsynonymous, lof])\n",
    "    \n",
    "    print({ancestry})\n",
    "    print('Total variants: ', len(gene))\n",
    "    print(\"Intronic: \", len(intronic))\n",
    "    print('UTR3: ', len(utr3))\n",
    "    print('UTR5: ', len(utr5))\n",
    "    print(\"Total exonic: \", len(exonic))\n",
    "    print('  Synonymous: ', len(coding_synonymous))\n",
    "    print(\"  Nonsynonymous: \", len(coding_nonsynonymous))\n",
    "    print(\"nonsynonymous_lof: \", len(nonsynonymous_lof))\n",
    "    results.append((gene, intronic, utr3, utr5, exonic, coding_synonymous, coding_nonsynonymous, nonsynonymous_lof))\n",
    "    print('\\n')\n",
    "    \n",
    "    # Save in PLINK format - coding nonsynonymous \n",
    "    # These are missense variants - other types of nonsynonymous variants (e.g stopgain/loss, or frameshift variants are coded differently in the ExonicFunc.refGene \n",
    "    variants_toKeep = nonsynonymous_lof[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "    variants_toKeep.to_csv(f'{WORK_DIR}/{ancestry}_TMEM175.nonsynonymous_lof.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)\n",
    "    \n",
    "    # Save in PLINK format - all coding variants\n",
    "    variants_toKeep2 = exonic[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "    variants_toKeep2.to_csv(f'{WORK_DIR}/{ancestry}_TMEM175.exonic.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a117d5ea-00f5-49aa-9199-b775529c28d9",
   "metadata": {},
   "source": [
    "## 5. Association analysis to compare allele frequencies between cases and controls\n",
    "* ALL variants\n",
    "\n",
    "* assoc\n",
    "\n",
    "* glm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7af229a1-2872-4b47-8957-8dff7d966fab",
   "metadata": {},
   "source": [
    "### ASSOC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14c14ecc-1aa1-47d4-bd32-c2d2592db62e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "3449 variants loaded from .bim file.\n",
      "1111 people (455 males, 656 females) loaded from .fam.\n",
      "1086 phenotype values loaded from .fam.\n",
      "--keep: 1111 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1111 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998498.\n",
      "2861 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "588 variants and 1111 people pass filters and QC.\n",
      "Among remaining phenotypes, 285 are cases and 801 are controls.  (25 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "3853 variants loaded from .bim file.\n",
      "2643 people (1480 males, 1163 females) loaded from .fam.\n",
      "2621 phenotype values loaded from .fam.\n",
      "--keep: 2643 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2643 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996619.\n",
      "3262 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "591 variants and 2643 people pass filters and QC.\n",
      "Among remaining phenotypes, 942 are cases and 1679 are controls.  (22\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "1285 variants loaded from .bim file.\n",
      "2655 people (1648 males, 1007 females) loaded from .fam.\n",
      "1703 phenotype values loaded from .fam.\n",
      "--keep: 2655 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2655 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998935.\n",
      "869 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "416 variants and 2655 people pass filters and QC.\n",
      "Among remaining phenotypes, 1292 are cases and 411 are controls.  (952\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "1987 variants loaded from .bim file.\n",
      "646 people (363 males, 283 females) loaded from .fam.\n",
      "613 phenotype values loaded from .fam.\n",
      "--keep: 646 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 646 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997492.\n",
      "1582 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "405 variants and 646 people pass filters and QC.\n",
      "Among remaining phenotypes, 458 are cases and 155 are controls.  (33 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "1955 variants loaded from .bim file.\n",
      "903 people (439 males, 464 females) loaded from .fam.\n",
      "895 phenotype values loaded from .fam.\n",
      "--keep: 903 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 903 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.995381.\n",
      "1526 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "429 variants and 903 people pass filters and QC.\n",
      "Among remaining phenotypes, 552 are cases and 343 are controls.  (8 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "3171 variants loaded from .bim file.\n",
      "5167 people (3278 males, 1889 females) loaded from .fam.\n",
      "5123 phenotype values loaded from .fam.\n",
      "--keep: 5167 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5167 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.994312.\n",
      "2794 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "377 variants and 5167 people pass filters and QC.\n",
      "Among remaining phenotypes, 2662 are cases and 2461 are controls.  (44\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "9504 variants loaded from .bim file.\n",
      "38839 people (23296 males, 15543 females) loaded from .fam.\n",
      "30412 phenotype values loaded from .fam.\n",
      "--keep: 38838 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 38838 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.760113.\n",
      "9107 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "397 variants and 38838 people pass filters and QC.\n",
      "Among remaining phenotypes, 21197 are cases and 9214 are controls.  (8427\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "605 variants loaded from .bim file.\n",
      "114 people (53 males, 61 females) loaded from .fam.\n",
      "106 phenotype values loaded from .fam.\n",
      "--keep: 114 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 114 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.991605.\n",
      "200 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "405 variants and 114 people pass filters and QC.\n",
      "Among remaining phenotypes, 98 are cases and 8 are controls.  (8 phenotypes are\n",
      "missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "1945 variants loaded from .bim file.\n",
      "581 people (369 males, 212 females) loaded from .fam.\n",
      "536 phenotype values loaded from .fam.\n",
      "--keep: 581 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 581 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.993732.\n",
      "1547 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "398 variants and 581 people pass filters and QC.\n",
      "Among remaining phenotypes, 311 are cases and 225 are controls.  (45 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "1728 variants loaded from .bim file.\n",
      "635 people (408 males, 227 females) loaded from .fam.\n",
      "609 phenotype values loaded from .fam.\n",
      "--keep: 635 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 635 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.994328.\n",
      "1251 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "477 variants and 635 people pass filters and QC.\n",
      "Among remaining phenotypes, 387 are cases and 222 are controls.  (26 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "3090 variants loaded from .bim file.\n",
      "851 people (446 males, 405 females) loaded from .fam.\n",
      "827 phenotype values loaded from .fam.\n",
      "--keep: 851 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 851 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997283.\n",
      "2542 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "548 variants and 851 people pass filters and QC.\n",
      "Among remaining phenotypes, 525 are cases and 302 are controls.  (24 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all.assoc ...\n",
      "done.\n"
     ]
    }
   ],
   "source": [
    "# Run case-control analysis using plink assoc for all variants, not adjusting for any covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    \n",
    "    ! /home/jupyter/tools/plink \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_TMEM175 \\\n",
    "    --keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "    --assoc \\\n",
    "    --allow-no-sex \\\n",
    "    --ci 0.95 \\\n",
    "    --maf 0.01 \\\n",
    "    --out {WORK_DIR}/{ancestry}_TMEM175.all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4092d28d-e7ad-495d-9cc3-cf53afa17488",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all.test.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.samplestoKeep\n",
      "  --max-maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all.test\n",
      "\n",
      "12984 MB RAM detected; reserving 6492 MB for main workspace.\n",
      "3853 variants loaded from .bim file.\n",
      "2643 people (1480 males, 1163 females) loaded from .fam.\n",
      "2621 phenotype values loaded from .fam.\n",
      "--keep: 2643 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2643 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996619.\n",
      "591 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "3262 variants and 2643 people pass filters and QC.\n",
      "Among remaining phenotypes, 942 are cases and 1679 are controls.  (22\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all.test.assoc\n",
      "... 316294done.\n"
     ]
    }
   ],
   "source": [
    "# Test if we can see TMEM175 p.M393T when set maf<0.01 in AFR\n",
    "WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_AFR'\n",
    "\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/AFR_TMEM175 \\\n",
    "--keep {WORK_DIR}/AFR.samplestoKeep \\\n",
    "--allow-no-sex \\\n",
    "--max-maf 0.01 \\\n",
    "--ci 0.95 \\\n",
    "--assoc \\\n",
    "--out {WORK_DIR}/AFR_TMEM175.all.test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ce9fa5a0-790b-48f0-b4c5-bf3693c9054f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_AFR'\n",
    "freq = pd.read_csv(f'{WORK_DIR}/AFR_TMEM175.all.test.assoc', sep='\\s+')\n",
    "freq.to_csv(f'{WORK_DIR}/AFR.all_nonadj.test.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "02ec7380-f232-46a4-bb9a-78600c917316",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AAC\n",
      "Variants with p-value < 0.05: (26, 13)\n",
      "WORKING ON: AFR\n",
      "Variants with p-value < 0.05: (90, 13)\n",
      "WORKING ON: AJ\n",
      "Variants with p-value < 0.05: (115, 13)\n",
      "WORKING ON: AMR\n",
      "Variants with p-value < 0.05: (7, 13)\n",
      "WORKING ON: CAS\n",
      "Variants with p-value < 0.05: (38, 13)\n",
      "WORKING ON: EAS\n",
      "Variants with p-value < 0.05: (200, 13)\n",
      "WORKING ON: EUR\n",
      "Variants with p-value < 0.05: (202, 13)\n",
      "WORKING ON: FIN\n",
      "Variants with p-value < 0.05: (9, 13)\n",
      "WORKING ON: MDE\n",
      "Variants with p-value < 0.05: (84, 13)\n",
      "WORKING ON: SAS\n",
      "Variants with p-value < 0.05: (24, 13)\n",
      "WORKING ON: CAH\n",
      "Variants with p-value < 0.05: (54, 13)\n"
     ]
    }
   ],
   "source": [
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    # L ook at assoc results, only variants with nominal p-value < 0.05\n",
    "    freq = pd.read_csv(f'{WORK_DIR}/{ancestry}_TMEM175.all.assoc', sep='\\s+')\n",
    "    sig_all_nonadj = freq[freq['P']<0.05]\n",
    "    \n",
    "    print(f'Variants with p-value < 0.05: {sig_all_nonadj.shape}')\n",
    "    \n",
    "    # Save FREQ to csv\n",
    "    freq.to_csv(f'{WORK_DIR}/{ancestry}.all_nonadj.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f539184",
   "metadata": {},
   "source": [
    "## 6. GLM analysis adjusting for gender, age, PC1-5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fb196a2e-2c9c-4154-8732-036513c49883",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all_adj\n",
      "\n",
      "Start time: Wed Jul 31 11:04:25 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50540 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1111 samples (656 females, 455 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam.\n",
      "3449 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim.\n",
      "1 binary phenotype loaded (285 cases, 801 controls).\n",
      "--keep: 1111 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_covariate_file.txt.\n",
      "1111 samples (656 females, 455 males; 1111 founders) remaining after main\n",
      "filters.\n",
      "285 cases and 801 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2861 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "588 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Wed Jul 31 11:04:25 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj\n",
      "\n",
      "Start time: Wed Jul 31 11:04:26 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50537 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam.\n",
      "3853 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim.\n",
      "1 binary phenotype loaded (942 cases, 1679 controls).\n",
      "--keep: 2643 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) remaining after main\n",
      "filters.\n",
      "942 cases and 1679 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "3262 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "591 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Wed Jul 31 11:04:26 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all_adj\n",
      "\n",
      "Start time: Wed Jul 31 11:04:26 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50516 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam.\n",
      "1285 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim.\n",
      "1 binary phenotype loaded (1292 cases, 411 controls).\n",
      "--keep: 2655 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_covariate_file.txt.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) remaining after main\n",
      "filters.\n",
      "1292 cases and 411 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "869 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "416 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Wed Jul 31 11:04:26 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all_adj\n",
      "\n",
      "Start time: Wed Jul 31 11:04:27 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50517 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "646 samples (283 females, 363 males; 646 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam.\n",
      "1987 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim.\n",
      "1 binary phenotype loaded (458 cases, 155 controls).\n",
      "--keep: 646 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_covariate_file.txt.\n",
      "646 samples (283 females, 363 males; 646 founders) remaining after main\n",
      "filters.\n",
      "458 cases and 155 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1582 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "405 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Wed Jul 31 11:04:27 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all_adj\n",
      "\n",
      "Start time: Wed Jul 31 11:04:27 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50516 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "903 samples (464 females, 439 males; 903 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam.\n",
      "1955 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim.\n",
      "1 binary phenotype loaded (552 cases, 343 controls).\n",
      "--keep: 903 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_covariate_file.txt.\n",
      "903 samples (464 females, 439 males; 903 founders) remaining after main\n",
      "filters.\n",
      "552 cases and 343 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1526 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "429 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Wed Jul 31 11:04:27 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj\n",
      "\n",
      "Start time: Wed Jul 31 11:04:28 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50529 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam.\n",
      "3171 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim.\n",
      "1 binary phenotype loaded (2662 cases, 2461 controls).\n",
      "--keep: 5167 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) remaining after main\n",
      "filters.\n",
      "2662 cases and 2461 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2794 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "377 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Wed Jul 31 11:04:28 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all_adj\n",
      "\n",
      "Start time: Wed Jul 31 11:04:28 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50516 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam.\n",
      "9504 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "--keep: 38838 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt.\n",
      "38838 samples (15543 females, 23295 males; 38838 founders) remaining after main\n",
      "filters.\n",
      "21197 cases and 9214 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "9107 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "397 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Wed Jul 31 11:04:29 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all_adj\n",
      "\n",
      "Start time: Wed Jul 31 11:04:29 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50517 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "114 samples (61 females, 53 males; 114 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam.\n",
      "605 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim.\n",
      "1 binary phenotype loaded (98 cases, 8 controls).\n",
      "--keep: 114 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_covariate_file.txt.\n",
      "114 samples (61 females, 53 males; 114 founders) remaining after main filters.\n",
      "98 cases and 8 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "200 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "405 variants remaining after main filters.\n",
      "Warning: --glm remaining control count is less than 10x predictor count for\n",
      "phenotype 'PHENO1'.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Wed Jul 31 11:04:30 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all_adj\n",
      "\n",
      "Start time: Wed Jul 31 11:04:30 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50517 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "581 samples (212 females, 369 males; 581 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam.\n",
      "1945 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim.\n",
      "1 binary phenotype loaded (311 cases, 225 controls).\n",
      "--keep: 581 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_covariate_file.txt.\n",
      "581 samples (212 females, 369 males; 581 founders) remaining after main\n",
      "filters.\n",
      "311 cases and 225 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1547 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "398 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Wed Jul 31 11:04:30 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all_adj\n",
      "\n",
      "Start time: Wed Jul 31 11:04:30 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50517 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "635 samples (227 females, 408 males; 635 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam.\n",
      "1728 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim.\n",
      "1 binary phenotype loaded (387 cases, 222 controls).\n",
      "--keep: 635 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_covariate_file.txt.\n",
      "635 samples (227 females, 408 males; 635 founders) remaining after main\n",
      "filters.\n",
      "387 cases and 222 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1251 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "477 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Wed Jul 31 11:04:31 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all_adj\n",
      "\n",
      "Start time: Wed Jul 31 11:04:31 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50518 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "851 samples (405 females, 446 males; 851 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam.\n",
      "3090 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim.\n",
      "1 binary phenotype loaded (525 cases, 302 controls).\n",
      "--keep: 851 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_covariate_file.txt.\n",
      "851 samples (405 females, 446 males; 851 founders) remaining after main\n",
      "filters.\n",
      "525 cases and 302 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2542 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "548 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Wed Jul 31 11:04:31 2024\n"
     ]
    }
   ],
   "source": [
    "# Run case-control analysis with covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_TMEM175 \\\n",
    "    --keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "    --allow-no-sex \\\n",
    "    --maf 0.01 \\\n",
    "    --ci 0.95 \\\n",
    "    --glm \\\n",
    "    --covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "    --covar-variance-standardize \\\n",
    "    --neg9-pheno-really-missing \\\n",
    "    --out {WORK_DIR}/{ancestry}_TMEM175.all_adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "957dbb8c-a86b-49b7-943d-d239e70bd49d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj.test.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.samplestoKeep\n",
      "  --max-maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj.test\n",
      "\n",
      "Start time: Wed Mar  5 10:28:20 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "12984 MiB RAM detected, ~11798 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam.\n",
      "3853 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim.\n",
      "1 binary phenotype loaded (942 cases, 1679 controls).\n",
      "--keep: 2643 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) remaining after main\n",
      "filters.\n",
      "942 cases and 1679 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "591 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "3262 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj.test.PHENO1.glm.logistic.hybrid .\n",
      "End time: Wed Mar  5 10:28:23 2025\n"
     ]
    }
   ],
   "source": [
    "# test if we can see TMEM175 p.M393T when set maf<0.01 in AFR\n",
    "WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_AFR'\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/AFR_TMEM175 \\\n",
    "--keep {WORK_DIR}/AFR.samplestoKeep \\\n",
    "--allow-no-sex \\\n",
    "--max-maf 0.01 \\\n",
    "--ci 0.95 \\\n",
    "--glm \\\n",
    "--covar {WORK_DIR}/AFR_covariate_file.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--covar-variance-standardize \\\n",
    "--neg9-pheno-really-missing \\\n",
    "--out {WORK_DIR}/AFR_TMEM175.all_adj.test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e0b25479-357c-45dc-9680-ba7cbb96cc7b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_2920/502712416.py:3: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    }
   ],
   "source": [
    "# Read in plink glm results\n",
    "WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_AFR'\n",
    "assoc = pd.read_csv(f'{WORK_DIR}/AFR_TMEM175.all_adj.test.PHENO1.glm.logistic.hybrid', delim_whitespace=True)\n",
    "assoc_add = assoc[assoc['TEST']==\"ADD\"]\n",
    "assoc_add.to_csv(f'{WORK_DIR}/AFR.all_adj.test.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e4e16bfa-a27e-4e6d-a941-31724185d0a5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AAC\n",
      "There are 24 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AFR\n",
      "There are 46 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AJ\n",
      "There are 121 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: AMR\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 7 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: CAS\n",
      "There are 26 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: EAS\n",
      "There are 44 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: EUR\n",
      "There are 148 variants with p-value < 0.05 in glm\n",
      "There are 40 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: FIN\n",
      "There are 15 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: MDE\n",
      "There are 124 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: SAS\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 42 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: CAH\n",
      "There are 27 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    }
   ],
   "source": [
    "# Process results from plink glm analysis including covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    # Read in plink glm results\n",
    "    assoc = pd.read_csv(f'{WORK_DIR}/{ancestry}_TMEM175.all_adj.PHENO1.glm.logistic.hybrid', delim_whitespace=True)\n",
    "    \n",
    "    # Filter for additive test only - this is the variant results\n",
    "    assoc_add = assoc[assoc['TEST']==\"ADD\"]\n",
    "    \n",
    "    # Check if there are any significant (p < 0.05) variants\n",
    "    significant = assoc_add[assoc_add['P']<0.05]\n",
    "\n",
    "    print(f'There are {len(significant)} variants with p-value < 0.05 in glm')\n",
    "    \n",
    "    # Check if there are any significant (p < 0.05) variants\n",
    "    GWsignificant = assoc_add[assoc_add['P']<5e-8]\n",
    "\n",
    "    print(f'There are {len(GWsignificant)} variants with p-value < 5e-8 in glm')\n",
    "    \n",
    "    # Save assoc_add to csv\n",
    "    assoc_add.to_csv(f'{WORK_DIR}/{ancestry}.all_adj.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d96aca21-208e-4d4b-a1a1-7e7510886504",
   "metadata": {},
   "source": [
    "## 7. Burden test (SkatO, Skat, cmc, zeggini, mb, fp, cmcWald)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "acb6ba3c-8ab5-4cd8-8599-74853bb7ede2",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "variant_classes = ['exonic', 'nonsynonymous_lof']\n",
    "\n",
    "# Loop over all the ancestries and the 2 variant classes - run rvtests for all coding and missense variants\n",
    "for ancestry in ancestries:\n",
    "    for variant_class in variant_classes:\n",
    "        \n",
    "        WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "        \n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running plink to extract {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        # Extract relevant variants\n",
    "        ! /home/jupyter/tools/plink2 \\\n",
    "        --pfile {REL7_PATH}/imputed_genotypes/{ancestry}/chr4_{ancestry}_release7_vwb \\\n",
    "        --keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "        --extract range {WORK_DIR}/{ancestry}_TMEM175.{variant_class}.variantstoKeep.txt \\\n",
    "        --recode vcf-iid \\\n",
    "        --out {WORK_DIR}/{ancestry}_TMEM175.{variant_class}\n",
    "        \n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running bgzip and tabix for {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        # Bgzip and Tabix (zip and index the file)\n",
    "        ! bgzip -f {WORK_DIR}/{ancestry}_TMEM175.{variant_class}.vcf\n",
    "        ! tabix -f -p vcf {WORK_DIR}/{ancestry}_TMEM175.{variant_class}.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1df08dc0-1dc4-4d35-af7f-4230011700bc",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Run RVtests\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "variant_classes = ['exonic', 'nonsynonymous_lof']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    for variant_class in variant_classes:\n",
    "        \n",
    "        WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "        \n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running RVtests for {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        # RVtests with covariates \n",
    "        # Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "        # The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "        ! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "        --out {WORK_DIR}/{ancestry}_TMEM175.burden.{variant_class} \\\n",
    "        --kernel skato \\\n",
    "        --inVcf {WORK_DIR}/{ancestry}_TMEM175.{variant_class}.vcf.gz \\\n",
    "        --pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "        --pheno-name PHENO \\\n",
    "        --gene TMEM175 \\\n",
    "        --geneFile ~/workspace/ws_files/TMEM175/refFlat.txt \\\n",
    "        --covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "        --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "        --freqUpper 0.01\n",
    "# --burden cmc,zeggini,mb,fp,cmcWald --kernel skat,skato \\"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9690b20-309e-48b8-99fe-59e28abacb2e",
   "metadata": {},
   "source": [
    "### EUR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "840337d9-30fa-4123-aef4-c003496b088d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/TMEM175\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ce2ab430-4375-40ed-a375-7d9e97d39449",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t174\t109\t323773\t0.9\t0.0304464\n"
     ]
    }
   ],
   "source": [
    "# Check EUR all_coding variant results\n",
    "# Next step - burden test\n",
    "! cat {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a2fac39-db8d-42ee-9137-ec6ff5c630a7",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "# The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--out {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.burdenTEST.exonic \\\n",
    "--burden cmc,zeggini,mb,fp,cmcWald \\\n",
    "--inVcf {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.exonic.vcf.gz \\\n",
    "--pheno {WORK_DIR}/TMEM175_EUR/EUR_covariate_file.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--gene TMEM175 \\\n",
    "--geneFile ~/workspace/ws_files/TMEM175/refFlat.txt \\\n",
    "--covar {WORK_DIR}/TMEM175_EUR/EUR_covariate_file.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--freqUpper 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f39966b1-733f-4ab7-8121-ab54deda9595",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tNonRefSite\tBeta\tSE\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t174\t109\t1500\t0.0708554\t0.0620621\t0.253585\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t174\t109\t1500\t-0.576489\t0.0316899\t6.02168e-74\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t174\t109\t1500\t0.0309305\t0.00133732\t2.37789e-118\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t174\t109\t1500\t-0.0224515\t0.0086344\t0.00931601\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t174\t109\t1500\t0.0140758\t0.00750917\t0.0608638\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t174\t109\t1500\t-0.0182897\t0.00785175\t0.019839\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t174\t109\t1500\t0.0119784\t0.00817279\t0.142746\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t174\t109\t1500\t-0.0137788\t0.00332666\t3.44415e-05\n"
     ]
    }
   ],
   "source": [
    "# See bureden test in exonic in EUR\n",
    "! cat {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.burdenTEST.exonic.CMCWald.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "09f35168-655d-4578-9781-31392635b4c9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tNonRefSite\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t174\t109\t1500\t0.253509\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.burdenTEST.exonic.CMC.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c04bab3a-965e-4a40-a628-a6a95bfce513",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t174\t109\t0.0365734\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.burdenTEST.exonic.Fp.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "eae9f761-d531-4751-b4e5-23897b50ac8a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t174\t109\t10000\t6504\t6.10068\t1000\t0\t0.153752\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.burdenTEST.exonic.MadsonBrowning.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "669f74ea-0f4c-406e-93c5-4e59ff95eced",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t174\t109\t0.227047\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.burdenTEST.exonic.Zeggini.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "73b80527-c554-440c-98f8-2912523823a6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t105\t65\t107233\t0.0560423\t10000\t10000\t107233\t537\t0\t0.0537\n"
     ]
    }
   ],
   "source": [
    "# Check EUR nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb15e4af-aa73-4093-a242-82034b94d596",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t20918\t105\t65\t53616.4\t0\t0.101898\n"
     ]
    }
   ],
   "source": [
    "# Check EUR nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "58037516-2b15-462d-97de-2e55f60e048f",
   "metadata": {},
   "source": [
    "### AAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e418c688-d599-45d1-82d8-2ebf530c3f4d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t925\t57\t45\t5938.41\t0.692352\t10000\t1323\t5938.41\t1000\t0\t0.755858\n"
     ]
    }
   ],
   "source": [
    "# Check AAC all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AAC/AAC_TMEM175.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bde40041-a3f2-4b81-a2b2-535674408424",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t925\t57\t45\t2977.49\t1\t0.5655\n"
     ]
    }
   ],
   "source": [
    "# Check AAC all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AAC/AAC_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94df761e-db76-4dff-907a-8eb1ed55d4d9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t925\t38\t31\t3451.67\t0.742505\t10000\t1275\t3451.67\t1000\t0\t0.784314\n"
     ]
    }
   ],
   "source": [
    "# Check AAC all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AAC/AAC_TMEM175.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8907bd9e-60b8-4007-9c88-48cb85fe7ea8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t925\t38\t31\t1792.92\t1\t0.607447\n"
     ]
    }
   ],
   "source": [
    "# Check AAC all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AAC/AAC_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a308e74f-7bd9-46e5-b0a8-b624954d2f3c",
   "metadata": {},
   "source": [
    "### AFR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "695a19a1-ab3f-400e-a349-9615998a5272",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t1062\t60\t43\t7846.37\t0.311869\t10000\t2587\t7846.37\t1000\t0\t0.386548\n"
     ]
    }
   ],
   "source": [
    "# Check AFR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AFR/AFR_TMEM175.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be66e3ac-1bec-4246-a2d7-257b772cbb8b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t1062\t60\t43\t3923.18\t0\t0.495811\n"
     ]
    }
   ],
   "source": [
    "# Check AFR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AFR/AFR_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "77c06bcc-9d2d-403c-988c-0c4fe3be46c3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t1062\t38\t27\t6145.09\t0.371636\t10000\t2581\t6145.09\t1000\t0\t0.387447\n"
     ]
    }
   ],
   "source": [
    "# Check AFR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AFR/AFR_TMEM175.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a5d0d75b-0de2-4692-81b3-617b53700861",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t1062\t38\t27\t3072.54\t0\t0.56138\n"
     ]
    }
   ],
   "source": [
    "# Check AFR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AFR/AFR_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "17bd6c61-04e4-4583-8c65-edef1c0c9e3e",
   "metadata": {},
   "source": [
    "### AJ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5156f450-8eae-43dc-839c-1fb9a10ceaf1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t1410\t20\t15\t2660.25\t0.548165\t10000\t1693\t2660.25\t1000\t0\t0.590667\n"
     ]
    }
   ],
   "source": [
    "# Check AJ all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AJ/AJ_TMEM175.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa3b6f44-4458-420d-a005-e1675d1d261d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t1410\t20\t15\t594.277\t1\t0.740559\n"
     ]
    }
   ],
   "source": [
    "# Check AJ all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AJ/AJ_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "df29028e-5840-4153-b257-195b5c98e2af",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t1410\t13\t10\t2189.49\t0.52697\t10000\t1745\t2189.49\t1000\t0\t0.573066\n"
     ]
    }
   ],
   "source": [
    "# Check AJ all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AJ/AJ_TMEM175.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4529c89f-c2c7-4852-be83-b5d5d049ef7e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t1410\t13\t10\t1094.74\t0\t0.708875\n"
     ]
    }
   ],
   "source": [
    "# Check AJ all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AJ/AJ_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3c707c32-e6bb-4b34-a5d3-88efc7549532",
   "metadata": {},
   "source": [
    "### AMR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "10133437-0ec7-4259-a009-99ca7c3c9e93",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t447\t36\t26\t4759.54\t0.274054\t10000\t4048\t4759.54\t1000\t0\t0.247036\n"
     ]
    }
   ],
   "source": [
    "# Check AMR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AMR/AMR_TMEM175.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "35000fab-b264-4606-8036-a58096047ccd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t447\t36\t26\t2379.77\t0\t0.433785\n"
     ]
    }
   ],
   "source": [
    "# Check AMR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AMR/AMR_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "875e300f-827c-407d-bac5-7e0bf851142b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t447\t18\t14\t3656.15\t0.188299\t10000\t6542\t3656.15\t1000\t0\t0.152858\n"
     ]
    }
   ],
   "source": [
    "# Check AMR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AMR/AMR_TMEM175.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "11e5bf7f-4893-4523-8597-17958132073d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t447\t18\t14\t1828.07\t0\t0.311882\n"
     ]
    }
   ],
   "source": [
    "# Check AMR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AMR/AMR_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92ac6585-9a94-48d5-9abd-583955d33981",
   "metadata": {},
   "source": [
    "### CAH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a1a97589-691f-4b68-ad26-ab250818a07e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t693\t43\t34\t4011.03\t0.742795\t10000\t1290\t4011.03\t1000\t0\t0.775194\n"
     ]
    }
   ],
   "source": [
    "# Check CAH all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAH/CAH_TMEM175.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e66124f6-8a3d-4266-be86-74f1aa587803",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t693\t43\t34\t14226.6\t1\t0.0312151\n"
     ]
    }
   ],
   "source": [
    "# Check CAH all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAH/CAH_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0427bf5a-edca-4bfc-8e71-5e095cc02db7",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t693\t26\t22\t3545.62\t0.531319\t10000\t1628\t3545.62\t1000\t0\t0.614251\n"
     ]
    }
   ],
   "source": [
    "# Check CAH all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAH/CAH_TMEM175.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9921df41-6a24-46dd-b22b-08852e1fc73e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t693\t26\t22\t10667.7\t1\t0.0308928\n"
     ]
    }
   ],
   "source": [
    "# Check CAH all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAH/CAH_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "79e545a3-e1ca-403c-aea5-cde44c7adf5d",
   "metadata": {},
   "source": [
    "### CAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f47c3562-960b-4ecd-89b2-80c452c27d5e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t658\t30\t19\t589.951\t0.984523\t10000\t1000\t589.951\t1000\t0\t1\n"
     ]
    }
   ],
   "source": [
    "# Check CAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAS/CAS_TMEM175.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "82279004-c38b-4dfc-b741-573b038eb4b3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t658\t30\t19\t447.27\t1\t0.767453\n"
     ]
    }
   ],
   "source": [
    "# Check CAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAS/CAS_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f05b96cd-2f32-4f05-838e-e91e111b6d7e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t658\t22\t11\t445.823\t0.237621\t10000\t4284\t445.823\t1000\t0\t0.233427\n"
     ]
    }
   ],
   "source": [
    "# Check CAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAS/CAS_TMEM175.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d3b936c8-fcaf-43db-b556-6ca05dc0b40f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t658\t22\t11\t222.912\t0\t0.351341\n"
     ]
    }
   ],
   "source": [
    "# Check CAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAS/CAS_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46739ace-d30a-45cb-8297-69b651306b5e",
   "metadata": {},
   "source": [
    "### EAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dae22fc7-d2fe-479a-a2e4-d95137a4ff72",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t2523\t45\t24\t6681.36\t0.298387\t10000\t3613\t6681.36\t1000\t0\t0.276778\n"
     ]
    }
   ],
   "source": [
    "# Check EAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_EAS/EAS_TMEM175.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f9d7fac-4a28-4889-a62c-f7e8deecca9f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t2523\t45\t24\t3340.68\t0\t0.45024\n"
     ]
    }
   ],
   "source": [
    "# Check EAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_EAS/EAS_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "563ae9e5-893b-47ac-a330-7391fbc43311",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t2523\t31\t15\t7571.83\t0.0863703\t10000\t10000\t7571.83\t648\t0\t0.0648\n"
     ]
    }
   ],
   "source": [
    "# Check EAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_EAS/EAS_TMEM175.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8cb94d61-f323-479a-a044-5798dbb22f62",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t2523\t31\t15\t3785.91\t0\t0.141074\n"
     ]
    }
   ],
   "source": [
    "# Check EAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_EAS/EAS_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7d1309d4-476a-468e-9049-8d309a4572c2",
   "metadata": {},
   "source": [
    "### FIN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ad2bbb1-20ca-4558-ae60-323a070657a1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t40\t2\t0\tNA\tNA\tNA\tNA\tNA\tNA\tNA\tNA\n"
     ]
    }
   ],
   "source": [
    "# Check FIN all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_FIN/FIN_TMEM175.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c122967-946b-40c5-8384-3b44eae1e8e6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t40\t2\t0\tNA\tNA\tNA\n"
     ]
    }
   ],
   "source": [
    "# Check FIN all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_FIN/FIN_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21fad881-5588-4641-8b87-ba18504048a1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t40\t2\t0\tNA\tNA\tNA\tNA\tNA\tNA\tNA\tNA\n"
     ]
    }
   ],
   "source": [
    "# Check FIN all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_FIN/FIN_TMEM175.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f5a4d36-a104-4811-88f1-12d60568e576",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t40\t2\t0\tNA\tNA\tNA\n"
     ]
    }
   ],
   "source": [
    "# Check FIN all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_FIN/FIN_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09845364-a730-4d2f-b11e-d1f1fe20be17",
   "metadata": {},
   "source": [
    "### MDE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "00afe19c-a5b1-4d97-b132-9d4df43d20c4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t379\t30\t13\t234.147\t0.708651\t10000\t1101\t234.147\t1000\t0\t0.908265\n"
     ]
    }
   ],
   "source": [
    "# Check MDE all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_MDE/MDE_TMEM175.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e58ca032-c70b-494b-b745-937682ecafd8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t379\t30\t13\t149.403\t1\t0.681695\n"
     ]
    }
   ],
   "source": [
    "# Check MDE all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_MDE/MDE_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d0f1ad4-b3e5-43dd-a8af-73248852600c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t379\t15\t6\t58.3182\t1\t10000\t1013\t58.3182\t1000\t0\t0.987167\n"
     ]
    }
   ],
   "source": [
    "# Check MDE all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_MDE/MDE_TMEM175.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "64ff3d48-8086-4c26-bfdf-03dd8b7eeac3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t379\t15\t6\t29.0677\t1\t0.839732\n"
     ]
    }
   ],
   "source": [
    "# Check MDE all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_MDE/MDE_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "23649329-4f8c-498b-a89b-68a6de9a8794",
   "metadata": {},
   "source": [
    "### SAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7dcf1de6-60d9-41ba-84bc-3526dec00edd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t296\t17\t10\t1810.68\t0.10187\t10000\t10000\t1810.68\t430\t0\t0.043\n"
     ]
    }
   ],
   "source": [
    "# Check SAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_SAS/SAS_TMEM175.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f175eda8-19b7-48af-b2bf-8e344347e422",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t296\t17\t10\t2890.96\t1\t0.0183843\n"
     ]
    }
   ],
   "source": [
    "# Check SAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_SAS/SAS_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39cd9af1-4adc-479e-8ed9-fa0a00d17d24",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t296\t12\t8\t1859.12\t0.0974177\t10000\t10000\t1859.12\t397\t0\t0.0397\n"
     ]
    }
   ],
   "source": [
    "# Check SAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_SAS/SAS_TMEM175.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f173e038-1b62-4d61-ae3e-2c94be0ce806",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655\t296\t12\t8\t2650.87\t1\t0.0230414\n"
     ]
    }
   ],
   "source": [
    "# Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_SAS/SAS_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bcc9c529-229b-4638-abcc-0916246180c5",
   "metadata": {},
   "source": [
    "### Change format of association file in EUR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "22c1f5b9-f9c8-4dc1-a815-9c3041e7c543",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If you need to read in the sumstats into python - you can skip this is if your data is already in python\n",
    "df = pd.read_csv(f'{WORK_DIR}/TMEM175_EUR/EUR.all_adj.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "12a52854-3f18-4e6c-92b4-f241364c8c54",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>PROVISIONAL_REF?</th>\n",
       "      <th>A1</th>\n",
       "      <th>OMITTED</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>FIRTH?</th>\n",
       "      <th>TEST</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>OR</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>ERRCODE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.164467</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20837</td>\n",
       "      <td>1.034410</td>\n",
       "      <td>0.030398</td>\n",
       "      <td>0.974580</td>\n",
       "      <td>1.097910</td>\n",
       "      <td>1.112890</td>\n",
       "      <td>0.265757</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>882611</td>\n",
       "      <td>chr4:882611:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.051185</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20885</td>\n",
       "      <td>0.931946</td>\n",
       "      <td>0.050102</td>\n",
       "      <td>0.844780</td>\n",
       "      <td>1.028110</td>\n",
       "      <td>-1.406740</td>\n",
       "      <td>0.159504</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>883018</td>\n",
       "      <td>chr4:883018:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.050599</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20880</td>\n",
       "      <td>0.928749</td>\n",
       "      <td>0.050369</td>\n",
       "      <td>0.841441</td>\n",
       "      <td>1.025120</td>\n",
       "      <td>-1.467500</td>\n",
       "      <td>0.142240</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>24</td>\n",
       "      <td>4</td>\n",
       "      <td>883133</td>\n",
       "      <td>chr4:883133:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.012734</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20811</td>\n",
       "      <td>0.809012</td>\n",
       "      <td>0.096557</td>\n",
       "      <td>0.669523</td>\n",
       "      <td>0.977562</td>\n",
       "      <td>-2.194990</td>\n",
       "      <td>0.028164</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>32</td>\n",
       "      <td>4</td>\n",
       "      <td>883483</td>\n",
       "      <td>chr4:883483:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.035357</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20901</td>\n",
       "      <td>0.883241</td>\n",
       "      <td>0.058911</td>\n",
       "      <td>0.786926</td>\n",
       "      <td>0.991344</td>\n",
       "      <td>-2.107530</td>\n",
       "      <td>0.035071</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>3136</td>\n",
       "      <td>4</td>\n",
       "      <td>1007954</td>\n",
       "      <td>chr4:1007954:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.279743</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>18971</td>\n",
       "      <td>0.913461</td>\n",
       "      <td>0.025919</td>\n",
       "      <td>0.868217</td>\n",
       "      <td>0.961063</td>\n",
       "      <td>-3.492280</td>\n",
       "      <td>0.000479</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393</th>\n",
       "      <td>3144</td>\n",
       "      <td>4</td>\n",
       "      <td>1008009</td>\n",
       "      <td>chr4:1008009:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.444961</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>18814</td>\n",
       "      <td>0.949197</td>\n",
       "      <td>0.023666</td>\n",
       "      <td>0.906174</td>\n",
       "      <td>0.994262</td>\n",
       "      <td>-2.203110</td>\n",
       "      <td>0.027587</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394</th>\n",
       "      <td>3152</td>\n",
       "      <td>4</td>\n",
       "      <td>1008209</td>\n",
       "      <td>chr4:1008209:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.036468</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20717</td>\n",
       "      <td>1.002060</td>\n",
       "      <td>0.059756</td>\n",
       "      <td>0.891315</td>\n",
       "      <td>1.126570</td>\n",
       "      <td>0.034499</td>\n",
       "      <td>0.972480</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>3160</td>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.279831</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>18965</td>\n",
       "      <td>0.913397</td>\n",
       "      <td>0.025912</td>\n",
       "      <td>0.868167</td>\n",
       "      <td>0.960983</td>\n",
       "      <td>-3.495880</td>\n",
       "      <td>0.000473</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>3168</td>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>chr4:1008480:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Y</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.053420</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20264</td>\n",
       "      <td>0.903891</td>\n",
       "      <td>0.049512</td>\n",
       "      <td>0.820297</td>\n",
       "      <td>0.996003</td>\n",
       "      <td>-2.040850</td>\n",
       "      <td>0.041266</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>397 rows √ó 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0  #CHROM      POS                ID REF ALT PROVISIONAL_REF?  \\\n",
       "0             0       4   882530   chr4:882530:C:T   C   T                Y   \n",
       "1             8       4   882611   chr4:882611:T:C   T   C                Y   \n",
       "2            16       4   883018   chr4:883018:G:A   G   A                Y   \n",
       "3            24       4   883133   chr4:883133:G:A   G   A                Y   \n",
       "4            32       4   883483   chr4:883483:G:A   G   A                Y   \n",
       "..          ...     ...      ...               ...  ..  ..              ...   \n",
       "392        3136       4  1007954  chr4:1007954:T:C   T   C                Y   \n",
       "393        3144       4  1008009  chr4:1008009:C:T   C   T                Y   \n",
       "394        3152       4  1008209  chr4:1008209:C:T   C   T                Y   \n",
       "395        3160       4  1008337  chr4:1008337:C:T   C   T                Y   \n",
       "396        3168       4  1008480  chr4:1008480:A:G   A   G                Y   \n",
       "\n",
       "    A1 OMITTED   A1_FREQ FIRTH? TEST  OBS_CT        OR  LOG(OR)_SE       L95  \\\n",
       "0    T       C  0.164467      N  ADD   20837  1.034410    0.030398  0.974580   \n",
       "1    C       T  0.051185      N  ADD   20885  0.931946    0.050102  0.844780   \n",
       "2    A       G  0.050599      N  ADD   20880  0.928749    0.050369  0.841441   \n",
       "3    A       G  0.012734      N  ADD   20811  0.809012    0.096557  0.669523   \n",
       "4    A       G  0.035357      N  ADD   20901  0.883241    0.058911  0.786926   \n",
       "..  ..     ...       ...    ...  ...     ...       ...         ...       ...   \n",
       "392  T       C  0.279743      N  ADD   18971  0.913461    0.025919  0.868217   \n",
       "393  C       T  0.444961      N  ADD   18814  0.949197    0.023666  0.906174   \n",
       "394  T       C  0.036468      N  ADD   20717  1.002060    0.059756  0.891315   \n",
       "395  C       T  0.279831      N  ADD   18965  0.913397    0.025912  0.868167   \n",
       "396  G       A  0.053420      N  ADD   20264  0.903891    0.049512  0.820297   \n",
       "\n",
       "          U95    Z_STAT         P ERRCODE  \n",
       "0    1.097910  1.112890  0.265757       .  \n",
       "1    1.028110 -1.406740  0.159504       .  \n",
       "2    1.025120 -1.467500  0.142240       .  \n",
       "3    0.977562 -2.194990  0.028164       .  \n",
       "4    0.991344 -2.107530  0.035071       .  \n",
       "..        ...       ...       ...     ...  \n",
       "392  0.961063 -3.492280  0.000479       .  \n",
       "393  0.994262 -2.203110  0.027587       .  \n",
       "394  1.126570  0.034499  0.972480       .  \n",
       "395  0.960983 -3.495880  0.000473       .  \n",
       "396  0.996003 -2.040850  0.041266       .  \n",
       "\n",
       "[397 rows x 20 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90b7a98c-2f17-48e5-9d12-acaec4a6c8ca",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# First just select relevant columns: chromosome, bp position and p-value\n",
    "export_ldassoc = df[['# CHROM', 'POS', 'P']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6ebfb5a7-78d0-4526-8165-77d1ecb82f82",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>0.265757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882611</td>\n",
       "      <td>0.159504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>883018</td>\n",
       "      <td>0.142240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>883133</td>\n",
       "      <td>0.028164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>883483</td>\n",
       "      <td>0.035071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>4</td>\n",
       "      <td>1007954</td>\n",
       "      <td>0.000479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393</th>\n",
       "      <td>4</td>\n",
       "      <td>1008009</td>\n",
       "      <td>0.027587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394</th>\n",
       "      <td>4</td>\n",
       "      <td>1008209</td>\n",
       "      <td>0.972480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>0.000473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>0.041266</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>397 rows √ó 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #CHROM      POS         P\n",
       "0         4   882530  0.265757\n",
       "1         4   882611  0.159504\n",
       "2         4   883018  0.142240\n",
       "3         4   883133  0.028164\n",
       "4         4   883483  0.035071\n",
       "..      ...      ...       ...\n",
       "392       4  1007954  0.000479\n",
       "393       4  1008009  0.027587\n",
       "394       4  1008209  0.972480\n",
       "395       4  1008337  0.000473\n",
       "396       4  1008480  0.041266\n",
       "\n",
       "[397 rows x 3 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "export_ldassoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "73408b68-2c18-40a2-982c-2ae3b8ece7e1",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Rename the # CHROM column to remove the hashtag as I think this might be confusing LDassoc\n",
    "export_ldassoc = export_ldassoc.rename(columns={'# CHROM': 'CHROM'}) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "11fef93b-f702-4da5-b66d-b81e5d0448e3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>0.265757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882611</td>\n",
       "      <td>0.159504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>883018</td>\n",
       "      <td>0.142240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>883133</td>\n",
       "      <td>0.028164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>883483</td>\n",
       "      <td>0.035071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>4</td>\n",
       "      <td>1007954</td>\n",
       "      <td>0.000479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393</th>\n",
       "      <td>4</td>\n",
       "      <td>1008009</td>\n",
       "      <td>0.027587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394</th>\n",
       "      <td>4</td>\n",
       "      <td>1008209</td>\n",
       "      <td>0.972480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>0.000473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>0.041266</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>397 rows √ó 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM      POS         P\n",
       "0        4   882530  0.265757\n",
       "1        4   882611  0.159504\n",
       "2        4   883018  0.142240\n",
       "3        4   883133  0.028164\n",
       "4        4   883483  0.035071\n",
       "..     ...      ...       ...\n",
       "392      4  1007954  0.000479\n",
       "393      4  1008009  0.027587\n",
       "394      4  1008209  0.972480\n",
       "395      4  1008337  0.000473\n",
       "396      4  1008480  0.041266\n",
       "\n",
       "[397 rows x 3 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "export_ldassoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c548f5fd-9b1a-47eb-b429-c7400cf02a10",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Then export as a tab-separated, not comma-separated file\n",
    "\n",
    "export_ldassoc.to_csv('EUR.all_adj.formatted.tab', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a60afaa-2363-4ac7-9a7a-5e01868a4127",
   "metadata": {},
   "source": [
    "## 8. Conditional analysis\n",
    "* Pearson and conditional analysis with GLM (rs6599388)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c136956e-9463-4ddd-94b7-5f7cadf5accc",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "64d8b199-e991-407b-adfe-7f042bde8ad1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs6599388.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/AJ_TMEM175\n",
      "  --chr 4\n",
      "  --double-id\n",
      "  --from-bp 945299\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs6599388\n",
      "  --recode A\n",
      "  --to-bp 945299\n",
      "\n",
      "12984 MB RAM detected; reserving 6492 MB for main workspace.\n",
      "1 out of 1285 variants loaded from .bim file.\n",
      "2655 people (1648 males, 1007 females) loaded from .fam.\n",
      "1703 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2655 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999623.\n",
      "1 variant and 2655 people pass filters and QC.\n",
      "Among remaining phenotypes, 1292 are cases and 411 are controls.  (952\n",
      "phenotypes are missing.)\n",
      "--recode A to\n",
      "/home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs6599388.raw ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/TMEM175/\"\n",
    "\n",
    "command = f\"\"\"\n",
    "/home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/TMEM175_AJ/AJ_TMEM175 \\\n",
    "--chr 4 \\\n",
    "--from-bp 945299 \\\n",
    "--to-bp 945299 \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/TMEM175_AJ/TMEM175_rs6599388 \\\n",
    "--double-id\n",
    "\"\"\"\n",
    "\n",
    "!{command}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1f4cd3e-3034-4ec8-8976-5a0c960400cd",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "rs6599388 = pd.read_csv(f'{WORK_DIR}/TMEM175_AJ/TMEM175_rs6599388.raw',sep=\" \")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e007e71b-9bad-4d37-9e27-a6c8b18c0684",
   "metadata": {},
   "source": [
    "### Pearson and conditional analysis with GLM (rs74391911)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3795916b-97ff-4efe-a365-f322d309c84f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs74391911.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/AJ_TMEM175\n",
      "  --chr 4\n",
      "  --double-id\n",
      "  --from-bp 956095\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs74391911\n",
      "  --recode A\n",
      "  --to-bp 956095\n",
      "\n",
      "12984 MB RAM detected; reserving 6492 MB for main workspace.\n",
      "1 out of 1285 variants loaded from .bim file.\n",
      "2655 people (1648 males, 1007 females) loaded from .fam.\n",
      "1703 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2655 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999247.\n",
      "1 variant and 2655 people pass filters and QC.\n",
      "Among remaining phenotypes, 1292 are cases and 411 are controls.  (952\n",
      "phenotypes are missing.)\n",
      "--recode A to\n",
      "/home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs74391911.raw ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/TMEM175/\"\n",
    "\n",
    "command = f\"\"\"\n",
    "/home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/TMEM175_AJ/AJ_TMEM175 \\\n",
    "--chr 4 \\\n",
    "--from-bp 956095 \\\n",
    "--to-bp 956095 \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/TMEM175_AJ/TMEM175_rs74391911 \\\n",
    "--double-id\n",
    "\"\"\"\n",
    "\n",
    "!{command}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ab5ff0a8-32e5-4596-89e8-1a16c7eaa14e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "rs74391911 = pd.read_csv(f'{WORK_DIR}/TMEM175_AJ/TMEM175_rs74391911.raw',sep=\" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5ed27a03-24c7-4bcd-b052-2468859c4b50",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID_x</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT_x</th>\n",
       "      <th>MAT_x</th>\n",
       "      <th>SEX_x</th>\n",
       "      <th>PHENOTYPE_x</th>\n",
       "      <th>chr4:945299:C:T_C</th>\n",
       "      <th>FID_y</th>\n",
       "      <th>PAT_y</th>\n",
       "      <th>MAT_y</th>\n",
       "      <th>SEX_y</th>\n",
       "      <th>PHENOTYPE_y</th>\n",
       "      <th>chr4:956095:C:G_G</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000021_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000475_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000486_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000646_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000808_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2650</th>\n",
       "      <td>0</td>\n",
       "      <td>VIPD_000746_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2651</th>\n",
       "      <td>0</td>\n",
       "      <td>VIPD_000748_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2652</th>\n",
       "      <td>0</td>\n",
       "      <td>YMS_000030_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2653</th>\n",
       "      <td>0</td>\n",
       "      <td>YMS_000031_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2654</th>\n",
       "      <td>0</td>\n",
       "      <td>YMS_000038_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2655 rows √ó 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID_x             IID  PAT_x  MAT_x  SEX_x  PHENOTYPE_x  \\\n",
       "0         0  APGS_000021_s1      0      0      1            2   \n",
       "1         0  APGS_000475_s1      0      0      1            2   \n",
       "2         0  APGS_000486_s1      0      0      1            2   \n",
       "3         0  APGS_000646_s1      0      0      2            2   \n",
       "4         0  APGS_000808_s1      0      0      1            2   \n",
       "...     ...             ...    ...    ...    ...          ...   \n",
       "2650      0  VIPD_000746_s1      0      0      1            2   \n",
       "2651      0  VIPD_000748_s1      0      0      1            2   \n",
       "2652      0   YMS_000030_s1      0      0      1            2   \n",
       "2653      0   YMS_000031_s1      0      0      1            1   \n",
       "2654      0   YMS_000038_s1      0      0      1            2   \n",
       "\n",
       "      chr4:945299:C:T_C  FID_y  PAT_y  MAT_y  SEX_y  PHENOTYPE_y  \\\n",
       "0                   0.0      0      0      0      1            2   \n",
       "1                   2.0      0      0      0      1            2   \n",
       "2                   1.0      0      0      0      1            2   \n",
       "3                   0.0      0      0      0      2            2   \n",
       "4                   1.0      0      0      0      1            2   \n",
       "...                 ...    ...    ...    ...    ...          ...   \n",
       "2650                1.0      0      0      0      1            2   \n",
       "2651                1.0      0      0      0      1            2   \n",
       "2652                0.0      0      0      0      1            2   \n",
       "2653                1.0      0      0      0      1            1   \n",
       "2654                1.0      0      0      0      1            2   \n",
       "\n",
       "      chr4:956095:C:G_G  \n",
       "0                   0.0  \n",
       "1                   0.0  \n",
       "2                   0.0  \n",
       "3                   0.0  \n",
       "4                   0.0  \n",
       "...                 ...  \n",
       "2650                0.0  \n",
       "2651                0.0  \n",
       "2652                0.0  \n",
       "2653                0.0  \n",
       "2654                0.0  \n",
       "\n",
       "[2655 rows x 13 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total = pd.merge(rs6599388, rs74391911, on=\"IID\")\n",
    "total"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e6166899-9a25-497d-bed6-2ca3ea0484bd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>FATID</th>\n",
       "      <th>MATID</th>\n",
       "      <th>SEX</th>\n",
       "      <th>AGE</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000002_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-29.483298</td>\n",
       "      <td>-38.473005</td>\n",
       "      <td>-9.398904</td>\n",
       "      <td>-10.842865</td>\n",
       "      <td>25.744648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000003_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-33.068958</td>\n",
       "      <td>-34.288862</td>\n",
       "      <td>-11.116951</td>\n",
       "      <td>-8.056146</td>\n",
       "      <td>22.924043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000004_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-28.204441</td>\n",
       "      <td>-30.969481</td>\n",
       "      <td>-15.883093</td>\n",
       "      <td>-7.004351</td>\n",
       "      <td>18.647007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000005_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-35.039753</td>\n",
       "      <td>-39.565822</td>\n",
       "      <td>-12.522753</td>\n",
       "      <td>-9.339179</td>\n",
       "      <td>21.956533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000006_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-30.279907</td>\n",
       "      <td>-34.988905</td>\n",
       "      <td>-10.664628</td>\n",
       "      <td>-7.407177</td>\n",
       "      <td>26.357548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58204</th>\n",
       "      <td>0</td>\n",
       "      <td>YMS_000049_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-33.320502</td>\n",
       "      <td>-37.170809</td>\n",
       "      <td>-9.867466</td>\n",
       "      <td>-6.818637</td>\n",
       "      <td>18.344976</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58205</th>\n",
       "      <td>0</td>\n",
       "      <td>YMS_000050_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-33.550740</td>\n",
       "      <td>-38.934634</td>\n",
       "      <td>-13.550840</td>\n",
       "      <td>-11.161991</td>\n",
       "      <td>26.601555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58206</th>\n",
       "      <td>0</td>\n",
       "      <td>YMS_000051_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-36.415950</td>\n",
       "      <td>-36.634710</td>\n",
       "      <td>-9.623981</td>\n",
       "      <td>-8.333470</td>\n",
       "      <td>19.165537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58207</th>\n",
       "      <td>0</td>\n",
       "      <td>YMS_000052_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>55.200734</td>\n",
       "      <td>-12.367759</td>\n",
       "      <td>-2.246785</td>\n",
       "      <td>-0.681236</td>\n",
       "      <td>5.281365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58208</th>\n",
       "      <td>0</td>\n",
       "      <td>YMS_000053_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-32.401921</td>\n",
       "      <td>-39.458156</td>\n",
       "      <td>-7.987121</td>\n",
       "      <td>-9.101960</td>\n",
       "      <td>19.051343</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>58209 rows √ó 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       FID             IID  FATID  MATID  SEX  AGE  PHENO        PC1  \\\n",
       "0        0  APGS_000002_s1      0      0  NaN  NaN    NaN -29.483298   \n",
       "1        0  APGS_000003_s1      0      0  NaN  NaN    NaN -33.068958   \n",
       "2        0  APGS_000004_s1      0      0  NaN  NaN    NaN -28.204441   \n",
       "3        0  APGS_000005_s1      0      0  NaN  NaN    NaN -35.039753   \n",
       "4        0  APGS_000006_s1      0      0  NaN  NaN    NaN -30.279907   \n",
       "...    ...             ...    ...    ...  ...  ...    ...        ...   \n",
       "58204    0   YMS_000049_s1      0      0  NaN  NaN    NaN -33.320502   \n",
       "58205    0   YMS_000050_s1      0      0  NaN  NaN    NaN -33.550740   \n",
       "58206    0   YMS_000051_s1      0      0  NaN  NaN    NaN -36.415950   \n",
       "58207    0   YMS_000052_s1      0      0  NaN  NaN    NaN  55.200734   \n",
       "58208    0   YMS_000053_s1      0      0  NaN  NaN    NaN -32.401921   \n",
       "\n",
       "             PC2        PC3        PC4        PC5  \n",
       "0     -38.473005  -9.398904 -10.842865  25.744648  \n",
       "1     -34.288862 -11.116951  -8.056146  22.924043  \n",
       "2     -30.969481 -15.883093  -7.004351  18.647007  \n",
       "3     -39.565822 -12.522753  -9.339179  21.956533  \n",
       "4     -34.988905 -10.664628  -7.407177  26.357548  \n",
       "...          ...        ...        ...        ...  \n",
       "58204 -37.170809  -9.867466  -6.818637  18.344976  \n",
       "58205 -38.934634 -13.550840 -11.161991  26.601555  \n",
       "58206 -36.634710  -9.623981  -8.333470  19.165537  \n",
       "58207 -12.367759  -2.246785  -0.681236   5.281365  \n",
       "58208 -39.458156  -7.987121  -9.101960  19.051343  \n",
       "\n",
       "[58209 rows x 12 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov = pd.read_csv(f'{WORK_DIR}/TMEM175_AJ/AJ_covariate_file.txt', sep = '\\t')\n",
    "cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "808533ee-d070-4922-82e6-3723db7f9d30",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID_x</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT_x</th>\n",
       "      <th>MAT_x</th>\n",
       "      <th>SEX_x</th>\n",
       "      <th>PHENOTYPE_x</th>\n",
       "      <th>chr4:945299:C:T_C</th>\n",
       "      <th>FID_y</th>\n",
       "      <th>PAT_y</th>\n",
       "      <th>MAT_y</th>\n",
       "      <th>SEX_y</th>\n",
       "      <th>PHENOTYPE_y</th>\n",
       "      <th>chr4:956095:C:G_G</th>\n",
       "      <th>FID</th>\n",
       "      <th>FATID</th>\n",
       "      <th>MATID</th>\n",
       "      <th>SEX</th>\n",
       "      <th>AGE</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000021_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-28.028682</td>\n",
       "      <td>-35.735363</td>\n",
       "      <td>-6.580637</td>\n",
       "      <td>-17.501411</td>\n",
       "      <td>-23.026102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000475_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-28.154728</td>\n",
       "      <td>-36.708128</td>\n",
       "      <td>-5.730325</td>\n",
       "      <td>-20.750956</td>\n",
       "      <td>-21.729629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000486_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-29.810620</td>\n",
       "      <td>-37.561248</td>\n",
       "      <td>-9.373941</td>\n",
       "      <td>-16.773748</td>\n",
       "      <td>-27.482223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000646_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-31.967732</td>\n",
       "      <td>-34.309571</td>\n",
       "      <td>-8.603258</td>\n",
       "      <td>-17.143810</td>\n",
       "      <td>-21.027547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000808_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-29.929420</td>\n",
       "      <td>-33.911640</td>\n",
       "      <td>-9.610071</td>\n",
       "      <td>-18.087631</td>\n",
       "      <td>-24.559511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2650</th>\n",
       "      <td>0</td>\n",
       "      <td>VIPD_000746_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-31.455498</td>\n",
       "      <td>-35.058387</td>\n",
       "      <td>-9.777408</td>\n",
       "      <td>-18.399794</td>\n",
       "      <td>-27.254806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2651</th>\n",
       "      <td>0</td>\n",
       "      <td>VIPD_000748_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-31.319358</td>\n",
       "      <td>-39.960596</td>\n",
       "      <td>-10.932457</td>\n",
       "      <td>-17.389794</td>\n",
       "      <td>-1.963005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2652</th>\n",
       "      <td>0</td>\n",
       "      <td>YMS_000030_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-29.635489</td>\n",
       "      <td>-34.956373</td>\n",
       "      <td>-6.466043</td>\n",
       "      <td>-14.761441</td>\n",
       "      <td>-20.758634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2653</th>\n",
       "      <td>0</td>\n",
       "      <td>YMS_000031_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-32.862131</td>\n",
       "      <td>-35.306282</td>\n",
       "      <td>-9.166195</td>\n",
       "      <td>-15.693077</td>\n",
       "      <td>-21.088431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2654</th>\n",
       "      <td>0</td>\n",
       "      <td>YMS_000038_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-31.121242</td>\n",
       "      <td>-36.512965</td>\n",
       "      <td>-5.037368</td>\n",
       "      <td>-16.566811</td>\n",
       "      <td>-24.178578</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2655 rows √ó 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID_x             IID  PAT_x  MAT_x  SEX_x  PHENOTYPE_x  \\\n",
       "0         0  APGS_000021_s1      0      0      1            2   \n",
       "1         0  APGS_000475_s1      0      0      1            2   \n",
       "2         0  APGS_000486_s1      0      0      1            2   \n",
       "3         0  APGS_000646_s1      0      0      2            2   \n",
       "4         0  APGS_000808_s1      0      0      1            2   \n",
       "...     ...             ...    ...    ...    ...          ...   \n",
       "2650      0  VIPD_000746_s1      0      0      1            2   \n",
       "2651      0  VIPD_000748_s1      0      0      1            2   \n",
       "2652      0   YMS_000030_s1      0      0      1            2   \n",
       "2653      0   YMS_000031_s1      0      0      1            1   \n",
       "2654      0   YMS_000038_s1      0      0      1            2   \n",
       "\n",
       "      chr4:945299:C:T_C  FID_y  PAT_y  MAT_y  SEX_y  PHENOTYPE_y  \\\n",
       "0                   0.0      0      0      0      1            2   \n",
       "1                   2.0      0      0      0      1            2   \n",
       "2                   1.0      0      0      0      1            2   \n",
       "3                   0.0      0      0      0      2            2   \n",
       "4                   1.0      0      0      0      1            2   \n",
       "...                 ...    ...    ...    ...    ...          ...   \n",
       "2650                1.0      0      0      0      1            2   \n",
       "2651                1.0      0      0      0      1            2   \n",
       "2652                0.0      0      0      0      1            2   \n",
       "2653                1.0      0      0      0      1            1   \n",
       "2654                1.0      0      0      0      1            2   \n",
       "\n",
       "      chr4:956095:C:G_G  FID  FATID  MATID  SEX   AGE  PHENO        PC1  \\\n",
       "0                   0.0    0      0      0  1.0   NaN    2.0 -28.028682   \n",
       "1                   0.0    0      0      0  1.0   NaN    2.0 -28.154728   \n",
       "2                   0.0    0      0      0  1.0   NaN    2.0 -29.810620   \n",
       "3                   0.0    0      0      0  2.0   NaN    2.0 -31.967732   \n",
       "4                   0.0    0      0      0  1.0   NaN    2.0 -29.929420   \n",
       "...                 ...  ...    ...    ...  ...   ...    ...        ...   \n",
       "2650                0.0    0      0      0  1.0  74.0    2.0 -31.455498   \n",
       "2651                0.0    0      0      0  1.0  73.0    2.0 -31.319358   \n",
       "2652                0.0    0      0      0  1.0  72.0    2.0 -29.635489   \n",
       "2653                0.0    0      0      0  1.0  69.0    1.0 -32.862131   \n",
       "2654                0.0    0      0      0  1.0  65.0    2.0 -31.121242   \n",
       "\n",
       "            PC2        PC3        PC4        PC5  \n",
       "0    -35.735363  -6.580637 -17.501411 -23.026102  \n",
       "1    -36.708128  -5.730325 -20.750956 -21.729629  \n",
       "2    -37.561248  -9.373941 -16.773748 -27.482223  \n",
       "3    -34.309571  -8.603258 -17.143810 -21.027547  \n",
       "4    -33.911640  -9.610071 -18.087631 -24.559511  \n",
       "...         ...        ...        ...        ...  \n",
       "2650 -35.058387  -9.777408 -18.399794 -27.254806  \n",
       "2651 -39.960596 -10.932457 -17.389794  -1.963005  \n",
       "2652 -34.956373  -6.466043 -14.761441 -20.758634  \n",
       "2653 -35.306282  -9.166195 -15.693077 -21.088431  \n",
       "2654 -36.512965  -5.037368 -16.566811 -24.178578  \n",
       "\n",
       "[2655 rows x 24 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total = pd.merge(total, cov, on=\"IID\")\n",
    "total"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "79333ff4-21da-437e-8e7a-146d2dce9979",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID_x</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT_x</th>\n",
       "      <th>MAT_x</th>\n",
       "      <th>SEX_x</th>\n",
       "      <th>PHENOTYPE_x</th>\n",
       "      <th>chr4:945299:C:T_C</th>\n",
       "      <th>FID_y</th>\n",
       "      <th>PAT_y</th>\n",
       "      <th>MAT_y</th>\n",
       "      <th>SEX_y</th>\n",
       "      <th>PHENOTYPE_y</th>\n",
       "      <th>chr4:956095:C:G_G</th>\n",
       "      <th>FID</th>\n",
       "      <th>FATID</th>\n",
       "      <th>MATID</th>\n",
       "      <th>SEX</th>\n",
       "      <th>AGE</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000021_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-28.028682</td>\n",
       "      <td>-35.735363</td>\n",
       "      <td>-6.580637</td>\n",
       "      <td>-17.501411</td>\n",
       "      <td>-23.026102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000475_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-28.154728</td>\n",
       "      <td>-36.708128</td>\n",
       "      <td>-5.730325</td>\n",
       "      <td>-20.750956</td>\n",
       "      <td>-21.729629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000486_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-29.810620</td>\n",
       "      <td>-37.561248</td>\n",
       "      <td>-9.373941</td>\n",
       "      <td>-16.773748</td>\n",
       "      <td>-27.482223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000646_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-31.967732</td>\n",
       "      <td>-34.309571</td>\n",
       "      <td>-8.603258</td>\n",
       "      <td>-17.143810</td>\n",
       "      <td>-21.027547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000808_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-29.929420</td>\n",
       "      <td>-33.911640</td>\n",
       "      <td>-9.610071</td>\n",
       "      <td>-18.087631</td>\n",
       "      <td>-24.559511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2650</th>\n",
       "      <td>0</td>\n",
       "      <td>VIPD_000746_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-31.455498</td>\n",
       "      <td>-35.058387</td>\n",
       "      <td>-9.777408</td>\n",
       "      <td>-18.399794</td>\n",
       "      <td>-27.254806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2651</th>\n",
       "      <td>0</td>\n",
       "      <td>VIPD_000748_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-31.319358</td>\n",
       "      <td>-39.960596</td>\n",
       "      <td>-10.932457</td>\n",
       "      <td>-17.389794</td>\n",
       "      <td>-1.963005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2652</th>\n",
       "      <td>0</td>\n",
       "      <td>YMS_000030_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-29.635489</td>\n",
       "      <td>-34.956373</td>\n",
       "      <td>-6.466043</td>\n",
       "      <td>-14.761441</td>\n",
       "      <td>-20.758634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2653</th>\n",
       "      <td>0</td>\n",
       "      <td>YMS_000031_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-32.862131</td>\n",
       "      <td>-35.306282</td>\n",
       "      <td>-9.166195</td>\n",
       "      <td>-15.693077</td>\n",
       "      <td>-21.088431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2654</th>\n",
       "      <td>0</td>\n",
       "      <td>YMS_000038_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-31.121242</td>\n",
       "      <td>-36.512965</td>\n",
       "      <td>-5.037368</td>\n",
       "      <td>-16.566811</td>\n",
       "      <td>-24.178578</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1703 rows √ó 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID_x             IID  PAT_x  MAT_x  SEX_x  PHENOTYPE_x  \\\n",
       "0         0  APGS_000021_s1      0      0      1            2   \n",
       "1         0  APGS_000475_s1      0      0      1            2   \n",
       "2         0  APGS_000486_s1      0      0      1            2   \n",
       "3         0  APGS_000646_s1      0      0      2            2   \n",
       "4         0  APGS_000808_s1      0      0      1            2   \n",
       "...     ...             ...    ...    ...    ...          ...   \n",
       "2650      0  VIPD_000746_s1      0      0      1            2   \n",
       "2651      0  VIPD_000748_s1      0      0      1            2   \n",
       "2652      0   YMS_000030_s1      0      0      1            2   \n",
       "2653      0   YMS_000031_s1      0      0      1            1   \n",
       "2654      0   YMS_000038_s1      0      0      1            2   \n",
       "\n",
       "      chr4:945299:C:T_C  FID_y  PAT_y  MAT_y  SEX_y  PHENOTYPE_y  \\\n",
       "0                   0.0      0      0      0      1            2   \n",
       "1                   2.0      0      0      0      1            2   \n",
       "2                   1.0      0      0      0      1            2   \n",
       "3                   0.0      0      0      0      2            2   \n",
       "4                   1.0      0      0      0      1            2   \n",
       "...                 ...    ...    ...    ...    ...          ...   \n",
       "2650                1.0      0      0      0      1            2   \n",
       "2651                1.0      0      0      0      1            2   \n",
       "2652                0.0      0      0      0      1            2   \n",
       "2653                1.0      0      0      0      1            1   \n",
       "2654                1.0      0      0      0      1            2   \n",
       "\n",
       "      chr4:956095:C:G_G  FID  FATID  MATID  SEX   AGE  PHENO        PC1  \\\n",
       "0                   0.0    0      0      0  1.0   NaN    2.0 -28.028682   \n",
       "1                   0.0    0      0      0  1.0   NaN    2.0 -28.154728   \n",
       "2                   0.0    0      0      0  1.0   NaN    2.0 -29.810620   \n",
       "3                   0.0    0      0      0  2.0   NaN    2.0 -31.967732   \n",
       "4                   0.0    0      0      0  1.0   NaN    2.0 -29.929420   \n",
       "...                 ...  ...    ...    ...  ...   ...    ...        ...   \n",
       "2650                0.0    0      0      0  1.0  74.0    2.0 -31.455498   \n",
       "2651                0.0    0      0      0  1.0  73.0    2.0 -31.319358   \n",
       "2652                0.0    0      0      0  1.0  72.0    2.0 -29.635489   \n",
       "2653                0.0    0      0      0  1.0  69.0    1.0 -32.862131   \n",
       "2654                0.0    0      0      0  1.0  65.0    2.0 -31.121242   \n",
       "\n",
       "            PC2        PC3        PC4        PC5  \n",
       "0    -35.735363  -6.580637 -17.501411 -23.026102  \n",
       "1    -36.708128  -5.730325 -20.750956 -21.729629  \n",
       "2    -37.561248  -9.373941 -16.773748 -27.482223  \n",
       "3    -34.309571  -8.603258 -17.143810 -21.027547  \n",
       "4    -33.911640  -9.610071 -18.087631 -24.559511  \n",
       "...         ...        ...        ...        ...  \n",
       "2650 -35.058387  -9.777408 -18.399794 -27.254806  \n",
       "2651 -39.960596 -10.932457 -17.389794  -1.963005  \n",
       "2652 -34.956373  -6.466043 -14.761441 -20.758634  \n",
       "2653 -35.306282  -9.166195 -15.693077 -21.088431  \n",
       "2654 -36.512965  -5.037368 -16.566811 -24.178578  \n",
       "\n",
       "[1703 rows x 24 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total = total[(total['PHENOTYPE_x']==1) | (total['PHENOTYPE_x']==2)]\n",
    "total"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f407949-666d-4b46-ba62-58ed341bc746",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "total = total[[\"IID\",'PHENOTYPE_x',\"chr4:945299:C:T_C\",\"chr4:956095:C:G_G\", \"SEX\", \"AGE\", \"PC1\",\"PC2\",\"PC3\",\"PC4\",\"PC5\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5eb8b198-a4f7-4e18-b5c0-7f29e3473167",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IID</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>TMEM175_rs6599388</th>\n",
       "      <th>TMEM175_rs74391911</th>\n",
       "      <th>SEX</th>\n",
       "      <th>AGE</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>APGS_000021_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-28.028682</td>\n",
       "      <td>-35.735363</td>\n",
       "      <td>-6.580637</td>\n",
       "      <td>-17.501411</td>\n",
       "      <td>-23.026102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>APGS_000475_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-28.154728</td>\n",
       "      <td>-36.708128</td>\n",
       "      <td>-5.730325</td>\n",
       "      <td>-20.750956</td>\n",
       "      <td>-21.729629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>APGS_000486_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-29.810620</td>\n",
       "      <td>-37.561248</td>\n",
       "      <td>-9.373941</td>\n",
       "      <td>-16.773748</td>\n",
       "      <td>-27.482223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>APGS_000646_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-31.967732</td>\n",
       "      <td>-34.309571</td>\n",
       "      <td>-8.603258</td>\n",
       "      <td>-17.143810</td>\n",
       "      <td>-21.027547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>APGS_000808_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-29.929420</td>\n",
       "      <td>-33.911640</td>\n",
       "      <td>-9.610071</td>\n",
       "      <td>-18.087631</td>\n",
       "      <td>-24.559511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2650</th>\n",
       "      <td>VIPD_000746_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>-31.455498</td>\n",
       "      <td>-35.058387</td>\n",
       "      <td>-9.777408</td>\n",
       "      <td>-18.399794</td>\n",
       "      <td>-27.254806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2651</th>\n",
       "      <td>VIPD_000748_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>-31.319358</td>\n",
       "      <td>-39.960596</td>\n",
       "      <td>-10.932457</td>\n",
       "      <td>-17.389794</td>\n",
       "      <td>-1.963005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2652</th>\n",
       "      <td>YMS_000030_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>-29.635489</td>\n",
       "      <td>-34.956373</td>\n",
       "      <td>-6.466043</td>\n",
       "      <td>-14.761441</td>\n",
       "      <td>-20.758634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2653</th>\n",
       "      <td>YMS_000031_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>-32.862131</td>\n",
       "      <td>-35.306282</td>\n",
       "      <td>-9.166195</td>\n",
       "      <td>-15.693077</td>\n",
       "      <td>-21.088431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2654</th>\n",
       "      <td>YMS_000038_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>-31.121242</td>\n",
       "      <td>-36.512965</td>\n",
       "      <td>-5.037368</td>\n",
       "      <td>-16.566811</td>\n",
       "      <td>-24.178578</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1703 rows √ó 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 IID  PHENOTYPE  TMEM175_rs6599388  TMEM175_rs74391911  SEX  \\\n",
       "0     APGS_000021_s1          2                0.0                 0.0  1.0   \n",
       "1     APGS_000475_s1          2                2.0                 0.0  1.0   \n",
       "2     APGS_000486_s1          2                1.0                 0.0  1.0   \n",
       "3     APGS_000646_s1          2                0.0                 0.0  2.0   \n",
       "4     APGS_000808_s1          2                1.0                 0.0  1.0   \n",
       "...              ...        ...                ...                 ...  ...   \n",
       "2650  VIPD_000746_s1          2                1.0                 0.0  1.0   \n",
       "2651  VIPD_000748_s1          2                1.0                 0.0  1.0   \n",
       "2652   YMS_000030_s1          2                0.0                 0.0  1.0   \n",
       "2653   YMS_000031_s1          1                1.0                 0.0  1.0   \n",
       "2654   YMS_000038_s1          2                1.0                 0.0  1.0   \n",
       "\n",
       "       AGE        PC1        PC2        PC3        PC4        PC5  \n",
       "0      NaN -28.028682 -35.735363  -6.580637 -17.501411 -23.026102  \n",
       "1      NaN -28.154728 -36.708128  -5.730325 -20.750956 -21.729629  \n",
       "2      NaN -29.810620 -37.561248  -9.373941 -16.773748 -27.482223  \n",
       "3      NaN -31.967732 -34.309571  -8.603258 -17.143810 -21.027547  \n",
       "4      NaN -29.929420 -33.911640  -9.610071 -18.087631 -24.559511  \n",
       "...    ...        ...        ...        ...        ...        ...  \n",
       "2650  74.0 -31.455498 -35.058387  -9.777408 -18.399794 -27.254806  \n",
       "2651  73.0 -31.319358 -39.960596 -10.932457 -17.389794  -1.963005  \n",
       "2652  72.0 -29.635489 -34.956373  -6.466043 -14.761441 -20.758634  \n",
       "2653  69.0 -32.862131 -35.306282  -9.166195 -15.693077 -21.088431  \n",
       "2654  65.0 -31.121242 -36.512965  -5.037368 -16.566811 -24.178578  \n",
       "\n",
       "[1703 rows x 11 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total.columns = [\"IID\",'PHENOTYPE',\"TMEM175_rs6599388\",\"TMEM175_rs74391911\",\"SEX\", \"AGE\", \"PC1\",\"PC2\",\"PC3\",\"PC4\",\"PC5\"]\n",
    "total"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7fd14645-f65b-42f6-b429-bc9a8db4d88d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "total.to_csv(f'{WORK_DIR}/TMEM175_AJ/TMEM175_rs6599388_rs74391911.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "778cb1de-4801-4c78-b732-931963267efd",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "total <- read.csv(\"~/workspace/ws_files/TMEM175/TMEM175_AJ/TMEM175_rs6599388_rs74391911.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c7ad5ef2-144e-4a8e-915f-5de4c90474a9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  X            IID PHENOTYPE TMEM175_rs6599388 TMEM175_rs74391911 SEX AGE\n",
      "1 0 APGS_000021_s1         2                 0                  0   1  NA\n",
      "2 1 APGS_000475_s1         2                 2                  0   1  NA\n",
      "3 2 APGS_000486_s1         2                 1                  0   1  NA\n",
      "4 3 APGS_000646_s1         2                 0                  0   2  NA\n",
      "5 4 APGS_000808_s1         2                 1                  0   1  NA\n",
      "6 5 APGS_001059_s1         2                 0                  0   1  NA\n",
      "        PC1       PC2       PC3       PC4       PC5\n",
      "1 -28.02868 -35.73536 -6.580637 -17.50141 -23.02610\n",
      "2 -28.15473 -36.70813 -5.730325 -20.75096 -21.72963\n",
      "3 -29.81062 -37.56125 -9.373941 -16.77375 -27.48222\n",
      "4 -31.96773 -34.30957 -8.603258 -17.14381 -21.02755\n",
      "5 -29.92942 -33.91164 -9.610071 -18.08763 -24.55951\n",
      "6 -28.19315 -35.13333 -4.791519 -16.36048 -22.63633\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "head(total)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8f1f0454-eda8-4c7e-acc4-1f87703b1e76",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Call:\n",
      "glm(formula = PHENOTYPE ~ TMEM175_rs6599388 + TMEM175_rs74391911 + \n",
      "    SEX + AGE + PC1 + PC2 + PC3 + PC4 + PC5, data = total)\n",
      "\n",
      "Coefficients:\n",
      "                     Estimate Std. Error t value Pr(>|t|)    \n",
      "(Intercept)         2.1225957  0.2629897   8.071 1.49e-15 ***\n",
      "TMEM175_rs6599388  -0.0564446  0.0137131  -4.116 4.08e-05 ***\n",
      "TMEM175_rs74391911 -0.2827724  0.0685212  -4.127 3.90e-05 ***\n",
      "SEX                -0.1090329  0.0201904  -5.400 7.81e-08 ***\n",
      "AGE                 0.0031262  0.0009613   3.252  0.00117 ** \n",
      "PC1                 0.0035181  0.0060532   0.581  0.56120    \n",
      "PC2                 0.0071335  0.0051950   1.373  0.16993    \n",
      "PC3                 0.0001873  0.0051913   0.036  0.97123    \n",
      "PC4                -0.0071116  0.0052951  -1.343  0.17947    \n",
      "PC5                 0.0022921  0.0025337   0.905  0.36581    \n",
      "---\n",
      "Signif. codes:  0 ‚Äò***‚Äô 0.001 ‚Äò**‚Äô 0.01 ‚Äò*‚Äô 0.05 ‚Äò.‚Äô 0.1 ‚Äò ‚Äô 1\n",
      "\n",
      "(Dispersion parameter for gaussian family taken to be 0.1266327)\n",
      "\n",
      "    Null deviance: 187.71  on 1408  degrees of freedom\n",
      "Residual deviance: 177.16  on 1399  degrees of freedom\n",
      "  (294 observations deleted due to missingness)\n",
      "AIC: 1098.9\n",
      "\n",
      "Number of Fisher Scoring iterations: 2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "dep_corr<- glm(PHENOTYPE ~ TMEM175_rs6599388 + TMEM175_rs74391911 + SEX + AGE + PC1 + PC2 + PC3 + PC4 + PC5, data=total)\n",
    "summary(dep_corr)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec5c5126-f45a-4380-87d5-72ba1e5f9b40",
   "metadata": {},
   "source": [
    "### GCTA conditional analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "25c84835-2b0b-4a18-95b6-a7c02cfeb266",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>PROVISIONAL_REF?</th>\n",
       "      <th>A1</th>\n",
       "      <th>OMITTED</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>FIRTH?</th>\n",
       "      <th>TEST</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>OR</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>ERRCODE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.164467</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20837</td>\n",
       "      <td>1.034410</td>\n",
       "      <td>0.030398</td>\n",
       "      <td>0.974580</td>\n",
       "      <td>1.097910</td>\n",
       "      <td>1.112890</td>\n",
       "      <td>0.265757</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>882611</td>\n",
       "      <td>chr4:882611:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.051185</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20885</td>\n",
       "      <td>0.931946</td>\n",
       "      <td>0.050102</td>\n",
       "      <td>0.844780</td>\n",
       "      <td>1.028110</td>\n",
       "      <td>-1.406740</td>\n",
       "      <td>0.159504</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>883018</td>\n",
       "      <td>chr4:883018:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.050599</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20880</td>\n",
       "      <td>0.928749</td>\n",
       "      <td>0.050369</td>\n",
       "      <td>0.841441</td>\n",
       "      <td>1.025120</td>\n",
       "      <td>-1.467500</td>\n",
       "      <td>0.142240</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>24</td>\n",
       "      <td>4</td>\n",
       "      <td>883133</td>\n",
       "      <td>chr4:883133:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.012734</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20811</td>\n",
       "      <td>0.809012</td>\n",
       "      <td>0.096557</td>\n",
       "      <td>0.669523</td>\n",
       "      <td>0.977562</td>\n",
       "      <td>-2.194990</td>\n",
       "      <td>0.028164</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>32</td>\n",
       "      <td>4</td>\n",
       "      <td>883483</td>\n",
       "      <td>chr4:883483:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.035357</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20901</td>\n",
       "      <td>0.883241</td>\n",
       "      <td>0.058911</td>\n",
       "      <td>0.786926</td>\n",
       "      <td>0.991344</td>\n",
       "      <td>-2.107530</td>\n",
       "      <td>0.035071</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>3136</td>\n",
       "      <td>4</td>\n",
       "      <td>1007954</td>\n",
       "      <td>chr4:1007954:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.279743</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>18971</td>\n",
       "      <td>0.913461</td>\n",
       "      <td>0.025919</td>\n",
       "      <td>0.868217</td>\n",
       "      <td>0.961063</td>\n",
       "      <td>-3.492280</td>\n",
       "      <td>0.000479</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393</th>\n",
       "      <td>3144</td>\n",
       "      <td>4</td>\n",
       "      <td>1008009</td>\n",
       "      <td>chr4:1008009:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.444961</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>18814</td>\n",
       "      <td>0.949197</td>\n",
       "      <td>0.023666</td>\n",
       "      <td>0.906174</td>\n",
       "      <td>0.994262</td>\n",
       "      <td>-2.203110</td>\n",
       "      <td>0.027587</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394</th>\n",
       "      <td>3152</td>\n",
       "      <td>4</td>\n",
       "      <td>1008209</td>\n",
       "      <td>chr4:1008209:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.036468</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20717</td>\n",
       "      <td>1.002060</td>\n",
       "      <td>0.059756</td>\n",
       "      <td>0.891315</td>\n",
       "      <td>1.126570</td>\n",
       "      <td>0.034499</td>\n",
       "      <td>0.972480</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>3160</td>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.279831</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>18965</td>\n",
       "      <td>0.913397</td>\n",
       "      <td>0.025912</td>\n",
       "      <td>0.868167</td>\n",
       "      <td>0.960983</td>\n",
       "      <td>-3.495880</td>\n",
       "      <td>0.000473</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>3168</td>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>chr4:1008480:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Y</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.053420</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20264</td>\n",
       "      <td>0.903891</td>\n",
       "      <td>0.049512</td>\n",
       "      <td>0.820297</td>\n",
       "      <td>0.996003</td>\n",
       "      <td>-2.040850</td>\n",
       "      <td>0.041266</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>397 rows √ó 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0  #CHROM      POS                ID REF ALT PROVISIONAL_REF?  \\\n",
       "0             0       4   882530   chr4:882530:C:T   C   T                Y   \n",
       "1             8       4   882611   chr4:882611:T:C   T   C                Y   \n",
       "2            16       4   883018   chr4:883018:G:A   G   A                Y   \n",
       "3            24       4   883133   chr4:883133:G:A   G   A                Y   \n",
       "4            32       4   883483   chr4:883483:G:A   G   A                Y   \n",
       "..          ...     ...      ...               ...  ..  ..              ...   \n",
       "392        3136       4  1007954  chr4:1007954:T:C   T   C                Y   \n",
       "393        3144       4  1008009  chr4:1008009:C:T   C   T                Y   \n",
       "394        3152       4  1008209  chr4:1008209:C:T   C   T                Y   \n",
       "395        3160       4  1008337  chr4:1008337:C:T   C   T                Y   \n",
       "396        3168       4  1008480  chr4:1008480:A:G   A   G                Y   \n",
       "\n",
       "    A1 OMITTED   A1_FREQ FIRTH? TEST  OBS_CT        OR  LOG(OR)_SE       L95  \\\n",
       "0    T       C  0.164467      N  ADD   20837  1.034410    0.030398  0.974580   \n",
       "1    C       T  0.051185      N  ADD   20885  0.931946    0.050102  0.844780   \n",
       "2    A       G  0.050599      N  ADD   20880  0.928749    0.050369  0.841441   \n",
       "3    A       G  0.012734      N  ADD   20811  0.809012    0.096557  0.669523   \n",
       "4    A       G  0.035357      N  ADD   20901  0.883241    0.058911  0.786926   \n",
       "..  ..     ...       ...    ...  ...     ...       ...         ...       ...   \n",
       "392  T       C  0.279743      N  ADD   18971  0.913461    0.025919  0.868217   \n",
       "393  C       T  0.444961      N  ADD   18814  0.949197    0.023666  0.906174   \n",
       "394  T       C  0.036468      N  ADD   20717  1.002060    0.059756  0.891315   \n",
       "395  C       T  0.279831      N  ADD   18965  0.913397    0.025912  0.868167   \n",
       "396  G       A  0.053420      N  ADD   20264  0.903891    0.049512  0.820297   \n",
       "\n",
       "          U95    Z_STAT         P ERRCODE  \n",
       "0    1.097910  1.112890  0.265757       .  \n",
       "1    1.028110 -1.406740  0.159504       .  \n",
       "2    1.025120 -1.467500  0.142240       .  \n",
       "3    0.977562 -2.194990  0.028164       .  \n",
       "4    0.991344 -2.107530  0.035071       .  \n",
       "..        ...       ...       ...     ...  \n",
       "392  0.961063 -3.492280  0.000479       .  \n",
       "393  0.994262 -2.203110  0.027587       .  \n",
       "394  1.126570  0.034499  0.972480       .  \n",
       "395  0.960983 -3.495880  0.000473       .  \n",
       "396  0.996003 -2.040850  0.041266       .  \n",
       "\n",
       "[397 rows x 20 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read in summary statistics\n",
    "sumstats = pd.read_csv(f'{WORK_DIR}/TMEM175_EUR/EUR.all_adj.csv')\n",
    "sumstats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f61dba53-18e0-41b3-9245-6b8769328c4a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Format summary statistics for GCTA-COJO\n",
    "# First get the log odds ratio - this is required for COJO\n",
    "# 1) For a case-control study, the effect size should be log(odds ratio) with its corresponding standard error.\n",
    "sumstats_formatted = sumstats.copy()\n",
    "sumstats_formatted['b'] = np.log(sumstats_formatted['OR'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "63194ead-379c-4203-8599-4d8a819f63f6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>PROVISIONAL_REF?</th>\n",
       "      <th>A1</th>\n",
       "      <th>OMITTED</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>FIRTH?</th>\n",
       "      <th>TEST</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>OR</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>ERRCODE</th>\n",
       "      <th>b</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.164467</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20837</td>\n",
       "      <td>1.034410</td>\n",
       "      <td>0.030398</td>\n",
       "      <td>0.974580</td>\n",
       "      <td>1.097910</td>\n",
       "      <td>1.112890</td>\n",
       "      <td>0.265757</td>\n",
       "      <td>.</td>\n",
       "      <td>0.033831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>882611</td>\n",
       "      <td>chr4:882611:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.051185</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20885</td>\n",
       "      <td>0.931946</td>\n",
       "      <td>0.050102</td>\n",
       "      <td>0.844780</td>\n",
       "      <td>1.028110</td>\n",
       "      <td>-1.406740</td>\n",
       "      <td>0.159504</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.070480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>883018</td>\n",
       "      <td>chr4:883018:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.050599</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20880</td>\n",
       "      <td>0.928749</td>\n",
       "      <td>0.050369</td>\n",
       "      <td>0.841441</td>\n",
       "      <td>1.025120</td>\n",
       "      <td>-1.467500</td>\n",
       "      <td>0.142240</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.073917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>24</td>\n",
       "      <td>4</td>\n",
       "      <td>883133</td>\n",
       "      <td>chr4:883133:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.012734</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20811</td>\n",
       "      <td>0.809012</td>\n",
       "      <td>0.096557</td>\n",
       "      <td>0.669523</td>\n",
       "      <td>0.977562</td>\n",
       "      <td>-2.194990</td>\n",
       "      <td>0.028164</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.211942</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>32</td>\n",
       "      <td>4</td>\n",
       "      <td>883483</td>\n",
       "      <td>chr4:883483:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.035357</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20901</td>\n",
       "      <td>0.883241</td>\n",
       "      <td>0.058911</td>\n",
       "      <td>0.786926</td>\n",
       "      <td>0.991344</td>\n",
       "      <td>-2.107530</td>\n",
       "      <td>0.035071</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.124157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>3136</td>\n",
       "      <td>4</td>\n",
       "      <td>1007954</td>\n",
       "      <td>chr4:1007954:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.279743</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>18971</td>\n",
       "      <td>0.913461</td>\n",
       "      <td>0.025919</td>\n",
       "      <td>0.868217</td>\n",
       "      <td>0.961063</td>\n",
       "      <td>-3.492280</td>\n",
       "      <td>0.000479</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.090515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393</th>\n",
       "      <td>3144</td>\n",
       "      <td>4</td>\n",
       "      <td>1008009</td>\n",
       "      <td>chr4:1008009:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.444961</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>18814</td>\n",
       "      <td>0.949197</td>\n",
       "      <td>0.023666</td>\n",
       "      <td>0.906174</td>\n",
       "      <td>0.994262</td>\n",
       "      <td>-2.203110</td>\n",
       "      <td>0.027587</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.052139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394</th>\n",
       "      <td>3152</td>\n",
       "      <td>4</td>\n",
       "      <td>1008209</td>\n",
       "      <td>chr4:1008209:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.036468</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20717</td>\n",
       "      <td>1.002060</td>\n",
       "      <td>0.059756</td>\n",
       "      <td>0.891315</td>\n",
       "      <td>1.126570</td>\n",
       "      <td>0.034499</td>\n",
       "      <td>0.972480</td>\n",
       "      <td>.</td>\n",
       "      <td>0.002058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>3160</td>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.279831</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>18965</td>\n",
       "      <td>0.913397</td>\n",
       "      <td>0.025912</td>\n",
       "      <td>0.868167</td>\n",
       "      <td>0.960983</td>\n",
       "      <td>-3.495880</td>\n",
       "      <td>0.000473</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.090585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>3168</td>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>chr4:1008480:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Y</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.053420</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>20264</td>\n",
       "      <td>0.903891</td>\n",
       "      <td>0.049512</td>\n",
       "      <td>0.820297</td>\n",
       "      <td>0.996003</td>\n",
       "      <td>-2.040850</td>\n",
       "      <td>0.041266</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.101047</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>397 rows √ó 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0  #CHROM      POS                ID REF ALT PROVISIONAL_REF?  \\\n",
       "0             0       4   882530   chr4:882530:C:T   C   T                Y   \n",
       "1             8       4   882611   chr4:882611:T:C   T   C                Y   \n",
       "2            16       4   883018   chr4:883018:G:A   G   A                Y   \n",
       "3            24       4   883133   chr4:883133:G:A   G   A                Y   \n",
       "4            32       4   883483   chr4:883483:G:A   G   A                Y   \n",
       "..          ...     ...      ...               ...  ..  ..              ...   \n",
       "392        3136       4  1007954  chr4:1007954:T:C   T   C                Y   \n",
       "393        3144       4  1008009  chr4:1008009:C:T   C   T                Y   \n",
       "394        3152       4  1008209  chr4:1008209:C:T   C   T                Y   \n",
       "395        3160       4  1008337  chr4:1008337:C:T   C   T                Y   \n",
       "396        3168       4  1008480  chr4:1008480:A:G   A   G                Y   \n",
       "\n",
       "    A1 OMITTED   A1_FREQ FIRTH? TEST  OBS_CT        OR  LOG(OR)_SE       L95  \\\n",
       "0    T       C  0.164467      N  ADD   20837  1.034410    0.030398  0.974580   \n",
       "1    C       T  0.051185      N  ADD   20885  0.931946    0.050102  0.844780   \n",
       "2    A       G  0.050599      N  ADD   20880  0.928749    0.050369  0.841441   \n",
       "3    A       G  0.012734      N  ADD   20811  0.809012    0.096557  0.669523   \n",
       "4    A       G  0.035357      N  ADD   20901  0.883241    0.058911  0.786926   \n",
       "..  ..     ...       ...    ...  ...     ...       ...         ...       ...   \n",
       "392  T       C  0.279743      N  ADD   18971  0.913461    0.025919  0.868217   \n",
       "393  C       T  0.444961      N  ADD   18814  0.949197    0.023666  0.906174   \n",
       "394  T       C  0.036468      N  ADD   20717  1.002060    0.059756  0.891315   \n",
       "395  C       T  0.279831      N  ADD   18965  0.913397    0.025912  0.868167   \n",
       "396  G       A  0.053420      N  ADD   20264  0.903891    0.049512  0.820297   \n",
       "\n",
       "          U95    Z_STAT         P ERRCODE         b  \n",
       "0    1.097910  1.112890  0.265757       .  0.033831  \n",
       "1    1.028110 -1.406740  0.159504       . -0.070480  \n",
       "2    1.025120 -1.467500  0.142240       . -0.073917  \n",
       "3    0.977562 -2.194990  0.028164       . -0.211942  \n",
       "4    0.991344 -2.107530  0.035071       . -0.124157  \n",
       "..        ...       ...       ...     ...       ...  \n",
       "392  0.961063 -3.492280  0.000479       . -0.090515  \n",
       "393  0.994262 -2.203110  0.027587       . -0.052139  \n",
       "394  1.126570  0.034499  0.972480       .  0.002058  \n",
       "395  0.960983 -3.495880  0.000473       . -0.090585  \n",
       "396  0.996003 -2.040850  0.041266       . -0.101047  \n",
       "\n",
       "[397 rows x 21 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sumstats_formatted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "854e5738-f644-4e64-8be4-2bdc9315ddd7",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Now select just the necessary columns for COJO\n",
    "sumstats_export = sumstats_formatted[['ID', 'A1', 'OMITTED', 'A1_FREQ', 'b', 'LOG(OR)_SE', 'P', 'OBS_CT']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e375d6e1-fd3e-46b2-b27b-938d0a760502",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>A1</th>\n",
       "      <th>OMITTED</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>b</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>P</th>\n",
       "      <th>OBS_CT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.164467</td>\n",
       "      <td>0.033831</td>\n",
       "      <td>0.030398</td>\n",
       "      <td>0.265757</td>\n",
       "      <td>20837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4:882611:T:C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.051185</td>\n",
       "      <td>-0.070480</td>\n",
       "      <td>0.050102</td>\n",
       "      <td>0.159504</td>\n",
       "      <td>20885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4:883018:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.050599</td>\n",
       "      <td>-0.073917</td>\n",
       "      <td>0.050369</td>\n",
       "      <td>0.142240</td>\n",
       "      <td>20880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4:883133:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.012734</td>\n",
       "      <td>-0.211942</td>\n",
       "      <td>0.096557</td>\n",
       "      <td>0.028164</td>\n",
       "      <td>20811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4:883483:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.035357</td>\n",
       "      <td>-0.124157</td>\n",
       "      <td>0.058911</td>\n",
       "      <td>0.035071</td>\n",
       "      <td>20901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>chr4:1007954:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.279743</td>\n",
       "      <td>-0.090515</td>\n",
       "      <td>0.025919</td>\n",
       "      <td>0.000479</td>\n",
       "      <td>18971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393</th>\n",
       "      <td>chr4:1008009:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.444961</td>\n",
       "      <td>-0.052139</td>\n",
       "      <td>0.023666</td>\n",
       "      <td>0.027587</td>\n",
       "      <td>18814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394</th>\n",
       "      <td>chr4:1008209:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.036468</td>\n",
       "      <td>0.002058</td>\n",
       "      <td>0.059756</td>\n",
       "      <td>0.972480</td>\n",
       "      <td>20717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.279831</td>\n",
       "      <td>-0.090585</td>\n",
       "      <td>0.025912</td>\n",
       "      <td>0.000473</td>\n",
       "      <td>18965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>chr4:1008480:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.053420</td>\n",
       "      <td>-0.101047</td>\n",
       "      <td>0.049512</td>\n",
       "      <td>0.041266</td>\n",
       "      <td>20264</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>397 rows √ó 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   ID A1 OMITTED   A1_FREQ         b  LOG(OR)_SE         P  \\\n",
       "0     chr4:882530:C:T  T       C  0.164467  0.033831    0.030398  0.265757   \n",
       "1     chr4:882611:T:C  C       T  0.051185 -0.070480    0.050102  0.159504   \n",
       "2     chr4:883018:G:A  A       G  0.050599 -0.073917    0.050369  0.142240   \n",
       "3     chr4:883133:G:A  A       G  0.012734 -0.211942    0.096557  0.028164   \n",
       "4     chr4:883483:G:A  A       G  0.035357 -0.124157    0.058911  0.035071   \n",
       "..                ... ..     ...       ...       ...         ...       ...   \n",
       "392  chr4:1007954:T:C  T       C  0.279743 -0.090515    0.025919  0.000479   \n",
       "393  chr4:1008009:C:T  C       T  0.444961 -0.052139    0.023666  0.027587   \n",
       "394  chr4:1008209:C:T  T       C  0.036468  0.002058    0.059756  0.972480   \n",
       "395  chr4:1008337:C:T  C       T  0.279831 -0.090585    0.025912  0.000473   \n",
       "396  chr4:1008480:A:G  G       A  0.053420 -0.101047    0.049512  0.041266   \n",
       "\n",
       "     OBS_CT  \n",
       "0     20837  \n",
       "1     20885  \n",
       "2     20880  \n",
       "3     20811  \n",
       "4     20901  \n",
       "..      ...  \n",
       "392   18971  \n",
       "393   18814  \n",
       "394   20717  \n",
       "395   18965  \n",
       "396   20264  \n",
       "\n",
       "[397 rows x 8 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sumstats_export"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03438e16-0c14-44e8-94f8-cc2fdcd8a693",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Rename columns following COJO format\n",
    "sumstats_export = sumstats_export.rename(columns = {'ID':'SNP', 'OMITTED':'A2', 'A1_FREQ':'freq', 'LOG(OR)_SE':'se', 'P':'p', 'OBS_CT':'N'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "46643f0a-4350-43ec-bb08-8fc162186075",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>freq</th>\n",
       "      <th>b</th>\n",
       "      <th>se</th>\n",
       "      <th>p</th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.164467</td>\n",
       "      <td>0.033831</td>\n",
       "      <td>0.030398</td>\n",
       "      <td>0.265757</td>\n",
       "      <td>20837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4:882611:T:C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.051185</td>\n",
       "      <td>-0.070480</td>\n",
       "      <td>0.050102</td>\n",
       "      <td>0.159504</td>\n",
       "      <td>20885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4:883018:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.050599</td>\n",
       "      <td>-0.073917</td>\n",
       "      <td>0.050369</td>\n",
       "      <td>0.142240</td>\n",
       "      <td>20880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4:883133:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.012734</td>\n",
       "      <td>-0.211942</td>\n",
       "      <td>0.096557</td>\n",
       "      <td>0.028164</td>\n",
       "      <td>20811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4:883483:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.035357</td>\n",
       "      <td>-0.124157</td>\n",
       "      <td>0.058911</td>\n",
       "      <td>0.035071</td>\n",
       "      <td>20901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>chr4:1007954:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.279743</td>\n",
       "      <td>-0.090515</td>\n",
       "      <td>0.025919</td>\n",
       "      <td>0.000479</td>\n",
       "      <td>18971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393</th>\n",
       "      <td>chr4:1008009:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.444961</td>\n",
       "      <td>-0.052139</td>\n",
       "      <td>0.023666</td>\n",
       "      <td>0.027587</td>\n",
       "      <td>18814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394</th>\n",
       "      <td>chr4:1008209:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.036468</td>\n",
       "      <td>0.002058</td>\n",
       "      <td>0.059756</td>\n",
       "      <td>0.972480</td>\n",
       "      <td>20717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.279831</td>\n",
       "      <td>-0.090585</td>\n",
       "      <td>0.025912</td>\n",
       "      <td>0.000473</td>\n",
       "      <td>18965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>chr4:1008480:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.053420</td>\n",
       "      <td>-0.101047</td>\n",
       "      <td>0.049512</td>\n",
       "      <td>0.041266</td>\n",
       "      <td>20264</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>397 rows √ó 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP A1 A2      freq         b        se         p      N\n",
       "0     chr4:882530:C:T  T  C  0.164467  0.033831  0.030398  0.265757  20837\n",
       "1     chr4:882611:T:C  C  T  0.051185 -0.070480  0.050102  0.159504  20885\n",
       "2     chr4:883018:G:A  A  G  0.050599 -0.073917  0.050369  0.142240  20880\n",
       "3     chr4:883133:G:A  A  G  0.012734 -0.211942  0.096557  0.028164  20811\n",
       "4     chr4:883483:G:A  A  G  0.035357 -0.124157  0.058911  0.035071  20901\n",
       "..                ... .. ..       ...       ...       ...       ...    ...\n",
       "392  chr4:1007954:T:C  T  C  0.279743 -0.090515  0.025919  0.000479  18971\n",
       "393  chr4:1008009:C:T  C  T  0.444961 -0.052139  0.023666  0.027587  18814\n",
       "394  chr4:1008209:C:T  T  C  0.036468  0.002058  0.059756  0.972480  20717\n",
       "395  chr4:1008337:C:T  C  T  0.279831 -0.090585  0.025912  0.000473  18965\n",
       "396  chr4:1008480:A:G  G  A  0.053420 -0.101047  0.049512  0.041266  20264\n",
       "\n",
       "[397 rows x 8 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sumstats_export"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5661a3d3-fc9d-454a-8966-28413c9e8327",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Export\n",
    "sumstats_export.to_csv(f'{WORK_DIR}/TMEM175_EUR/EUR.all_adj.sumstats.ma', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f5b3666",
   "metadata": {},
   "source": [
    "### Install GCTA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "00c788a1-627d-4944-ab1c-e33a4c3eae30",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-08-07 11:14:11--  https://yanglab.westlake.edu.cn/software/gcta/bin/gcta-1.94.1-linux-kernel-3-x86_64.zip\n",
      "Resolving yanglab.westlake.edu.cn (yanglab.westlake.edu.cn)... 124.160.108.195, 42.247.30.189, 2001:250:6413:1002:250:56ff:10:195\n",
      "Connecting to yanglab.westlake.edu.cn (yanglab.westlake.edu.cn)|124.160.108.195|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 14333133 (14M) [application/zip]\n",
      "Saving to: ‚Äògcta-1.94.1-linux-kernel-3-x86_64.zip‚Äô\n",
      "\n",
      "gcta-1.94.1-linux-k 100%[===================>]  13.67M   844KB/s    in 17s     \n",
      "\n",
      "2024-08-07 11:14:32 (824 KB/s) - ‚Äògcta-1.94.1-linux-kernel-3-x86_64.zip‚Äô saved [14333133/14333133]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! wget https://yanglab.westlake.edu.cn/software/gcta/bin/gcta-1.94.1-linux-kernel-3-x86_64.zip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "043b2e49-cc83-46ba-81a8-056813ce5eda",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Archive:  gcta-1.94.1-linux-kernel-3-x86_64.zip\n",
      "   creating: gcta-1.94.1-linux-kernel-3-x86_64/\n",
      "  inflating: __MACOSX/._gcta-1.94.1-linux-kernel-3-x86_64  \n",
      "  inflating: gcta-1.94.1-linux-kernel-3-x86_64/gcta64  \n",
      "  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._gcta64  \n",
      "  inflating: gcta-1.94.1-linux-kernel-3-x86_64/.DS_Store  \n",
      "  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._.DS_Store  \n",
      "  inflating: gcta-1.94.1-linux-kernel-3-x86_64/test.bim  \n",
      "  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._test.bim  \n",
      "  inflating: gcta-1.94.1-linux-kernel-3-x86_64/MIT_License.txt  \n",
      "  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._MIT_License.txt  \n",
      "  inflating: gcta-1.94.1-linux-kernel-3-x86_64/test.fam  \n",
      "  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._test.fam  \n",
      "  inflating: gcta-1.94.1-linux-kernel-3-x86_64/README.txt  \n",
      "  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._README.txt  \n",
      "  inflating: gcta-1.94.1-linux-kernel-3-x86_64/test.bed  \n",
      "  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._test.bed  \n",
      "  inflating: gcta-1.94.1-linux-kernel-3-x86_64/test.phen  \n",
      "  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._test.phen  \n"
     ]
    }
   ],
   "source": [
    "! unzip gcta-1.94.1-linux-kernel-3-x86_64.zip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cbac1270-5ff7-4db2-9bf2-ad7e397b271d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "! mv ~/workspace/ws_files/gcta-1.94.1-linux-kernel-3-x86_64 /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db0e57b4-9d75-48c7-b2f5-a73cad87dc68",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LICENSE\t\t\t\t   plink\t\t\t    prettify\n",
      "annovar\t\t\t\t   plink2\t\t\t    rvtests\n",
      "annovar.latest.tar.gz\t\t   plink2_linux_x86_64_latest.zip   toy.map\n",
      "gcta-1.94.1-linux-kernel-3-x86_64  plink_linux_x86_64_20190304.zip  toy.ped\n"
     ]
    }
   ],
   "source": [
    "! ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4246a7e0-b0f5-43a7-b95f-3a9b08a9fef8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: line 1: /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64: Permission denied\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64 --version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "22598760-e0a7-4e3b-bcfe-64fad5c2d1f0",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "! chmod u+x /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1110e4f9-7bab-4bd2-9441-4a94a15ccb62",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 11:22:48 UTC on Wed Aug 07 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;31mError: \u001b[0mthe --out option is missing.\n",
      "\u001b[0m\u001b[0mAn error occurs, please check the options or data\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64 --version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "390d7f06-93d0-41e6-b6a4-b35b0d674abb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 11:25:36 UTC on Wed Aug 07 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma\n",
      "--cojo-p 5e-08\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.fam].\n",
      "38839 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bim].\n",
      "9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 38839 individuals and 9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 397 SNPs (9107 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 397 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 397 SNPs: 5.20346 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "397 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 397 SNPs to select association signals ... (p cutoff = 5e-08; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Finally, 1 associated SNPs are selected.\n",
      "Performing joint analysis on all the 1 selected signals ...\n",
      "Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.COJO.jma.cojo] ...\n",
      "Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.COJO.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 396 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.COJO.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 11:25:42 UTC on Wed Aug 07 2024\n",
      "Overall computational time: 6.04 sec.\n"
     ]
    }
   ],
   "source": [
    "# Select multiple associated SNPs based on significance p-value 5e-08\n",
    "# Can change the p-value for significance if needed\n",
    "# Bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS\n",
    "\n",
    "! /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64 --bfile {WORK_DIR}/TMEM175_EUR/EUR_TMEM175 \\\n",
    "--maf 0.01 \\\n",
    "--cojo-file {WORK_DIR}/TMEM175_EUR/EUR.all_adj.sumstats.ma \\\n",
    "--cojo-p 5e-8 \\\n",
    "--cojo-slct \\\n",
    "--out {WORK_DIR}/TMEM175_EUR/EUR.all_adj.COJO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "84db68f5-0f9e-476c-8880-ac714ef0fb79",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chr\tSNP\tbp\trefA\tfreq\tb\tse\tp\tn\tfreq_geno\tbC\tbC_se\tpC\n",
      "4\tchr4:882530:C:T\t882530\tT\t0.164467\t0.0338312\t0.0303975\t0.265725\t20489.9\t0.164537\t-0.0279995\t0.0303977\t0.356994\n",
      "4\tchr4:882611:T:C\t882611\tC\t0.0511851\t-0.0704804\t0.0501022\t0.159507\t21340.5\t0.0504798\t-0.0330724\t0.0501033\t0.5092\n",
      "4\tchr4:883018:G:A\t883018\tA\t0.0505987\t-0.0739168\t0.0503694\t0.142242\t21346\t0.0499974\t-0.0364219\t0.0503708\t0.469634\n",
      "4\tchr4:883133:G:A\t883133\tA\t0.0127337\t-0.211942\t0.0965569\t0.0281642\t22193.8\t0.0132536\t-0.170603\t0.0965652\t0.0772761\n",
      "4\tchr4:883483:G:A\t883483\tA\t0.0353572\t-0.124157\t0.0589113\t0.0350719\t21976.2\t0.0366916\t-0.0786346\t0.0589159\t0.181977\n",
      "4\tchr4:883624:C:T\t883624\tT\t0.181816\t0.0947191\t0.0295052\t0.00132619\t20080.8\t0.179007\t-0.0154726\t0.029512\t0.600084\n",
      "4\tchr4:883837:G:A\t883837\tA\t0.400635\t-0.031951\t0.022935\t0.163586\t20597\t0.404776\t-0.028599\t0.0229355\t0.212423\n",
      "4\tchr4:883921:T:C\t883921\tC\t0.448883\t-0.0438296\t0.02274\t0.0539267\t20335.1\t0.452722\t-0.0401386\t0.0227415\t0.0775649\n",
      "4\tchr4:884131:C:T\t884131\tT\t0.0117174\t0.0247414\t0.104187\t0.812291\t20698.7\t0.0122093\t0.0706551\t0.104185\t0.497663\n"
     ]
    }
   ],
   "source": [
    "! head {WORK_DIR}/TMEM175_EUR/EUR.all_adj.COJO.cma.cojo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cb6ea87d-dbd9-4e25-8091-c2dca0a9cf91",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chr\tSNP\tbp\trefA\tfreq\tb\tse\tp\tn\tfreq_geno\tbJ\tbJ_se\tpJ\tLD_r\n",
      "4\tchr4:958159:T:C\t958159\tC\t0.203811\t0.258912\t0.0290424\t4.87943e-19\t18930.3\t0.198229\t0.258912\t0.0291025\t5.76097e-19\t0\n"
     ]
    }
   ],
   "source": [
    "! head {WORK_DIR}/TMEM175_EUR/EUR.all_adj.COJO.jma.cojo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ed8ea88-eabb-461f-89b5-57e1cc4fa4f8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SNP\tchr4:958159:T:C\t\n",
      "chr4:958159:T:C\t1\t\n"
     ]
    }
   ],
   "source": [
    "! head {WORK_DIR}/TMEM175_EUR/EUR.all_adj.COJO.ldr.cojo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "144e2490-d240-492d-9d1a-f3824696e2f1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 11:25:36 UTC on Wed Aug 07 2024.\n",
      "Hostname: jupyterlabvertexai20240708\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! head {WORK_DIR}/TMEM175_EUR/EUR.all_adj.COJO.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "259ef615-8e97-4538-accc-04b835dfbba8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 12:48:26 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma\n",
      "--cojo-p 0.000459\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.LDprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.fam].\n",
      "38839 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bim].\n",
      "9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 38839 individuals and 9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 397 SNPs (9107 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 397 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 397 SNPs: 5.20346 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "397 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 397 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Finally, 1 associated SNPs are selected.\n",
      "Performing joint analysis on all the 1 selected signals ...\n",
      "Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.LDprune.COJO.jma.cojo] ...\n",
      "Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.LDprune.COJO.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 396 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.LDprune.COJO.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 12:48:32 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 6.05 sec.\n"
     ]
    }
   ],
   "source": [
    "# Select multiple associated SNPs based on significance p-value 5e-08\n",
    "# Can change the p-value for significance if needed\n",
    "# Bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS\n",
    "\n",
    "! /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64 --bfile {WORK_DIR}/TMEM175_EUR/EUR_TMEM175 \\\n",
    "--maf 0.01 \\\n",
    "--cojo-file {WORK_DIR}/TMEM175_EUR/EUR.all_adj.sumstats.ma \\\n",
    "--cojo-p 4.59e-4 \\\n",
    "--cojo-slct \\\n",
    "--out {WORK_DIR}/TMEM175_EUR/EUR.all_adj.LDprune.COJO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e2bdfb2f-7e69-4222-b616-98e23aad57d1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 11:33:10 UTC on Wed Aug 07 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.fam].\n",
      "38839 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bim].\n",
      "9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 38839 individuals and 9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 397 SNPs (9107 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 397 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 397 SNPs: 5.20346 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "397 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 397 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 387 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 11:33:17 UTC on Wed Aug 07 2024\n",
      "Overall computational time: 7.85 sec.\n"
     ]
    }
   ],
   "source": [
    "# Or you can run COJO and select just the top 10 independent SNPs (can be below GWAS significance p-value)\n",
    "# Select top 10 SNPs\n",
    "! /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64 --bfile {WORK_DIR}/TMEM175_EUR/EUR_TMEM175 \\\n",
    "--maf 0.01 \\\n",
    "--cojo-file {WORK_DIR}/TMEM175_EUR/EUR.all_adj.sumstats.ma \\\n",
    "--cojo-top-SNPs 10 \\\n",
    "--out {WORK_DIR}/TMEM175_EUR/EUR.all_adj.top10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90f9be7c-83a3-4993-a219-014162e0db6e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chr\tSNP\tbp\trefA\tfreq\tb\tse\tp\tn\tfreq_geno\tbC\tbC_se\tpC\n",
      "4\tchr4:882530:C:T\t882530\tT\t0.164467\t0.0338312\t0.0303975\t0.265725\t20489.9\t0.164537\t-0.277937\t0.105335\t0.00832474\n",
      "4\tchr4:882611:T:C\t882611\tC\t0.0511851\t-0.0704804\t0.0501022\t0.159507\t21340.5\t0.0504798\t-0.122516\t0.0814926\t0.132737\n",
      "4\tchr4:883018:G:A\t883018\tA\t0.0505987\t-0.0739168\t0.0503694\t0.142242\t21346\t0.0499974\t-0.114817\t0.0418044\t0.00602305\n",
      "4\tchr4:883483:G:A\t883483\tA\t0.0353572\t-0.124157\t0.0589113\t0.0350719\t21976.2\t0.0366916\t0.177534\t0.0999408\t0.075669\n",
      "4\tchr4:883624:C:T\t883624\tT\t0.181816\t0.0947191\t0.0295052\t0.00132619\t20080.8\t0.179007\t-0.0724453\t0.0497149\t0.145057\n",
      "4\tchr4:883837:G:A\t883837\tA\t0.400635\t-0.031951\t0.022935\t0.163586\t20597\t0.404776\t0.282062\t0.0325328\t4.31799e-18\n",
      "4\tchr4:883921:T:C\t883921\tC\t0.448883\t-0.0438296\t0.02274\t0.0539267\t20335.1\t0.452722\t0.110611\t0.0509097\t0.0298032\n",
      "4\tchr4:884131:C:T\t884131\tT\t0.0117174\t0.0247414\t0.104187\t0.812291\t20698.7\t0.0122093\t-0.136355\t0.0806028\t0.0907057\n",
      "4\tchr4:884260:C:T\t884260\tT\t0.335987\t0.0328447\t0.024008\t0.17129\t20231.7\t0.337507\t0.151407\t0.0726496\t0.0371541\n"
     ]
    }
   ],
   "source": [
    "! head {WORK_DIR}/TMEM175_EUR/EUR.all_adj.top10.cma.cojo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e490c7c-cd01-488d-8ee5-11c1967c2b8a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chr\tSNP\tbp\trefA\tfreq\tb\tse\tp\tn\tfreq_geno\tbJ\tbJ_se\tpJ\tLD_r\n",
      "4\tchr4:883133:G:A\t883133\tA\t0.0127337\t-0.211942\t0.0965569\t0.0281642\t22193.8\t0.0132536\t-0.277937\t0.105335\t0.00832474\t-0.0174097\n",
      "4\tchr4:905058:T:C\t905058\tC\t0.0220098\t0.103396\t0.0776335\t0.182912\t20053.8\t0.0211291\t-0.122516\t0.0814926\t0.132737\t-0.0460043\n",
      "4\tchr4:931588:C:T\t931588\tT\t0.098972\t-0.14224\t0.0364747\t9.63155e-05\t21915.3\t0.0994334\t-0.114817\t0.0418044\t0.00602305\t0.372451\n",
      "4\tchr4:933370:G:A\t933370\tA\t0.0153177\t0.0255118\t0.0925403\t0.782792\t20143.3\t0.0149901\t0.177534\t0.0999408\t0.075669\t-0.0298242\n",
      "4\tchr4:939809:A:G\t939809\tG\t0.0536428\t-0.111124\t0.048938\t0.0231648\t21395.3\t0.0540681\t-0.0724453\t0.0497149\t0.145057\t-0.117727\n",
      "4\tchr4:958159:T:C\t958159\tC\t0.203811\t0.258912\t0.0290424\t4.87943e-19\t18930.3\t0.198229\t0.282062\t0.0325328\t4.31799e-18\t-0.121267\n",
      "4\tchr4:983473:C:T\t983473\tT\t0.0623297\t0.02634\t0.0460307\t0.567168\t21010.5\t0.063255\t0.110611\t0.0509097\t0.0298032\t-0.0387902\n",
      "4\tchr4:988557:C:T\t988557\tT\t0.022547\t0.0935541\t0.076603\t0.221978\t20117.6\t0.0216671\t-0.136355\t0.0806028\t0.0907057\t-0.0231418\n",
      "4\tchr4:1001112:T:G\t1001112\tG\t0.0262346\t0.119018\t0.0721329\t0.0989471\t19571.7\t0.0264005\t0.151407\t0.0726496\t0.0371541\t-0.0354378\n"
     ]
    }
   ],
   "source": [
    "! head {WORK_DIR}/TMEM175_EUR/EUR.all_adj.top10.jma.cojo"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ada1374a-d186-40a4-9e8b-24df94291a8f",
   "metadata": {},
   "source": [
    "### Use P-value threshold of 4.59e-4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60408d55-c614-4be0-8891-63902efe79ee",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:32:01 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma\n",
      "--cojo-p 0.000459\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam].\n",
      "646 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim].\n",
      "1987 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 646 individuals and 1987 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 405 SNPs (1582 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 405 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 405 SNPs: 6.34329 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "405 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 405 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:32:02 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.29 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:32:02 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.000459\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam].\n",
      "903 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim].\n",
      "1955 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 903 individuals and 1955 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 429 SNPs (1526 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 429 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 429 SNPs: 4.98286 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "429 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 429 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:32:02 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.37 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:32:02 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma\n",
      "--cojo-p 0.000459\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam].\n",
      "581 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim].\n",
      "1945 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 581 individuals and 1945 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 398 SNPs (1547 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 398 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 398 SNPs: 5.25039 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "398 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 398 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:32:03 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.30 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:32:03 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma\n",
      "--cojo-p 0.000459\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam].\n",
      "2655 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim].\n",
      "1285 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 2655 individuals and 1285 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 416 SNPs (869 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 416 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 416 SNPs: 7.98751 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "416 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 416 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Finally, 2 associated SNPs are selected.\n",
      "Performing joint analysis on all the 2 selected signals ...\n",
      "Saving the 2 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.ldprune.COJO.jma.cojo] ...\n",
      "Saving the LD structure of 2 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.ldprune.COJO.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 414 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.ldprune.COJO.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:32:04 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.74 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:32:04 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma\n",
      "--cojo-p 0.000459\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam].\n",
      "2643 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim].\n",
      "3853 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 2643 individuals and 3853 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 591 SNPs (3262 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 591 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 591 SNPs: 7.37522 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "591 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 591 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:32:04 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.53 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:32:05 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.000459\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam].\n",
      "5167 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim].\n",
      "3171 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 5167 individuals and 3171 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 377 SNPs (2794 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 377 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 377 SNPs: 5.163 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "377 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 377 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:32:05 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.54 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:32:05 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma\n",
      "--cojo-p 0.000459\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam].\n",
      "114 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim].\n",
      "605 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 114 individuals and 605 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 405 SNPs (200 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 405 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 405 SNPs: 7.93715 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "405 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 405 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:32:06 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.23 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:32:06 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma\n",
      "--cojo-p 0.000459\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam].\n",
      "1111 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim].\n",
      "3449 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 1111 individuals and 3449 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 588 SNPs (2861 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 588 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 588 SNPs: 7.51272 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "588 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 588 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:32:06 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.18 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:32:06 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma\n",
      "--cojo-p 0.000459\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam].\n",
      "851 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim].\n",
      "3090 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 851 individuals and 3090 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 547 SNPs (2543 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 548 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 548 SNPs: 5.36028 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "547 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 547 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:32:06 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.32 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:32:07 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.000459\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam].\n",
      "635 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim].\n",
      "1728 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 635 individuals and 1728 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 477 SNPs (1251 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 477 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 477 SNPs: 5.50087 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "477 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 477 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:32:07 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.27 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:32:07 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma\n",
      "--cojo-p 0.000459\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam].\n",
      "38839 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim].\n",
      "9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 38839 individuals and 9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 397 SNPs (9107 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 397 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 397 SNPs: 5.20346 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "397 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 397 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Finally, 1 associated SNPs are selected.\n",
      "Performing joint analysis on all the 1 selected signals ...\n",
      "Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.ldprune.COJO.jma.cojo] ...\n",
      "Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.ldprune.COJO.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 396 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.ldprune.COJO.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:32:13 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 6.33 sec.\n"
     ]
    }
   ],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    \n",
    "    # Select multiple associated SNPs based on LD pruning p-value 0.00238\n",
    "    # Can change the p-value for significance if needed\n",
    "    # Bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS\n",
    "    ! /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_TMEM175 \\\n",
    "    --maf 0.01 \\\n",
    "    --cojo-file {WORK_DIR}/{ancestry}.all_adj.sumstats.ma \\\n",
    "    --cojo-p 4.59e-4 \\\n",
    "    --cojo-slct \\\n",
    "    --out {WORK_DIR}/{ancestry}.all_adj.ldprune.COJO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d8d8d4c-42e6-4a3e-b8b4-7c20800c094f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/TMEM175/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5a2ae318-13ad-4017-84a9-52b0341a16c6",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Prepare file for COJO\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    # Read in summary statistics\n",
    "    sumstats = pd.read_csv(f'{WORK_DIR}/TMEM175_{ancestry}/{ancestry}.all_adj.csv')\n",
    "    \n",
    "    # Format summary statistics for GCTA-COJO\n",
    "    # First get the log odds ratio - this is required for COJO\n",
    "    # 1) For a case-control study, the effect size should be log (odds ratio) with its corresponding standard error.\n",
    "    sumstats_formatted = sumstats.copy()\n",
    "    sumstats_formatted['b'] = np.log(sumstats_formatted['OR'])\n",
    "    \n",
    "    # Now select just the necessary columns for COJO\n",
    "    sumstats_export = sumstats_formatted[['ID', 'A1', 'OMITTED', 'A1_FREQ', 'b', 'LOG(OR)_SE', 'P', 'OBS_CT']].copy()\n",
    "    \n",
    "    # Rename columns following COJO format\n",
    "    sumstats_export = sumstats_export.rename(columns = {'ID':'SNP', 'OMITTED':'A2', 'A1_FREQ':'freq', 'LOG(OR)_SE':'se', 'P':'p', 'OBS_CT':'N'})\n",
    "    \n",
    "    # Export\n",
    "    sumstats_export.to_csv(f'{WORK_DIR}/TMEM175_{ancestry}/{ancestry}.all_adj.sumstats.ma', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db3be3bf-a31c-4e49-b358-4f68e0440015",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:04:39 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam].\n",
      "646 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim].\n",
      "1987 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 646 individuals and 1987 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 405 SNPs (1582 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 405 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 405 SNPs: 6.34329 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "405 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 405 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 395 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:04:40 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.84 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:04:40 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam].\n",
      "903 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim].\n",
      "1955 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 903 individuals and 1955 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 429 SNPs (1526 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 429 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 429 SNPs: 4.98286 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "429 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 429 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 419 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:04:41 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.92 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:04:41 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam].\n",
      "581 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim].\n",
      "1945 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 581 individuals and 1945 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 398 SNPs (1547 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 398 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 398 SNPs: 5.25039 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "398 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 398 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 388 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:04:42 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.96 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:04:43 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam].\n",
      "2655 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim].\n",
      "1285 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 2655 individuals and 1285 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 416 SNPs (869 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 416 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 416 SNPs: 7.98751 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "416 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 416 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 406 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:04:44 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 1.06 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:04:44 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam].\n",
      "2643 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim].\n",
      "3853 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 2643 individuals and 3853 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 591 SNPs (3262 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 591 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 591 SNPs: 7.37522 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "591 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 591 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 581 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:04:45 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 1.20 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:04:45 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam].\n",
      "5167 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim].\n",
      "3171 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 5167 individuals and 3171 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 377 SNPs (2794 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 377 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 377 SNPs: 5.163 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "377 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 377 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 367 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:04:47 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 1.43 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:04:47 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam].\n",
      "114 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim].\n",
      "605 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 114 individuals and 605 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 405 SNPs (200 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 405 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 405 SNPs: 7.93715 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "405 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 405 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 395 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:04:48 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.74 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:04:48 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam].\n",
      "1111 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim].\n",
      "3449 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 1111 individuals and 3449 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 588 SNPs (2861 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 588 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 588 SNPs: 7.51272 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "588 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 588 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 578 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:04:49 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.93 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:04:49 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam].\n",
      "851 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim].\n",
      "3090 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 851 individuals and 3090 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 547 SNPs (2543 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 548 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 548 SNPs: 5.36028 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "547 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 547 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 537 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:04:50 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.86 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:04:50 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam].\n",
      "635 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim].\n",
      "1728 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 635 individuals and 1728 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 477 SNPs (1251 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 477 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 477 SNPs: 5.50087 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "477 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 477 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 467 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:04:51 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.68 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:04:51 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam].\n",
      "38839 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim].\n",
      "9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 38839 individuals and 9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 397 SNPs (9107 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 397 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 397 SNPs: 5.20346 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "397 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 397 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 387 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:05:00 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 8.86 sec.\n"
     ]
    }
   ],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    \n",
    "    # Select multiple associated SNPs based on significance p-value 5e-08\n",
    "    # Can change the p-value for significance if needed\n",
    "    # Bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS\n",
    "    ! /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_TMEM175 \\\n",
    "    --maf 0.01 \\\n",
    "    --cojo-file {WORK_DIR}/{ancestry}.all_adj.sumstats.ma \\\n",
    "    --cojo-top-SNPs 10 \\\n",
    "    --out {WORK_DIR}/{ancestry}.all_adj.top10"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0b081052-1e4b-4a64-8193-9ad339846750",
   "metadata": {},
   "source": [
    "### Change format of association file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5dc5cebc-5b19-44d6-9e10-12a2e57110dd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AMR\n",
      "WORKING ON: CAS\n",
      "WORKING ON: MDE\n",
      "WORKING ON: AJ\n",
      "WORKING ON: AFR\n",
      "WORKING ON: EAS\n",
      "WORKING ON: FIN\n",
      "WORKING ON: AAC\n",
      "WORKING ON: CAH\n",
      "WORKING ON: SAS\n",
      "WORKING ON: EUR\n"
     ]
    }
   ],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    WORK_DIR = \"~/workspace/ws_files/TMEM175/\"\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    df = pd.read_csv(f'{WORK_DIR}/TMEM175_{ancestry}/{ancestry}.all_adj.csv')\n",
    "    \n",
    "    # First just select relevant columns: chromosome, bp position and p-value\n",
    "    export_ldassoc = df[['# CHROM', 'POS', 'P']].copy()\n",
    "    \n",
    "    # Rename the # CHROM column to remove the hashtag as I think this might be confusing LDassoc\n",
    "    export_ldassoc = export_ldassoc.rename(columns={'# CHROM': 'CHROM'}) \n",
    "    \n",
    "    # Then export as a tab-separated, not comma-separated file\n",
    "    export_ldassoc.to_csv(f'{WORK_DIR}/TMEM175_{ancestry}/{ancestry}.all_adj.formatted.tab', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd3f5bd2-61b7-4506-a86b-660b89c43cf7",
   "metadata": {},
   "source": [
    "### Validate GLM output file in EAS ancestry again"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "27d15018-1c04-43ec-9d14-4dd87f789bdc",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_EAS'\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--pfile {REL7_PATH}/imputed_genotypes/EAS/chr4_EAS_release7_vwb \\\n",
    "--chr 4 \\\n",
    "--from-bp 882387 \\\n",
    "--to-bp 1008656 \\\n",
    "--make-bed \\\n",
    "--out {WORK_DIR}/EAS_TMEM175"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d202a6b-4743-4181-aafa-ae44334e73d0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "3171 variants loaded from .bim file.\n",
      "5167 people (3278 males, 1889 females) loaded from .fam.\n",
      "5123 phenotype values loaded from .fam.\n",
      "--keep: 5167 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5167 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.994312.\n",
      "2794 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "377 variants and 5167 people pass filters and QC.\n",
      "Among remaining phenotypes, 2662 are cases and 2461 are controls.  (44\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all.assoc ...\n",
      "done.\n"
     ]
    }
   ],
   "source": [
    "# ASSOC analysis\n",
    "\n",
    "WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_EAS'\n",
    "    \n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/EAS_TMEM175 \\\n",
    "--keep {WORK_DIR}/EAS.samplestoKeep \\\n",
    "--assoc \\\n",
    "--allow-no-sex \\\n",
    "--ci 0.95 \\\n",
    "--maf 0.01 \\\n",
    "--out {WORK_DIR}/EAS_TMEM175.all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "25ef9e7c-cf2c-4736-b9fa-39c9e9bcd612",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>882530</td>\n",
       "      <td>T</td>\n",
       "      <td>0.29880</td>\n",
       "      <td>0.28300</td>\n",
       "      <td>C</td>\n",
       "      <td>3.0680</td>\n",
       "      <td>0.07984</td>\n",
       "      <td>1.0800</td>\n",
       "      <td>0.04382</td>\n",
       "      <td>0.9909</td>\n",
       "      <td>1.1770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>chr4:882611:T:C</td>\n",
       "      <td>882611</td>\n",
       "      <td>C</td>\n",
       "      <td>0.06649</td>\n",
       "      <td>0.07835</td>\n",
       "      <td>T</td>\n",
       "      <td>5.3640</td>\n",
       "      <td>0.02056</td>\n",
       "      <td>0.8379</td>\n",
       "      <td>0.07645</td>\n",
       "      <td>0.7213</td>\n",
       "      <td>0.9733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>chr4:883018:G:A</td>\n",
       "      <td>883018</td>\n",
       "      <td>A</td>\n",
       "      <td>0.06649</td>\n",
       "      <td>0.07814</td>\n",
       "      <td>G</td>\n",
       "      <td>5.1880</td>\n",
       "      <td>0.02275</td>\n",
       "      <td>0.8403</td>\n",
       "      <td>0.07649</td>\n",
       "      <td>0.7233</td>\n",
       "      <td>0.9762</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>chr4:883483:G:A</td>\n",
       "      <td>883483</td>\n",
       "      <td>A</td>\n",
       "      <td>0.07547</td>\n",
       "      <td>0.06904</td>\n",
       "      <td>G</td>\n",
       "      <td>1.5690</td>\n",
       "      <td>0.21040</td>\n",
       "      <td>1.1010</td>\n",
       "      <td>0.07663</td>\n",
       "      <td>0.9472</td>\n",
       "      <td>1.2790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>chr4:883624:C:T</td>\n",
       "      <td>883624</td>\n",
       "      <td>T</td>\n",
       "      <td>0.40600</td>\n",
       "      <td>0.41950</td>\n",
       "      <td>C</td>\n",
       "      <td>1.8970</td>\n",
       "      <td>0.16840</td>\n",
       "      <td>0.9456</td>\n",
       "      <td>0.04061</td>\n",
       "      <td>0.8733</td>\n",
       "      <td>1.0240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>372</th>\n",
       "      <td>4</td>\n",
       "      <td>chr4:1007954:T:C</td>\n",
       "      <td>1007954</td>\n",
       "      <td>T</td>\n",
       "      <td>0.42810</td>\n",
       "      <td>0.42160</td>\n",
       "      <td>C</td>\n",
       "      <td>0.4384</td>\n",
       "      <td>0.50790</td>\n",
       "      <td>1.0270</td>\n",
       "      <td>0.04030</td>\n",
       "      <td>0.9490</td>\n",
       "      <td>1.1110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373</th>\n",
       "      <td>4</td>\n",
       "      <td>chr4:1008009:C:T</td>\n",
       "      <td>1008009</td>\n",
       "      <td>T</td>\n",
       "      <td>0.37580</td>\n",
       "      <td>0.37980</td>\n",
       "      <td>C</td>\n",
       "      <td>0.1604</td>\n",
       "      <td>0.68880</td>\n",
       "      <td>0.9831</td>\n",
       "      <td>0.04266</td>\n",
       "      <td>0.9042</td>\n",
       "      <td>1.0690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>374</th>\n",
       "      <td>4</td>\n",
       "      <td>chr4:1008064:G:C</td>\n",
       "      <td>1008064</td>\n",
       "      <td>C</td>\n",
       "      <td>0.02647</td>\n",
       "      <td>0.02219</td>\n",
       "      <td>G</td>\n",
       "      <td>1.9000</td>\n",
       "      <td>0.16810</td>\n",
       "      <td>1.1980</td>\n",
       "      <td>0.13130</td>\n",
       "      <td>0.9263</td>\n",
       "      <td>1.5500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>375</th>\n",
       "      <td>4</td>\n",
       "      <td>chr4:1008209:C:T</td>\n",
       "      <td>1008209</td>\n",
       "      <td>T</td>\n",
       "      <td>0.10880</td>\n",
       "      <td>0.11340</td>\n",
       "      <td>C</td>\n",
       "      <td>0.5450</td>\n",
       "      <td>0.46040</td>\n",
       "      <td>0.9539</td>\n",
       "      <td>0.06397</td>\n",
       "      <td>0.8415</td>\n",
       "      <td>1.0810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>376</th>\n",
       "      <td>4</td>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>1008337</td>\n",
       "      <td>C</td>\n",
       "      <td>0.42970</td>\n",
       "      <td>0.42260</td>\n",
       "      <td>T</td>\n",
       "      <td>0.5250</td>\n",
       "      <td>0.46870</td>\n",
       "      <td>1.0290</td>\n",
       "      <td>0.04012</td>\n",
       "      <td>0.9516</td>\n",
       "      <td>1.1140</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>377 rows √ó 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHR               SNP       BP A1      F_A      F_U A2   CHISQ        P  \\\n",
       "0      4   chr4:882530:C:T   882530  T  0.29880  0.28300  C  3.0680  0.07984   \n",
       "1      4   chr4:882611:T:C   882611  C  0.06649  0.07835  T  5.3640  0.02056   \n",
       "2      4   chr4:883018:G:A   883018  A  0.06649  0.07814  G  5.1880  0.02275   \n",
       "3      4   chr4:883483:G:A   883483  A  0.07547  0.06904  G  1.5690  0.21040   \n",
       "4      4   chr4:883624:C:T   883624  T  0.40600  0.41950  C  1.8970  0.16840   \n",
       "..   ...               ...      ... ..      ...      ... ..     ...      ...   \n",
       "372    4  chr4:1007954:T:C  1007954  T  0.42810  0.42160  C  0.4384  0.50790   \n",
       "373    4  chr4:1008009:C:T  1008009  T  0.37580  0.37980  C  0.1604  0.68880   \n",
       "374    4  chr4:1008064:G:C  1008064  C  0.02647  0.02219  G  1.9000  0.16810   \n",
       "375    4  chr4:1008209:C:T  1008209  T  0.10880  0.11340  C  0.5450  0.46040   \n",
       "376    4  chr4:1008337:C:T  1008337  C  0.42970  0.42260  T  0.5250  0.46870   \n",
       "\n",
       "         OR       SE     L95     U95  \n",
       "0    1.0800  0.04382  0.9909  1.1770  \n",
       "1    0.8379  0.07645  0.7213  0.9733  \n",
       "2    0.8403  0.07649  0.7233  0.9762  \n",
       "3    1.1010  0.07663  0.9472  1.2790  \n",
       "4    0.9456  0.04061  0.8733  1.0240  \n",
       "..      ...      ...     ...     ...  \n",
       "372  1.0270  0.04030  0.9490  1.1110  \n",
       "373  0.9831  0.04266  0.9042  1.0690  \n",
       "374  1.1980  0.13130  0.9263  1.5500  \n",
       "375  0.9539  0.06397  0.8415  1.0810  \n",
       "376  1.0290  0.04012  0.9516  1.1140  \n",
       "\n",
       "[377 rows x 13 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Look at assoc results\n",
    "WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_EAS'\n",
    "freq = pd.read_csv(f'{WORK_DIR}/EAS_TMEM175.all.assoc', sep='\\s+')\n",
    "freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b207f773-84b8-41cc-8fc0-5050029d4e54",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Save FREQ to csv\n",
    "freq.to_csv(f'{WORK_DIR}/EAS.all_nonadj.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ccfeb4f-aa13-4b44-87cf-21727d6d1b54",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj\n",
      "\n",
      "Start time: Thu Aug 29 12:42:49 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50486 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam.\n",
      "3171 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim.\n",
      "1 binary phenotype loaded (2662 cases, 2461 controls).\n",
      "--keep: 5167 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) remaining after main\n",
      "filters.\n",
      "2662 cases and 2461 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2794 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "377 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Thu Aug 29 12:42:49 2024\n"
     ]
    }
   ],
   "source": [
    "# GLM analysis\n",
    "\n",
    "WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_EAS'\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/EAS_TMEM175 \\\n",
    "--keep {WORK_DIR}/EAS.samplestoKeep \\\n",
    "--allow-no-sex \\\n",
    "--maf 0.01 \\\n",
    "--ci 0.95 \\\n",
    "--glm \\\n",
    "--covar {WORK_DIR}/EAS_covariate_file.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--covar-variance-standardize \\\n",
    "--neg9-pheno-really-missing \\\n",
    "--out {WORK_DIR}/EAS_TMEM175.all_adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a3e6154-6528-4a51-92c0-4b24983a2431",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_3382/2767392481.py:5: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>PROVISIONAL_REF?</th>\n",
       "      <th>A1</th>\n",
       "      <th>OMITTED</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>FIRTH?</th>\n",
       "      <th>TEST</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>OR</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>ERRCODE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.278626</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2489</td>\n",
       "      <td>1.079170</td>\n",
       "      <td>0.071422</td>\n",
       "      <td>0.938201</td>\n",
       "      <td>1.241320</td>\n",
       "      <td>1.066800</td>\n",
       "      <td>2.860620e-01</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.278626</td>\n",
       "      <td>N</td>\n",
       "      <td>SEX</td>\n",
       "      <td>2489</td>\n",
       "      <td>0.804336</td>\n",
       "      <td>0.044091</td>\n",
       "      <td>0.737746</td>\n",
       "      <td>0.876936</td>\n",
       "      <td>-4.938340</td>\n",
       "      <td>7.878890e-07</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.278626</td>\n",
       "      <td>N</td>\n",
       "      <td>AGE</td>\n",
       "      <td>2489</td>\n",
       "      <td>1.510690</td>\n",
       "      <td>0.047695</td>\n",
       "      <td>1.375870</td>\n",
       "      <td>1.658720</td>\n",
       "      <td>8.650160</td>\n",
       "      <td>5.142550e-18</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.278626</td>\n",
       "      <td>N</td>\n",
       "      <td>PC1</td>\n",
       "      <td>2489</td>\n",
       "      <td>0.978250</td>\n",
       "      <td>0.044623</td>\n",
       "      <td>0.896327</td>\n",
       "      <td>1.067660</td>\n",
       "      <td>-0.492802</td>\n",
       "      <td>6.221530e-01</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.278626</td>\n",
       "      <td>N</td>\n",
       "      <td>PC2</td>\n",
       "      <td>2489</td>\n",
       "      <td>0.997227</td>\n",
       "      <td>0.116869</td>\n",
       "      <td>0.793077</td>\n",
       "      <td>1.253930</td>\n",
       "      <td>-0.023759</td>\n",
       "      <td>9.810450e-01</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3011</th>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.419561</td>\n",
       "      <td>N</td>\n",
       "      <td>PC1</td>\n",
       "      <td>2505</td>\n",
       "      <td>0.974019</td>\n",
       "      <td>0.044508</td>\n",
       "      <td>0.892651</td>\n",
       "      <td>1.062800</td>\n",
       "      <td>-0.591460</td>\n",
       "      <td>5.542120e-01</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3012</th>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.419561</td>\n",
       "      <td>N</td>\n",
       "      <td>PC2</td>\n",
       "      <td>2505</td>\n",
       "      <td>0.980706</td>\n",
       "      <td>0.117453</td>\n",
       "      <td>0.779045</td>\n",
       "      <td>1.234570</td>\n",
       "      <td>-0.165874</td>\n",
       "      <td>8.682560e-01</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3013</th>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.419561</td>\n",
       "      <td>N</td>\n",
       "      <td>PC3</td>\n",
       "      <td>2505</td>\n",
       "      <td>0.679168</td>\n",
       "      <td>0.102293</td>\n",
       "      <td>0.555782</td>\n",
       "      <td>0.829945</td>\n",
       "      <td>-3.782130</td>\n",
       "      <td>1.554910e-04</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3014</th>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.419561</td>\n",
       "      <td>N</td>\n",
       "      <td>PC4</td>\n",
       "      <td>2505</td>\n",
       "      <td>0.562607</td>\n",
       "      <td>0.066139</td>\n",
       "      <td>0.494206</td>\n",
       "      <td>0.640475</td>\n",
       "      <td>-8.696510</td>\n",
       "      <td>3.422460e-18</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3015</th>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.419561</td>\n",
       "      <td>N</td>\n",
       "      <td>PC5</td>\n",
       "      <td>2505</td>\n",
       "      <td>1.009840</td>\n",
       "      <td>0.045912</td>\n",
       "      <td>0.922934</td>\n",
       "      <td>1.104920</td>\n",
       "      <td>0.213190</td>\n",
       "      <td>8.311790e-01</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3016 rows √ó 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #CHROM      POS                ID REF ALT PROVISIONAL_REF? A1 OMITTED  \\\n",
       "0          4   882530   chr4:882530:C:T   C   T                Y  T       C   \n",
       "1          4   882530   chr4:882530:C:T   C   T                Y  T       C   \n",
       "2          4   882530   chr4:882530:C:T   C   T                Y  T       C   \n",
       "3          4   882530   chr4:882530:C:T   C   T                Y  T       C   \n",
       "4          4   882530   chr4:882530:C:T   C   T                Y  T       C   \n",
       "...      ...      ...               ...  ..  ..              ... ..     ...   \n",
       "3011       4  1008337  chr4:1008337:C:T   C   T                Y  C       T   \n",
       "3012       4  1008337  chr4:1008337:C:T   C   T                Y  C       T   \n",
       "3013       4  1008337  chr4:1008337:C:T   C   T                Y  C       T   \n",
       "3014       4  1008337  chr4:1008337:C:T   C   T                Y  C       T   \n",
       "3015       4  1008337  chr4:1008337:C:T   C   T                Y  C       T   \n",
       "\n",
       "       A1_FREQ FIRTH? TEST  OBS_CT        OR  LOG(OR)_SE       L95       U95  \\\n",
       "0     0.278626      N  ADD    2489  1.079170    0.071422  0.938201  1.241320   \n",
       "1     0.278626      N  SEX    2489  0.804336    0.044091  0.737746  0.876936   \n",
       "2     0.278626      N  AGE    2489  1.510690    0.047695  1.375870  1.658720   \n",
       "3     0.278626      N  PC1    2489  0.978250    0.044623  0.896327  1.067660   \n",
       "4     0.278626      N  PC2    2489  0.997227    0.116869  0.793077  1.253930   \n",
       "...        ...    ...  ...     ...       ...         ...       ...       ...   \n",
       "3011  0.419561      N  PC1    2505  0.974019    0.044508  0.892651  1.062800   \n",
       "3012  0.419561      N  PC2    2505  0.980706    0.117453  0.779045  1.234570   \n",
       "3013  0.419561      N  PC3    2505  0.679168    0.102293  0.555782  0.829945   \n",
       "3014  0.419561      N  PC4    2505  0.562607    0.066139  0.494206  0.640475   \n",
       "3015  0.419561      N  PC5    2505  1.009840    0.045912  0.922934  1.104920   \n",
       "\n",
       "        Z_STAT             P ERRCODE  \n",
       "0     1.066800  2.860620e-01       .  \n",
       "1    -4.938340  7.878890e-07       .  \n",
       "2     8.650160  5.142550e-18       .  \n",
       "3    -0.492802  6.221530e-01       .  \n",
       "4    -0.023759  9.810450e-01       .  \n",
       "...        ...           ...     ...  \n",
       "3011 -0.591460  5.542120e-01       .  \n",
       "3012 -0.165874  8.682560e-01       .  \n",
       "3013 -3.782130  1.554910e-04       .  \n",
       "3014 -8.696510  3.422460e-18       .  \n",
       "3015  0.213190  8.311790e-01       .  \n",
       "\n",
       "[3016 rows x 19 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read output file of GLM\n",
    "\n",
    "# Read in plink GLM results\n",
    "WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_EAS'\n",
    "assoc = pd.read_csv(f'{WORK_DIR}/EAS_TMEM175.all_adj.PHENO1.glm.logistic.hybrid', delim_whitespace=True)\n",
    "assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5db2b5e3-f4ef-4c47-873c-5624ee91bba0",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter for additive test only - this is the variant results\n",
    "assoc_add = assoc[assoc['TEST']==\"ADD\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aff36f76-a89a-40c2-9f1a-3de16a25ba2a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Save assoc_add to csv\n",
    "assoc_add.to_csv(f'{WORK_DIR}/EAS.all_adj.csv')"
   ]
  }
 ],
 "metadata": {
  "environment": {
   "kernel": "python3",
   "name": ".m125",
   "type": "gcloud",
   "uri": "us-docker.pkg.dev/deeplearning-platform-release/gcr.io/:m125"
  },
  "kernelspec": {
   "display_name": "Python 3 (Local)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
