{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fe7fbe3d-77c1-46a4-be72-30e47b94a50b",
   "metadata": {
    "tags": []
   },
   "source": [
    "SCARB2 - Single gene analysis in GP2 Neurobooster genotyping data (all ancestries)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6357db9f-081c-4762-a59e-6afe6ec5d1ea",
   "metadata": {
    "tags": []
   },
   "source": [
    "Project: GP2 SCARB2\n",
    "\n",
    "Version: Python/3.10.15, R/4.3.3"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d18fbe46-bd97-4cc9-953c-cf67b0e57bef",
   "metadata": {},
   "source": [
    "Notebook Overview\n",
    "\n",
    "1. Description\n",
    "Loading Python libraries\n",
    "Set paths\n",
    "Make working directory\n",
    "\n",
    "2. Installing packages\n",
    "\n",
    "3. LD pruning(TMEM175/SCARB2/CTSB)\n",
    "\n",
    "4. Create a covariate file with GP2 data\n",
    "\n",
    "5. Annotation of the gene SCARB2\n",
    "\n",
    "6. Association analysis to compare allele frequencies between cases and controls\n",
    "\n",
    "7. GLM analysis adjusting for gender, age, PC1-5\n",
    "\n",
    "8. Burden test(SkatO, Skat, cmc,zeggini,mb,fp,cmcWald)\n",
    "\n",
    "9. Conditional analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c695fa50-4fd9-46ae-923b-a77f78bb2769",
   "metadata": {},
   "source": [
    "Loading Python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "56c961b1-1aa3-4aa3-9faf-78233c643aec",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Use pathlib for file path manipulation\n",
    "import pathlib\n",
    "\n",
    "# Install numpy\n",
    "import numpy as np\n",
    "\n",
    "# Install Pandas for tabular data\n",
    "import pandas as pd\n",
    "\n",
    "# Install plotnine: a ggplot2-compatible Python plotting package\n",
    "from plotnine import *\n",
    "\n",
    "# Always show all columns in a Pandas DataFrame\n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e5ac0f8f-b73f-472f-902a-c7f836af9ca4",
   "metadata": {},
   "source": [
    "Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "34bf7f59-d019-4b85-ba00-ccf42d208bd9",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "REL7_PATH = pathlib.Path(pathlib.Path.home(), 'workspace/gp2_tier2_eu_release7_30042024')\n",
    "!ls -hal {REL7_PATH}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "957183b5-ff79-4272-8269-023876f780ce",
   "metadata": {},
   "source": [
    "Make working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "20088528-1426-4736-b645-5f157e97eaee",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/home/jupyter/workspace/ws_files/SCARB2’: File exists\n"
     ]
    }
   ],
   "source": [
    "! mkdir ~/workspace/ws_files/SCARB2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1a1f447d-76a8-43a4-b168-4b09e186f09f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/SCARB2/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3961ff39-3cae-4082-9459-fe020845a2fd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LICENSE\t\t\t\t   plink\t\t\t    prettify\n",
      "annovar\t\t\t\t   plink2\t\t\t    rvtests\n",
      "annovar.latest.tar.gz\t\t   plink2_linux_x86_64_latest.zip   toy.map\n",
      "gcta-1.94.1-linux-kernel-3-x86_64  plink_linux_x86_64_20190304.zip  toy.ped\n"
     ]
    }
   ],
   "source": [
    "# make sure all tools installed\n",
    "! ls /home/jupyter/tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "bd9a9e05-45cf-493b-ba48-75d27fc5502c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# give permission\n",
    "\n",
    "# chmod to make sure you have permission to run the program\n",
    "! chmod u+x /home/jupyter/tools/plink\n",
    "! chmod u+x /home/jupyter/tools/plink2\n",
    "! chmod 777 /home/jupyter/tools/rvtests/executable/rvtest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "33cc1871-7fcb-42bd-959c-5df43e28e0eb",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "# making working directory\n",
    "#Loop over all the ancestries\n",
    "for ancestry in {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'} ;\n",
    "do\n",
    "\n",
    "#Make a folder for each ancestry\n",
    "mkdir ~/workspace/ws_files/SCARB2/SCARB2_\"$ancestry\"\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "afd021ff-34bb-4b84-aaf3-b39676d36e8c",
   "metadata": {},
   "source": [
    "LD pruning(in EUR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "84c9b3d7-61fc-47db-86dc-2a784a2a82ce",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ce10ade9-8cc3-434a-866c-e2976240b8f2",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to TMEM175_UNIMPUTED.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files//TMEM175/TMEM175_EUR/EUR_TMEM175\n",
      "  --exclude /home/jupyter/workspace/ws_files//exclusion_regions_hg38.txt\n",
      "  --geno 0.05\n",
      "  --hwe 1E-6\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out TMEM175_UNIMPUTED\n",
      "\n",
      "Start time: Tue Aug 20 12:38:19 2024\n",
      "52223 MiB RAM detected, ~50840 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files//TMEM175/TMEM175_EUR/EUR_TMEM175.fam.\n",
      "9504 variants loaded from\n",
      "/home/jupyter/workspace/ws_files//TMEM175/TMEM175_EUR/EUR_TMEM175.bim.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "--exclude: 9504 variants remaining.\n",
      "Calculating allele frequencies... done.\n",
      "--geno: 4576 variants removed due to missing genotype data.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test (founders only).\n",
      "4562 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "366 variants remaining after main filters.\n",
      "Writing TMEM175_UNIMPUTED.fam ... done.\n",
      "Writing TMEM175_UNIMPUTED.bim ... done.\n",
      "Writing TMEM175_UNIMPUTED.bed ... done.\n",
      "End time: Tue Aug 20 12:38:20 2024\n"
     ]
    }
   ],
   "source": [
    "# Make sure to use high-quality SNPs\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/TMEM175/TMEM175_EUR/EUR_TMEM175 \\\n",
    "--maf 0.01 \\\n",
    "--geno 0.05 \\\n",
    "--hwe 1E-6 \\\n",
    "--make-bed \\\n",
    "--exclude {WORK_DIR}/exclusion_regions_hg38.txt \\\n",
    "--out TMEM175_UNIMPUTED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "29a0c452-74b3-43f2-9a98-11c36c620c1e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files//LDpruning/prune_TMEM175.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files//LDpruning/TMEM175_UNIMPUTED\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out /home/jupyter/workspace/ws_files//LDpruning/prune_TMEM175\n",
      "\n",
      "Start time: Tue Aug 20 12:40:52 2024\n",
      "52223 MiB RAM detected, ~50788 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files//LDpruning/TMEM175_UNIMPUTED.fam.\n",
      "366 variants loaded from\n",
      "/home/jupyter/workspace/ws_files//LDpruning/TMEM175_UNIMPUTED.bim.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "Calculating allele frequencies... done.\n",
      "--indep-pairwise (1 compute thread): 50257/366 variants removed.\n",
      "Variant lists written to\n",
      "/home/jupyter/workspace/ws_files//LDpruning/prune_TMEM175.prune.in and\n",
      "/home/jupyter/workspace/ws_files//LDpruning/prune_TMEM175.prune.out .\n",
      "End time: Tue Aug 20 12:40:53 2024\n"
     ]
    }
   ],
   "source": [
    "# Prune out unnecessary SNPs (only need to do this to generate PCs)\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/LDpruning/TMEM175_UNIMPUTED \\\n",
    "--indep-pairwise 50 5 0.5 \\\n",
    "--out {WORK_DIR}/LDpruning/prune_TMEM175"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2e0d546a-91fb-4c73-8c7f-8fa160f79a3d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "109 /home/jupyter/workspace/ws_files//LDpruning/prune_TMEM175.prune.in\n"
     ]
    }
   ],
   "source": [
    "!wc -l {WORK_DIR}/LDpruning/prune_TMEM175.prune.in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "5297c7fd-72c6-4ee6-bccb-ee0b287b4e9d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files//LDpruning/SCARB2_UNIMPUTED.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files//SCARB2/SCARB2_EUR/EUR_SCARB2\n",
      "  --exclude /home/jupyter/workspace/ws_files//LDpruning/exclusion_regions_hg38.txt\n",
      "  --geno 0.05\n",
      "  --hwe 1E-6\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files//LDpruning/SCARB2_UNIMPUTED\n",
      "\n",
      "Start time: Tue Aug 20 13:59:30 2024\n",
      "52223 MiB RAM detected, ~50600 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files//SCARB2/SCARB2_EUR/EUR_SCARB2.fam.\n",
      "9570 variants loaded from\n",
      "/home/jupyter/workspace/ws_files//SCARB2/SCARB2_EUR/EUR_SCARB2.bim.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "--exclude: 9570 variants remaining.\n",
      "Calculating allele frequencies... done.\n",
      "--geno: 4070 variants removed due to missing genotype data.\n",
      "--hwe: 1 variant removed due to Hardy-Weinberg exact test (founders only).\n",
      "4580 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "919 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files//LDpruning/SCARB2_UNIMPUTED.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files//LDpruning/SCARB2_UNIMPUTED.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files//LDpruning/SCARB2_UNIMPUTED.bed ...\n",
      "done.\n",
      "End time: Tue Aug 20 13:59:31 2024\n"
     ]
    }
   ],
   "source": [
    "# SCARB2 LD pruning\n",
    "# Make sure to use high-quality SNPs\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/SCARB2/SCARB2_EUR/EUR_SCARB2 \\\n",
    "--maf 0.01 \\\n",
    "--geno 0.05 \\\n",
    "--hwe 1E-6 \\\n",
    "--make-bed \\\n",
    "--exclude {WORK_DIR}/LDpruning/exclusion_regions_hg38.txt \\\n",
    "--out {WORK_DIR}/LDpruning/SCARB2_UNIMPUTED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "b0559709-b54c-476c-abe9-a9a3af902c93",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files//LDpruning/prune_SCARB2.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files//LDpruning/SCARB2_UNIMPUTED\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out /home/jupyter/workspace/ws_files//LDpruning/prune_SCARB2\n",
      "\n",
      "Start time: Tue Aug 20 13:59:36 2024\n",
      "52223 MiB RAM detected, ~50591 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files//LDpruning/SCARB2_UNIMPUTED.fam.\n",
      "919 variants loaded from\n",
      "/home/jupyter/workspace/ws_files//LDpruning/SCARB2_UNIMPUTED.bim.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "Calculating allele frequencies... done.\n",
      "--indep-pairwise (1 compute thread): 50776/919 variants removed.\n",
      "Variant lists written to\n",
      "/home/jupyter/workspace/ws_files//LDpruning/prune_SCARB2.prune.in and\n",
      "/home/jupyter/workspace/ws_files//LDpruning/prune_SCARB2.prune.out .\n",
      "End time: Tue Aug 20 13:59:36 2024\n"
     ]
    }
   ],
   "source": [
    "# Prune out unnecessary SNPs (only need to do this to generate PCs)\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/LDpruning/SCARB2_UNIMPUTED \\\n",
    "--indep-pairwise 50 5 0.5 \\\n",
    "--out {WORK_DIR}/LDpruning/prune_SCARB2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "bd54d546-0715-4aa4-8ffe-61a7ffbecb30",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "143 /home/jupyter/workspace/ws_files//LDpruning/prune_SCARB2.prune.in\n"
     ]
    }
   ],
   "source": [
    "!wc -l {WORK_DIR}/LDpruning/prune_SCARB2.prune.in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "11310259-35d6-4388-a844-8376e69e2e24",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files//LDpruning/CTSB_UNIMPUTED.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files//CTSB_EUR/EUR_CTSB\n",
      "  --exclude /home/jupyter/workspace/ws_files//LDpruning/exclusion_regions_hg38.txt\n",
      "  --geno 0.05\n",
      "  --hwe 1E-6\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files//LDpruning/CTSB_UNIMPUTED\n",
      "\n",
      "Start time: Tue Aug 20 14:02:15 2024\n",
      "52223 MiB RAM detected, ~50614 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files//CTSB_EUR/EUR_CTSB.fam.\n",
      "13481 variants loaded from\n",
      "/home/jupyter/workspace/ws_files//CTSB_EUR/EUR_CTSB.bim.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "--exclude: 13481 variants remaining.\n",
      "Calculating allele frequencies... done.\n",
      "--geno: 6697 variants removed due to missing genotype data.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test (founders only).\n",
      "6190 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "594 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files//LDpruning/CTSB_UNIMPUTED.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files//LDpruning/CTSB_UNIMPUTED.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files//LDpruning/CTSB_UNIMPUTED.bed ...\n",
      "done.\n",
      "End time: Tue Aug 20 14:02:17 2024\n"
     ]
    }
   ],
   "source": [
    "# CTSB LD pruning\n",
    "# Make sure to use high-quality SNPs\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/CTSB_EUR/EUR_CTSB \\\n",
    "--maf 0.01 \\\n",
    "--geno 0.05 \\\n",
    "--hwe 1E-6 \\\n",
    "--make-bed \\\n",
    "--exclude {WORK_DIR}/LDpruning/exclusion_regions_hg38.txt \\\n",
    "--out {WORK_DIR}/LDpruning/CTSB_UNIMPUTED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "83509f1a-d4bd-4ed9-b970-5ac5f6f6296d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files//LDpruning/prune_CTSB.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files//LDpruning/CTSB_UNIMPUTED\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out /home/jupyter/workspace/ws_files//LDpruning/prune_CTSB\n",
      "\n",
      "Start time: Tue Aug 20 14:02:51 2024\n",
      "52223 MiB RAM detected, ~50604 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files//LDpruning/CTSB_UNIMPUTED.fam.\n",
      "594 variants loaded from\n",
      "/home/jupyter/workspace/ws_files//LDpruning/CTSB_UNIMPUTED.bim.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "Calculating allele frequencies... done.\n",
      "--indep-pairwise (1 compute thread): 50477/594 variants removed.\n",
      "Variant lists written to\n",
      "/home/jupyter/workspace/ws_files//LDpruning/prune_CTSB.prune.in and\n",
      "/home/jupyter/workspace/ws_files//LDpruning/prune_CTSB.prune.out .\n",
      "End time: Tue Aug 20 14:02:52 2024\n"
     ]
    }
   ],
   "source": [
    "# Prune out unnecessary SNPs (only need to do this to generate PCs)\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/LDpruning/CTSB_UNIMPUTED \\\n",
    "--indep-pairwise 50 5 0.5 \\\n",
    "--out {WORK_DIR}/LDpruning/prune_CTSB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "23a7c18f-8ddd-4bba-b53c-8f73ead0cee8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "117 /home/jupyter/workspace/ws_files//LDpruning/prune_CTSB.prune.in\n"
     ]
    }
   ],
   "source": [
    "!wc -l {WORK_DIR}/LDpruning/prune_CTSB.prune.in"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0d56f08-f350-4307-aed4-3d92b578075d",
   "metadata": {},
   "source": [
    "Create a covariate file with GP2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "be08a62f-d5eb-487e-9a49-58110f77f99e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "CLINICAL_DATA_PATH = pathlib.Path(REL7_PATH, 'clinical_data/master_key_release7_final_vwb.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b234d81-537d-4c4c-aabb-91fa771be215",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Let's load the master key\n",
    "key = pd.read_csv(CLINICAL_DATA_PATH, low_memory=False)\n",
    "print(key.shape)\n",
    "key.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6f2323d1-1d6e-4c7d-89aa-28ac02f01e5a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Subsetting to keep only a few columns \n",
    "key = key[['GP2sampleID', 'baseline_GP2_phenotype_for_qc', 'biological_sex_for_qc', 'age_at_sample_collection', 'age_of_onset', 'label']]\n",
    "# Renaming the columns\n",
    "key.rename(columns = {'GP2sampleID':'IID',\n",
    "                                     'baseline_GP2_phenotype_for_qc':'phenotype',\n",
    "                                     'biological_sex_for_qc':'SEX', \n",
    "                                     'age_at_sample_collection':'AGE', \n",
    "                                     'age_of_onset':'AAO'}, inplace = True)\n",
    "key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "7e443482-c420-4e2a-a6c7-f2322ac35cfb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/jupyter/workspace/ws_files\n"
     ]
    }
   ],
   "source": [
    "! pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7ad706d2-25cb-4d24-b3da-074b1221aa5d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AJ\n",
      "Related individuals: (245, 9)\n",
      "Unrelated individuals: (2675, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          1655\n",
      "Female                        1009\n",
      "Other/Unknown/Not Reported      11\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       1655\n",
      "2       1009\n",
      "<NA>      11\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 95\n",
      "Number of controls missing age: 195\n",
      "WORKING ON: EAS\n",
      "Related individuals: (350, 9)\n",
      "Unrelated individuals: (5303, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          3398\n",
      "Female                        1899\n",
      "Other/Unknown/Not Reported       6\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       3398\n",
      "2       1899\n",
      "<NA>       6\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 1349\n",
      "Number of controls missing age: 1268\n",
      "WORKING ON: SAS\n",
      "Related individuals: (55, 9)\n",
      "Unrelated individuals: (624, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          397\n",
      "Female                        226\n",
      "Other/Unknown/Not Reported      1\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       397\n",
      "2       226\n",
      "<NA>      1\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 203\n",
      "Number of controls missing age: 83\n",
      "WORKING ON: AMR\n",
      "Related individuals: (51, 9)\n",
      "Unrelated individuals: (672, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          374\n",
      "Female                        296\n",
      "Other/Unknown/Not Reported      2\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       374\n",
      "2       296\n",
      "<NA>      2\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 106\n",
      "Number of controls missing age: 55\n",
      "WORKING ON: MDE\n",
      "Related individuals: (19, 9)\n",
      "Unrelated individuals: (589, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          368\n",
      "Female                        219\n",
      "Other/Unknown/Not Reported      2\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       368\n",
      "2       219\n",
      "<NA>      2\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 140\n",
      "Number of controls missing age: 12\n",
      "WORKING ON: CAH\n",
      "Related individuals: (65, 9)\n",
      "Unrelated individuals: (1108, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          585\n",
      "Female                        498\n",
      "Other/Unknown/Not Reported     25\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       585\n",
      "2       498\n",
      "<NA>     25\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 136\n",
      "Number of controls missing age: 52\n",
      "WORKING ON: FIN\n",
      "Related individuals: (7, 9)\n",
      "Unrelated individuals: (117, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Female    63\n",
      "Male      54\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "2    63\n",
      "1    54\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 62\n",
      "Number of controls missing age: 3\n",
      "WORKING ON: CAS\n",
      "Related individuals: (137, 9)\n",
      "Unrelated individuals: (933, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Female    483\n",
      "Male      450\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "2    483\n",
      "1    450\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 111\n",
      "Number of controls missing age: 88\n",
      "WORKING ON: AFR\n",
      "Related individuals: (228, 9)\n",
      "Unrelated individuals: (2752, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          1553\n",
      "Female                        1197\n",
      "Other/Unknown/Not Reported       2\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       1553\n",
      "2       1197\n",
      "<NA>       2\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 835\n",
      "Number of controls missing age: 809\n",
      "WORKING ON: EUR\n",
      "Related individuals: (3662, 9)\n",
      "Unrelated individuals: (39710, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          23733\n",
      "Female                        15784\n",
      "Other/Unknown/Not Reported      193\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       23733\n",
      "2       15784\n",
      "<NA>      193\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 5947\n",
      "Number of controls missing age: 3347\n",
      "WORKING ON: AAC\n",
      "Related individuals: (28, 9)\n",
      "Unrelated individuals: (1181, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Female                        690\n",
      "Male                          489\n",
      "Other/Unknown/Not Reported      2\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "2       690\n",
      "1       489\n",
      "<NA>      2\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 91\n",
      "Number of controls missing age: 96\n"
     ]
    }
   ],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    ## Subset to keep ancestry of interest \n",
    "    ancestry_key = key[key['label']==ancestry].copy()\n",
    "    ancestry_key.reset_index(drop=True)\n",
    "    \n",
    "    # Load information about related individuals in the ancestry analyzed\n",
    "    related_df = pd.read_csv(f'{REL7_PATH}/meta_data/related_samples/{ancestry}_release7_vwb.related')\n",
    "    print(f'Related individuals: {related_df.shape}')\n",
    "    \n",
    "    # Make a list of just one set of related people\n",
    "    related_list = list(related_df['IID1'])\n",
    "    \n",
    "    # Check value counts of related and remove only one related individual\n",
    "    ancestry_key = ancestry_key[~ancestry_key[\"IID\"].isin(related_list)]\n",
    "    \n",
    "    # Check size\n",
    "    print(f'Unrelated individuals: {ancestry_key.shape}')\n",
    "    \n",
    "    # Convert phenotype to binary (1/2)\n",
    "    ## Assign conditions so case=2 and controls=1, and -9 otherwise (matching PLINK convention)\n",
    "    # PD = 2; control = 1\n",
    "    pheno_mapping = {\"PD\": 2, \"Control\": 1}\n",
    "    ancestry_key['PHENO'] = ancestry_key['phenotype'].map(pheno_mapping).astype('Int64')\n",
    "    \n",
    "    # Check value counts of pheno\n",
    "    ancestry_key['PHENO'].value_counts(dropna=False)\n",
    "    \n",
    "    ## Get the PCs\n",
    "    pcs = pd.read_csv(f'{REL7_PATH}/meta_data/qc_metrics/projected_pcs_vwb.csv')\n",
    "    \n",
    "     #Select just first 5 PCs\n",
    "    selected_columns = ['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5']\n",
    "    pcs = pd.DataFrame(data=pcs.iloc[:, 1:7].values, columns=selected_columns)\n",
    "    \n",
    "     # Drop the first row (since it's now the column names)\n",
    "    pcs = pcs.drop(0)\n",
    "    \n",
    "    # Reset the index to remove any potential issues\n",
    "    pcs = pcs.reset_index(drop=True)\n",
    "    \n",
    "    # Check size\n",
    "    print(f'PCs: {pcs.shape}')\n",
    "    \n",
    "     # Check value counts of SEX\n",
    "    sex_og_values = ancestry_key['SEX'].value_counts(dropna=False)\n",
    "    print(f'Sex value counts - original:\\n {sex_og_values.to_string()}')\n",
    "    \n",
    "     # Convert sex to binary (1/2)\n",
    "    ## Assign conditions so female=2 and men=1, and -9 otherwise (matching PLINK convention)\n",
    "    # Female = 2; Male = 1\n",
    "    sex_mapping = {\"Female\": 2, \"Male\": 1}\n",
    "    ancestry_key['SEX'] = ancestry_key['SEX'].map(sex_mapping).astype('Int64')\n",
    "    \n",
    "    # Check value counts of SEX after recoding\n",
    "    sex_recode_values = ancestry_key['SEX'].value_counts(dropna=False)\n",
    "    print(f'Sex value counts - recoded:\\n{sex_recode_values.to_string()}')\n",
    "    \n",
    "    ## Make covariate file\n",
    "    df = pd.merge(pcs, ancestry_key, on='IID', how='left')\n",
    "    print(f'Check columns for covariate file: {df.columns}')\n",
    "    \n",
    "    #Make additional columns - FID, fatid and matid - these are needed for RVtests!!\n",
    "    #RVtests needs the first 5 columns to be fid, iid, fatid, matid and sex otherwise it does not run correctly\n",
    "    #Uppercase column name is ok\n",
    "    #See https://zhanxw.github.io/rvtests/#phenotype-file\n",
    "    df['FID'] = 0\n",
    "    df['FATID'] = 0\n",
    "    df['MATID'] = 0\n",
    "    \n",
    "    ## Clean up and keep columns we need \n",
    "    final_df = df[['FID','IID', 'FATID', 'MATID', 'SEX', 'AGE', 'PHENO','PC1', 'PC2', 'PC3', 'PC4', 'PC5']].copy()\n",
    "    \n",
    "    ##DO NOT replace missing values with -9 as this is misinterpreted by RVtests - needs to be nonnumeric\n",
    "    #Leave missing values as NA\n",
    "    \n",
    "    #Check number of PD cases missing age\n",
    "    pd_missAge = final_df[(final_df['PHENO']==2)&(final_df['AGE'].isna())]\n",
    "    print(f'Number of PD cases missing age: {pd_missAge.shape[0]}')\n",
    "    \n",
    "    #Check number of controls missing age\n",
    "    control_missAge = final_df[(final_df['PHENO']==1)&(final_df['AGE'].isna())]\n",
    "    print(f'Number of controls missing age: {control_missAge.shape[0]}')\n",
    "    \n",
    "    ## Make file of sample IDs to keep \n",
    "    samples_toKeep = final_df[['FID', 'IID']].copy()\n",
    "    samples_toKeep.to_csv(f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}/{ancestry}.samplestoKeep', sep = '\\t', index=False, header=None)\n",
    "    \n",
    "    ## Make your covariate file\n",
    "    #Included na_rep to write out missing/NA values explicitly as string/text, not as blank otherwise they are misread in RVtests\n",
    "    final_df.to_csv(f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}/{ancestry}_covariate_file.txt', sep = '\\t', na_rep='NA', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be3cb231-35ac-4d1a-8aa0-8f46d18d66d8",
   "metadata": {},
   "source": [
    "Annotation of the gene\n",
    "\n",
    "Extract the region using PLINK\n",
    "\n",
    "Extract SCARB2 gene\n",
    "\n",
    "SCARB2 coordinates: Chromosome 4: 76,158,737-76,234,536(GRCh38/hg38)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "8d01527f-0f81-45a5-806a-3a5b14dff51d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Tue Aug 27 14:03:25 2024\n",
      "52223 MiB RAM detected, ~50462 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb.psam.\n",
      "10419161 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "9570 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.bed ...\n",
      "54done.\n",
      "End time: Tue Aug 27 14:03:33 2024\n"
     ]
    }
   ],
   "source": [
    "# generate binary file in eur ancestry\n",
    "\n",
    "WORK_DIR = \"~/workspace/ws_files/SCARB2/\"\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--pfile {REL7_PATH}/imputed_genotypes/EUR/chr4_EUR_release7_vwb \\\n",
    "--chr 4 \\\n",
    "--from-bp 76108737 \\\n",
    "--to-bp 76284536 \\\n",
    "--make-bed \\\n",
    "--out {WORK_DIR}/SCARB2_EUR/EUR_SCARB2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "1d7cd78a-550c-4162-9d96-6b1033606842",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Tue Aug 20 09:35:27 2024\n",
      "52223 MiB RAM detected, ~50313 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb.psam.\n",
      "1669760 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (1292 cases, 411 controls).\n",
      "1849 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.bed ...\n",
      "done.\n",
      "End time: Tue Aug 20 09:35:29 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Tue Aug 20 09:35:30 2024\n",
      "52223 MiB RAM detected, ~50291 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb.psam.\n",
      "3235052 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (2662 cases, 2461 controls).\n",
      "3094 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.bed ...\n",
      "done.\n",
      "End time: Tue Aug 20 09:35:32 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Tue Aug 20 09:35:33 2024\n",
      "52223 MiB RAM detected, ~50272 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "635 samples (227 females, 408 males; 635 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb.psam.\n",
      "1933318 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (387 cases, 222 controls).\n",
      "2058 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.bed ...\n",
      "done.\n",
      "End time: Tue Aug 20 09:35:34 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Tue Aug 20 09:35:35 2024\n",
      "52223 MiB RAM detected, ~50265 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "646 samples (283 females, 363 males; 646 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb.psam.\n",
      "2407746 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (458 cases, 155 controls).\n",
      "2516 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.bed ...\n",
      "done.\n",
      "End time: Tue Aug 20 09:35:37 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Tue Aug 20 09:35:37 2024\n",
      "52223 MiB RAM detected, ~50251 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "581 samples (212 females, 369 males; 581 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb.psam.\n",
      "2161336 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (311 cases, 225 controls).\n",
      "2380 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.bed ...\n",
      "done.\n",
      "End time: Tue Aug 20 09:35:39 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Tue Aug 20 09:35:39 2024\n",
      "52223 MiB RAM detected, ~50246 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "851 samples (405 females, 446 males; 851 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb.psam.\n",
      "3678288 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (525 cases, 302 controls).\n",
      "3550 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.bed ...\n",
      "done.\n",
      "End time: Tue Aug 20 09:35:42 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Tue Aug 20 09:35:42 2024\n",
      "52223 MiB RAM detected, ~50245 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "114 samples (61 females, 53 males; 114 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb.psam.\n",
      "891614 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (98 cases, 8 controls).\n",
      "1166 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.bed ...\n",
      "done.\n",
      "End time: Tue Aug 20 09:35:44 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Tue Aug 20 09:35:44 2024\n",
      "52223 MiB RAM detected, ~50240 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "903 samples (464 females, 439 males; 903 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb.psam.\n",
      "2200655 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (552 cases, 343 controls).\n",
      "2219 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.bed ...\n",
      "done.\n",
      "End time: Tue Aug 20 09:35:46 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Tue Aug 20 09:35:46 2024\n",
      "52223 MiB RAM detected, ~50236 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb.psam.\n",
      "4555051 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (942 cases, 1679 controls).\n",
      "4304 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.bed ...\n",
      "done.\n",
      "End time: Tue Aug 20 09:35:50 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Tue Aug 20 09:35:50 2024\n",
      "52223 MiB RAM detected, ~50236 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb.psam.\n",
      "10419161 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "9570 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.bed ...\n",
      "54done.\n",
      "End time: Tue Aug 20 09:35:58 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Tue Aug 20 09:35:58 2024\n",
      "52223 MiB RAM detected, ~50226 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1111 samples (656 females, 455 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb.psam.\n",
      "3986699 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (285 cases, 801 controls).\n",
      "3848 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.bed ...\n",
      "done.\n",
      "End time: Tue Aug 20 09:36:01 2024\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = \"~/workspace/ws_files/SCARB2/\"\n",
    "\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --pfile {REL7_PATH}/imputed_genotypes/{ancestry}/chr4_{ancestry}_release7_vwb \\\n",
    "    --chr 4 \\\n",
    "    --from-bp 76108737 \\\n",
    "    --to-bp 76284536 \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/SCARB2_{ancestry}/{ancestry}_SCARB2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "7414bf85-051c-451c-9689-69dc75152bae",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4\tchr4:76108753:T:G\t0\t76108753\tG\tT\n",
      "4\tchr4:76108755:T:C\t0\t76108755\tC\tT\n",
      "4\tchr4:76108761:G:A\t0\t76108761\tA\tG\n",
      "4\tchr4:76108794:G:A\t0\t76108794\tA\tG\n",
      "4\tchr4:76108804:T:A\t0\t76108804\tA\tT\n",
      "4\tchr4:76108804:T:C\t0\t76108804\tC\tT\n",
      "4\tchr4:76108807:A:C\t0\t76108807\tC\tA\n",
      "4\tchr4:76108828:G:T\t0\t76108828\tT\tG\n",
      "4\tchr4:76108844:G:A\t0\t76108844\tA\tG\n",
      "4\tchr4:76108848:C:T\t0\t76108848\tT\tC\n"
     ]
    }
   ],
   "source": [
    "# Visualize bim file\n",
    "! head {WORK_DIR}/SCARB2_EUR/EUR_SCARB2.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "794583e6-a634-4945-a7f4-491f26fe6ec8",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Visualize bim file\n",
    "! head {WORK_DIR}/SCARB2_EUR/EUR_SCARB2.fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "43e42230-1861-4aba-b88f-6cde131ebfb9",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = \"~/workspace/ws_files/SCARB2/\"\n",
    "    \n",
    "    ! head -n 1 {WORK_DIR}/SCARB2_{ancestry}/{ancestry}_SCARB2.fam > {WORK_DIR}/SCARB2_{ancestry}/{ancestry}_s1.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "773ff6a8-6eca-440a-b4f2-7509b50c8eba",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "! head {WORK_DIR}/SCARB2_EUR/EUR_s1.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "425b003b-e6eb-4a9a-b0a3-aa5e4d3ca43f",
   "metadata": {},
   "source": [
    "Turn binary files into VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "adc62294-0899-4c9b-a8b4-a20f24a8adee",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:38:51 2024\n",
      "52223 MiB RAM detected, ~50233 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.fam.\n",
      "1849 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.bim.\n",
      "1 binary phenotype loaded (1292 cases, 411 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2_v1.bed ...\n",
      "done.\n",
      "End time: Tue Aug 20 09:38:52 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:38:52 2024\n",
      "52223 MiB RAM detected, ~50243 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.fam.\n",
      "3094 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.bim.\n",
      "1 binary phenotype loaded (2662 cases, 2461 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2_v1.bed\n",
      "... done.\n",
      "End time: Tue Aug 20 09:38:52 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:38:52 2024\n",
      "52223 MiB RAM detected, ~50244 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "635 samples (227 females, 408 males; 635 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.fam.\n",
      "2058 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.bim.\n",
      "1 binary phenotype loaded (387 cases, 222 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2_v1.bed\n",
      "... done.\n",
      "End time: Tue Aug 20 09:38:53 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:38:53 2024\n",
      "52223 MiB RAM detected, ~50228 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "646 samples (283 females, 363 males; 646 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.fam.\n",
      "2516 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.bim.\n",
      "1 binary phenotype loaded (458 cases, 155 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2_v1.bed\n",
      "... done.\n",
      "End time: Tue Aug 20 09:38:54 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:38:54 2024\n",
      "52223 MiB RAM detected, ~50229 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "581 samples (212 females, 369 males; 581 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.fam.\n",
      "2380 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.bim.\n",
      "1 binary phenotype loaded (311 cases, 225 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2_v1.bed\n",
      "... done.\n",
      "End time: Tue Aug 20 09:38:54 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:38:55 2024\n",
      "52223 MiB RAM detected, ~50229 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "851 samples (405 females, 446 males; 851 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.fam.\n",
      "3550 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.bim.\n",
      "1 binary phenotype loaded (525 cases, 302 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2_v1.bed\n",
      "... done.\n",
      "End time: Tue Aug 20 09:38:55 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:38:55 2024\n",
      "52223 MiB RAM detected, ~50230 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "114 samples (61 females, 53 males; 114 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.fam.\n",
      "1166 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.bim.\n",
      "1 binary phenotype loaded (98 cases, 8 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2_v1.bed\n",
      "... done.\n",
      "End time: Tue Aug 20 09:38:56 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:38:56 2024\n",
      "52223 MiB RAM detected, ~50231 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "903 samples (464 females, 439 males; 903 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.fam.\n",
      "2219 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.bim.\n",
      "1 binary phenotype loaded (552 cases, 343 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2_v1.bed\n",
      "... done.\n",
      "End time: Tue Aug 20 09:38:57 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:38:57 2024\n",
      "52223 MiB RAM detected, ~50231 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.fam.\n",
      "4304 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.bim.\n",
      "1 binary phenotype loaded (942 cases, 1679 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2_v1.bed\n",
      "... done.\n",
      "End time: Tue Aug 20 09:38:57 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:38:57 2024\n",
      "52223 MiB RAM detected, ~50232 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.fam.\n",
      "9570 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.bim.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2_v1.bed\n",
      "... done.\n",
      "End time: Tue Aug 20 09:38:58 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:38:58 2024\n",
      "52223 MiB RAM detected, ~50232 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1111 samples (656 females, 455 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.fam.\n",
      "3848 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.bim.\n",
      "1 binary phenotype loaded (285 cases, 801 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2_v1.fam\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2_v1.bim\n",
      "... done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2_v1.bed\n",
      "... done.\n",
      "End time: Tue Aug 20 09:38:58 2024\n"
     ]
    }
   ],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = \"~/workspace/ws_files/SCARB2/\"\n",
    "    \n",
    "    ## Turn binary files into VCF\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/SCARB2_{ancestry}/{ancestry}_SCARB2 \\\n",
    "    --keep {WORK_DIR}/SCARB2_{ancestry}/{ancestry}_s1.txt \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/SCARB2_{ancestry}/{ancestry}_SCARB2_v1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "6ecbf289-3a9e-46f4-ad22-40dde9c18cb1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:39:45 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50245 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2_v1.fam.\n",
      "1849 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2_v1.vcf ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Tue Aug 20 09:39:45 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:39:46 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50245 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2_v1.fam.\n",
      "3094 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2_v1.vcf ... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Tue Aug 20 09:39:46 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:39:46 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50231 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2_v1.fam.\n",
      "2058 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2_v1.vcf ... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Tue Aug 20 09:39:46 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:39:47 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50232 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2_v1.fam.\n",
      "2516 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2_v1.vcf ... 1010111112121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373738383939404041414242434344444545464647474848495050515152525353545455555656575758585959606061616262636364646565666667676868696970707171727273737475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Tue Aug 20 09:39:47 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:39:47 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50233 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2_v1.fam.\n",
      "2380 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2_v1.vcf ... 101011111212131314151516161717181819202021212222232324252526262727282829303031313232333334353536363737383839404041414242434344454546464747484849505051515252535354555556565757585859606061616262636364656566666767686869707071717272737374757576767777787879808081818282838384858586868787888889909091919292939394959596969797989899done.\n",
      "End time: Tue Aug 20 09:39:47 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:39:48 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50231 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2_v1.fam.\n",
      "3550 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2_v1.vcf ... 101011121213141415161617181819202021222223242425262627282829303031323233343435363637383839404041424243444445464647484849505051525253545455565657585859606061626263646465666667686869707071727273747475767677787879808081828283848485868687888889909091929293949495969697989899done.\n",
      "End time: Tue Aug 20 09:39:48 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:39:48 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50231 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2_v1.fam.\n",
      "1166 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2_v1.vcf ... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Tue Aug 20 09:39:48 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:39:49 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50231 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2_v1.fam.\n",
      "2219 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2_v1.vcf ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Tue Aug 20 09:39:49 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:39:49 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50232 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2_v1.fam.\n",
      "4304 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2_v1.vcf ... 1010111112121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373738383939404041414242434344444545464647474848495050515152525353545455555656575758585959606061616262636364646565666667676868696970707171727273737475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Tue Aug 20 09:39:49 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:39:50 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50233 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2_v1.fam.\n",
      "9570 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2_v1.vcf ... 101011111212131314141515161617171818192020212122222323242425252626272728282930303131323233333434353536363737383839404041414242434344444545464647474848495050515152525353545455555656575758585960606161626263636464656566666767686869707071717272737374747575767677777878798080818182828383848485858686878788888990909191929293939494959596969797989899done.\n",
      "End time: Tue Aug 20 09:39:50 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2_v1\n",
      "\n",
      "Start time: Tue Aug 20 09:39:50 2024\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "52223 MiB RAM detected, ~50233 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2_v1.fam.\n",
      "3848 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2_v1.vcf ... 1010111112121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373738383939404041414242434344444545464647474848495050515152525353545455555656575758585959606061616262636364646565666667676868696970707171727273737475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Tue Aug 20 09:39:50 2024\n"
     ]
    }
   ],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = \"~/workspace/ws_files/SCARB2/\"\n",
    "    \n",
    "    ## Turn binary files into VCF\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/SCARB2_{ancestry}/{ancestry}_SCARB2_v1 \\\n",
    "    --recode vcf-fid \\\n",
    "    --out {WORK_DIR}/SCARB2_{ancestry}/{ancestry}_SCARB2_v1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "3b7f1d14-99e8-4096-aea5-ed0b5759ca87",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = \"~/workspace/ws_files/SCARB2/\"\n",
    "    ! bgzip -f {WORK_DIR}/SCARB2_{ancestry}/{ancestry}_SCARB2_v1.vcf\n",
    "    ! tabix -f -p vcf {WORK_DIR}/SCARB2_{ancestry}/{ancestry}_SCARB2_v1.vcf.gz "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "58841071-9e46-4a7d-bba5-48e925f86ff4",
   "metadata": {},
   "source": [
    "Annotate using ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "87ed12e7-35fb-486e-ad98-8d0e8aa35583",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 1856 lines from VCF file\n",
      "NOTICE: A total of 1849 locus in VCF file passed QC threshold, representing 1707 SNPs (1174 transitions and 533 transversions) and 142 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 1707 SNP genotypes (1174 transitions and 533 transversions) and 142 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 1849 unique variants in 1849 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 1849 unique variants in 1849 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AJ/AJ_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 3101 lines from VCF file\n",
      "NOTICE: A total of 3094 locus in VCF file passed QC threshold, representing 2838 SNPs (2001 transitions and 837 transversions) and 256 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2838 SNP genotypes (2001 transitions and 837 transversions) and 256 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3094 unique variants in 3094 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3094 unique variants in 3094 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EAS/EAS_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 2065 lines from VCF file\n",
      "NOTICE: A total of 2058 locus in VCF file passed QC threshold, representing 1908 SNPs (1329 transitions and 579 transversions) and 150 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 1908 SNP genotypes (1329 transitions and 579 transversions) and 150 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2058 unique variants in 2058 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2058 unique variants in 2058 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_SAS/SAS_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 2523 lines from VCF file\n",
      "NOTICE: A total of 2516 locus in VCF file passed QC threshold, representing 2313 SNPs (1613 transitions and 700 transversions) and 203 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2313 SNP genotypes (1613 transitions and 700 transversions) and 203 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2516 unique variants in 2516 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2516 unique variants in 2516 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AMR/AMR_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 2387 lines from VCF file\n",
      "NOTICE: A total of 2380 locus in VCF file passed QC threshold, representing 2192 SNPs (1535 transitions and 657 transversions) and 188 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2192 SNP genotypes (1535 transitions and 657 transversions) and 188 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2380 unique variants in 2380 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2380 unique variants in 2380 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_MDE/MDE_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 3557 lines from VCF file\n",
      "NOTICE: A total of 3550 locus in VCF file passed QC threshold, representing 3252 SNPs (2279 transitions and 973 transversions) and 298 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3252 SNP genotypes (2279 transitions and 973 transversions) and 298 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3550 unique variants in 3550 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3550 unique variants in 3550 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAH/CAH_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 1173 lines from VCF file\n",
      "NOTICE: A total of 1166 locus in VCF file passed QC threshold, representing 1080 SNPs (735 transitions and 345 transversions) and 86 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 1080 SNP genotypes (735 transitions and 345 transversions) and 86 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 1166 unique variants in 1166 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 1166 unique variants in 1166 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_FIN/FIN_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 2226 lines from VCF file\n",
      "NOTICE: A total of 2219 locus in VCF file passed QC threshold, representing 2036 SNPs (1417 transitions and 619 transversions) and 183 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2036 SNP genotypes (1417 transitions and 619 transversions) and 183 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2219 unique variants in 2219 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2219 unique variants in 2219 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_CAS/CAS_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 4311 lines from VCF file\n",
      "NOTICE: A total of 4304 locus in VCF file passed QC threshold, representing 3886 SNPs (2705 transitions and 1181 transversions) and 418 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3886 SNP genotypes (2705 transitions and 1181 transversions) and 418 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 4304 unique variants in 4304 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 4304 unique variants in 4304 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AFR/AFR_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 9577 lines from VCF file\n",
      "NOTICE: A total of 9570 locus in VCF file passed QC threshold, representing 8744 SNPs (6039 transitions and 2705 transversions) and 826 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 8744 SNP genotypes (6039 transitions and 2705 transversions) and 826 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 9570 unique variants in 9570 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 9570 unique variants in 9570 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_EUR/EUR_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 3855 lines from VCF file\n",
      "NOTICE: A total of 3848 locus in VCF file passed QC threshold, representing 3499 SNPs (2424 transitions and 1075 transversions) and 349 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3499 SNP genotypes (2424 transitions and 1075 transversions) and 349 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3848 unique variants in 3848 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3848 unique variants in 3848 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2//SCARB2_AAC/AAC_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "## annotate using ANNOVAR\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = \"~/workspace/ws_files/SCARB2/\"\n",
    "    \n",
    "    ! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/SCARB2_{ancestry}/{ancestry}_SCARB2_v1.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "    -out {WORK_DIR}/SCARB2_{ancestry}/{ancestry}_SCARB2.annovar \\\n",
    "    -remove -protocol refGene,clinvar_20140902 \\\n",
    "    -operation g,f \\\n",
    "    --nopolish \\\n",
    "    -nastring . \\\n",
    "    -vcfinput"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "07dcd456-5741-428c-9e22-5f37a67023fb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>76108753</td>\n",
       "      <td>76108753</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108753</td>\n",
       "      <td>chr4:76108753:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>76108755</td>\n",
       "      <td>76108755</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108755</td>\n",
       "      <td>chr4:76108755:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>76108761</td>\n",
       "      <td>76108761</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108761</td>\n",
       "      <td>chr4:76108761:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>76108804</td>\n",
       "      <td>76108804</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108804</td>\n",
       "      <td>chr4:76108804:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>76108828</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>chr4:76108828:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3843</th>\n",
       "      <td>4</td>\n",
       "      <td>76284266</td>\n",
       "      <td>76284266</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284266</td>\n",
       "      <td>chr4:76284266:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3844</th>\n",
       "      <td>4</td>\n",
       "      <td>76284298</td>\n",
       "      <td>76284298</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284298</td>\n",
       "      <td>chr4:76284298:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3845</th>\n",
       "      <td>4</td>\n",
       "      <td>76284397</td>\n",
       "      <td>76284397</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284397</td>\n",
       "      <td>chr4:76284397:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3846</th>\n",
       "      <td>4</td>\n",
       "      <td>76284490</td>\n",
       "      <td>76284490</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284490</td>\n",
       "      <td>chr4:76284490:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3847</th>\n",
       "      <td>4</td>\n",
       "      <td>76284509</td>\n",
       "      <td>76284509</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284509</td>\n",
       "      <td>chr4:76284509:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3848 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       4  76108753  76108753   T   C     intronic          ART3   \n",
       "1       4  76108755  76108755   T   C     intronic          ART3   \n",
       "2       4  76108761  76108761   G   A     intronic          ART3   \n",
       "3       4  76108804  76108804   T   C     intronic          ART3   \n",
       "4       4  76108828  76108828   G   T     intronic          ART3   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "3843    4  76284266  76284266   G   A     intronic  FAM47E-STBD1   \n",
       "3844    4  76284298  76284298   C   T     intronic  FAM47E-STBD1   \n",
       "3845    4  76284397  76284397   A   G     intronic  FAM47E-STBD1   \n",
       "3846    4  76284490  76284490   A   G     intronic  FAM47E-STBD1   \n",
       "3847    4  76284509  76284509   G   A     intronic  FAM47E-STBD1   \n",
       "\n",
       "     GeneDetail.refGene ExonicFunc.refGene AAChange.refGene clinvar_20140902  \\\n",
       "0                     .                  .                .                .   \n",
       "1                     .                  .                .                .   \n",
       "2                     .                  .                .                .   \n",
       "3                     .                  .                .                .   \n",
       "4                     .                  .                .                .   \n",
       "...                 ...                ...              ...              ...   \n",
       "3843                  .                  .                .                .   \n",
       "3844                  .                  .                .                .   \n",
       "3845                  .                  .                .                .   \n",
       "3846                  .                  .                .                .   \n",
       "3847                  .                  .                .                .   \n",
       "\n",
       "      Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  Otherinfo5  \\\n",
       "0            0.0          .          .           4    76108753   \n",
       "1            0.0          .          .           4    76108755   \n",
       "2            0.0          .          .           4    76108761   \n",
       "3            0.0          .          .           4    76108804   \n",
       "4            0.5          .          .           4    76108828   \n",
       "...          ...        ...        ...         ...         ...   \n",
       "3843         0.0          .          .           4    76284266   \n",
       "3844         0.0          .          .           4    76284298   \n",
       "3845         0.0          .          .           4    76284397   \n",
       "3846         0.0          .          .           4    76284490   \n",
       "3847         0.0          .          .           4    76284509   \n",
       "\n",
       "             Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9 Otherinfo10  \\\n",
       "0     chr4:76108753:T:C          T          C          .           .   \n",
       "1     chr4:76108755:T:C          T          C          .           .   \n",
       "2     chr4:76108761:G:A          G          A          .           .   \n",
       "3     chr4:76108804:T:C          T          C          .           .   \n",
       "4     chr4:76108828:G:T          G          T          .           .   \n",
       "...                 ...        ...        ...        ...         ...   \n",
       "3843  chr4:76284266:G:A          G          A          .           .   \n",
       "3844  chr4:76284298:C:T          C          T          .           .   \n",
       "3845  chr4:76284397:A:G          A          G          .           .   \n",
       "3846  chr4:76284490:A:G          A          G          .           .   \n",
       "3847  chr4:76284509:G:A          G          A          .           .   \n",
       "\n",
       "     Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0             PR          GT         0/0  \n",
       "1             PR          GT         0/0  \n",
       "2             PR          GT         0/0  \n",
       "3             PR          GT         0/0  \n",
       "4             PR          GT         0/1  \n",
       "...          ...         ...         ...  \n",
       "3843          PR          GT         0/0  \n",
       "3844          PR          GT         0/0  \n",
       "3845          PR          GT         0/0  \n",
       "3846          PR          GT         0/0  \n",
       "3847          PR          GT         0/0  \n",
       "\n",
       "[3848 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/SCARB2_AAC/AAC_SCARB2.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "aec18645-1d29-4e59-9d30-299a2bf7ef6f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      3385\n",
       "intergenic     186\n",
       "UTR3            89\n",
       "downstream      71\n",
       "exonic          59\n",
       "upstream        33\n",
       "UTR5            22\n",
       "splicing         3\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c374e710-faf7-4b8c-b146-7157f2d1dd68",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "92740e50-473f-4bbe-8e48-8d4d4c5d8899",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    32\n",
       "synonymous SNV       25\n",
       "stopgain              1\n",
       "stoploss              1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "d47c2a2f-fbb0-4901-8f46-52dfff41e834",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>76108750</td>\n",
       "      <td>76108750</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108750</td>\n",
       "      <td>chr4:76108750:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>76108755</td>\n",
       "      <td>76108755</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108755</td>\n",
       "      <td>chr4:76108755:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>76108804</td>\n",
       "      <td>76108804</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108804</td>\n",
       "      <td>chr4:76108804:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>76108828</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>chr4:76108828:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>76108858</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>chr4:76108858:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4299</th>\n",
       "      <td>4</td>\n",
       "      <td>76284274</td>\n",
       "      <td>76284274</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284274</td>\n",
       "      <td>chr4:76284274:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4300</th>\n",
       "      <td>4</td>\n",
       "      <td>76284298</td>\n",
       "      <td>76284298</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284298</td>\n",
       "      <td>chr4:76284298:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4301</th>\n",
       "      <td>4</td>\n",
       "      <td>76284438</td>\n",
       "      <td>76284438</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284438</td>\n",
       "      <td>chr4:76284438:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4302</th>\n",
       "      <td>4</td>\n",
       "      <td>76284476</td>\n",
       "      <td>76284476</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284476</td>\n",
       "      <td>chr4:76284476:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4303</th>\n",
       "      <td>4</td>\n",
       "      <td>76284490</td>\n",
       "      <td>76284490</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284490</td>\n",
       "      <td>chr4:76284490:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4304 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       4  76108750  76108750   T   C     intronic          ART3   \n",
       "1       4  76108755  76108755   T   C     intronic          ART3   \n",
       "2       4  76108804  76108804   T   C     intronic          ART3   \n",
       "3       4  76108828  76108828   G   T     intronic          ART3   \n",
       "4       4  76108858  76108858   G   A     intronic          ART3   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "4299    4  76284274  76284274   T   C     intronic  FAM47E-STBD1   \n",
       "4300    4  76284298  76284298   C   T     intronic  FAM47E-STBD1   \n",
       "4301    4  76284438  76284438   C   T     intronic  FAM47E-STBD1   \n",
       "4302    4  76284476  76284476   C   G     intronic  FAM47E-STBD1   \n",
       "4303    4  76284490  76284490   A   G     intronic  FAM47E-STBD1   \n",
       "\n",
       "     GeneDetail.refGene ExonicFunc.refGene AAChange.refGene clinvar_20140902  \\\n",
       "0                     .                  .                .                .   \n",
       "1                     .                  .                .                .   \n",
       "2                     .                  .                .                .   \n",
       "3                     .                  .                .                .   \n",
       "4                     .                  .                .                .   \n",
       "...                 ...                ...              ...              ...   \n",
       "4299                  .                  .                .                .   \n",
       "4300                  .                  .                .                .   \n",
       "4301                  .                  .                .                .   \n",
       "4302                  .                  .                .                .   \n",
       "4303                  .                  .                .                .   \n",
       "\n",
       "      Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  Otherinfo5  \\\n",
       "0            0.0          .          .           4    76108750   \n",
       "1            0.0          .          .           4    76108755   \n",
       "2            0.0          .          .           4    76108804   \n",
       "3            0.5          .          .           4    76108828   \n",
       "4            1.0          .          .           4    76108858   \n",
       "...          ...        ...        ...         ...         ...   \n",
       "4299         0.0          .          .           4    76284274   \n",
       "4300         0.0          .          .           4    76284298   \n",
       "4301         0.0          .          .           4    76284438   \n",
       "4302         1.0          .          .           4    76284476   \n",
       "4303         0.0          .          .           4    76284490   \n",
       "\n",
       "             Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9 Otherinfo10  \\\n",
       "0     chr4:76108750:T:C          T          C          .           .   \n",
       "1     chr4:76108755:T:C          T          C          .           .   \n",
       "2     chr4:76108804:T:C          T          C          .           .   \n",
       "3     chr4:76108828:G:T          G          T          .           .   \n",
       "4     chr4:76108858:G:A          G          A          .           .   \n",
       "...                 ...        ...        ...        ...         ...   \n",
       "4299  chr4:76284274:T:C          T          C          .           .   \n",
       "4300  chr4:76284298:C:T          C          T          .           .   \n",
       "4301  chr4:76284438:C:T          C          T          .           .   \n",
       "4302  chr4:76284476:C:G          C          G          .           .   \n",
       "4303  chr4:76284490:A:G          A          G          .           .   \n",
       "\n",
       "     Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0             PR          GT         0/0  \n",
       "1             PR          GT         0/0  \n",
       "2             PR          GT         0/0  \n",
       "3             PR          GT         0/1  \n",
       "4             PR          GT         1/1  \n",
       "...          ...         ...         ...  \n",
       "4299          PR          GT         0/0  \n",
       "4300          PR          GT         0/0  \n",
       "4301          PR          GT         0/0  \n",
       "4302          PR          GT         1/1  \n",
       "4303          PR          GT         0/0  \n",
       "\n",
       "[4304 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/SCARB2_AFR/AFR_SCARB2.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "3a767e6e-4ca3-4873-948b-1e91fd394c31",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      3759\n",
       "intergenic     242\n",
       "UTR3            95\n",
       "downstream      81\n",
       "exonic          64\n",
       "upstream        34\n",
       "UTR5            26\n",
       "splicing         3\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3430b340-d9c8-4552-afd0-324bc779a3d1",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "248d2a70-91e7-4438-ae23-819d2d72ebf5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV      36\n",
       "synonymous SNV         26\n",
       "frameshift deletion     1\n",
       "stoploss                1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "28bce744-2810-4ed6-9cc1-2fb90533d9e4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>76108753</td>\n",
       "      <td>76108753</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108753</td>\n",
       "      <td>chr4:76108753:T:G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>76108828</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>chr4:76108828:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>76108858</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>chr4:76108858:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>76108994</td>\n",
       "      <td>76108994</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108994</td>\n",
       "      <td>chr4:76108994:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>76109007</td>\n",
       "      <td>76109007</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76109007</td>\n",
       "      <td>chr4:76109007:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1844</th>\n",
       "      <td>4</td>\n",
       "      <td>76283418</td>\n",
       "      <td>76283418</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>FAM47E</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>FAM47E:NM_001136570:exon8:c.G1142A:p.R381H,FAM...</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76283418</td>\n",
       "      <td>chr4:76283418:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1845</th>\n",
       "      <td>4</td>\n",
       "      <td>76283566</td>\n",
       "      <td>76283566</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>FAM47E</td>\n",
       "      <td>NM_001242936:c.*108A&gt;T;NM_001136570:c.*108A&gt;T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76283566</td>\n",
       "      <td>chr4:76283566:A:T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1846</th>\n",
       "      <td>4</td>\n",
       "      <td>76284014</td>\n",
       "      <td>76284014</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284014</td>\n",
       "      <td>chr4:76284014:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1847</th>\n",
       "      <td>4</td>\n",
       "      <td>76284108</td>\n",
       "      <td>76284108</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284108</td>\n",
       "      <td>chr4:76284108:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1848</th>\n",
       "      <td>4</td>\n",
       "      <td>76284318</td>\n",
       "      <td>76284318</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284318</td>\n",
       "      <td>chr4:76284318:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1849 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       4  76108753  76108753   T   G     intronic          ART3   \n",
       "1       4  76108828  76108828   G   T     intronic          ART3   \n",
       "2       4  76108858  76108858   G   A     intronic          ART3   \n",
       "3       4  76108994  76108994   G   A     intronic          ART3   \n",
       "4       4  76109007  76109007   G   A     intronic          ART3   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "1844    4  76283418  76283418   G   A       exonic        FAM47E   \n",
       "1845    4  76283566  76283566   A   T         UTR3        FAM47E   \n",
       "1846    4  76284014  76284014   A   G     intronic  FAM47E-STBD1   \n",
       "1847    4  76284108  76284108   G   C     intronic  FAM47E-STBD1   \n",
       "1848    4  76284318  76284318   G   T     intronic  FAM47E-STBD1   \n",
       "\n",
       "                                 GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                                                 .                  .   \n",
       "1                                                 .                  .   \n",
       "2                                                 .                  .   \n",
       "3                                                 .                  .   \n",
       "4                                                 .                  .   \n",
       "...                                             ...                ...   \n",
       "1844                                              .  nonsynonymous SNV   \n",
       "1845  NM_001242936:c.*108A>T;NM_001136570:c.*108A>T                  .   \n",
       "1846                                              .                  .   \n",
       "1847                                              .                  .   \n",
       "1848                                              .                  .   \n",
       "\n",
       "                                       AAChange.refGene clinvar_20140902  \\\n",
       "0                                                     .                .   \n",
       "1                                                     .                .   \n",
       "2                                                     .                .   \n",
       "3                                                     .                .   \n",
       "4                                                     .                .   \n",
       "...                                                 ...              ...   \n",
       "1844  FAM47E:NM_001136570:exon8:c.G1142A:p.R381H,FAM...                .   \n",
       "1845                                                  .                .   \n",
       "1846                                                  .                .   \n",
       "1847                                                  .                .   \n",
       "1848                                                  .                .   \n",
       "\n",
       "      Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  Otherinfo5  \\\n",
       "0            0.0          .          .           4    76108753   \n",
       "1            0.5          .          .           4    76108828   \n",
       "2            0.5          .          .           4    76108858   \n",
       "3            0.5          .          .           4    76108994   \n",
       "4            0.5          .          .           4    76109007   \n",
       "...          ...        ...        ...         ...         ...   \n",
       "1844         0.0          .          .           4    76283418   \n",
       "1845         0.0          .          .           4    76283566   \n",
       "1846         0.0          .          .           4    76284014   \n",
       "1847         0.0          .          .           4    76284108   \n",
       "1848         0.0          .          .           4    76284318   \n",
       "\n",
       "             Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9 Otherinfo10  \\\n",
       "0     chr4:76108753:T:G          T          G          .           .   \n",
       "1     chr4:76108828:G:T          G          T          .           .   \n",
       "2     chr4:76108858:G:A          G          A          .           .   \n",
       "3     chr4:76108994:G:A          G          A          .           .   \n",
       "4     chr4:76109007:G:A          G          A          .           .   \n",
       "...                 ...        ...        ...        ...         ...   \n",
       "1844  chr4:76283418:G:A          G          A          .           .   \n",
       "1845  chr4:76283566:A:T          A          T          .           .   \n",
       "1846  chr4:76284014:A:G          A          G          .           .   \n",
       "1847  chr4:76284108:G:C          G          C          .           .   \n",
       "1848  chr4:76284318:G:T          G          T          .           .   \n",
       "\n",
       "     Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0             PR          GT         0/0  \n",
       "1             PR          GT         0/1  \n",
       "2             PR          GT         0/1  \n",
       "3             PR          GT         0/1  \n",
       "4             PR          GT         0/1  \n",
       "...          ...         ...         ...  \n",
       "1844          PR          GT         0/0  \n",
       "1845          PR          GT         0/0  \n",
       "1846          PR          GT         0/0  \n",
       "1847          PR          GT         0/0  \n",
       "1848          PR          GT         0/0  \n",
       "\n",
       "[1849 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/SCARB2_AJ/AJ_SCARB2.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "ba04da14-e82f-4c1f-b87d-09f46cfdd715",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      1605\n",
       "intergenic     111\n",
       "UTR3            42\n",
       "downstream      33\n",
       "exonic          31\n",
       "upstream        21\n",
       "UTR5             5\n",
       "splicing         1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20547643-a8ac-4bef-b5f4-44151be0305d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "51662fec-213a-4902-b106-bcc8a2e503bf",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    19\n",
       "synonymous SNV       12\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "181990e3-c6fd-4864-a7b5-cc42c183ae3e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>76108753</td>\n",
       "      <td>76108753</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108753</td>\n",
       "      <td>chr4:76108753:T:G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>76108755</td>\n",
       "      <td>76108755</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108755</td>\n",
       "      <td>chr4:76108755:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>76108804</td>\n",
       "      <td>76108804</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108804</td>\n",
       "      <td>chr4:76108804:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>76108828</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>chr4:76108828:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>76108858</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>chr4:76108858:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2511</th>\n",
       "      <td>4</td>\n",
       "      <td>76283751</td>\n",
       "      <td>76283751</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>FAM47E</td>\n",
       "      <td>NM_001242936:c.*293G&gt;A;NM_001136570:c.*293G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76283751</td>\n",
       "      <td>chr4:76283751:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2512</th>\n",
       "      <td>4</td>\n",
       "      <td>76283911</td>\n",
       "      <td>76283911</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76283911</td>\n",
       "      <td>chr4:76283911:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2513</th>\n",
       "      <td>4</td>\n",
       "      <td>76284105</td>\n",
       "      <td>76284105</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284104</td>\n",
       "      <td>chr4:76284104:TA:T</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2514</th>\n",
       "      <td>4</td>\n",
       "      <td>76284108</td>\n",
       "      <td>76284108</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284108</td>\n",
       "      <td>chr4:76284108:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2515</th>\n",
       "      <td>4</td>\n",
       "      <td>76284509</td>\n",
       "      <td>76284509</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284509</td>\n",
       "      <td>chr4:76284509:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2516 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       4  76108753  76108753   T   G     intronic          ART3   \n",
       "1       4  76108755  76108755   T   C     intronic          ART3   \n",
       "2       4  76108804  76108804   T   C     intronic          ART3   \n",
       "3       4  76108828  76108828   G   T     intronic          ART3   \n",
       "4       4  76108858  76108858   G   A     intronic          ART3   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "2511    4  76283751  76283751   G   A         UTR3        FAM47E   \n",
       "2512    4  76283911  76283911   C   T     intronic  FAM47E-STBD1   \n",
       "2513    4  76284105  76284105   A   -     intronic  FAM47E-STBD1   \n",
       "2514    4  76284108  76284108   G   C     intronic  FAM47E-STBD1   \n",
       "2515    4  76284509  76284509   G   A     intronic  FAM47E-STBD1   \n",
       "\n",
       "                                 GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                                                 .                  .   \n",
       "1                                                 .                  .   \n",
       "2                                                 .                  .   \n",
       "3                                                 .                  .   \n",
       "4                                                 .                  .   \n",
       "...                                             ...                ...   \n",
       "2511  NM_001242936:c.*293G>A;NM_001136570:c.*293G>A                  .   \n",
       "2512                                              .                  .   \n",
       "2513                                              .                  .   \n",
       "2514                                              .                  .   \n",
       "2515                                              .                  .   \n",
       "\n",
       "     AAChange.refGene clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  \\\n",
       "0                   .                .          0          .          .   \n",
       "1                   .                .          0          .          .   \n",
       "2                   .                .          0          .          .   \n",
       "3                   .                .        0.5          .          .   \n",
       "4                   .                .        0.5          .          .   \n",
       "...               ...              ...        ...        ...        ...   \n",
       "2511                .                .          0          .          .   \n",
       "2512                .                .          0          .          .   \n",
       "2513                .                .          0          .          .   \n",
       "2514                .                .          0          .          .   \n",
       "2515                .                .          0          .          .   \n",
       "\n",
       "      Otherinfo4  Otherinfo5          Otherinfo6 Otherinfo7 Otherinfo8  \\\n",
       "0              4    76108753   chr4:76108753:T:G          T          G   \n",
       "1              4    76108755   chr4:76108755:T:C          T          C   \n",
       "2              4    76108804   chr4:76108804:T:C          T          C   \n",
       "3              4    76108828   chr4:76108828:G:T          G          T   \n",
       "4              4    76108858   chr4:76108858:G:A          G          A   \n",
       "...          ...         ...                 ...        ...        ...   \n",
       "2511           4    76283751   chr4:76283751:G:A          G          A   \n",
       "2512           4    76283911   chr4:76283911:C:T          C          T   \n",
       "2513           4    76284104  chr4:76284104:TA:T         TA          T   \n",
       "2514           4    76284108   chr4:76284108:G:C          G          C   \n",
       "2515           4    76284509   chr4:76284509:G:A          G          A   \n",
       "\n",
       "     Otherinfo9 Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0             .           .          PR          GT         0/0  \n",
       "1             .           .          PR          GT         0/0  \n",
       "2             .           .          PR          GT         0/0  \n",
       "3             .           .          PR          GT         0/1  \n",
       "4             .           .          PR          GT         0/1  \n",
       "...         ...         ...         ...         ...         ...  \n",
       "2511          .           .          PR          GT         0/0  \n",
       "2512          .           .          PR          GT         0/0  \n",
       "2513          .           .          PR          GT         0/0  \n",
       "2514          .           .          PR          GT         0/0  \n",
       "2515          .           .          PR          GT         0/0  \n",
       "\n",
       "[2516 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/SCARB2_AMR/AMR_SCARB2.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "44536728-4775-46eb-bd83-2e62eb848fe5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      2212\n",
       "intergenic     137\n",
       "UTR3            47\n",
       "downstream      44\n",
       "exonic          43\n",
       "upstream        22\n",
       "UTR5             9\n",
       "splicing         2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1e74d5cb-39ea-44bc-981d-a98db467e9ef",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "364a976b-e4b1-4798-ba6c-41f2ce80fb03",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    29\n",
       "synonymous SNV       14\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "1cc4eeb4-212f-405d-b7d7-5e040c61c5e7",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>76108753</td>\n",
       "      <td>76108753</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108753</td>\n",
       "      <td>chr4:76108753:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>76108755</td>\n",
       "      <td>76108755</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108755</td>\n",
       "      <td>chr4:76108755:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>76108804</td>\n",
       "      <td>76108804</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108804</td>\n",
       "      <td>chr4:76108804:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>76108828</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>chr4:76108828:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>76108848</td>\n",
       "      <td>76108848</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108848</td>\n",
       "      <td>chr4:76108848:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3545</th>\n",
       "      <td>4</td>\n",
       "      <td>76284241</td>\n",
       "      <td>76284241</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284241</td>\n",
       "      <td>chr4:76284241:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3546</th>\n",
       "      <td>4</td>\n",
       "      <td>76284281</td>\n",
       "      <td>76284281</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284281</td>\n",
       "      <td>chr4:76284281:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3547</th>\n",
       "      <td>4</td>\n",
       "      <td>76284298</td>\n",
       "      <td>76284298</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284298</td>\n",
       "      <td>chr4:76284298:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3548</th>\n",
       "      <td>4</td>\n",
       "      <td>76284303</td>\n",
       "      <td>76284303</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284303</td>\n",
       "      <td>chr4:76284303:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3549</th>\n",
       "      <td>4</td>\n",
       "      <td>76284397</td>\n",
       "      <td>76284397</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284397</td>\n",
       "      <td>chr4:76284397:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3550 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       4  76108753  76108753   T   C     intronic          ART3   \n",
       "1       4  76108755  76108755   T   C     intronic          ART3   \n",
       "2       4  76108804  76108804   T   C     intronic          ART3   \n",
       "3       4  76108828  76108828   G   T     intronic          ART3   \n",
       "4       4  76108848  76108848   C   T     intronic          ART3   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "3545    4  76284241  76284241   C   T     intronic  FAM47E-STBD1   \n",
       "3546    4  76284281  76284281   C   A     intronic  FAM47E-STBD1   \n",
       "3547    4  76284298  76284298   C   T     intronic  FAM47E-STBD1   \n",
       "3548    4  76284303  76284303   C   A     intronic  FAM47E-STBD1   \n",
       "3549    4  76284397  76284397   A   G     intronic  FAM47E-STBD1   \n",
       "\n",
       "     GeneDetail.refGene ExonicFunc.refGene AAChange.refGene clinvar_20140902  \\\n",
       "0                     .                  .                .                .   \n",
       "1                     .                  .                .                .   \n",
       "2                     .                  .                .                .   \n",
       "3                     .                  .                .                .   \n",
       "4                     .                  .                .                .   \n",
       "...                 ...                ...              ...              ...   \n",
       "3545                  .                  .                .                .   \n",
       "3546                  .                  .                .                .   \n",
       "3547                  .                  .                .                .   \n",
       "3548                  .                  .                .                .   \n",
       "3549                  .                  .                .                .   \n",
       "\n",
       "      Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  Otherinfo5  \\\n",
       "0            0.0          .          .           4    76108753   \n",
       "1            0.0          .          .           4    76108755   \n",
       "2            0.0          .          .           4    76108804   \n",
       "3            0.0          .          .           4    76108828   \n",
       "4            0.0          .          .           4    76108848   \n",
       "...          ...        ...        ...         ...         ...   \n",
       "3545         0.0          .          .           4    76284241   \n",
       "3546         0.0          .          .           4    76284281   \n",
       "3547         0.0          .          .           4    76284298   \n",
       "3548         0.0          .          .           4    76284303   \n",
       "3549         0.0          .          .           4    76284397   \n",
       "\n",
       "             Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9 Otherinfo10  \\\n",
       "0     chr4:76108753:T:C          T          C          .           .   \n",
       "1     chr4:76108755:T:C          T          C          .           .   \n",
       "2     chr4:76108804:T:C          T          C          .           .   \n",
       "3     chr4:76108828:G:T          G          T          .           .   \n",
       "4     chr4:76108848:C:T          C          T          .           .   \n",
       "...                 ...        ...        ...        ...         ...   \n",
       "3545  chr4:76284241:C:T          C          T          .           .   \n",
       "3546  chr4:76284281:C:A          C          A          .           .   \n",
       "3547  chr4:76284298:C:T          C          T          .           .   \n",
       "3548  chr4:76284303:C:A          C          A          .           .   \n",
       "3549  chr4:76284397:A:G          A          G          .           .   \n",
       "\n",
       "     Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0             PR          GT         0/0  \n",
       "1             PR          GT         0/0  \n",
       "2             PR          GT         0/0  \n",
       "3             PR          GT         0/0  \n",
       "4             PR          GT         0/0  \n",
       "...          ...         ...         ...  \n",
       "3545          PR          GT         0/0  \n",
       "3546          PR          GT         0/0  \n",
       "3547          PR          GT         0/0  \n",
       "3548          PR          GT         0/0  \n",
       "3549          PR          GT         0/0  \n",
       "\n",
       "[3550 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/SCARB2_CAH/CAH_SCARB2.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "7f426f3c-6a86-4502-b145-0f6e73a20568",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      3125\n",
       "intergenic     189\n",
       "UTR3            71\n",
       "downstream      66\n",
       "exonic          52\n",
       "upstream        25\n",
       "UTR5            19\n",
       "splicing         3\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1b87e6c6-ddbe-4f89-ac22-21c6bab82039",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "d11b0870-76bc-497a-9e5c-633c847f2417",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    29\n",
       "synonymous SNV       22\n",
       "stopgain              1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "c9149147-c452-48da-933a-03692f101fab",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>76108828</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>chr4:76108828:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>76108858</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>chr4:76108858:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>76108861</td>\n",
       "      <td>76108864</td>\n",
       "      <td>GTAA</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108860</td>\n",
       "      <td>chr4:76108860:GGTAA:G</td>\n",
       "      <td>GGTAA</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>76108994</td>\n",
       "      <td>76108994</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108994</td>\n",
       "      <td>chr4:76108994:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>76109007</td>\n",
       "      <td>76109007</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76109007</td>\n",
       "      <td>chr4:76109007:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2214</th>\n",
       "      <td>4</td>\n",
       "      <td>76283347</td>\n",
       "      <td>76283347</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E;FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76283347</td>\n",
       "      <td>chr4:76283347:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2215</th>\n",
       "      <td>4</td>\n",
       "      <td>76283509</td>\n",
       "      <td>76283509</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>FAM47E</td>\n",
       "      <td>NM_001242936:c.*51A&gt;G;NM_001136570:c.*51A&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76283509</td>\n",
       "      <td>chr4:76283509:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2216</th>\n",
       "      <td>4</td>\n",
       "      <td>76284107</td>\n",
       "      <td>76284107</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284107</td>\n",
       "      <td>chr4:76284107:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2217</th>\n",
       "      <td>4</td>\n",
       "      <td>76284122</td>\n",
       "      <td>76284122</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284122</td>\n",
       "      <td>chr4:76284122:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2218</th>\n",
       "      <td>4</td>\n",
       "      <td>76284241</td>\n",
       "      <td>76284241</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284241</td>\n",
       "      <td>chr4:76284241:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2219 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End   Ref Alt Func.refGene         Gene.refGene  \\\n",
       "0       4  76108828  76108828     G   T     intronic                 ART3   \n",
       "1       4  76108858  76108858     G   A     intronic                 ART3   \n",
       "2       4  76108861  76108864  GTAA   -     intronic                 ART3   \n",
       "3       4  76108994  76108994     G   A     intronic                 ART3   \n",
       "4       4  76109007  76109007     G   A     intronic                 ART3   \n",
       "...   ...       ...       ...   ...  ..          ...                  ...   \n",
       "2214    4  76283347  76283347     C   T     intronic  FAM47E;FAM47E-STBD1   \n",
       "2215    4  76283509  76283509     A   G         UTR3               FAM47E   \n",
       "2216    4  76284107  76284107     A   G     intronic         FAM47E-STBD1   \n",
       "2217    4  76284122  76284122     C   G     intronic         FAM47E-STBD1   \n",
       "2218    4  76284241  76284241     C   T     intronic         FAM47E-STBD1   \n",
       "\n",
       "                               GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                                               .                  .   \n",
       "1                                               .                  .   \n",
       "2                                               .                  .   \n",
       "3                                               .                  .   \n",
       "4                                               .                  .   \n",
       "...                                           ...                ...   \n",
       "2214                                            .                  .   \n",
       "2215  NM_001242936:c.*51A>G;NM_001136570:c.*51A>G                  .   \n",
       "2216                                            .                  .   \n",
       "2217                                            .                  .   \n",
       "2218                                            .                  .   \n",
       "\n",
       "     AAChange.refGene clinvar_20140902  Otherinfo1 Otherinfo2 Otherinfo3  \\\n",
       "0                   .                .         0.5          .          .   \n",
       "1                   .                .         0.5          .          .   \n",
       "2                   .                .         0.0          .          .   \n",
       "3                   .                .         0.0          .          .   \n",
       "4                   .                .         0.0          .          .   \n",
       "...               ...              ...         ...        ...        ...   \n",
       "2214                .                .         0.0          .          .   \n",
       "2215                .                .         0.0          .          .   \n",
       "2216                .                .         0.0          .          .   \n",
       "2217                .                .         0.0          .          .   \n",
       "2218                .                .         0.0          .          .   \n",
       "\n",
       "      Otherinfo4  Otherinfo5             Otherinfo6 Otherinfo7 Otherinfo8  \\\n",
       "0              4    76108828      chr4:76108828:G:T          G          T   \n",
       "1              4    76108858      chr4:76108858:G:A          G          A   \n",
       "2              4    76108860  chr4:76108860:GGTAA:G      GGTAA          G   \n",
       "3              4    76108994      chr4:76108994:G:A          G          A   \n",
       "4              4    76109007      chr4:76109007:G:A          G          A   \n",
       "...          ...         ...                    ...        ...        ...   \n",
       "2214           4    76283347      chr4:76283347:C:T          C          T   \n",
       "2215           4    76283509      chr4:76283509:A:G          A          G   \n",
       "2216           4    76284107      chr4:76284107:A:G          A          G   \n",
       "2217           4    76284122      chr4:76284122:C:G          C          G   \n",
       "2218           4    76284241      chr4:76284241:C:T          C          T   \n",
       "\n",
       "     Otherinfo9 Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0             .           .          PR          GT         0/1  \n",
       "1             .           .          PR          GT         0/1  \n",
       "2             .           .          PR          GT         0/0  \n",
       "3             .           .          PR          GT         0/0  \n",
       "4             .           .          PR          GT         0/0  \n",
       "...         ...         ...         ...         ...         ...  \n",
       "2214          .           .          PR          GT         0/0  \n",
       "2215          .           .          PR          GT         0/0  \n",
       "2216          .           .          PR          GT         0/0  \n",
       "2217          .           .          PR          GT         0/0  \n",
       "2218          .           .          PR          GT         0/0  \n",
       "\n",
       "[2219 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/SCARB2_CAS/CAS_SCARB2.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "e5896149-1cca-4fa8-b56a-52534e6b332d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      1945\n",
       "intergenic     120\n",
       "UTR3            51\n",
       "downstream      37\n",
       "exonic          36\n",
       "upstream        19\n",
       "UTR5            10\n",
       "splicing         1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed5c25d0-4516-4c7c-a02d-3226cf5f1cd9",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "9521b3f6-ca50-496d-825f-7d18cd14c33c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    21\n",
       "synonymous SNV       14\n",
       "stopgain              1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "51622636-f490-42f6-9e6a-19b37e2919b7",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>76108828</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>chr4:76108828:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>76108858</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>chr4:76108858:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>76108994</td>\n",
       "      <td>76108994</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108994</td>\n",
       "      <td>chr4:76108994:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>76109007</td>\n",
       "      <td>76109007</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76109007</td>\n",
       "      <td>chr4:76109007:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>76109074</td>\n",
       "      <td>76109074</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76109074</td>\n",
       "      <td>chr4:76109074:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3089</th>\n",
       "      <td>4</td>\n",
       "      <td>76284241</td>\n",
       "      <td>76284241</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284241</td>\n",
       "      <td>chr4:76284241:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3090</th>\n",
       "      <td>4</td>\n",
       "      <td>76284298</td>\n",
       "      <td>76284298</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284298</td>\n",
       "      <td>chr4:76284298:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3091</th>\n",
       "      <td>4</td>\n",
       "      <td>76284478</td>\n",
       "      <td>76284478</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284478</td>\n",
       "      <td>chr4:76284478:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3092</th>\n",
       "      <td>4</td>\n",
       "      <td>76284489</td>\n",
       "      <td>76284489</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284489</td>\n",
       "      <td>chr4:76284489:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3093</th>\n",
       "      <td>4</td>\n",
       "      <td>76284493</td>\n",
       "      <td>76284493</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284493</td>\n",
       "      <td>chr4:76284493:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3094 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       4  76108828  76108828   G   T     intronic          ART3   \n",
       "1       4  76108858  76108858   G   A     intronic          ART3   \n",
       "2       4  76108994  76108994   G   A     intronic          ART3   \n",
       "3       4  76109007  76109007   G   A     intronic          ART3   \n",
       "4       4  76109074  76109074   A   G     intronic          ART3   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "3089    4  76284241  76284241   C   T     intronic  FAM47E-STBD1   \n",
       "3090    4  76284298  76284298   C   T     intronic  FAM47E-STBD1   \n",
       "3091    4  76284478  76284478   C   T     intronic  FAM47E-STBD1   \n",
       "3092    4  76284489  76284489   C   T     intronic  FAM47E-STBD1   \n",
       "3093    4  76284493  76284493   C   T     intronic  FAM47E-STBD1   \n",
       "\n",
       "     GeneDetail.refGene ExonicFunc.refGene AAChange.refGene clinvar_20140902  \\\n",
       "0                     .                  .                .                .   \n",
       "1                     .                  .                .                .   \n",
       "2                     .                  .                .                .   \n",
       "3                     .                  .                .                .   \n",
       "4                     .                  .                .                .   \n",
       "...                 ...                ...              ...              ...   \n",
       "3089                  .                  .                .                .   \n",
       "3090                  .                  .                .                .   \n",
       "3091                  .                  .                .                .   \n",
       "3092                  .                  .                .                .   \n",
       "3093                  .                  .                .                .   \n",
       "\n",
       "     Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  Otherinfo5  \\\n",
       "0             0          .          .           4    76108828   \n",
       "1             0          .          .           4    76108858   \n",
       "2             1          .          .           4    76108994   \n",
       "3             1          .          .           4    76109007   \n",
       "4             0          .          .           4    76109074   \n",
       "...         ...        ...        ...         ...         ...   \n",
       "3089          0          .          .           4    76284241   \n",
       "3090          0          .          .           4    76284298   \n",
       "3091          0          .          .           4    76284478   \n",
       "3092          0          .          .           4    76284489   \n",
       "3093          0          .          .           4    76284493   \n",
       "\n",
       "             Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9 Otherinfo10  \\\n",
       "0     chr4:76108828:G:T          G          T          .           .   \n",
       "1     chr4:76108858:G:A          G          A          .           .   \n",
       "2     chr4:76108994:G:A          G          A          .           .   \n",
       "3     chr4:76109007:G:A          G          A          .           .   \n",
       "4     chr4:76109074:A:G          A          G          .           .   \n",
       "...                 ...        ...        ...        ...         ...   \n",
       "3089  chr4:76284241:C:T          C          T          .           .   \n",
       "3090  chr4:76284298:C:T          C          T          .           .   \n",
       "3091  chr4:76284478:C:T          C          T          .           .   \n",
       "3092  chr4:76284489:C:T          C          T          .           .   \n",
       "3093  chr4:76284493:C:T          C          T          .           .   \n",
       "\n",
       "     Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0             PR          GT         0/0  \n",
       "1             PR          GT         0/0  \n",
       "2             PR          GT         1/1  \n",
       "3             PR          GT         1/1  \n",
       "4             PR          GT         0/0  \n",
       "...          ...         ...         ...  \n",
       "3089          PR          GT         0/0  \n",
       "3090          PR          GT         0/0  \n",
       "3091          PR          GT         0/0  \n",
       "3092          PR          GT         0/0  \n",
       "3093          PR          GT         0/0  \n",
       "\n",
       "[3094 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/SCARB2_EAS/EAS_SCARB2.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "90ffa388-b6a9-4395-b020-2bde2b3334c4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      2670\n",
       "intergenic     170\n",
       "UTR3            84\n",
       "downstream      59\n",
       "exonic          58\n",
       "UTR5            26\n",
       "upstream        25\n",
       "splicing         2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ab70fc2a-9bdc-4d52-9ce8-edf6f89fe7cd",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "2bf0839b-9cbe-40c2-90e0-3ee617001d3a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV      35\n",
       "synonymous SNV         21\n",
       "stopgain                1\n",
       "frameshift deletion     1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "58d97421-dd7b-4ca7-90d7-ea3258b12d44",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>76108755</td>\n",
       "      <td>76108755</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108755</td>\n",
       "      <td>chr4:76108755:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>76108804</td>\n",
       "      <td>76108804</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108804</td>\n",
       "      <td>chr4:76108804:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>76108820</td>\n",
       "      <td>76108820</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108820</td>\n",
       "      <td>chr4:76108820:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>76108828</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>chr4:76108828:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>76108858</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>chr4:76108858:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2375</th>\n",
       "      <td>4</td>\n",
       "      <td>76283774</td>\n",
       "      <td>76283774</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>FAM47E</td>\n",
       "      <td>NM_001242936:c.*316T&gt;G;NM_001136570:c.*316T&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76283774</td>\n",
       "      <td>chr4:76283774:T:G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2376</th>\n",
       "      <td>4</td>\n",
       "      <td>76284107</td>\n",
       "      <td>76284107</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284107</td>\n",
       "      <td>chr4:76284107:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2377</th>\n",
       "      <td>4</td>\n",
       "      <td>76284108</td>\n",
       "      <td>76284108</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284108</td>\n",
       "      <td>chr4:76284108:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2378</th>\n",
       "      <td>4</td>\n",
       "      <td>76284127</td>\n",
       "      <td>76284127</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284127</td>\n",
       "      <td>chr4:76284127:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2379</th>\n",
       "      <td>4</td>\n",
       "      <td>76284298</td>\n",
       "      <td>76284298</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284298</td>\n",
       "      <td>chr4:76284298:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2380 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       4  76108755  76108755   T   C     intronic          ART3   \n",
       "1       4  76108804  76108804   T   C     intronic          ART3   \n",
       "2       4  76108820  76108820   A   G     intronic          ART3   \n",
       "3       4  76108828  76108828   G   T     intronic          ART3   \n",
       "4       4  76108858  76108858   G   A     intronic          ART3   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "2375    4  76283774  76283774   T   G         UTR3        FAM47E   \n",
       "2376    4  76284107  76284107   A   G     intronic  FAM47E-STBD1   \n",
       "2377    4  76284108  76284108   G   C     intronic  FAM47E-STBD1   \n",
       "2378    4  76284127  76284127   C   T     intronic  FAM47E-STBD1   \n",
       "2379    4  76284298  76284298   C   T     intronic  FAM47E-STBD1   \n",
       "\n",
       "                                 GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                                                 .                  .   \n",
       "1                                                 .                  .   \n",
       "2                                                 .                  .   \n",
       "3                                                 .                  .   \n",
       "4                                                 .                  .   \n",
       "...                                             ...                ...   \n",
       "2375  NM_001242936:c.*316T>G;NM_001136570:c.*316T>G                  .   \n",
       "2376                                              .                  .   \n",
       "2377                                              .                  .   \n",
       "2378                                              .                  .   \n",
       "2379                                              .                  .   \n",
       "\n",
       "     AAChange.refGene clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  \\\n",
       "0                   .                .          0          .          .   \n",
       "1                   .                .          0          .          .   \n",
       "2                   .                .          0          .          .   \n",
       "3                   .                .        0.5          .          .   \n",
       "4                   .                .        0.5          .          .   \n",
       "...               ...              ...        ...        ...        ...   \n",
       "2375                .                .          0          .          .   \n",
       "2376                .                .          0          .          .   \n",
       "2377                .                .          0          .          .   \n",
       "2378                .                .          0          .          .   \n",
       "2379                .                .          0          .          .   \n",
       "\n",
       "      Otherinfo4  Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8  \\\n",
       "0              4    76108755  chr4:76108755:T:C          T          C   \n",
       "1              4    76108804  chr4:76108804:T:C          T          C   \n",
       "2              4    76108820  chr4:76108820:A:G          A          G   \n",
       "3              4    76108828  chr4:76108828:G:T          G          T   \n",
       "4              4    76108858  chr4:76108858:G:A          G          A   \n",
       "...          ...         ...                ...        ...        ...   \n",
       "2375           4    76283774  chr4:76283774:T:G          T          G   \n",
       "2376           4    76284107  chr4:76284107:A:G          A          G   \n",
       "2377           4    76284108  chr4:76284108:G:C          G          C   \n",
       "2378           4    76284127  chr4:76284127:C:T          C          T   \n",
       "2379           4    76284298  chr4:76284298:C:T          C          T   \n",
       "\n",
       "     Otherinfo9 Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0             .           .          PR          GT         0/0  \n",
       "1             .           .          PR          GT         0/0  \n",
       "2             .           .          PR          GT         0/0  \n",
       "3             .           .          PR          GT         0/1  \n",
       "4             .           .          PR          GT         0/1  \n",
       "...         ...         ...         ...         ...         ...  \n",
       "2375          .           .          PR          GT         0/0  \n",
       "2376          .           .          PR          GT         0/0  \n",
       "2377          .           .          PR          GT         0/0  \n",
       "2378          .           .          PR          GT         0/0  \n",
       "2379          .           .          PR          GT         0/0  \n",
       "\n",
       "[2380 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/SCARB2_MDE/MDE_SCARB2.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "31b8427b-245f-44de-9140-73dd22925a2b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      2116\n",
       "intergenic     116\n",
       "UTR3            42\n",
       "downstream      37\n",
       "exonic          36\n",
       "upstream        20\n",
       "UTR5            11\n",
       "splicing         2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "22a100aa-df60-402c-a912-15896fef84ea",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "6862edaa-bbfc-4178-8e40-c293b8121fdb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    26\n",
       "synonymous SNV       10\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "c964554b-62d2-4159-bb37-815fc5512dce",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>76108828</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>chr4:76108828:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>76108858</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>chr4:76108858:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>76108861</td>\n",
       "      <td>76108864</td>\n",
       "      <td>GTAA</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108860</td>\n",
       "      <td>chr4:76108860:GGTAA:G</td>\n",
       "      <td>GGTAA</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>76108972</td>\n",
       "      <td>76108972</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108971</td>\n",
       "      <td>chr4:76108971:GT:G</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>76108974</td>\n",
       "      <td>76108974</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108974</td>\n",
       "      <td>chr4:76108974:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2053</th>\n",
       "      <td>4</td>\n",
       "      <td>76283677</td>\n",
       "      <td>76283677</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>FAM47E</td>\n",
       "      <td>NM_001242936:c.*219T&gt;G;NM_001136570:c.*219T&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76283677</td>\n",
       "      <td>chr4:76283677:T:G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2054</th>\n",
       "      <td>4</td>\n",
       "      <td>76283721</td>\n",
       "      <td>76283721</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>FAM47E</td>\n",
       "      <td>NM_001242936:c.*263G&gt;A;NM_001136570:c.*263G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76283721</td>\n",
       "      <td>chr4:76283721:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2055</th>\n",
       "      <td>4</td>\n",
       "      <td>76284022</td>\n",
       "      <td>76284022</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284022</td>\n",
       "      <td>chr4:76284022:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2056</th>\n",
       "      <td>4</td>\n",
       "      <td>76284108</td>\n",
       "      <td>76284108</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284108</td>\n",
       "      <td>chr4:76284108:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2057</th>\n",
       "      <td>4</td>\n",
       "      <td>76284298</td>\n",
       "      <td>76284298</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284298</td>\n",
       "      <td>chr4:76284298:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2058 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End   Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       4  76108828  76108828     G   T     intronic          ART3   \n",
       "1       4  76108858  76108858     G   A     intronic          ART3   \n",
       "2       4  76108861  76108864  GTAA   -     intronic          ART3   \n",
       "3       4  76108972  76108972     T   -     intronic          ART3   \n",
       "4       4  76108974  76108974     G   A     intronic          ART3   \n",
       "...   ...       ...       ...   ...  ..          ...           ...   \n",
       "2053    4  76283677  76283677     T   G         UTR3        FAM47E   \n",
       "2054    4  76283721  76283721     G   A         UTR3        FAM47E   \n",
       "2055    4  76284022  76284022     G   A     intronic  FAM47E-STBD1   \n",
       "2056    4  76284108  76284108     G   C     intronic  FAM47E-STBD1   \n",
       "2057    4  76284298  76284298     C   T     intronic  FAM47E-STBD1   \n",
       "\n",
       "                                 GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                                                 .                  .   \n",
       "1                                                 .                  .   \n",
       "2                                                 .                  .   \n",
       "3                                                 .                  .   \n",
       "4                                                 .                  .   \n",
       "...                                             ...                ...   \n",
       "2053  NM_001242936:c.*219T>G;NM_001136570:c.*219T>G                  .   \n",
       "2054  NM_001242936:c.*263G>A;NM_001136570:c.*263G>A                  .   \n",
       "2055                                              .                  .   \n",
       "2056                                              .                  .   \n",
       "2057                                              .                  .   \n",
       "\n",
       "     AAChange.refGene clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  \\\n",
       "0                   .                .        0.5          .          .   \n",
       "1                   .                .        0.5          .          .   \n",
       "2                   .                .          0          .          .   \n",
       "3                   .                .          0          .          .   \n",
       "4                   .                .          0          .          .   \n",
       "...               ...              ...        ...        ...        ...   \n",
       "2053                .                .          0          .          .   \n",
       "2054                .                .          0          .          .   \n",
       "2055                .                .          0          .          .   \n",
       "2056                .                .          0          .          .   \n",
       "2057                .                .          0          .          .   \n",
       "\n",
       "      Otherinfo4  Otherinfo5             Otherinfo6 Otherinfo7 Otherinfo8  \\\n",
       "0              4    76108828      chr4:76108828:G:T          G          T   \n",
       "1              4    76108858      chr4:76108858:G:A          G          A   \n",
       "2              4    76108860  chr4:76108860:GGTAA:G      GGTAA          G   \n",
       "3              4    76108971     chr4:76108971:GT:G         GT          G   \n",
       "4              4    76108974      chr4:76108974:G:A          G          A   \n",
       "...          ...         ...                    ...        ...        ...   \n",
       "2053           4    76283677      chr4:76283677:T:G          T          G   \n",
       "2054           4    76283721      chr4:76283721:G:A          G          A   \n",
       "2055           4    76284022      chr4:76284022:G:A          G          A   \n",
       "2056           4    76284108      chr4:76284108:G:C          G          C   \n",
       "2057           4    76284298      chr4:76284298:C:T          C          T   \n",
       "\n",
       "     Otherinfo9 Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0             .           .          PR          GT         0/1  \n",
       "1             .           .          PR          GT         0/1  \n",
       "2             .           .          PR          GT         0/0  \n",
       "3             .           .          PR          GT         0/0  \n",
       "4             .           .          PR          GT         0/0  \n",
       "...         ...         ...         ...         ...         ...  \n",
       "2053          .           .          PR          GT         0/0  \n",
       "2054          .           .          PR          GT         0/0  \n",
       "2055          .           .          PR          GT         0/0  \n",
       "2056          .           .          PR          GT         0/0  \n",
       "2057          .           .          PR          GT         0/0  \n",
       "\n",
       "[2058 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/SCARB2_SAS/SAS_SCARB2.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "80c5954e-72f2-41b7-8c6a-5095b42f67a5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      1794\n",
       "intergenic     105\n",
       "UTR3            53\n",
       "downstream      40\n",
       "exonic          34\n",
       "upstream        22\n",
       "UTR5             9\n",
       "splicing         1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1a9e0215-d415-4938-bca4-d8349c8076a3",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "a80c39e4-2669-4a7a-aa31-afc7da296842",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    19\n",
       "synonymous SNV       15\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "fa7f06c2-46af-40e1-ac17-2780d50ed9ab",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>76108828</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>chr4:76108828:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>76108858</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>chr4:76108858:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>76108994</td>\n",
       "      <td>76108994</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108994</td>\n",
       "      <td>chr4:76108994:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>76109007</td>\n",
       "      <td>76109007</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76109007</td>\n",
       "      <td>chr4:76109007:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>76109102</td>\n",
       "      <td>76109102</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76109102</td>\n",
       "      <td>chr4:76109102:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1161</th>\n",
       "      <td>4</td>\n",
       "      <td>76283246</td>\n",
       "      <td>76283247</td>\n",
       "      <td>AT</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E;FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76283245</td>\n",
       "      <td>chr4:76283245:CAT:C</td>\n",
       "      <td>CAT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1162</th>\n",
       "      <td>4</td>\n",
       "      <td>76283278</td>\n",
       "      <td>76283279</td>\n",
       "      <td>AG</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E;FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76283277</td>\n",
       "      <td>chr4:76283277:TAG:T</td>\n",
       "      <td>TAG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1163</th>\n",
       "      <td>4</td>\n",
       "      <td>76283347</td>\n",
       "      <td>76283347</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E;FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76283347</td>\n",
       "      <td>chr4:76283347:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1164</th>\n",
       "      <td>4</td>\n",
       "      <td>76283587</td>\n",
       "      <td>76283587</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>FAM47E</td>\n",
       "      <td>NM_001242936:c.*129A&gt;G;NM_001136570:c.*129A&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76283587</td>\n",
       "      <td>chr4:76283587:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1165</th>\n",
       "      <td>4</td>\n",
       "      <td>76284108</td>\n",
       "      <td>76284108</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284108</td>\n",
       "      <td>chr4:76284108:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1166 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene         Gene.refGene  \\\n",
       "0       4  76108828  76108828   G   T     intronic                 ART3   \n",
       "1       4  76108858  76108858   G   A     intronic                 ART3   \n",
       "2       4  76108994  76108994   G   A     intronic                 ART3   \n",
       "3       4  76109007  76109007   G   A     intronic                 ART3   \n",
       "4       4  76109102  76109102   C   T     intronic                 ART3   \n",
       "...   ...       ...       ...  ..  ..          ...                  ...   \n",
       "1161    4  76283246  76283247  AT   -     intronic  FAM47E;FAM47E-STBD1   \n",
       "1162    4  76283278  76283279  AG   -     intronic  FAM47E;FAM47E-STBD1   \n",
       "1163    4  76283347  76283347   C   T     intronic  FAM47E;FAM47E-STBD1   \n",
       "1164    4  76283587  76283587   A   G         UTR3               FAM47E   \n",
       "1165    4  76284108  76284108   G   C     intronic         FAM47E-STBD1   \n",
       "\n",
       "                                 GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                                                 .                  .   \n",
       "1                                                 .                  .   \n",
       "2                                                 .                  .   \n",
       "3                                                 .                  .   \n",
       "4                                                 .                  .   \n",
       "...                                             ...                ...   \n",
       "1161                                              .                  .   \n",
       "1162                                              .                  .   \n",
       "1163                                              .                  .   \n",
       "1164  NM_001242936:c.*129A>G;NM_001136570:c.*129A>G                  .   \n",
       "1165                                              .                  .   \n",
       "\n",
       "     AAChange.refGene clinvar_20140902  Otherinfo1 Otherinfo2 Otherinfo3  \\\n",
       "0                   .                .         0.0          .          .   \n",
       "1                   .                .         0.0          .          .   \n",
       "2                   .                .         0.5          .          .   \n",
       "3                   .                .         0.5          .          .   \n",
       "4                   .                .         0.5          .          .   \n",
       "...               ...              ...         ...        ...        ...   \n",
       "1161                .                .         0.5          .          .   \n",
       "1162                .                .         0.0          .          .   \n",
       "1163                .                .         0.5          .          .   \n",
       "1164                .                .         0.0          .          .   \n",
       "1165                .                .         0.0          .          .   \n",
       "\n",
       "      Otherinfo4  Otherinfo5           Otherinfo6 Otherinfo7 Otherinfo8  \\\n",
       "0              4    76108828    chr4:76108828:G:T          G          T   \n",
       "1              4    76108858    chr4:76108858:G:A          G          A   \n",
       "2              4    76108994    chr4:76108994:G:A          G          A   \n",
       "3              4    76109007    chr4:76109007:G:A          G          A   \n",
       "4              4    76109102    chr4:76109102:C:T          C          T   \n",
       "...          ...         ...                  ...        ...        ...   \n",
       "1161           4    76283245  chr4:76283245:CAT:C        CAT          C   \n",
       "1162           4    76283277  chr4:76283277:TAG:T        TAG          T   \n",
       "1163           4    76283347    chr4:76283347:C:T          C          T   \n",
       "1164           4    76283587    chr4:76283587:A:G          A          G   \n",
       "1165           4    76284108    chr4:76284108:G:C          G          C   \n",
       "\n",
       "     Otherinfo9 Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0             .           .          PR          GT         0/0  \n",
       "1             .           .          PR          GT         0/0  \n",
       "2             .           .          PR          GT         0/1  \n",
       "3             .           .          PR          GT         0/1  \n",
       "4             .           .          PR          GT         0/1  \n",
       "...         ...         ...         ...         ...         ...  \n",
       "1161          .           .          PR          GT         0/1  \n",
       "1162          .           .          PR          GT         0/0  \n",
       "1163          .           .          PR          GT         0/1  \n",
       "1164          .           .          PR          GT         0/0  \n",
       "1165          .           .          PR          GT         0/0  \n",
       "\n",
       "[1166 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/SCARB2_FIN/FIN_SCARB2.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "ed37d506-31e8-4f52-94b8-18768f3f1fdb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      999\n",
       "intergenic     84\n",
       "downstream     24\n",
       "UTR3           23\n",
       "exonic         16\n",
       "upstream       15\n",
       "UTR5            4\n",
       "splicing        1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9a8c0d8f-8995-4471-995b-0b9ca830b016",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "06ffcd0f-9f85-40b5-ab95-5394900bb0f6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    10\n",
       "synonymous SNV        6\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "d3da38bf-1766-4a98-8c56-728550de35bb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>76108753</td>\n",
       "      <td>76108753</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108753</td>\n",
       "      <td>chr4:76108753:T:G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>./.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>76108755</td>\n",
       "      <td>76108755</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108755</td>\n",
       "      <td>chr4:76108755:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>76108761</td>\n",
       "      <td>76108761</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108761</td>\n",
       "      <td>chr4:76108761:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>76108794</td>\n",
       "      <td>76108794</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108794</td>\n",
       "      <td>chr4:76108794:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>76108804</td>\n",
       "      <td>76108804</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108804</td>\n",
       "      <td>chr4:76108804:T:A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9565</th>\n",
       "      <td>4</td>\n",
       "      <td>76284318</td>\n",
       "      <td>76284318</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284318</td>\n",
       "      <td>chr4:76284318:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9566</th>\n",
       "      <td>4</td>\n",
       "      <td>76284341</td>\n",
       "      <td>76284341</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284341</td>\n",
       "      <td>chr4:76284341:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9567</th>\n",
       "      <td>4</td>\n",
       "      <td>76284385</td>\n",
       "      <td>76284385</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284385</td>\n",
       "      <td>chr4:76284385:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9568</th>\n",
       "      <td>4</td>\n",
       "      <td>76284405</td>\n",
       "      <td>76284405</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284405</td>\n",
       "      <td>chr4:76284405:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9569</th>\n",
       "      <td>4</td>\n",
       "      <td>76284509</td>\n",
       "      <td>76284509</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284509</td>\n",
       "      <td>chr4:76284509:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>./.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9570 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       4  76108753  76108753   T   G     intronic          ART3   \n",
       "1       4  76108755  76108755   T   C     intronic          ART3   \n",
       "2       4  76108761  76108761   G   A     intronic          ART3   \n",
       "3       4  76108794  76108794   G   A     intronic          ART3   \n",
       "4       4  76108804  76108804   T   A     intronic          ART3   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "9565    4  76284318  76284318   G   T     intronic  FAM47E-STBD1   \n",
       "9566    4  76284341  76284341   C   T     intronic  FAM47E-STBD1   \n",
       "9567    4  76284385  76284385   G   A     intronic  FAM47E-STBD1   \n",
       "9568    4  76284405  76284405   A   G     intronic  FAM47E-STBD1   \n",
       "9569    4  76284509  76284509   G   A     intronic  FAM47E-STBD1   \n",
       "\n",
       "     GeneDetail.refGene ExonicFunc.refGene AAChange.refGene clinvar_20140902  \\\n",
       "0                     .                  .                .                .   \n",
       "1                     .                  .                .                .   \n",
       "2                     .                  .                .                .   \n",
       "3                     .                  .                .                .   \n",
       "4                     .                  .                .                .   \n",
       "...                 ...                ...              ...              ...   \n",
       "9565                  .                  .                .                .   \n",
       "9566                  .                  .                .                .   \n",
       "9567                  .                  .                .                .   \n",
       "9568                  .                  .                .                .   \n",
       "9569                  .                  .                .                .   \n",
       "\n",
       "     Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  Otherinfo5  \\\n",
       "0             .          .          .           4    76108753   \n",
       "1             0          .          .           4    76108755   \n",
       "2             0          .          .           4    76108761   \n",
       "3             0          .          .           4    76108794   \n",
       "4             0          .          .           4    76108804   \n",
       "...         ...        ...        ...         ...         ...   \n",
       "9565          0          .          .           4    76284318   \n",
       "9566          0          .          .           4    76284341   \n",
       "9567          0          .          .           4    76284385   \n",
       "9568          0          .          .           4    76284405   \n",
       "9569          .          .          .           4    76284509   \n",
       "\n",
       "             Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9 Otherinfo10  \\\n",
       "0     chr4:76108753:T:G          T          G          .           .   \n",
       "1     chr4:76108755:T:C          T          C          .           .   \n",
       "2     chr4:76108761:G:A          G          A          .           .   \n",
       "3     chr4:76108794:G:A          G          A          .           .   \n",
       "4     chr4:76108804:T:A          T          A          .           .   \n",
       "...                 ...        ...        ...        ...         ...   \n",
       "9565  chr4:76284318:G:T          G          T          .           .   \n",
       "9566  chr4:76284341:C:T          C          T          .           .   \n",
       "9567  chr4:76284385:G:A          G          A          .           .   \n",
       "9568  chr4:76284405:A:G          A          G          .           .   \n",
       "9569  chr4:76284509:G:A          G          A          .           .   \n",
       "\n",
       "     Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0             PR          GT         ./.  \n",
       "1             PR          GT         0/0  \n",
       "2             PR          GT         0/0  \n",
       "3             PR          GT         0/0  \n",
       "4             PR          GT         0/0  \n",
       "...          ...         ...         ...  \n",
       "9565          PR          GT         0/0  \n",
       "9566          PR          GT         0/0  \n",
       "9567          PR          GT         0/0  \n",
       "9568          PR          GT         0/0  \n",
       "9569          PR          GT         ./.  \n",
       "\n",
       "[9570 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/SCARB2_EUR/EUR_SCARB2.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "6fdac814-807d-493b-a227-5f19276e2f71",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      8370\n",
       "intergenic     473\n",
       "UTR3           225\n",
       "exonic         194\n",
       "downstream     156\n",
       "upstream        83\n",
       "UTR5            66\n",
       "splicing         3\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "c636ee15-3688-41dd-8fe8-f0a55da69a83",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "dd6b1db6-959e-4d81-a4ae-8f1878ad729d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV         128\n",
       "synonymous SNV             58\n",
       "stopgain                    4\n",
       "nonframeshift deletion      1\n",
       "frameshift insertion        1\n",
       "frameshift deletion         1\n",
       "stoploss                    1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "9ca15b00-9c4b-403e-9211-68255a4dd296",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AJ\n",
      "{'AJ'}\n",
      "Total variants:  1849\n",
      "Intronic:  1605\n",
      "UTR3:  42\n",
      "UTR5:  5\n",
      "Total exonic:  31\n",
      "  Synonymous:  12\n",
      "  Nonsynonymous:  19\n",
      "nonsynonymous_lof:  19\n",
      "\n",
      "\n",
      "WORKING ON: EAS\n",
      "{'EAS'}\n",
      "Total variants:  3094\n",
      "Intronic:  2670\n",
      "UTR3:  84\n",
      "UTR5:  26\n",
      "Total exonic:  58\n",
      "  Synonymous:  23\n",
      "  Nonsynonymous:  35\n",
      "nonsynonymous_lof:  37\n",
      "\n",
      "\n",
      "WORKING ON: SAS\n",
      "{'SAS'}\n",
      "Total variants:  2058\n",
      "Intronic:  1794\n",
      "UTR3:  53\n",
      "UTR5:  9\n",
      "Total exonic:  34\n",
      "  Synonymous:  15\n",
      "  Nonsynonymous:  19\n",
      "nonsynonymous_lof:  19\n",
      "\n",
      "\n",
      "WORKING ON: AMR\n",
      "{'AMR'}\n",
      "Total variants:  2516\n",
      "Intronic:  2212\n",
      "UTR3:  47\n",
      "UTR5:  9\n",
      "Total exonic:  43\n",
      "  Synonymous:  14\n",
      "  Nonsynonymous:  29\n",
      "nonsynonymous_lof:  29\n",
      "\n",
      "\n",
      "WORKING ON: MDE\n",
      "{'MDE'}\n",
      "Total variants:  2380\n",
      "Intronic:  2116\n",
      "UTR3:  42\n",
      "UTR5:  11\n",
      "Total exonic:  36\n",
      "  Synonymous:  10\n",
      "  Nonsynonymous:  26\n",
      "nonsynonymous_lof:  26\n",
      "\n",
      "\n",
      "WORKING ON: CAH\n",
      "{'CAH'}\n",
      "Total variants:  3550\n",
      "Intronic:  3125\n",
      "UTR3:  71\n",
      "UTR5:  19\n",
      "Total exonic:  52\n",
      "  Synonymous:  23\n",
      "  Nonsynonymous:  29\n",
      "nonsynonymous_lof:  30\n",
      "\n",
      "\n",
      "WORKING ON: FIN\n",
      "{'FIN'}\n",
      "Total variants:  1166\n",
      "Intronic:  999\n",
      "UTR3:  23\n",
      "UTR5:  4\n",
      "Total exonic:  16\n",
      "  Synonymous:  6\n",
      "  Nonsynonymous:  10\n",
      "nonsynonymous_lof:  10\n",
      "\n",
      "\n",
      "WORKING ON: CAS\n",
      "{'CAS'}\n",
      "Total variants:  2219\n",
      "Intronic:  1945\n",
      "UTR3:  51\n",
      "UTR5:  10\n",
      "Total exonic:  36\n",
      "  Synonymous:  15\n",
      "  Nonsynonymous:  21\n",
      "nonsynonymous_lof:  22\n",
      "\n",
      "\n",
      "WORKING ON: AFR\n",
      "{'AFR'}\n",
      "Total variants:  4304\n",
      "Intronic:  3759\n",
      "UTR3:  95\n",
      "UTR5:  26\n",
      "Total exonic:  64\n",
      "  Synonymous:  28\n",
      "  Nonsynonymous:  36\n",
      "nonsynonymous_lof:  38\n",
      "\n",
      "\n",
      "WORKING ON: EUR\n",
      "{'EUR'}\n",
      "Total variants:  9570\n",
      "Intronic:  8370\n",
      "UTR3:  225\n",
      "UTR5:  66\n",
      "Total exonic:  194\n",
      "  Synonymous:  66\n",
      "  Nonsynonymous:  128\n",
      "nonsynonymous_lof:  135\n",
      "\n",
      "\n",
      "WORKING ON: AAC\n",
      "{'AAC'}\n",
      "Total variants:  3848\n",
      "Intronic:  3385\n",
      "UTR3:  89\n",
      "UTR5:  22\n",
      "Total exonic:  59\n",
      "  Synonymous:  27\n",
      "  Nonsynonymous:  32\n",
      "nonsynonymous_lof:  34\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Make lists of variants to keep - all coding, coding nonsynonymous (missense - as they are coded in ANNOVAR), deleterious (CADD_phred > 20)\n",
    "\n",
    "for ancestry in ancestries:\n",
    "        \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    # Read in ANNOVAR multianno file\n",
    "    gene = pd.read_csv(f'{WORK_DIR}/{ancestry}_SCARB2.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "    \n",
    "    #Print number of variants in the different categories\n",
    "    results = [] \n",
    "    \n",
    "    utr5 = gene[gene['Func.refGene']== 'UTR5']\n",
    "    intronic = gene[gene['Func.refGene']== 'intronic']\n",
    "    exonic = gene[gene['Func.refGene']== 'exonic']\n",
    "    utr3 = gene[gene['Func.refGene']== 'UTR3']\n",
    "    coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "    coding_synonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] != 'nonsynonymous SNV')]\n",
    "    lof = exonic[(exonic['ExonicFunc.refGene'] == 'stopgain') | (exonic['ExonicFunc.refGene'] == 'stoploss') | (exonic['ExonicFunc.refGene'] == 'frameshift deletion') | (exonic['ExonicFunc.refGene'] == 'frameshift insertion')]\n",
    "    nonsynonymous_lof = pd.concat([coding_nonsynonymous, lof])\n",
    "    \n",
    "    print({ancestry})\n",
    "    print('Total variants: ', len(gene))\n",
    "    print(\"Intronic: \", len(intronic))\n",
    "    print('UTR3: ', len(utr3))\n",
    "    print('UTR5: ', len(utr5))\n",
    "    print(\"Total exonic: \", len(exonic))\n",
    "    print('  Synonymous: ', len(coding_synonymous))\n",
    "    print(\"  Nonsynonymous: \", len(coding_nonsynonymous))\n",
    "    print(\"nonsynonymous_lof: \", len(nonsynonymous_lof))\n",
    "    results.append((gene, intronic, utr3, utr5, exonic, coding_synonymous, coding_nonsynonymous, nonsynonymous_lof))\n",
    "    print('\\n')\n",
    "    \n",
    "    # Save in PLINK format - coding nonsynonymous \n",
    "    # These are missense variants - other types of nonsynonymous variants (e.g stopgain/loss, or frameshift variants are coded differently in the ExonicFunc.refGene \n",
    "    variants_toKeep = nonsynonymous_lof[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "    variants_toKeep.to_csv(f'{WORK_DIR}/{ancestry}_SCARB2.nonsynonymous_lof.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)\n",
    "    \n",
    "    # Save in PLINK format - all coding variants\n",
    "    variants_toKeep2 = exonic[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "    variants_toKeep2.to_csv(f'{WORK_DIR}/{ancestry}_SCARB2.exonic.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ed11958-0450-4539-bf27-20b72a47afe5",
   "metadata": {},
   "source": [
    "ALL variants\n",
    "\n",
    "assoc\n",
    "\n",
    "glm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc2e5d52-5d6f-408d-9150-80a11efaf80d",
   "metadata": {},
   "source": [
    "ASSOC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "ccbddcd1-e46b-417b-a12c-a807aaf7a495",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "3848 variants loaded from .bim file.\n",
      "1111 people (455 males, 656 females) loaded from .fam.\n",
      "1086 phenotype values loaded from .fam.\n",
      "--keep: 1111 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1111 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99922.\n",
      "2388 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1460 variants and 1111 people pass filters and QC.\n",
      "Among remaining phenotypes, 285 are cases and 801 are controls.  (25 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.all.assoc ...\n",
      "70done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "4304 variants loaded from .bim file.\n",
      "2643 people (1480 males, 1163 females) loaded from .fam.\n",
      "2621 phenotype values loaded from .fam.\n",
      "--keep: 2643 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2643 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998648.\n",
      "2791 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1513 variants and 2643 people pass filters and QC.\n",
      "Among remaining phenotypes, 942 are cases and 1679 are controls.  (22\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.all.assoc ...\n",
      "67done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "1849 variants loaded from .bim file.\n",
      "2655 people (1648 males, 1007 females) loaded from .fam.\n",
      "1703 phenotype values loaded from .fam.\n",
      "--keep: 2655 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2655 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99951.\n",
      "816 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1033 variants and 2655 people pass filters and QC.\n",
      "Among remaining phenotypes, 1292 are cases and 411 are controls.  (952\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.all.assoc ...\n",
      "99done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "2516 variants loaded from .bim file.\n",
      "646 people (363 males, 283 females) loaded from .fam.\n",
      "613 phenotype values loaded from .fam.\n",
      "--keep: 646 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 646 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99861.\n",
      "1569 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "947 variants and 646 people pass filters and QC.\n",
      "Among remaining phenotypes, 458 are cases and 155 are controls.  (33 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "2219 variants loaded from .bim file.\n",
      "903 people (439 males, 464 females) loaded from .fam.\n",
      "895 phenotype values loaded from .fam.\n",
      "--keep: 903 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 903 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996859.\n",
      "1248 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "971 variants and 903 people pass filters and QC.\n",
      "Among remaining phenotypes, 552 are cases and 343 are controls.  (8 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "3094 variants loaded from .bim file.\n",
      "5167 people (3278 males, 1889 females) loaded from .fam.\n",
      "5123 phenotype values loaded from .fam.\n",
      "--keep: 5167 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5167 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.995158.\n",
      "2143 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "951 variants and 5167 people pass filters and QC.\n",
      "Among remaining phenotypes, 2662 are cases and 2461 are controls.  (44\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "9570 variants loaded from .bim file.\n",
      "38839 people (23296 males, 15543 females) loaded from .fam.\n",
      "30412 phenotype values loaded from .fam.\n",
      "--keep: 38838 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 38838 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.787043.\n",
      "8651 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "919 variants and 38838 people pass filters and QC.\n",
      "Among remaining phenotypes, 21197 are cases and 9214 are controls.  (8427\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "1166 variants loaded from .bim file.\n",
      "114 people (53 males, 61 females) loaded from .fam.\n",
      "106 phenotype values loaded from .fam.\n",
      "--keep: 114 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 114 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99827.\n",
      "226 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "940 variants and 114 people pass filters and QC.\n",
      "Among remaining phenotypes, 98 are cases and 8 are controls.  (8 phenotypes are\n",
      "missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "2380 variants loaded from .bim file.\n",
      "581 people (369 males, 212 females) loaded from .fam.\n",
      "536 phenotype values loaded from .fam.\n",
      "--keep: 581 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 581 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997371.\n",
      "1367 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1013 variants and 581 people pass filters and QC.\n",
      "Among remaining phenotypes, 311 are cases and 225 are controls.  (45 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "2058 variants loaded from .bim file.\n",
      "635 people (408 males, 227 females) loaded from .fam.\n",
      "609 phenotype values loaded from .fam.\n",
      "--keep: 635 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 635 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997379.\n",
      "1147 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "911 variants and 635 people pass filters and QC.\n",
      "Among remaining phenotypes, 387 are cases and 222 are controls.  (26 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.all\n",
      "\n",
      "52223 MB RAM detected; reserving 26111 MB for main workspace.\n",
      "3550 variants loaded from .bim file.\n",
      "851 people (446 males, 405 females) loaded from .fam.\n",
      "827 phenotype values loaded from .fam.\n",
      "--keep: 851 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 851 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998183.\n",
      "2257 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1293 variants and 851 people pass filters and QC.\n",
      "Among remaining phenotypes, 525 are cases and 302 are controls.  (24 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.all.assoc ...\n",
      "79done.\n"
     ]
    }
   ],
   "source": [
    "#Run case-control analysis using plink assoc for all variants, not adjusting for any covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    \n",
    "    ! /home/jupyter/tools/plink \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_SCARB2 \\\n",
    "    --keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "    --assoc \\\n",
    "    --allow-no-sex \\\n",
    "    --ci 0.95 \\\n",
    "    --maf 0.01 \\\n",
    "    --out {WORK_DIR}/{ancestry}_SCARB2.all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "2138b8b1-4817-4bcc-8f2a-beb1c790b090",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AAC\n",
      "Variants with p-value < 0.05: (106, 13)\n",
      "WORKING ON: AFR\n",
      "Variants with p-value < 0.05: (368, 13)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_5192/3754789743.py:9: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/var/tmp/ipykernel_5192/3754789743.py:9: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AJ\n",
      "Variants with p-value < 0.05: (56, 13)\n",
      "WORKING ON: AMR\n",
      "Variants with p-value < 0.05: (10, 13)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_5192/3754789743.py:9: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/var/tmp/ipykernel_5192/3754789743.py:9: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: CAS\n",
      "Variants with p-value < 0.05: (96, 13)\n",
      "WORKING ON: EAS\n",
      "Variants with p-value < 0.05: (403, 13)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_5192/3754789743.py:9: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/var/tmp/ipykernel_5192/3754789743.py:9: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: EUR\n",
      "Variants with p-value < 0.05: (277, 13)\n",
      "WORKING ON: FIN\n",
      "Variants with p-value < 0.05: (41, 13)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_5192/3754789743.py:9: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/var/tmp/ipykernel_5192/3754789743.py:9: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: MDE\n",
      "Variants with p-value < 0.05: (132, 13)\n",
      "WORKING ON: SAS\n",
      "Variants with p-value < 0.05: (20, 13)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_5192/3754789743.py:9: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/var/tmp/ipykernel_5192/3754789743.py:9: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: CAH\n",
      "Variants with p-value < 0.05: (72, 13)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_5192/3754789743.py:9: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    }
   ],
   "source": [
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    #Look at assoc results, only variants with nominal p-value < 0.05\n",
    "    freq = pd.read_csv(f'{WORK_DIR}/{ancestry}_SCARB2.all.assoc', delim_whitespace=True)\n",
    "    sig_all_nonadj = freq[freq['P']<0.05]\n",
    "    \n",
    "    print(f'Variants with p-value < 0.05: {sig_all_nonadj.shape}')\n",
    "    \n",
    "    #Save FREQ to csv\n",
    "    freq.to_csv(f'{WORK_DIR}/{ancestry}.all_nonadj.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bcfcc2e4-5209-4013-b54f-a27be2c2eb49",
   "metadata": {},
   "source": [
    "GLM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "2a475074-32ed-451e-b556-ffcf5851ac08",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.all_adj\n",
      "\n",
      "Start time: Tue Aug 20 09:48:42 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50235 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1111 samples (656 females, 455 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.fam.\n",
      "3848 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bim.\n",
      "1 binary phenotype loaded (285 cases, 801 controls).\n",
      "--keep: 1111 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_covariate_file.txt.\n",
      "1111 samples (656 females, 455 males; 1111 founders) remaining after main\n",
      "filters.\n",
      "285 cases and 801 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2388 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1460 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Tue Aug 20 09:48:43 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.all_adj\n",
      "\n",
      "Start time: Tue Aug 20 09:48:43 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50236 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.fam.\n",
      "4304 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bim.\n",
      "1 binary phenotype loaded (942 cases, 1679 controls).\n",
      "--keep: 2643 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_covariate_file.txt.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) remaining after main\n",
      "filters.\n",
      "942 cases and 1679 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2791 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1513 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Tue Aug 20 09:48:43 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.all_adj\n",
      "\n",
      "Start time: Tue Aug 20 09:48:43 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50235 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.fam.\n",
      "1849 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bim.\n",
      "1 binary phenotype loaded (1292 cases, 411 controls).\n",
      "--keep: 2655 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_covariate_file.txt.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) remaining after main\n",
      "filters.\n",
      "1292 cases and 411 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "816 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1033 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Tue Aug 20 09:48:44 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.all_adj\n",
      "\n",
      "Start time: Tue Aug 20 09:48:44 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50237 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "646 samples (283 females, 363 males; 646 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.fam.\n",
      "2516 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bim.\n",
      "1 binary phenotype loaded (458 cases, 155 controls).\n",
      "--keep: 646 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_covariate_file.txt.\n",
      "646 samples (283 females, 363 males; 646 founders) remaining after main\n",
      "filters.\n",
      "458 cases and 155 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1569 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "947 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Tue Aug 20 09:48:44 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.all_adj\n",
      "\n",
      "Start time: Tue Aug 20 09:48:44 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50236 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "903 samples (464 females, 439 males; 903 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.fam.\n",
      "2219 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bim.\n",
      "1 binary phenotype loaded (552 cases, 343 controls).\n",
      "--keep: 903 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_covariate_file.txt.\n",
      "903 samples (464 females, 439 males; 903 founders) remaining after main\n",
      "filters.\n",
      "552 cases and 343 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1248 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "971 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Tue Aug 20 09:48:44 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.all_adj\n",
      "\n",
      "Start time: Tue Aug 20 09:48:45 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50236 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.fam.\n",
      "3094 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bim.\n",
      "1 binary phenotype loaded (2662 cases, 2461 controls).\n",
      "--keep: 5167 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_covariate_file.txt.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) remaining after main\n",
      "filters.\n",
      "2662 cases and 2461 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2143 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "951 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Tue Aug 20 09:48:45 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.all_adj\n",
      "\n",
      "Start time: Tue Aug 20 09:48:45 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50254 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.fam.\n",
      "9570 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bim.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "--keep: 38838 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_covariate_file.txt.\n",
      "38838 samples (15543 females, 23295 males; 38838 founders) remaining after main\n",
      "filters.\n",
      "21197 cases and 9214 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "8651 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "919 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Tue Aug 20 09:48:46 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.all_adj\n",
      "\n",
      "Start time: Tue Aug 20 09:48:47 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50237 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "114 samples (61 females, 53 males; 114 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.fam.\n",
      "1166 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bim.\n",
      "1 binary phenotype loaded (98 cases, 8 controls).\n",
      "--keep: 114 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_covariate_file.txt.\n",
      "114 samples (61 females, 53 males; 114 founders) remaining after main filters.\n",
      "98 cases and 8 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "226 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "940 variants remaining after main filters.\n",
      "Warning: --glm remaining control count is less than 10x predictor count for\n",
      "phenotype 'PHENO1'.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Tue Aug 20 09:48:47 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.all_adj\n",
      "\n",
      "Start time: Tue Aug 20 09:48:47 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50237 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "581 samples (212 females, 369 males; 581 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.fam.\n",
      "2380 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bim.\n",
      "1 binary phenotype loaded (311 cases, 225 controls).\n",
      "--keep: 581 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_covariate_file.txt.\n",
      "581 samples (212 females, 369 males; 581 founders) remaining after main\n",
      "filters.\n",
      "311 cases and 225 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1367 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1013 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Tue Aug 20 09:48:47 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.all_adj\n",
      "\n",
      "Start time: Tue Aug 20 09:48:48 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50235 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "635 samples (227 females, 408 males; 635 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.fam.\n",
      "2058 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bim.\n",
      "1 binary phenotype loaded (387 cases, 222 controls).\n",
      "--keep: 635 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_covariate_file.txt.\n",
      "635 samples (227 females, 408 males; 635 founders) remaining after main\n",
      "filters.\n",
      "387 cases and 222 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1147 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "911 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Tue Aug 20 09:48:48 2024\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.all_adj\n",
      "\n",
      "Start time: Tue Aug 20 09:48:48 2024\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "52223 MiB RAM detected, ~50237 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "851 samples (405 females, 446 males; 851 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.fam.\n",
      "3550 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bim.\n",
      "1 binary phenotype loaded (525 cases, 302 controls).\n",
      "--keep: 851 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_covariate_file.txt.\n",
      "851 samples (405 females, 446 males; 851 founders) remaining after main\n",
      "filters.\n",
      "525 cases and 302 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2257 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1293 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Tue Aug 20 09:48:48 2024\n"
     ]
    }
   ],
   "source": [
    "#Run case-control analysis with covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_SCARB2 \\\n",
    "    --keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "    --allow-no-sex \\\n",
    "    --maf 0.01 \\\n",
    "    --ci 0.95 \\\n",
    "    --glm \\\n",
    "    --covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "    --covar-variance-standardize \\\n",
    "    --neg9-pheno-really-missing \\\n",
    "    --out {WORK_DIR}/{ancestry}_SCARB2.all_adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "d2fa0aaa-038d-479e-ab55-6e44a2dc47f3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AAC\n",
      "There are 82 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: AFR\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_5192/882271930.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/var/tmp/ipykernel_5192/882271930.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 62 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: AJ\n",
      "There are 52 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_5192/882271930.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/var/tmp/ipykernel_5192/882271930.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AMR\n",
      "There are 19 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: CAS\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_5192/882271930.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/var/tmp/ipykernel_5192/882271930.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 1 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: EAS\n",
      "There are 123 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: EUR\n",
      "There are 279 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: FIN\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_5192/882271930.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/var/tmp/ipykernel_5192/882271930.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 21 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: MDE\n",
      "There are 58 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_5192/882271930.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/var/tmp/ipykernel_5192/882271930.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: SAS\n",
      "There are 8 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: CAH\n",
      "There are 104 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_5192/882271930.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    }
   ],
   "source": [
    "#Process results from plink glm analysis including covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    #Read in plink glm results\n",
    "    assoc = pd.read_csv(f'{WORK_DIR}/{ancestry}_SCARB2.all_adj.PHENO1.glm.logistic.hybrid', delim_whitespace=True)\n",
    "    \n",
    "    #Filter for additive test only - this is the variant results\n",
    "    assoc_add = assoc[assoc['TEST']==\"ADD\"]\n",
    "    \n",
    "    #Check if there are any significant (p < 0.05) variants\n",
    "    significant = assoc_add[assoc_add['P']<0.05]\n",
    "\n",
    "    print(f'There are {len(significant)} variants with p-value < 0.05 in glm')\n",
    "    \n",
    "    #Check if there are any significant (p < 0.05) variants\n",
    "    GWsignificant = assoc_add[assoc_add['P']<5e-8]\n",
    "\n",
    "    print(f'There are {len(GWsignificant)} variants with p-value < 5e-8 in glm')\n",
    "    \n",
    "    #Save assoc_add to csv\n",
    "    assoc_add.to_csv(f'{WORK_DIR}/{ancestry}.all_adj.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0043dbe2-3d23-48bf-a2b6-64b231c7babd",
   "metadata": {},
   "source": [
    "change format of association file in EUR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "b94d5d22-cfc0-4aa5-9846-95a8d26679bd",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/SCARB2/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "84f8e8d8-b0af-4ee3-b406-8be6108d7acd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AJ\n",
      "WORKING ON: EAS\n",
      "WORKING ON: SAS\n",
      "WORKING ON: AMR\n",
      "WORKING ON: MDE\n",
      "WORKING ON: CAH\n",
      "WORKING ON: FIN\n",
      "WORKING ON: CAS\n",
      "WORKING ON: AFR\n",
      "WORKING ON: EUR\n",
      "WORKING ON: AAC\n"
     ]
    }
   ],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    WORK_DIR = \"~/workspace/ws_files/SCARB2/\"\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    df = pd.read_csv(f'{WORK_DIR}/SCARB2_{ancestry}/{ancestry}.all_adj.csv')\n",
    "    \n",
    "    #First just select relevant columns: chromosome, bp position and p-value\n",
    "    export_ldassoc = df[['#CHROM', 'POS', 'P']].copy()\n",
    "    \n",
    "    #Rename the #CHROM column to remove the hashtag as I think this might be confusing LDassoc\n",
    "    export_ldassoc = export_ldassoc.rename(columns={'#CHROM': 'CHROM'}) \n",
    "    \n",
    "    #Then export as a tab-separated, not comma-separated file\n",
    "\n",
    "    export_ldassoc.to_csv(f'{WORK_DIR}/SCARB2_{ancestry}/{ancestry}.all_adj.formatted.tab', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a5dcf4e1-e393-43aa-bbd9-eb522e481065",
   "metadata": {},
   "source": [
    "GCTA conditional analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "df45c099-780c-431b-879a-b44d4fa05de6",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# prepare file for COJO\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    #Read in summary statistics\n",
    "    sumstats = pd.read_csv(f'{WORK_DIR}/SCARB2_{ancestry}/{ancestry}.all_adj.csv')\n",
    "    \n",
    "    #Format summary statistics for GCTA-COJO\n",
    "    #First get the log odds ratio - this is required for COJO\n",
    "    #1) For a case-control study, the effect size should be log(odds ratio) with its corresponding standard error.\n",
    "    sumstats_formatted = sumstats.copy()\n",
    "    sumstats_formatted['b'] = np.log(sumstats_formatted['OR'])\n",
    "    \n",
    "    #Now select just the necessary columns for COJO\n",
    "    sumstats_export = sumstats_formatted[['ID', 'A1', 'OMITTED', 'A1_FREQ', 'b', 'LOG(OR)_SE', 'P', 'OBS_CT']].copy()\n",
    "    \n",
    "    #Rename columns following COJO format\n",
    "    sumstats_export = sumstats_export.rename(columns = {'ID':'SNP', 'OMITTED':'A2', 'A1_FREQ':'freq', 'LOG(OR)_SE':'se', 'P':'p', 'OBS_CT':'N'})\n",
    "    \n",
    "    #Export\n",
    "    sumstats_export.to_csv(f'{WORK_DIR}/SCARB2_{ancestry}/{ancestry}.all_adj.sumstats.ma', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "f893db0f-c2ae-4d50-ad95-f7165c65d2e3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:23:43 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.sumstats.ma\n",
      "--cojo-p 5e-08\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.fam].\n",
      "2655 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bim].\n",
      "1849 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 2655 individuals and 1849 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1032 SNPs (817 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1033 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1033 SNPs: 7.65083 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1032 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1032 SNPs to select association signals ... (p cutoff = 5e-08; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:23:43 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.34 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:23:43 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.sumstats.ma\n",
      "--cojo-p 5e-08\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.fam].\n",
      "5167 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bim].\n",
      "3094 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 5167 individuals and 3094 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 951 SNPs (2143 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 951 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 951 SNPs: 5.17038 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "951 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 951 SNPs to select association signals ... (p cutoff = 5e-08; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:23:44 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.51 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:23:44 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.sumstats.ma\n",
      "--cojo-p 5e-08\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.fam].\n",
      "635 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bim].\n",
      "2058 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 635 individuals and 2058 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 911 SNPs (1147 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 911 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 911 SNPs: 5.52256 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "911 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 911 SNPs to select association signals ... (p cutoff = 5e-08; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:23:44 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.22 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:23:44 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.sumstats.ma\n",
      "--cojo-p 5e-08\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.fam].\n",
      "646 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bim].\n",
      "2516 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 646 individuals and 2516 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 947 SNPs (1569 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 947 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 947 SNPs: 6.27 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "947 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 947 SNPs to select association signals ... (p cutoff = 5e-08; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:23:45 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.19 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:23:45 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.sumstats.ma\n",
      "--cojo-p 5e-08\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.fam].\n",
      "581 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bim].\n",
      "2380 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 581 individuals and 2380 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1013 SNPs (1367 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1013 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1013 SNPs: 5.2279 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1013 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1013 SNPs to select association signals ... (p cutoff = 5e-08; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:23:45 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.19 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:23:45 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.sumstats.ma\n",
      "--cojo-p 5e-08\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.fam].\n",
      "851 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bim].\n",
      "3550 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 851 individuals and 3550 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1289 SNPs (2261 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1293 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1293 SNPs: 5.49246 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1289 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1289 SNPs to select association signals ... (p cutoff = 5e-08; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:23:46 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.24 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:23:46 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.sumstats.ma\n",
      "--cojo-p 5e-08\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.fam].\n",
      "114 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bim].\n",
      "1166 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 114 individuals and 1166 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 940 SNPs (226 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 940 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 940 SNPs: 10.5175 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "940 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 940 SNPs to select association signals ... (p cutoff = 5e-08; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:23:46 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.19 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:23:46 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.sumstats.ma\n",
      "--cojo-p 5e-08\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.fam].\n",
      "903 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bim].\n",
      "2219 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 903 individuals and 2219 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 971 SNPs (1248 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 971 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 971 SNPs: 5.10318 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "971 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 971 SNPs to select association signals ... (p cutoff = 5e-08; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:23:46 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.22 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:23:47 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.sumstats.ma\n",
      "--cojo-p 5e-08\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.fam].\n",
      "2643 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bim].\n",
      "4304 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 2643 individuals and 4304 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1513 SNPs (2791 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1513 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1513 SNPs: 7.08346 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1513 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1513 SNPs to select association signals ... (p cutoff = 5e-08; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:23:47 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.40 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:23:47 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.sumstats.ma\n",
      "--cojo-p 5e-08\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.fam].\n",
      "38839 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bim].\n",
      "9570 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 38839 individuals and 9570 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 919 SNPs (8651 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 919 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 919 SNPs: 5.22946 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "919 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 919 SNPs to select association signals ... (p cutoff = 5e-08; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:23:53 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 5.85 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:23:53 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.sumstats.ma\n",
      "--cojo-p 5e-08\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.fam].\n",
      "1111 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bim].\n",
      "3848 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 1111 individuals and 3848 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1460 SNPs (2388 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1460 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1460 SNPs: 7.52793 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1460 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1460 SNPs to select association signals ... (p cutoff = 5e-08; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:23:53 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.25 sec.\n"
     ]
    }
   ],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    \n",
    "    #Select multiple associated SNPs based on significance p-value 5e-08\n",
    "    #Can change the p-value for significance if needed\n",
    "    #bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS\n",
    "    ! /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_SCARB2 \\\n",
    "    --maf 0.01 \\\n",
    "    --cojo-file {WORK_DIR}/{ancestry}.all_adj.sumstats.ma \\\n",
    "    --cojo-p 5e-8 \\\n",
    "    --cojo-slct \\\n",
    "    --out {WORK_DIR}/{ancestry}.all_adj.COJO"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "522cf191-16ba-4596-a765-7417285892d6",
   "metadata": {},
   "source": [
    "use p threshold(0.00238) from LD pruning"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "43aeda9e-c030-47cc-9b0e-7e6519fa97ae",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 14:22:46 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.sumstats.ma\n",
      "--cojo-p 0.00035\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.LDprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.fam].\n",
      "851 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bim].\n",
      "3550 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 851 individuals and 3550 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1289 SNPs (2261 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1293 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1293 SNPs: 5.49246 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1289 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1289 SNPs to select association signals ... (p cutoff = 0.00035; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 14:22:46 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.26 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 14:22:46 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.sumstats.ma\n",
      "--cojo-p 0.00035\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.LDprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.fam].\n",
      "114 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bim].\n",
      "1166 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 114 individuals and 1166 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 940 SNPs (226 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 940 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 940 SNPs: 10.5175 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "940 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 940 SNPs to select association signals ... (p cutoff = 0.00035; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 14:22:47 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.16 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 14:22:47 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.00035\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.LDprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.fam].\n",
      "635 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bim].\n",
      "2058 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 635 individuals and 2058 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 911 SNPs (1147 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 911 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 911 SNPs: 5.52256 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "911 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 911 SNPs to select association signals ... (p cutoff = 0.00035; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 14:22:47 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.28 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 14:22:47 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.sumstats.ma\n",
      "--cojo-p 0.00035\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.LDprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.fam].\n",
      "581 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bim].\n",
      "2380 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 581 individuals and 2380 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1013 SNPs (1367 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1013 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1013 SNPs: 5.2279 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1013 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1013 SNPs to select association signals ... (p cutoff = 0.00035; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 14:22:48 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.19 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 14:22:48 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.sumstats.ma\n",
      "--cojo-p 0.00035\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.LDprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.fam].\n",
      "1111 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bim].\n",
      "3848 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 1111 individuals and 3848 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1460 SNPs (2388 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1460 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1460 SNPs: 7.52793 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1460 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1460 SNPs to select association signals ... (p cutoff = 0.00035; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 14:22:48 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.26 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 14:22:48 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.sumstats.ma\n",
      "--cojo-p 0.00035\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.LDprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.fam].\n",
      "38839 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bim].\n",
      "9570 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 38839 individuals and 9570 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 919 SNPs (8651 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 919 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 919 SNPs: 5.22946 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "919 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 919 SNPs to select association signals ... (p cutoff = 0.00035; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Finally, 1 associated SNPs are selected.\n",
      "Performing joint analysis on all the 1 selected signals ...\n",
      "Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.LDprune.COJO.jma.cojo] ...\n",
      "Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.LDprune.COJO.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 918 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.LDprune.COJO.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 14:22:55 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 6.63 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 14:22:55 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.00035\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.LDprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.fam].\n",
      "5167 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bim].\n",
      "3094 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 5167 individuals and 3094 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 951 SNPs (2143 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 951 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 951 SNPs: 5.17038 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "951 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 951 SNPs to select association signals ... (p cutoff = 0.00035; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 14:22:56 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.49 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 14:22:56 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.sumstats.ma\n",
      "--cojo-p 0.00035\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.LDprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.fam].\n",
      "646 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bim].\n",
      "2516 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 646 individuals and 2516 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 947 SNPs (1569 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 947 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 947 SNPs: 6.27 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "947 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 947 SNPs to select association signals ... (p cutoff = 0.00035; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 14:22:56 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.20 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 14:22:56 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.sumstats.ma\n",
      "--cojo-p 0.00035\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.LDprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.fam].\n",
      "2643 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bim].\n",
      "4304 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 2643 individuals and 4304 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1513 SNPs (2791 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1513 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1513 SNPs: 7.08346 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1513 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1513 SNPs to select association signals ... (p cutoff = 0.00035; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 14:22:57 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.39 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 14:22:57 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.00035\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.LDprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.fam].\n",
      "903 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bim].\n",
      "2219 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 903 individuals and 2219 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 971 SNPs (1248 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 971 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 971 SNPs: 5.10318 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "971 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 971 SNPs to select association signals ... (p cutoff = 0.00035; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 14:22:57 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.24 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 14:22:57 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.sumstats.ma\n",
      "--cojo-p 0.00035\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.LDprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.fam].\n",
      "2655 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bim].\n",
      "1849 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 2655 individuals and 1849 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1032 SNPs (817 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1033 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1033 SNPs: 7.65083 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1032 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1032 SNPs to select association signals ... (p cutoff = 0.00035; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 14:22:57 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.28 sec.\n"
     ]
    }
   ],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    \n",
    "    #Select multiple associated SNPs based on significance p-value 5e-08\n",
    "    #Can change the p-value for significance if needed\n",
    "    #bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS\n",
    "    ! /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_SCARB2 \\\n",
    "    --maf 0.01 \\\n",
    "    --cojo-file {WORK_DIR}/{ancestry}.all_adj.sumstats.ma \\\n",
    "    --cojo-p 3.5e-4 \\\n",
    "    --cojo-slct \\\n",
    "    --out {WORK_DIR}/{ancestry}.all_adj.LDprune.COJO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "3c2420a3-f94a-4ca9-907d-f36847a7e96e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:29:53 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.fam].\n",
      "2655 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bim].\n",
      "1849 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 2655 individuals and 1849 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1032 SNPs (817 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1033 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1033 SNPs: 7.65083 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1032 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 1032 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 1022 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:29:54 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 1.21 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:29:54 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.fam].\n",
      "5167 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bim].\n",
      "3094 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 5167 individuals and 3094 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 951 SNPs (2143 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 951 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 951 SNPs: 5.17038 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "951 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 951 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 941 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:29:56 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 1.83 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:29:56 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.fam].\n",
      "635 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bim].\n",
      "2058 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 635 individuals and 2058 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 911 SNPs (1147 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 911 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 911 SNPs: 5.52256 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "911 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 911 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 901 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:29:57 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.64 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:29:57 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.fam].\n",
      "646 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bim].\n",
      "2516 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 646 individuals and 2516 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 947 SNPs (1569 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 947 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 947 SNPs: 6.27 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "947 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 947 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 937 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:29:58 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.65 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:29:58 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.fam].\n",
      "581 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bim].\n",
      "2380 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 581 individuals and 2380 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1013 SNPs (1367 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1013 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1013 SNPs: 5.2279 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1013 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 1013 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 1003 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:29:58 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.66 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:29:59 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.fam].\n",
      "851 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bim].\n",
      "3550 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 851 individuals and 3550 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1289 SNPs (2261 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1293 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1293 SNPs: 5.49246 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1289 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 1289 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 1279 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:29:59 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.78 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:30:00 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.fam].\n",
      "114 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bim].\n",
      "1166 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 114 individuals and 1166 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 940 SNPs (226 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 940 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 940 SNPs: 10.5175 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "940 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 940 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 930 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:30:00 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.62 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:30:00 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.fam].\n",
      "903 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bim].\n",
      "2219 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 903 individuals and 2219 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 971 SNPs (1248 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 971 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 971 SNPs: 5.10318 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "971 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 971 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 961 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:30:01 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 0.87 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:30:02 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.fam].\n",
      "2643 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bim].\n",
      "4304 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 2643 individuals and 4304 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1513 SNPs (2791 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1513 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1513 SNPs: 7.08346 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1513 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 1513 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 1503 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:30:03 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 1.58 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:30:03 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.fam].\n",
      "38839 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bim].\n",
      "9570 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 38839 individuals and 9570 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 919 SNPs (8651 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 919 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 919 SNPs: 5.22946 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "919 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 919 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 909 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:30:16 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 12.43 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 10:30:16 UTC on Tue Aug 20 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: jupyterlabvertexai20240708\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.sumstats.ma\n",
      "--cojo-top-SNPs 10\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.top10\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.fam].\n",
      "1111 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bim].\n",
      "3848 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 1111 individuals and 3848 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1460 SNPs (2388 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1460 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1460 SNPs: 7.52793 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1460 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.\n",
      "\n",
      "Performing stepwise model selection on 1460 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Switched to perform a forward model selection because the significance level is too low...\n",
      "5 associated SNPs have been selected.\n",
      "10 associated SNPs have been selected.\n",
      "Performing backward elimination...\n",
      "Finally, 10 associated SNPs are selected.\n",
      "Performing joint analysis on all the 10 selected signals ...\n",
      "Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.top10.jma.cojo] ...\n",
      "Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.top10.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 1450 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.top10.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 10:30:17 UTC on Tue Aug 20 2024\n",
      "Overall computational time: 1.09 sec.\n"
     ]
    }
   ],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    \n",
    "    #Select multiple associated SNPs based on significance p-value 5e-08\n",
    "    #Can change the p-value for significance if needed\n",
    "    #bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS\n",
    "    ! /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_SCARB2 \\\n",
    "    --maf 0.01 \\\n",
    "    --cojo-file {WORK_DIR}/{ancestry}.all_adj.sumstats.ma \\\n",
    "    --cojo-top-SNPs 10 \\\n",
    "    --out {WORK_DIR}/{ancestry}.all_adj.top10"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e16f133c-0179-45b5-a2df-b64909555d20",
   "metadata": {},
   "source": [
    "Burden Analyses using RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "a7078ad2-9967-4d9b-b3ab-efd6b9705746",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running plink to extract exonic variants for ancestry: AAC\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:25 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50656 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1111 samples (656 females, 455 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb.psam.\n",
      "3986699 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (285 cases, 801 controls).\n",
      "--extract bed1: 3986640 variants excluded.\n",
      "--keep: 1111 samples remaining.\n",
      "1111 samples (656 females, 455 males; 1111 founders) remaining after main\n",
      "filters.\n",
      "285 cases and 801 controls remaining after main filters.\n",
      "59 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.exonic.vcf ...\n",
      "1011131516182022232527283032333537384042444547495052545557596162646667697172747677798183848688899193949698done.\n",
      "End time: Mon Jul 29 15:19:26 2024\n",
      "Running bgzip and tabix for exonic variants for ancestry: AAC\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AAC\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:26 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50617 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "1111 samples (656 females, 455 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb.psam.\n",
      "3986699 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (285 cases, 801 controls).\n",
      "--extract bed1: 3986664 variants excluded.\n",
      "--keep: 1111 samples remaining.\n",
      "1111 samples (656 females, 455 males; 1111 founders) remaining after main\n",
      "filters.\n",
      "285 cases and 801 controls remaining after main filters.\n",
      "35 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.nonsynonymous_lof.vcf\n",
      "... 11141720222528313437404245485154576062656871747780828588919497done.\n",
      "End time: Mon Jul 29 15:19:27 2024\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AAC\n",
      "Running plink to extract exonic variants for ancestry: AFR\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:28 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50623 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb.psam.\n",
      "4555051 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (942 cases, 1679 controls).\n",
      "--extract bed1: 4554988 variants excluded.\n",
      "--keep: 2643 samples remaining.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) remaining after main\n",
      "filters.\n",
      "942 cases and 1679 controls remaining after main filters.\n",
      "63 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.exonic.vcf ...\n",
      "1112141517192022232526283031333436383941424446474950525355575860616365666869717374767779808284858788909293959698done.\n",
      "End time: Mon Jul 29 15:19:29 2024\n",
      "Running bgzip and tabix for exonic variants for ancestry: AFR\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AFR\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:30 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50623 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb.psam.\n",
      "4555051 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (942 cases, 1679 controls).\n",
      "--extract bed1: 4555014 variants excluded.\n",
      "--keep: 2643 samples remaining.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) remaining after main\n",
      "filters.\n",
      "942 cases and 1679 controls remaining after main filters.\n",
      "37 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.nonsynonymous_lof.vcf\n",
      "... 101316182124272932353740434548515456596264677072757881838689919497done.\n",
      "End time: Mon Jul 29 15:19:31 2024\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AFR\n",
      "Running plink to extract exonic variants for ancestry: AJ\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:32 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50622 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb.psam.\n",
      "1669760 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (1292 cases, 411 controls).\n",
      "--extract bed1: 1669729 variants excluded.\n",
      "--keep: 2655 samples remaining.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) remaining after main\n",
      "filters.\n",
      "1292 cases and 411 controls remaining after main filters.\n",
      "31 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.exonic.vcf ...\n",
      "121619222529323538414548515458616467707477808387909396done.\n",
      "End time: Mon Jul 29 15:19:33 2024\n",
      "Running bgzip and tabix for exonic variants for ancestry: AJ\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AJ\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:34 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50617 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb.psam.\n",
      "1669760 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (1292 cases, 411 controls).\n",
      "--extract bed1: 1669741 variants excluded.\n",
      "--keep: 2655 samples remaining.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) remaining after main\n",
      "filters.\n",
      "1292 cases and 411 controls remaining after main filters.\n",
      "19 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.nonsynonymous_lof.vcf\n",
      "... 1015212631364247525763687378848994done.\n",
      "End time: Mon Jul 29 15:19:34 2024\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AJ\n",
      "Running plink to extract exonic variants for ancestry: AMR\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:35 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50618 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "646 samples (283 females, 363 males; 646 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb.psam.\n",
      "2407746 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (458 cases, 155 controls).\n",
      "--extract bed1: 2407703 variants excluded.\n",
      "--keep: 646 samples remaining.\n",
      "646 samples (283 females, 363 males; 646 founders) remaining after main\n",
      "filters.\n",
      "458 cases and 155 controls remaining after main filters.\n",
      "43 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.exonic.vcf ...\n",
      "1113161820232527303234373941444648515355586062656769727476798183868890939597done.\n",
      "End time: Mon Jul 29 15:19:36 2024\n",
      "Running bgzip and tabix for exonic variants for ancestry: AMR\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AMR\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:37 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50617 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "646 samples (283 females, 363 males; 646 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb.psam.\n",
      "2407746 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (458 cases, 155 controls).\n",
      "--extract bed1: 2407717 variants excluded.\n",
      "--keep: 646 samples remaining.\n",
      "646 samples (283 females, 363 males; 646 founders) remaining after main\n",
      "filters.\n",
      "458 cases and 155 controls remaining after main filters.\n",
      "29 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.nonsynonymous_lof.vcf\n",
      "... 1013172024273134374144485155586265687275798286899396done.\n",
      "End time: Mon Jul 29 15:19:37 2024\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AMR\n",
      "Running plink to extract exonic variants for ancestry: CAS\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:38 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50616 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "903 samples (464 females, 439 males; 903 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb.psam.\n",
      "2200655 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (552 cases, 343 controls).\n",
      "--extract bed1: 2200619 variants excluded.\n",
      "--keep: 903 samples remaining.\n",
      "903 samples (464 females, 439 males; 903 founders) remaining after main\n",
      "filters.\n",
      "552 cases and 343 controls remaining after main filters.\n",
      "36 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.exonic.vcf ...\n",
      "1113161922252730333638414447505255586163666972757780838688919497done.\n",
      "End time: Mon Jul 29 15:19:39 2024\n",
      "Running bgzip and tabix for exonic variants for ancestry: CAS\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: CAS\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:40 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50617 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "903 samples (464 females, 439 males; 903 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb.psam.\n",
      "2200655 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (552 cases, 343 controls).\n",
      "--extract bed1: 2200633 variants excluded.\n",
      "--keep: 903 samples remaining.\n",
      "903 samples (464 females, 439 males; 903 founders) remaining after main\n",
      "filters.\n",
      "552 cases and 343 controls remaining after main filters.\n",
      "22 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.nonsynonymous_lof.vcf\n",
      "... 13182227313640455054596368727781869095done.\n",
      "End time: Mon Jul 29 15:19:41 2024\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: CAS\n",
      "Running plink to extract exonic variants for ancestry: EAS\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:41 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50606 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb.psam.\n",
      "3235052 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (2662 cases, 2461 controls).\n",
      "--extract bed1: 3234995 variants excluded.\n",
      "--keep: 5167 samples remaining.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) remaining after main\n",
      "filters.\n",
      "2662 cases and 2461 controls remaining after main filters.\n",
      "57 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.exonic.vcf ...\n",
      "101214151719212224262829313335363840424345474950525456575961636466687071737577788082848587899192949698done.\n",
      "End time: Mon Jul 29 15:19:42 2024\n",
      "Running bgzip and tabix for exonic variants for ancestry: EAS\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: EAS\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:43 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50615 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb.psam.\n",
      "3235052 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (2662 cases, 2461 controls).\n",
      "--extract bed1: 3235016 variants excluded.\n",
      "--keep: 5167 samples remaining.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) remaining after main\n",
      "filters.\n",
      "2662 cases and 2461 controls remaining after main filters.\n",
      "36 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.nonsynonymous_lof.vcf\n",
      "... 1113161922252730333638414447505255586163666972757780838688919497done.\n",
      "End time: Mon Jul 29 15:19:44 2024\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: EAS\n",
      "Running plink to extract exonic variants for ancestry: EUR\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:45 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50613 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb.psam.\n",
      "10419161 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "--extract bed1: 10418969 variants excluded.\n",
      "--keep: 38838 samples remaining.\n",
      "38838 samples (15543 females, 23295 males; 38838 founders) remaining after main\n",
      "filters.\n",
      "21197 cases and 9214 controls remaining after main filters.\n",
      "192 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.exonic.vcf ...\n",
      "10101111121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626363646465656666676768686969707071717272737374757576767777787879798080818182828383848485858686878888898990909191929293939494959596969797989899done.\n",
      "End time: Mon Jul 29 15:19:47 2024\n",
      "Running bgzip and tabix for exonic variants for ancestry: EUR\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: EUR\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:48 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50595 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb.psam.\n",
      "10419161 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "--extract bed1: 10419026 variants excluded.\n",
      "--keep: 38838 samples remaining.\n",
      "38838 samples (15543 females, 23295 males; 38838 founders) remaining after main\n",
      "filters.\n",
      "21197 cases and 9214 controls remaining after main filters.\n",
      "135 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.nonsynonymous_lof.vcf\n",
      "... 10111112131414151617171819202021222223242525262728282930313132333434353637373839404041424243444545464748484950515152535454555657575859606061626263646565666768686970717172737474757677777879808081828283848585868788888990919192939494959697979899done.\n",
      "End time: Mon Jul 29 15:19:50 2024\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: EUR\n",
      "Running plink to extract exonic variants for ancestry: FIN\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:51 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50596 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "114 samples (61 females, 53 males; 114 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb.psam.\n",
      "891614 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (98 cases, 8 controls).\n",
      "--extract bed1: 891598 variants excluded.\n",
      "--keep: 114 samples remaining.\n",
      "114 samples (61 females, 53 males; 114 founders) remaining after main filters.\n",
      "98 cases and 8 controls remaining after main filters.\n",
      "16 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.exonic.vcf ...\n",
      "1218253137435056626875818793done.\n",
      "End time: Mon Jul 29 15:19:52 2024\n",
      "Running bgzip and tabix for exonic variants for ancestry: FIN\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: FIN\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:53 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50597 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "114 samples (61 females, 53 males; 114 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb.psam.\n",
      "891614 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (98 cases, 8 controls).\n",
      "--extract bed1: 891604 variants excluded.\n",
      "--keep: 114 samples remaining.\n",
      "114 samples (61 females, 53 males; 114 founders) remaining after main filters.\n",
      "98 cases and 8 controls remaining after main filters.\n",
      "10 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.nonsynonymous_lof.vcf\n",
      "... 102030405060708090done.\n",
      "End time: Mon Jul 29 15:19:53 2024\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: FIN\n",
      "Running plink to extract exonic variants for ancestry: MDE\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:54 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50598 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "581 samples (212 females, 369 males; 581 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb.psam.\n",
      "2161336 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (311 cases, 225 controls).\n",
      "--extract bed1: 2161300 variants excluded.\n",
      "--keep: 581 samples remaining.\n",
      "581 samples (212 females, 369 males; 581 founders) remaining after main\n",
      "filters.\n",
      "311 cases and 225 controls remaining after main filters.\n",
      "36 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.exonic.vcf ...\n",
      "1113161922252730333638414447505255586163666972757780838688919497done.\n",
      "End time: Mon Jul 29 15:19:55 2024\n",
      "Running bgzip and tabix for exonic variants for ancestry: MDE\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: MDE\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:56 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50606 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "581 samples (212 females, 369 males; 581 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb.psam.\n",
      "2161336 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (311 cases, 225 controls).\n",
      "--extract bed1: 2161310 variants excluded.\n",
      "--keep: 581 samples remaining.\n",
      "581 samples (212 females, 369 males; 581 founders) remaining after main\n",
      "filters.\n",
      "311 cases and 225 controls remaining after main filters.\n",
      "26 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.nonsynonymous_lof.vcf\n",
      "... 1115192326303438424650535761656973768084889296done.\n",
      "End time: Mon Jul 29 15:19:56 2024\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: MDE\n",
      "Running plink to extract exonic variants for ancestry: SAS\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:57 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50596 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "635 samples (227 females, 408 males; 635 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb.psam.\n",
      "1933318 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (387 cases, 222 controls).\n",
      "--extract bed1: 1933284 variants excluded.\n",
      "--keep: 635 samples remaining.\n",
      "635 samples (227 females, 408 males; 635 founders) remaining after main\n",
      "filters.\n",
      "387 cases and 222 controls remaining after main filters.\n",
      "34 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.exonic.vcf ...\n",
      "111417202326293235384144475052555861646770737679828588919497done.\n",
      "End time: Mon Jul 29 15:19:58 2024\n",
      "Running bgzip and tabix for exonic variants for ancestry: SAS\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: SAS\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:19:58 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50596 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "635 samples (227 females, 408 males; 635 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb.psam.\n",
      "1933318 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (387 cases, 222 controls).\n",
      "--extract bed1: 1933299 variants excluded.\n",
      "--keep: 635 samples remaining.\n",
      "635 samples (227 females, 408 males; 635 founders) remaining after main\n",
      "filters.\n",
      "387 cases and 222 controls remaining after main filters.\n",
      "19 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.nonsynonymous_lof.vcf\n",
      "... 1015212631364247525763687378848994done.\n",
      "End time: Mon Jul 29 15:19:59 2024\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: SAS\n",
      "Running plink to extract exonic variants for ancestry: CAH\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:20:00 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50596 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "851 samples (405 females, 446 males; 851 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb.psam.\n",
      "3678288 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (525 cases, 302 controls).\n",
      "--extract bed1: 3678236 variants excluded.\n",
      "--keep: 851 samples remaining.\n",
      "851 samples (405 females, 446 males; 851 founders) remaining after main\n",
      "filters.\n",
      "525 cases and 302 controls remaining after main filters.\n",
      "52 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.exonic.vcf ...\n",
      "11131517192123252628303234363840424446485051535557596163656769717375767880828486889092949698done.\n",
      "End time: Mon Jul 29 15:20:01 2024\n",
      "Running bgzip and tabix for exonic variants for ancestry: CAH\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: CAH\n",
      "PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb\n",
      "\n",
      "Start time: Mon Jul 29 15:20:02 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "52223 MiB RAM detected, ~50580 available; reserving 26111 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "851 samples (405 females, 446 males; 851 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb.psam.\n",
      "3678288 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (525 cases, 302 controls).\n",
      "--extract bed1: 3678258 variants excluded.\n",
      "--keep: 851 samples remaining.\n",
      "851 samples (405 females, 446 males; 851 founders) remaining after main\n",
      "filters.\n",
      "525 cases and 302 controls remaining after main filters.\n",
      "30 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.nonsynonymous_lof.vcf\n",
      "... 101316202326303336404346505356606366707376808386909396done.\n",
      "End time: Mon Jul 29 15:20:02 2024\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: CAH\n"
     ]
    }
   ],
   "source": [
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "variant_classes = ['exonic', 'nonsynonymous_lof']\n",
    "\n",
    "#Loop over all the ancestries and the 2 variant classes - run rvtests for all coding and missense variants\n",
    "for ancestry in ancestries:\n",
    "    for variant_class in variant_classes:\n",
    "        \n",
    "        WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "        \n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running plink to extract {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        #Extract relevant variants\n",
    "        ! /home/jupyter/tools/plink2 \\\n",
    "        --pfile {REL7_PATH}/imputed_genotypes/{ancestry}/chr4_{ancestry}_release7_vwb \\\n",
    "        --keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "        --extract range {WORK_DIR}/{ancestry}_SCARB2.{variant_class}.variantstoKeep.txt \\\n",
    "        --recode vcf-iid \\\n",
    "        --out {WORK_DIR}/{ancestry}_SCARB2.{variant_class}\n",
    "        \n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running bgzip and tabix for {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        ## Bgzip and Tabix (zip and index the file)\n",
    "        ! bgzip -f {WORK_DIR}/{ancestry}_SCARB2.{variant_class}.vcf\n",
    "        ! tabix -f -p vcf {WORK_DIR}/{ancestry}_SCARB2.{variant_class}.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ac289e03-1dce-4de6-80a4-250ec0cca394",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#Run RVtests\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "variant_classes = ['exonic', 'nonsynonymous_lof']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    for variant_class in variant_classes:\n",
    "        \n",
    "        WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "        \n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running RVtests for {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        ## RVtests with covariates \n",
    "        #Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "        #The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "        ! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "        --out {WORK_DIR}/{ancestry}_SCARB2.burden.{variant_class} \\\n",
    "        --kernel skat,skato \\\n",
    "        --inVcf {WORK_DIR}/{ancestry}_SCARB2.{variant_class}.vcf.gz \\\n",
    "        --pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "        --pheno-name PHENO \\\n",
    "        --gene SCARB2 \\\n",
    "        --geneFile ~/workspace/ws_files/TMEM175/refFlat.txt \\\n",
    "        --covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "        --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "        --freqUpper 0.01"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3491eba0-7295-4ea3-ad39-45b9b1902e56",
   "metadata": {},
   "source": [
    "EUR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "18a80a31-db85-4fb3-98d9-291bb051c243",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t20918\t64\t43\t8475.92\t0.891794\t10000\t1109\t8475.92\t1000\t0\t0.901713\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_EUR/EUR_SCARB2.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "439bd264-261f-4df8-a859-3bc176fc303d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t20918\t64\t43\t892.339\t1\t1\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_EUR/EUR_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "a70f9822-275e-45ae-8a4b-73625d323155",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t20918\t43\t26\t3091.69\t0.731763\t10000\t1389\t3091.69\t1000\t0\t0.719942\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_EUR/EUR_SCARB2.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "23ea33b7-11c6-4929-919a-d8e31a361656",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t20918\t43\t26\t4256.15\t1\t0.345116\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_EUR/EUR_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "87b5e623-5968-494b-88b7-39827fdeab9f",
   "metadata": {},
   "source": [
    "AAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "1011879d-4c34-42b6-963c-148c6ccab239",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t925\t20\t16\t626.538\t0.971408\t10000\t1062\t626.538\t1000\t0\t0.94162\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AAC/AAC_SCARB2.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "2e82bde0-f5e1-42f7-b1c3-ea5366184be0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t925\t20\t16\t473.593\t1\t0.719617\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AAC/AAC_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "1709914c-f1d5-4fae-b002-74a2bf8818dc",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t925\t11\t9\t547.367\t0.85572\t10000\t1263\t547.367\t1000\t0\t0.791766\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AAC/AAC_SCARB2.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "b8b9930f-b29c-4102-a542-28ae2620470f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t925\t11\t9\t462.384\t1\t0.638153\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AAC/AAC_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6335fd78-185c-4c3b-9b65-8bd96eb717a7",
   "metadata": {},
   "source": [
    "AFR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "4db24441-03bd-46ba-ba6a-19bd127e9c4e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t1062\t17\t12\t2540.23\t0.518996\t10000\t1937\t2540.23\t1000\t0\t0.516262\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AFR/AFR_SCARB2.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "c4c734f2-4d70-44ac-b6af-5538b9e3b710",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t1062\t17\t12\t1066.59\t1\t0.590661\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AFR/AFR_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "a3855fef-285e-4288-a608-489badf3967b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t1062\t9\t6\t2386.29\t0.293053\t10000\t4078\t2386.29\t1000\t0\t0.245218\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AFR/AFR_SCARB2.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "c95eeb89-cabf-4520-9de1-a69070efb841",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t1062\t9\t6\t1141.95\t0.6\t0.402744\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AFR/AFR_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc22d3e0-945a-4744-9b1e-0f132053d4e5",
   "metadata": {},
   "source": [
    "AJ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "0aeb3b88-35de-432b-82c3-f63d6a81c074",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t1410\t5\t3\t187.979\t0.360916\t10000\t3606\t187.979\t1000\t0\t0.277316\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AJ/AJ_SCARB2.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "8eaed7bb-1912-43b1-86a8-3f75b5edc2ad",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t1410\t5\t3\tNA\tNA\tNA\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AJ/AJ_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "6a2a79d5-75fd-4286-b630-d9a2b18eacff",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t1410\t4\t2\t187.979\t0.360916\t10000\t3606\t187.979\t1000\t0\t0.277316\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AJ/AJ_SCARB2.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "32855a82-0885-47e5-a588-824a5c99f3d8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t1410\t4\t2\tNA\tNA\tNA\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AJ/AJ_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "285f2c71-b279-40b9-b5ae-36bb88740ce0",
   "metadata": {},
   "source": [
    "AMR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "dba864d3-0882-4807-a564-11ad8c76de5f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t447\t15\t5\t76.8106\t0.466648\t10000\t1730\t76.8106\t1000\t0\t0.578035\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AMR/AMR_SCARB2.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "55a696ed-24e8-4b03-a5e5-dc8763b28084",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t447\t15\t5\t38.4053\t0.1\t0.466649\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AMR/AMR_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "5f43d156-8c50-48f9-9789-7e3b5c2251d9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t447\t11\t4\t0\t-nan\t10000\t1000\t0\t0\t1000\t0.5\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AMR/AMR_SCARB2.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "87b4a348-e81e-4d2c-b10d-af8c62997f42",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t447\t11\t4\tNA\tNA\tNA\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AMR/AMR_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "847b984a-a3a0-4df4-9bed-375cfea5af4c",
   "metadata": {},
   "source": [
    "CAH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "0cb8fd48-4d0d-4dda-ba57-71b00845d16c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t693\t15\t10\t4916.94\t0.185781\t10000\t6012\t4916.94\t1000\t0\t0.166334\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_CAH/CAH_SCARB2.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "f1f280cd-b192-49ac-8181-671287514f36",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t693\t15\t10\t2458.47\t0\t0.300097\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_CAH/CAH_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "9e5f5661-f806-4a93-83a5-a3dd948d9d90",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t693\t8\t6\t1734.19\t0.428815\t10000\t2561\t1734.19\t1000\t0\t0.390472\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_CAH/CAH_SCARB2.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "05ff4b14-3956-4e73-aeb8-bbfb036c13ff",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t693\t8\t6\t867.093\t0\t0.581358\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_CAH/CAH_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "02b1643c-c3d6-4f3a-b192-73d24c0247a1",
   "metadata": {},
   "source": [
    "CAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "8eae3a82-bd5e-46a0-a53c-66d5c70773cc",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t658\t10\t7\t1034.61\t0.149244\t10000\t8942\t1034.61\t1000\t0\t0.111832\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_CAS/CAS_SCARB2.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "7bd8e79c-c58a-4007-8f54-efd366e20b9c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t658\t10\t7\t517.304\t0\t0.233515\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_CAS/CAS_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "2ab82d42-68c3-40a7-b9aa-fa8b8df70ff6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t658\t7\t4\t53.0168\t0.514372\t10000\t1537\t53.0168\t1000\t0\t0.650618\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_CAS/CAS_SCARB2.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "fc5982ae-eb97-48c4-b86f-00a899a8e0c2",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t658\t7\t4\t26.5084\t0.4\t0.517254\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_CAS/CAS_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2a23af17-07e0-4935-9586-5ae82a5f2acc",
   "metadata": {},
   "source": [
    "EAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "45d30339-7a5f-4947-8bb9-1439ccce698c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t2523\t17\t12\t3848.26\t0.728263\t10000\t1379\t3848.26\t1000\t0\t0.725163\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_EAS/EAS_SCARB2.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "29fadce7-62c8-4881-a4b2-0a299e4d52fd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t2523\t17\t12\t2026.13\t1\t0.666468\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_EAS/EAS_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "cbbfe83c-ea9f-4cc5-b601-99be970997d2",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t2523\t12\t9\t4024.31\t0.60396\t10000\t1692\t4024.31\t1000\t0\t0.591017\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_EAS/EAS_SCARB2.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "b28adc22-0dcd-4df7-bd0a-a168a1dbeb7f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t2523\t12\t9\t3800.46\t1\t0.447802\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_EAS/EAS_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28f0d151-bac9-4aa2-9870-10352ff67bcc",
   "metadata": {},
   "source": [
    "FIN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "267de414-7d8b-4e77-8529-ba162066cb18",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_FIN/FIN_SCARB2.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "ff3c658c-8284-4951-8f17-d067c5909e86",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_FIN/FIN_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "b2677539-2d5f-471a-ab63-58a254353c23",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_FIN/FIN_SCARB2.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "ccc98875-7ad0-4712-b732-ce268afc0bf0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_FIN/FIN_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "677bf4c8-823b-41b8-aeff-ddd80632a1e0",
   "metadata": {},
   "source": [
    "MDE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "383c6497-a427-4585-b3a1-20c6c9683a06",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t379\t10\t5\t243.498\t0.489121\t10000\t1430\t243.498\t1000\t0\t0.699301\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_MDE/MDE_SCARB2.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "b2c1a367-193e-43de-a04f-75cd59e9d993",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t379\t10\t5\t274.609\t1\t0.276794\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_MDE/MDE_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "849ce8ea-fd61-4dc1-9cce-e05b0367c1ba",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t379\t7\t3\t183.965\t0.361618\t10000\t1971\t183.965\t1000\t0\t0.507357\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_MDE/MDE_SCARB2.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "f4d8f87f-1e6e-4e00-b5a4-bf5a977a64d2",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t379\t7\t3\t123.624\t1\t0.350369\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_MDE/MDE_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4e9ea40-3c62-4a33-a83b-d35003777198",
   "metadata": {},
   "source": [
    "SAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "c7410f34-95cd-4ad2-9633-17d4db4ca72a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t296\t9\t3\t464.996\t0.0027725\t10000\t10000\t464.996\t128\t37\t0.01465\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_SAS/SAS_SCARB2.burden.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "d5e787d9-2396-47d2-9a58-284f708e0ca0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t296\t9\t3\tNA\tNA\tNA\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_SAS/SAS_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "a7d143a1-9d7f-427d-bbee-62a9556180d1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t296\t6\t2\t464.996\t0.0027725\t10000\t10000\t464.996\t128\t37\t0.01465\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_SAS/SAS_SCARB2.burden.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "77cf7ba4-7600-48d6-82c7-647c6963d928",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213824\t296\t6\t2\t232.498\t0\t0.000884866\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_SAS/SAS_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1b2ebd42-d6ed-439d-9f1d-8391aeb78de0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "environment": {
   "kernel": "python3",
   "name": ".m125",
   "type": "gcloud",
   "uri": "us-docker.pkg.dev/deeplearning-platform-release/gcr.io/:m125"
  },
  "kernelspec": {
   "display_name": "Python 3 (Local)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
