{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "GBA1 - Single gene analysis in AMP-PD WGS data (EUR population)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notebook Overview\n",
    "\n",
    "1. Loading Python libraries \n",
    "\n",
    "2. Installing packages\n",
    "\n",
    "3. copy data\n",
    "\n",
    "4. Create a covariate file with AMP-PD data\n",
    "\n",
    "5. Annotation of the gene GBA1\n",
    "\n",
    "6. Burden test(SkatO, Skat, cmc,zeggini,mb,fp,cmcWald)\n",
    "\n",
    "7. Save out results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Loading Python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2369/2693492717.py:22: DeprecationWarning: Importing display from IPython.core.display is deprecated since IPython 7.14, please import from IPython display\n",
      "  from IPython.core.display import display, HTML\n"
     ]
    }
   ],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "\n",
    "# Numpy for basics\n",
    "\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery\n",
    "\n",
    "#Import Sys\n",
    "import sys as sys"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Defining functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mannovar\u001b[0m/               \u001b[01;32mplink2\u001b[0m*                          \u001b[34;42mrvtests\u001b[0m/\r\n",
      "annovar.latest.tar.gz  plink2_linux_x86_64_latest.zip   toy.map\r\n",
      "LICENSE                plink_linux_x86_64_20190304.zip  toy.ped\r\n",
      "\u001b[01;32mplink\u001b[0m*                 \u001b[01;32mprettify\u001b[0m*\r\n"
     ]
    }
   ],
   "source": [
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 1.9 to make sure you have permission to run the program\n",
    "chmod u+x /home/jupyter/tools/plink\n",
    "# chmod plink 2.0 to make sure you have permission to run the program\n",
    "chmod u+x /home/jupyter/tools/plink2\n",
    "# Change permissions\n",
    "chmod 777 /home/jupyter/tools/rvtests/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Copy data from AMP-PD bucket to workspace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = \"/home/jupyter/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-ddf4a6f6 -m cp -r gs://amp-pd-genomics/releases/2023_v4release_1027/wgs-WB-DWGS/plink/pfiles/chr1.* /home/jupyter/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://amp-pd-genomics/releases/2023_v4release_1027/wgs-WB-DWGS/plink/pfiles/chr1.pgen...\n",
      "Copying gs://amp-pd-genomics/releases/2023_v4release_1027/wgs-WB-DWGS/plink/pfiles/chr1.psam...\n",
      "Copying gs://amp-pd-genomics/releases/2023_v4release_1027/wgs-WB-DWGS/plink/pfiles/chr1.pvar...\n",
      "/ [3/3 files][ 18.7 GiB/ 18.7 GiB] 100% Done  39.3 MiB/s ETA 00:00:00           \n",
      "Operation completed over 3 objects/18.7 GiB.                                     \n"
     ]
    }
   ],
   "source": [
    "#Copy data\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {AMP_WGS_RELEASE_PLINK_PATH}/pfiles/chr1.* {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1_updated_formatted.log   chr1_updated.log\tCOVS.txt\r\n",
      "chr1_updated_formatted.pgen  chr1_updated.pgen\tGBA.log\r\n",
      "chr1_updated_formatted.psam  chr1_updated.psam\r\n",
      "chr1_updated_formatted.pvar  chr1_updated.pvar\r\n"
     ]
    }
   ],
   "source": [
    "! ls /home/jupyter/GBA_AMPPD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r /home/jupyter/COVS.txt /home/jupyter/GBA_AMPPD/COVS.txt')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Update sex and pheno info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Update sex and phenotype in plink files, using the COVS.txt file that we made\n",
    "#-update-sex and column number tells plink where to look for the sex information in the file\n",
    "\n",
    "WORK_DIR = \"/home/jupyter/\"\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--pfile {WORK_DIR}/chr1 \\\n",
    "--update-sex {WORK_DIR}/COVS.txt col-num=5 \\\n",
    "--pheno {WORK_DIR}/COVS.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--make-pgen \\\n",
    "--out {WORK_DIR}/GBA_AMPPD/chr1_updated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#FID\tIID\tSEX\tPHENO\n",
      "HB-PD_INVAB109VHC\tHB-PD_INVAB109VHC\t2\t1\n",
      "HB-PD_INVAB289LG3\tHB-PD_INVAB289LG3\t2\t1\n",
      "HB-PD_INVAC488AAF\tHB-PD_INVAC488AAF\t1\t1\n",
      "HB-PD_INVAE296YP8\tHB-PD_INVAE296YP8\t2\t1\n",
      "HB-PD_INVAJ549VWD\tHB-PD_INVAJ549VWD\t2\t1\n",
      "HB-PD_INVAK106DV5\tHB-PD_INVAK106DV5\t1\t1\n",
      "HB-PD_INVAK293NGR\tHB-PD_INVAK293NGR\t1\t1\n",
      "HB-PD_INVAM835KNR\tHB-PD_INVAM835KNR\t2\t1\n",
      "HB-PD_INVAM864YC9\tHB-PD_INVAM864YC9\t2\t1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "head /home/jupyter/GBA_AMPPD/chr1_updated.psam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BF-1001\tBF-1001\t1\t1\r\n"
     ]
    }
   ],
   "source": [
    "! grep \"BF-1001\" /home/jupyter/GBA_AMPPD/chr1_updated.psam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Extract and annotate the gene Extract the region using PLINK Extract GBA gene GBA coordinates: Chromosome 1:155234452-1:155244699(GRCh38/hg38)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## update the variant IDs\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--pfile /home/jupyter/GBA_AMPPD/chr1_updated \\\n",
    "--fa /home/jupyter/hg38.fa \\\n",
    "--memory 25000 \\\n",
    "--set-all-var-ids \"chr@:#:\\$r:\\$a\" \\\n",
    "--new-id-max-allele-len 999 --sort-vars \\\n",
    "--make-pgen \\\n",
    "--out /home/jupyter/GBA_AMPPD/chr1_updated_formatted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (17 Jun 2024)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter//GBA_AMPPD/GBA.log.\n",
      "Options in effect:\n",
      "  --chr 1\n",
      "  --from-bp 155184452\n",
      "  --make-pgen\n",
      "  --max-alleles 2\n",
      "  --out /home/jupyter//GBA_AMPPD/GBA\n",
      "  --pfile /home/jupyter//GBA_AMPPD/chr1_updated_formatted\n",
      "  --to-bp 155294699\n",
      "\n",
      "Start time: Wed Nov 20 13:32:49 2024\n",
      "30093 MiB RAM detected, ~28416 available; reserving 15046 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "10418 samples (4462 females, 5627 males, 329 ambiguous; 10418 founders) loaded\n",
      "from /home/jupyter//GBA_AMPPD/chr1_updated_formatted.psam.\n",
      "11357160 out of 12349094 variants loaded from\n",
      "/home/jupyter//GBA_AMPPD/chr1_updated_formatted.pvar.\n",
      "1 binary phenotype loaded (3188 cases, 4014 controls).\n",
      "5190 variants remaining after main filters.\n",
      "Writing /home/jupyter//GBA_AMPPD/GBA.psam ... done.\n",
      "Writing /home/jupyter//GBA_AMPPD/GBA.pvar ... 101011111212131314141515161617171818192020212122222323242425252626272728282930303131323233333434353536363737383839404041414242434344444545464647474848495050515152525353545455555656575758585960606161626263636464656566666767686869707071717272737374747575767677777878798080818182828383848485858686878788888990909191929293939494959596969797989899done.\n",
      "Writing /home/jupyter//GBA_AMPPD/GBA.pgen ... done.\n",
      "End time: Wed Nov 20 13:38:53 2024\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink and make pgen files\n",
    "WORK_DIR = \"/home/jupyter/\"\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--pfile {WORK_DIR}/GBA_AMPPD/chr1_updated_formatted \\\n",
    "--chr 1 \\\n",
    "--from-bp 155184452 \\\n",
    "--to-bp 155294699 \\\n",
    "--max-alleles 2 \\\n",
    "--make-pgen \\\n",
    "--out {WORK_DIR}/GBA_AMPPD/GBA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (17 Jun 2024)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter//GBA_AMPPD/GBA.log.\n",
      "Options in effect:\n",
      "  --chr 1\n",
      "  --from-bp 155184452\n",
      "  --make-bed\n",
      "  --max-alleles 2\n",
      "  --out /home/jupyter//GBA_AMPPD/GBA\n",
      "  --pfile /home/jupyter//GBA_AMPPD/chr1_updated_formatted\n",
      "  --to-bp 155294699\n",
      "\n",
      "Start time: Wed Nov 20 13:41:36 2024\n",
      "30093 MiB RAM detected, ~28438 available; reserving 15046 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "10418 samples (4462 females, 5627 males, 329 ambiguous; 10418 founders) loaded\n",
      "from /home/jupyter//GBA_AMPPD/chr1_updated_formatted.psam.\n",
      "11357160 out of 12349094 variants loaded from\n",
      "/home/jupyter//GBA_AMPPD/chr1_updated_formatted.pvar.\n",
      "1 binary phenotype loaded (3188 cases, 4014 controls).\n",
      "5190 variants remaining after main filters.\n",
      "Writing /home/jupyter//GBA_AMPPD/GBA.fam ... done.\n",
      "Writing /home/jupyter//GBA_AMPPD/GBA.bim ... done.\n",
      "Writing /home/jupyter//GBA_AMPPD/GBA.bed ... done.\n",
      "End time: Wed Nov 20 13:43:10 2024\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink and make binary files\n",
    "WORK_DIR = \"/home/jupyter/\"\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--pfile {WORK_DIR}/GBA_AMPPD/chr1_updated_formatted \\\n",
    "--chr 1 \\\n",
    "--from-bp 155184452 \\\n",
    "--to-bp 155294699 \\\n",
    "--max-alleles 2 \\\n",
    "--make-bed \\\n",
    "--out {WORK_DIR}/GBA_AMPPD/GBA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\tchr1:155184542:ACTGGGCATGAT:A\t0\t155184542\tA\tACTGGGCATGAT\r\n",
      "1\tchr1:155184559:G:T\t0\t155184559\tT\tG\r\n",
      "1\tchr1:155184604:C:T\t0\t155184604\tT\tC\r\n",
      "1\tchr1:155184621:G:A\t0\t155184621\tA\tG\r\n",
      "1\tchr1:155184622:C:A\t0\t155184622\tA\tC\r\n",
      "1\tchr1:155184639:C:T\t0\t155184639\tT\tC\r\n",
      "1\tchr1:155184670:G:A\t0\t155184670\tA\tG\r\n",
      "1\tchr1:155184680:A:T\t0\t155184680\tT\tA\r\n",
      "1\tchr1:155184709:C:A\t0\t155184709\tA\tC\r\n",
      "1\tchr1:155184712:C:T\t0\t155184712\tT\tC\r\n"
     ]
    }
   ],
   "source": [
    "# Visualize bim file\n",
    "! head {WORK_DIR}/GBA_AMPPD/GBA.bim"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Turn binary files into VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (17 Jun 2024)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter//GBA_AMPPD/GBA.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter//GBA_AMPPD/GBA\n",
      "  --export vcf id-paste=iid\n",
      "  --mac 2\n",
      "  --out /home/jupyter//GBA_AMPPD/GBA\n",
      "\n",
      "Start time: Wed Nov 20 13:45:12 2024\n",
      "30093 MiB RAM detected, ~28443 available; reserving 15046 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "10418 samples (4462 females, 5627 males, 329 ambiguous; 10418 founders) loaded\n",
      "from /home/jupyter//GBA_AMPPD/GBA.fam.\n",
      "5190 variants loaded from /home/jupyter//GBA_AMPPD/GBA.bim.\n",
      "1 binary phenotype loaded (3188 cases, 4014 controls).\n",
      "Calculating allele frequencies... done.\n",
      "3174 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "2016 variants remaining after main filters.\n",
      "--export vcf to /home/jupyter//GBA_AMPPD/GBA.vcf ... 1010111112121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373738383939404041414242434344444545464647474848495050515152525353545455555656575758585959606061616262636364646565666667676868696970707171727273737475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Wed Nov 20 13:45:12 2024\n"
     ]
    }
   ],
   "source": [
    "## Turn binary files into VCF\n",
    "WORK_DIR = \"/home/jupyter/\"\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/GBA_AMPPD/GBA \\\n",
    "--recode vcf id-paste=iid \\\n",
    "--mac 2 \\\n",
    "--out {WORK_DIR}/GBA_AMPPD/GBA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "\n",
    "! bgzip -f {WORK_DIR}/GBA_AMPPD/GBA.vcf\n",
    "! tabix -f -p vcf {WORK_DIR}/GBA_AMPPD/GBA.vcf.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Annotate using ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter//GBA_AMPPD/GBA.vcf.gz > /home/jupyter//GBA_AMPPD/GBA.annovar.avinput>\n",
      "NOTICE: Finished reading 2023 lines from VCF file\n",
      "NOTICE: A total of 2016 locus in VCF file passed QC threshold, representing 1867 SNPs (1321 transitions and 546 transversions) and 149 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 19450406 SNP genotypes (13762178 transitions and 5688228 transversions) and 1552282 indels/substitutions for 10418 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter//GBA_AMPPD/GBA.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter//GBA_AMPPD/GBA.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter//GBA_AMPPD/GBA.annovar.refGene -exonsort -nofirstcodondel /home/jupyter//GBA_AMPPD/GBA.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter//GBA_AMPPD/GBA.annovar.refGene.variant_function, /home/jupyter//GBA_AMPPD/GBA.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2016 unique variants in 2016 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 43 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter//GBA_AMPPD/GBA.annovar /home/jupyter//GBA_AMPPD/GBA.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter//GBA_AMPPD/GBA.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter//GBA_AMPPD/GBA.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2016 unique variants in 2016 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 14\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter//GBA_AMPPD/GBA.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter//GBA_AMPPD/GBA.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter//GBA_AMPPD/GBA.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "## annotate using ANNOVAR\n",
    "! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/GBA_AMPPD/GBA.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out {WORK_DIR}/GBA_AMPPD/GBA.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo10421</th>\n",
       "      <th>Otherinfo10422</th>\n",
       "      <th>Otherinfo10423</th>\n",
       "      <th>Otherinfo10424</th>\n",
       "      <th>Otherinfo10425</th>\n",
       "      <th>Otherinfo10426</th>\n",
       "      <th>Otherinfo10427</th>\n",
       "      <th>Otherinfo10428</th>\n",
       "      <th>Otherinfo10429</th>\n",
       "      <th>Otherinfo10430</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>155184604</td>\n",
       "      <td>155184604</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>TRIM46</td>\n",
       "      <td>NM_001282378:c.*414C&gt;T;NM_025058:c.*414C&gt;T;NM_...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>155184622</td>\n",
       "      <td>155184622</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>TRIM46</td>\n",
       "      <td>NM_001282378:c.*432C&gt;A;NM_025058:c.*432C&gt;A;NM_...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>155184763</td>\n",
       "      <td>155184763</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>TRIM46</td>\n",
       "      <td>NM_001282378:c.*573G&gt;T;NM_025058:c.*573G&gt;T;NM_...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>155184969</td>\n",
       "      <td>155184969</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>TRIM46</td>\n",
       "      <td>NM_001282378:c.*779G&gt;A;NM_025058:c.*779G&gt;A;NM_...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>155185007</td>\n",
       "      <td>155185007</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>downstream</td>\n",
       "      <td>MUC1;TRIM46</td>\n",
       "      <td>dist=817</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2011</th>\n",
       "      <td>1</td>\n",
       "      <td>155294470</td>\n",
       "      <td>155294470</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PKLR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2012</th>\n",
       "      <td>1</td>\n",
       "      <td>155294538</td>\n",
       "      <td>155294538</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PKLR</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>PKLR:NM_000298:exon6:c.G909C:p.P303P,PKLR:NM_1...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2013</th>\n",
       "      <td>1</td>\n",
       "      <td>155294560</td>\n",
       "      <td>155294560</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PKLR</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PKLR:NM_000298:exon6:c.C887A:p.A296D,PKLR:NM_1...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2014</th>\n",
       "      <td>1</td>\n",
       "      <td>155294618</td>\n",
       "      <td>155294618</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PKLR</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PKLR:NM_000298:exon6:c.G829A:p.E277K,PKLR:NM_1...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2015</th>\n",
       "      <td>1</td>\n",
       "      <td>155294633</td>\n",
       "      <td>155294633</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PKLR</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PKLR:NM_000298:exon6:c.C814G:p.L272V,PKLR:NM_1...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2016 rows × 10441 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       1  155184604  155184604   C   T         UTR3       TRIM46   \n",
       "1       1  155184622  155184622   C   A         UTR3       TRIM46   \n",
       "2       1  155184763  155184763   G   T         UTR3       TRIM46   \n",
       "3       1  155184969  155184969   G   A         UTR3       TRIM46   \n",
       "4       1  155185007  155185007   G   A   downstream  MUC1;TRIM46   \n",
       "...   ...        ...        ...  ..  ..          ...          ...   \n",
       "2011    1  155294470  155294470   C   T     intronic         PKLR   \n",
       "2012    1  155294538  155294538   C   G       exonic         PKLR   \n",
       "2013    1  155294560  155294560   G   T       exonic         PKLR   \n",
       "2014    1  155294618  155294618   C   T       exonic         PKLR   \n",
       "2015    1  155294633  155294633   G   C       exonic         PKLR   \n",
       "\n",
       "                                     GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0     NM_001282378:c.*414C>T;NM_025058:c.*414C>T;NM_...                  .   \n",
       "1     NM_001282378:c.*432C>A;NM_025058:c.*432C>A;NM_...                  .   \n",
       "2     NM_001282378:c.*573G>T;NM_025058:c.*573G>T;NM_...                  .   \n",
       "3     NM_001282378:c.*779G>A;NM_025058:c.*779G>A;NM_...                  .   \n",
       "4                                              dist=817                  .   \n",
       "...                                                 ...                ...   \n",
       "2011                                                  .                  .   \n",
       "2012                                                  .     synonymous SNV   \n",
       "2013                                                  .  nonsynonymous SNV   \n",
       "2014                                                  .  nonsynonymous SNV   \n",
       "2015                                                  .  nonsynonymous SNV   \n",
       "\n",
       "                                       AAChange.refGene  ... Otherinfo10421  \\\n",
       "0                                                     .  ...            0/0   \n",
       "1                                                     .  ...            0/0   \n",
       "2                                                     .  ...            0/0   \n",
       "3                                                     .  ...            0/0   \n",
       "4                                                     .  ...            0/0   \n",
       "...                                                 ...  ...            ...   \n",
       "2011                                                  .  ...            0/0   \n",
       "2012  PKLR:NM_000298:exon6:c.G909C:p.P303P,PKLR:NM_1...  ...            0/0   \n",
       "2013  PKLR:NM_000298:exon6:c.C887A:p.A296D,PKLR:NM_1...  ...            0/0   \n",
       "2014  PKLR:NM_000298:exon6:c.G829A:p.E277K,PKLR:NM_1...  ...            0/0   \n",
       "2015  PKLR:NM_000298:exon6:c.C814G:p.L272V,PKLR:NM_1...  ...            0/0   \n",
       "\n",
       "      Otherinfo10422 Otherinfo10423 Otherinfo10424  Otherinfo10425  \\\n",
       "0                0/0            0/0            0/0             0/0   \n",
       "1                0/0            0/0            0/0             0/0   \n",
       "2                0/0            0/0            0/0             0/0   \n",
       "3                0/0            0/1            0/0             0/0   \n",
       "4                0/0            0/0            0/0             0/0   \n",
       "...              ...            ...            ...             ...   \n",
       "2011             0/0            0/0            0/0             0/0   \n",
       "2012             0/0            0/0            0/0             0/0   \n",
       "2013             0/0            0/0            0/0             0/0   \n",
       "2014             0/0            0/0            0/0             0/0   \n",
       "2015             0/0            0/0            0/0             0/0   \n",
       "\n",
       "      Otherinfo10426 Otherinfo10427 Otherinfo10428 Otherinfo10429  \\\n",
       "0                0/0            0/0            0/0            0/0   \n",
       "1                0/0            0/0            0/0            0/0   \n",
       "2                0/0            0/0            0/0            0/0   \n",
       "3                0/0            0/0            0/0            0/0   \n",
       "4                0/0            0/0            0/0            0/0   \n",
       "...              ...            ...            ...            ...   \n",
       "2011             0/0            0/0            0/0            0/0   \n",
       "2012             0/0            0/0            0/0            0/0   \n",
       "2013             0/0            0/0            0/0            0/0   \n",
       "2014             0/0            0/0            0/0            0/0   \n",
       "2015             0/0            0/0            0/0            0/0   \n",
       "\n",
       "     Otherinfo10430  \n",
       "0               0/0  \n",
       "1               0/0  \n",
       "2               0/0  \n",
       "3               0/0  \n",
       "4               0/0  \n",
       "...             ...  \n",
       "2011            0/0  \n",
       "2012            0/0  \n",
       "2013            0/0  \n",
       "2014            0/0  \n",
       "2015            0/0  \n",
       "\n",
       "[2016 rows x 10441 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/GBA_AMPPD/GBA.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total variants:  2016\n",
      "exonic:  267\n",
      "Intronic:  1026\n",
      "Upstream:  111\n",
      "Downstream:  49\n",
      "UTR3:  75\n",
      "UTR5:  42\n",
      "Splicing:  3\n",
      "Total exonic:  267\n",
      "Stopgain:  3\n",
      "Stoploss:  0\n",
      "Startloss:  0\n",
      "Frameshift deletion:  5\n",
      "Frameshift insertion:  4\n",
      "Non-frameshift insertion:  1\n",
      "Non-frameshift deletion:  1\n",
      "Synonymous:  77\n",
      "Nonsynonymous:  176\n",
      "nonsynonymous_lof:  188\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_84/996593315.py:20: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  lof = exonic[(gene['ExonicFunc.refGene'] == 'stopgain') | (gene['ExonicFunc.refGene'] == 'stoploss') | (gene['ExonicFunc.refGene'] == 'frameshift deletion') | (gene['ExonicFunc.refGene'] == 'frameshift insertion') | (gene['ExonicFunc.refGene'] == 'startloss')]\n"
     ]
    }
   ],
   "source": [
    "results = [] \n",
    "\n",
    "\n",
    "intronic = gene[gene['Func.refGene']== 'intronic']\n",
    "upstream = gene[gene['Func.refGene']== 'upstream']\n",
    "downstream = gene[gene['Func.refGene']== 'downstream']\n",
    "utr5 = gene[gene['Func.refGene']== 'UTR5']\n",
    "utr3 = gene[gene['Func.refGene']== 'UTR3']\n",
    "splicing = gene[gene['Func.refGene']== 'splicing']\n",
    "exonic = gene[gene['Func.refGene']== 'exonic']\n",
    "stopgain = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'stopgain')]\n",
    "stoploss = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'stoploss')]\n",
    "startloss = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'startloss')]\n",
    "frameshift_deletion = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'frameshift deletion')]\n",
    "frameshift_insertion = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'frameshift insertion')]\n",
    "nonframeshift_deletion = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonframeshift deletion')]\n",
    "nonframeshift_insertion = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonframeshift insertion')]\n",
    "coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_synonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'synonymous SNV')]\n",
    "lof = exonic[(gene['ExonicFunc.refGene'] == 'stopgain') | (gene['ExonicFunc.refGene'] == 'stoploss') | (gene['ExonicFunc.refGene'] == 'frameshift deletion') | (gene['ExonicFunc.refGene'] == 'frameshift insertion') | (gene['ExonicFunc.refGene'] == 'startloss')]\n",
    "nonsynonymous_lof = pd.concat([coding_nonsynonymous, lof])\n",
    "\n",
    "print('Total variants: ', len(gene))\n",
    "print(\"exonic: \", len(exonic))\n",
    "print(\"Intronic: \", len(intronic))\n",
    "print(\"Upstream: \", len(upstream))\n",
    "print(\"Downstream: \", len(downstream))\n",
    "print('UTR3: ', len(utr3))\n",
    "print('UTR5: ', len(utr5))\n",
    "print(\"Splicing: \", len(splicing))\n",
    "print(\"Total exonic: \", len(exonic))\n",
    "print(\"Stopgain: \", len(stopgain))\n",
    "print(\"Stoploss: \", len(stoploss))\n",
    "print(\"Startloss: \", len(startloss))\n",
    "print(\"Frameshift deletion: \", len(frameshift_deletion))\n",
    "print(\"Frameshift insertion: \", len(frameshift_insertion))\n",
    "print(\"Non-frameshift insertion: \", len(nonframeshift_insertion))\n",
    "print(\"Non-frameshift deletion: \", len(nonframeshift_deletion))\n",
    "print('Synonymous: ', len(coding_synonymous))\n",
    "print(\"Nonsynonymous: \", len(coding_nonsynonymous))\n",
    "print(\"nonsynonymous_lof: \", len(nonsynonymous_lof))\n",
    "results.append((gene, intronic, upstream, downstream, utr3, utr5, splicing,exonic,stopgain,stoploss,startloss, frameshift_deletion,frameshift_insertion,nonframeshift_deletion,nonframeshift_insertion,coding_synonymous, coding_nonsynonymous, nonsynonymous_lof))\n",
    "print('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save in PLINK format\n",
    "variants_toKeep = exonic[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "variants_toKeep.to_csv(f'{WORK_DIR}/GBA_AMPPD/GBA.exonic.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save in PLINK format\n",
    "variants_toKeep = nonsynonymous_lof[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "variants_toKeep.to_csv(f'{WORK_DIR}/GBA_AMPPD/GBA.nonsynonymous_lof.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Burden Analyses using RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running plink to extract exonic variants for ancestry: AMPPD\n",
      "PLINK v2.00a6LM 64-bit Intel (17 Jun 2024)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter//GBA_AMPPD/GBA.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter//GBA_AMPPD/GBA.exonic.variantstoKeep.txt\n",
      "  --out /home/jupyter//GBA_AMPPD/GBA.exonic\n",
      "  --pfile /home/jupyter//GBA_AMPPD/GBA\n",
      "\n",
      "Start time: Wed Nov 20 13:58:11 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "30093 MiB RAM detected, ~27911 available; reserving 15046 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "10418 samples (4462 females, 5627 males, 329 ambiguous; 10418 founders) loaded\n",
      "from /home/jupyter//GBA_AMPPD/GBA.psam.\n",
      "5190 variants loaded from /home/jupyter//GBA_AMPPD/GBA.pvar.\n",
      "1 binary phenotype loaded (3188 cases, 4014 controls).\n",
      "--extract bed1: 4929 variants excluded.\n",
      "261 variants remaining after main filters.\n",
      "--export vcf to /home/jupyter//GBA_AMPPD/GBA.exonic.vcf ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Wed Nov 20 13:58:11 2024\n",
      "Running bgzip and tabix for exonic variants for ancestry: AMPPD\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AMPPD\n",
      "PLINK v2.00a6LM 64-bit Intel (17 Jun 2024)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter//GBA_AMPPD/GBA.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter//GBA_AMPPD/GBA.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --out /home/jupyter//GBA_AMPPD/GBA.nonsynonymous_lof\n",
      "  --pfile /home/jupyter//GBA_AMPPD/GBA\n",
      "\n",
      "Start time: Wed Nov 20 13:58:14 2024\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "30093 MiB RAM detected, ~27918 available; reserving 15046 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "10418 samples (4462 females, 5627 males, 329 ambiguous; 10418 founders) loaded\n",
      "from /home/jupyter//GBA_AMPPD/GBA.psam.\n",
      "5190 variants loaded from /home/jupyter//GBA_AMPPD/GBA.pvar.\n",
      "1 binary phenotype loaded (3188 cases, 4014 controls).\n",
      "--extract bed1: 5007 variants excluded.\n",
      "183 variants remaining after main filters.\n",
      "--export vcf to /home/jupyter//GBA_AMPPD/GBA.nonsynonymous_lof.vcf ... 1010111212131314141515161617181819192020212122222324242525262627272828293030313132323333343435363637373838393940404142424343444445454646474848494950505151525353545455555656575758595960606161626263636465656666676768686969707171727273737474757576777778787979808081818283838484858586868787888989909091919292939394959596969797989899done.\n",
      "End time: Wed Nov 20 13:58:14 2024\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AMPPD\n"
     ]
    }
   ],
   "source": [
    "#Loop over 2 variant classes, extract variants and run rvtests\n",
    "##GBA\n",
    "variant_classes = ['exonic', 'nonsynonymous_lof']\n",
    "\n",
    "WORK_DIR = \"/home/jupyter/\"\n",
    "\n",
    "for variant_class in variant_classes:\n",
    "    \n",
    "    # Print the command to be executed (for debugging purposes)\n",
    "    print(f'Running plink to extract {variant_class} variants for ancestry: AMPPD')\n",
    "    \n",
    "    #Extract relevant variants\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --pfile {WORK_DIR}/GBA_AMPPD/GBA \\\n",
    "    --extract range {WORK_DIR}/GBA_AMPPD/GBA.{variant_class}.variantstoKeep.txt \\\n",
    "    --recode vcf-iid \\\n",
    "    --out {WORK_DIR}/GBA_AMPPD/GBA.{variant_class}\n",
    "    \n",
    "    # Print the command to be executed (for debugging purposes)\n",
    "    print(f'Running bgzip and tabix for {variant_class} variants for ancestry: AMPPD')\n",
    "        \n",
    "    ## Bgzip and Tabix (zip and index the file)\n",
    "    ! bgzip -f {WORK_DIR}/GBA_AMPPD/GBA.{variant_class}.vcf\n",
    "    ! tabix -f -p vcf {WORK_DIR}/GBA_AMPPD/GBA.{variant_class}.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = \"/home/jupyter/\"\n",
    "\n",
    "variant_classes = ['exonic', 'nonsynonymous_lof']\n",
    "\n",
    "for variant_class in variant_classes:\n",
    "    \n",
    "    # Print the command to be executed (for debugging purposes)\n",
    "    print(f'Running RVtests for {variant_class} variants for ancestry: AMPPD')\n",
    "        \n",
    "     ## RVtests with covariates \n",
    "    #Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "    #The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "    ! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "    --out {WORK_DIR}/GBA_AMPPD/GBA.burden.{variant_class} \\\n",
    "    --kernel skato \\\n",
    "    --inVcf {WORK_DIR}/GBA_AMPPD/GBA.{variant_class}.vcf.gz \\\n",
    "    --pheno {WORK_DIR}/AMPPD_EUR.COVS.txt \\\n",
    "    --pheno-name PHENO \\\n",
    "    --gene GBA \\\n",
    "    --geneFile {WORK_DIR}/refFlat_HG38.txt \\\n",
    "    --covar {WORK_DIR}/AMPPD_EUR.COVS.txt \\\n",
    "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "    --freqUpper 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t42\t29\t95921.2\t1\t0.00861309\r\n"
     ]
    }
   ],
   "source": [
    "## look at results in exonic\n",
    "! cat {WORK_DIR}/GBA_AMPPD/GBA.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t28\t20\t62217\t1\t0.0202248\r\n"
     ]
    }
   ],
   "source": [
    "## look at results in nonsynonymous_lof\n",
    "! cat {WORK_DIR}/GBA_AMPPD/GBA.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# burden test in AMPPD ancestry\n",
    "#Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "        #The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--out {WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTEST.exonic \\\n",
    "--burden cmc,zeggini,mb,fp,cmcWald \\\n",
    "--inVcf {WORK_DIR}/GBA_AMPPD/GBA.exonic.vcf.gz \\\n",
    "--pheno {WORK_DIR}/AMPPD_EUR.COVS.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--gene GBA \\\n",
    "--geneFile {WORK_DIR}/refFlat_HG38.txt \\\n",
    "--covar {WORK_DIR}/AMPPD_EUR.COVS.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--freqUpper 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tNonRefSite\tBeta\tSE\tPvalue\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t42\t29\t203\t0.415272\t0.154824\t0.00731365\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t42\t29\t203\t-0.640712\t0.0613669\t1.61654e-25\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t42\t29\t203\t-0.0459337\t0.0027196\t5.34302e-64\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t42\t29\t203\t-42.8899\t2.52673\t1.26894e-64\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t42\t29\t203\t-4.69764\t2.10785\t0.0258377\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t42\t29\t203\t2.56574\t2.14628\t0.231915\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t42\t29\t203\t-2.12616\t2.12684\t0.317464\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t42\t29\t203\t-1.10186\t2.15252\t0.608725\r\n"
     ]
    }
   ],
   "source": [
    "# see bureden test in exonic in AMPPD\n",
    "! cat {WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTEST.exonic.CMCWald.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tNonRefSite\tPvalue\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t42\t29\t203\t0.00705455\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTEST.exonic.CMC.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tPvalue\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t42\t29\t0.00144637\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTEST.exonic.Fp.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t7202\t41\t34\t10000\t10000\t18.4115\t4\t0\t0.0004\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTEST.exonic.MadsonBrowning.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tPvalue\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t42\t29\t0.01077\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTEST.exonic.Zeggini.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# burden test in AMPPD ancestry\n",
    "#Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "        #The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--out {WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTEST.nonsynonymous_lof \\\n",
    "--burden cmc,zeggini,mb,fp,cmcWald \\\n",
    "--inVcf {WORK_DIR}/GBA_AMPPD/GBA.nonsynonymous_lof.vcf.gz \\\n",
    "--pheno {WORK_DIR}/AMPPD_EUR.COVS.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--gene GBA \\\n",
    "--geneFile {WORK_DIR}/refFlat_HG38.txt \\\n",
    "--covar {WORK_DIR}/AMPPD_EUR.COVS.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--freqUpper 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tNonRefSite\tBeta\tSE\tPvalue\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t28\t20\t173\t0.419539\t0.167773\t0.0123968\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t28\t20\t173\t-0.640728\t0.0613631\t1.60119e-25\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t28\t20\t173\t-0.0459663\t0.00271961\t4.36317e-64\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t28\t20\t173\t-42.8292\t2.52519\t1.60136e-64\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t28\t20\t173\t-4.79295\t2.10744\t0.0229478\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t28\t20\t173\t2.60351\t2.1463\t0.225121\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t28\t20\t173\t-2.14297\t2.12681\t0.313648\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t28\t20\t173\t-1.12219\t2.15256\t0.602139\r\n"
     ]
    }
   ],
   "source": [
    "# see bureden test in nonsynonymous_lof in AMPPD\n",
    "! cat {WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTEST.nonsynonymous_lof.CMCWald.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tNonRefSite\tPvalue\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t28\t20\t173\t0.012\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTEST.nonsynonymous_lof.CMC.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tPvalue\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t28\t20\t0.00465715\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTEST.nonsynonymous_lof.Fp.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t28\t20\t10000\t10000\t10.6587\t46\t0\t0.0046\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTEST.nonsynonymous_lof.MadsonBrowning.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tPvalue\r\n",
      "GBA\t1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-2561,1_GL383519v1_alt:0-3358,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3426,1_GL383519v1_alt:39474-46272,1_GL383519v1_alt:0-3339,1_GL383519v1_alt:39474-49650,1_GL383519v1_alt:0-3426,1:155234451-155241249,1:155234451-155241249,1:155234451-155244627,1:155234451-155244627,1:155234451-155244627\t5086\t28\t20\t0.0122216\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTEST.nonsynonymous_lof.Zeggini.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# run single variant burden analysis\n",
    "# single variant analysis\n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--inVcf {WORK_DIR}/GBA_AMPPD/GBA.exonic.vcf.gz \\\n",
    "--single score \\\n",
    "--pheno {WORK_DIR}/AMPPD_EUR.COVS.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--geneFile {WORK_DIR}/refFlat_HG38.txt \\\n",
    "--gene GBA \\\n",
    "--out {WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTESTSV.exonic \\\n",
    "--covar {WORK_DIR}/AMPPD_EUR.COVS.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--freqUpper 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "SV_AMPPDexonic = pd.read_csv(f'{WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTESTSV.exonic.SingleScore.assoc', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>N_INFORMATIVE</th>\n",
       "      <th>AF</th>\n",
       "      <th>U</th>\n",
       "      <th>V</th>\n",
       "      <th>STAT</th>\n",
       "      <th>DIRECTION</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>SE</th>\n",
       "      <th>PVALUE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235196</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000197</td>\n",
       "      <td>0.644930</td>\n",
       "      <td>0.429459</td>\n",
       "      <td>0.968507</td>\n",
       "      <td>+</td>\n",
       "      <td>1.501730</td>\n",
       "      <td>1.525950</td>\n",
       "      <td>0.325053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235203</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.002064</td>\n",
       "      <td>0.568009</td>\n",
       "      <td>4.718310</td>\n",
       "      <td>0.068379</td>\n",
       "      <td>+</td>\n",
       "      <td>0.120384</td>\n",
       "      <td>0.460370</td>\n",
       "      <td>0.793711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235206</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.632278</td>\n",
       "      <td>0.232255</td>\n",
       "      <td>1.721280</td>\n",
       "      <td>+</td>\n",
       "      <td>2.722340</td>\n",
       "      <td>2.075000</td>\n",
       "      <td>0.189529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235211</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235217</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.001475</td>\n",
       "      <td>0.025041</td>\n",
       "      <td>3.522080</td>\n",
       "      <td>0.000178</td>\n",
       "      <td>+</td>\n",
       "      <td>0.007110</td>\n",
       "      <td>0.532844</td>\n",
       "      <td>0.989354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235227</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235245</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000197</td>\n",
       "      <td>-0.951925</td>\n",
       "      <td>0.389797</td>\n",
       "      <td>2.324700</td>\n",
       "      <td>-</td>\n",
       "      <td>-2.442100</td>\n",
       "      <td>1.601700</td>\n",
       "      <td>0.127335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235256</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235269</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235727</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000295</td>\n",
       "      <td>-0.524750</td>\n",
       "      <td>0.739292</td>\n",
       "      <td>0.372468</td>\n",
       "      <td>-</td>\n",
       "      <td>-0.709801</td>\n",
       "      <td>1.163030</td>\n",
       "      <td>0.541662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235772</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235776</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.596818</td>\n",
       "      <td>0.240267</td>\n",
       "      <td>1.482480</td>\n",
       "      <td>+</td>\n",
       "      <td>2.483970</td>\n",
       "      <td>2.040110</td>\n",
       "      <td>0.223387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235777</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.218746</td>\n",
       "      <td>0.170490</td>\n",
       "      <td>0.280660</td>\n",
       "      <td>+</td>\n",
       "      <td>1.283040</td>\n",
       "      <td>2.421870</td>\n",
       "      <td>0.596269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235790</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000295</td>\n",
       "      <td>0.496820</td>\n",
       "      <td>0.389246</td>\n",
       "      <td>0.634125</td>\n",
       "      <td>+</td>\n",
       "      <td>1.276370</td>\n",
       "      <td>1.602830</td>\n",
       "      <td>0.425846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235820</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.666422</td>\n",
       "      <td>0.222039</td>\n",
       "      <td>2.000180</td>\n",
       "      <td>+</td>\n",
       "      <td>3.001370</td>\n",
       "      <td>2.122190</td>\n",
       "      <td>0.157281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235824</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000197</td>\n",
       "      <td>-0.039608</td>\n",
       "      <td>0.413151</td>\n",
       "      <td>0.003797</td>\n",
       "      <td>-</td>\n",
       "      <td>-0.095868</td>\n",
       "      <td>1.555770</td>\n",
       "      <td>0.950865</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155235843</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.003244</td>\n",
       "      <td>7.466830</td>\n",
       "      <td>6.716640</td>\n",
       "      <td>8.300810</td>\n",
       "      <td>+</td>\n",
       "      <td>1.111690</td>\n",
       "      <td>0.385855</td>\n",
       "      <td>0.003963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155236245</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000393</td>\n",
       "      <td>1.369860</td>\n",
       "      <td>0.889397</td>\n",
       "      <td>2.109860</td>\n",
       "      <td>+</td>\n",
       "      <td>1.540210</td>\n",
       "      <td>1.060360</td>\n",
       "      <td>0.146352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155236246</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.009339</td>\n",
       "      <td>4.090250</td>\n",
       "      <td>20.220700</td>\n",
       "      <td>0.827377</td>\n",
       "      <td>+</td>\n",
       "      <td>0.202280</td>\n",
       "      <td>0.222383</td>\n",
       "      <td>0.363032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155236269</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155236366</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.462734</td>\n",
       "      <td>0.248273</td>\n",
       "      <td>0.862448</td>\n",
       "      <td>+</td>\n",
       "      <td>1.863810</td>\n",
       "      <td>2.006940</td>\n",
       "      <td>0.353055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155236377</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155236469</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155237423</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000197</td>\n",
       "      <td>-0.465445</td>\n",
       "      <td>0.350959</td>\n",
       "      <td>0.617278</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.326210</td>\n",
       "      <td>1.688000</td>\n",
       "      <td>0.432061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155237438</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000197</td>\n",
       "      <td>0.960708</td>\n",
       "      <td>0.457038</td>\n",
       "      <td>2.019440</td>\n",
       "      <td>+</td>\n",
       "      <td>2.102030</td>\n",
       "      <td>1.479190</td>\n",
       "      <td>0.155297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155237451</td>\n",
       "      <td>C</td>\n",
       "      <td>CT</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.345027</td>\n",
       "      <td>0.225632</td>\n",
       "      <td>0.527601</td>\n",
       "      <td>+</td>\n",
       "      <td>1.529160</td>\n",
       "      <td>2.105230</td>\n",
       "      <td>0.467617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155237453</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000393</td>\n",
       "      <td>0.279937</td>\n",
       "      <td>0.798504</td>\n",
       "      <td>0.098140</td>\n",
       "      <td>+</td>\n",
       "      <td>0.350577</td>\n",
       "      <td>1.119080</td>\n",
       "      <td>0.754074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155237458</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>-0.115823</td>\n",
       "      <td>0.102269</td>\n",
       "      <td>0.131173</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.132540</td>\n",
       "      <td>3.127010</td>\n",
       "      <td>0.717218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155238174</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.740153</td>\n",
       "      <td>0.191988</td>\n",
       "      <td>2.853440</td>\n",
       "      <td>+</td>\n",
       "      <td>3.855200</td>\n",
       "      <td>2.282250</td>\n",
       "      <td>0.091179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155238189</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.236705</td>\n",
       "      <td>0.180482</td>\n",
       "      <td>0.310441</td>\n",
       "      <td>+</td>\n",
       "      <td>1.311510</td>\n",
       "      <td>2.353870</td>\n",
       "      <td>0.577409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155238200</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000197</td>\n",
       "      <td>0.682175</td>\n",
       "      <td>0.443058</td>\n",
       "      <td>1.050340</td>\n",
       "      <td>+</td>\n",
       "      <td>1.539700</td>\n",
       "      <td>1.502350</td>\n",
       "      <td>0.305428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155238208</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155238265</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155238570</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155238631</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.001081</td>\n",
       "      <td>0.140238</td>\n",
       "      <td>2.476900</td>\n",
       "      <td>0.007940</td>\n",
       "      <td>+</td>\n",
       "      <td>0.056618</td>\n",
       "      <td>0.635398</td>\n",
       "      <td>0.928997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155238646</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.287433</td>\n",
       "      <td>0.204549</td>\n",
       "      <td>0.403901</td>\n",
       "      <td>+</td>\n",
       "      <td>1.405200</td>\n",
       "      <td>2.211060</td>\n",
       "      <td>0.525081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155239700</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>-0.074611</td>\n",
       "      <td>0.068987</td>\n",
       "      <td>0.080693</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.081520</td>\n",
       "      <td>3.807300</td>\n",
       "      <td>0.776361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155239946</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155239961</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000197</td>\n",
       "      <td>1.086160</td>\n",
       "      <td>0.467005</td>\n",
       "      <td>2.526200</td>\n",
       "      <td>+</td>\n",
       "      <td>2.325800</td>\n",
       "      <td>1.463320</td>\n",
       "      <td>0.111970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155240076</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.647492</td>\n",
       "      <td>0.227302</td>\n",
       "      <td>1.844440</td>\n",
       "      <td>+</td>\n",
       "      <td>2.848600</td>\n",
       "      <td>2.097480</td>\n",
       "      <td>0.174431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155240660</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GBA</td>\n",
       "      <td>1</td>\n",
       "      <td>155240707</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.000590</td>\n",
       "      <td>-1.769080</td>\n",
       "      <td>1.358250</td>\n",
       "      <td>2.304190</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.302480</td>\n",
       "      <td>0.858046</td>\n",
       "      <td>0.129025</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene  CHROM        POS REF ALT  N_INFORMATIVE        AF         U  \\\n",
       "0   GBA      1  155235196   G   A           5086  0.000197  0.644930   \n",
       "1   GBA      1  155235203   C   G           5086  0.002064  0.568009   \n",
       "2   GBA      1  155235206   G   A           5086  0.000098  0.632278   \n",
       "3   GBA      1  155235211   C   G           5086  0.000000       NaN   \n",
       "4   GBA      1  155235217   C   G           5086  0.001475  0.025041   \n",
       "5   GBA      1  155235227   G   A           5086  0.000000       NaN   \n",
       "6   GBA      1  155235245   T   C           5086  0.000197 -0.951925   \n",
       "7   GBA      1  155235256   C   T           5086  0.000000       NaN   \n",
       "8   GBA      1  155235269   G   C           5086  0.000000       NaN   \n",
       "9   GBA      1  155235727   C   G           5086  0.000295 -0.524750   \n",
       "10  GBA      1  155235772   C   A           5086  0.000000       NaN   \n",
       "11  GBA      1  155235776   A   G           5086  0.000098  0.596818   \n",
       "12  GBA      1  155235777   T   C           5086  0.000098  0.218746   \n",
       "13  GBA      1  155235790   C   T           5086  0.000295  0.496820   \n",
       "14  GBA      1  155235820   A   C           5086  0.000098  0.666422   \n",
       "15  GBA      1  155235824   G   A           5086  0.000197 -0.039608   \n",
       "16  GBA      1  155235843   T   C           5086  0.003244  7.466830   \n",
       "17  GBA      1  155236245   C   T           5086  0.000393  1.369860   \n",
       "18  GBA      1  155236246   G   A           5086  0.009339  4.090250   \n",
       "19  GBA      1  155236269   C   T           5086  0.000000       NaN   \n",
       "20  GBA      1  155236366   C   T           5086  0.000098  0.462734   \n",
       "21  GBA      1  155236377   C   T           5086  0.000000       NaN   \n",
       "22  GBA      1  155236469   C   T           5086  0.000000       NaN   \n",
       "23  GBA      1  155237423   G   A           5086  0.000197 -0.465445   \n",
       "24  GBA      1  155237438   C   T           5086  0.000197  0.960708   \n",
       "25  GBA      1  155237451   C  CT           5086  0.000098  0.345027   \n",
       "26  GBA      1  155237453   C   T           5086  0.000393  0.279937   \n",
       "27  GBA      1  155237458   A   C           5086  0.000098 -0.115823   \n",
       "28  GBA      1  155238174   C   T           5086  0.000098  0.740153   \n",
       "29  GBA      1  155238189   G   A           5086  0.000098  0.236705   \n",
       "30  GBA      1  155238200   C   T           5086  0.000197  0.682175   \n",
       "31  GBA      1  155238208   C   T           5086  0.000000       NaN   \n",
       "32  GBA      1  155238265   G   A           5086  0.000000       NaN   \n",
       "33  GBA      1  155238570   C   G           5086  0.000000       NaN   \n",
       "34  GBA      1  155238631   G   A           5086  0.001081  0.140238   \n",
       "35  GBA      1  155238646   G   A           5086  0.000098  0.287433   \n",
       "36  GBA      1  155239700   T   C           5086  0.000098 -0.074611   \n",
       "37  GBA      1  155239946   G   A           5086  0.000000       NaN   \n",
       "38  GBA      1  155239961   G   A           5086  0.000197  1.086160   \n",
       "39  GBA      1  155240076   A   C           5086  0.000098  0.647492   \n",
       "40  GBA      1  155240660   G  GC           5086  0.000000       NaN   \n",
       "41  GBA      1  155240707   T   C           5086  0.000590 -1.769080   \n",
       "\n",
       "            V      STAT DIRECTION    EFFECT        SE    PVALUE  \n",
       "0    0.429459  0.968507         +  1.501730  1.525950  0.325053  \n",
       "1    4.718310  0.068379         +  0.120384  0.460370  0.793711  \n",
       "2    0.232255  1.721280         +  2.722340  2.075000  0.189529  \n",
       "3         NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "4    3.522080  0.000178         +  0.007110  0.532844  0.989354  \n",
       "5         NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "6    0.389797  2.324700         - -2.442100  1.601700  0.127335  \n",
       "7         NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "8         NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "9    0.739292  0.372468         - -0.709801  1.163030  0.541662  \n",
       "10        NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "11   0.240267  1.482480         +  2.483970  2.040110  0.223387  \n",
       "12   0.170490  0.280660         +  1.283040  2.421870  0.596269  \n",
       "13   0.389246  0.634125         +  1.276370  1.602830  0.425846  \n",
       "14   0.222039  2.000180         +  3.001370  2.122190  0.157281  \n",
       "15   0.413151  0.003797         - -0.095868  1.555770  0.950865  \n",
       "16   6.716640  8.300810         +  1.111690  0.385855  0.003963  \n",
       "17   0.889397  2.109860         +  1.540210  1.060360  0.146352  \n",
       "18  20.220700  0.827377         +  0.202280  0.222383  0.363032  \n",
       "19        NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "20   0.248273  0.862448         +  1.863810  2.006940  0.353055  \n",
       "21        NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "22        NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "23   0.350959  0.617278         - -1.326210  1.688000  0.432061  \n",
       "24   0.457038  2.019440         +  2.102030  1.479190  0.155297  \n",
       "25   0.225632  0.527601         +  1.529160  2.105230  0.467617  \n",
       "26   0.798504  0.098140         +  0.350577  1.119080  0.754074  \n",
       "27   0.102269  0.131173         - -1.132540  3.127010  0.717218  \n",
       "28   0.191988  2.853440         +  3.855200  2.282250  0.091179  \n",
       "29   0.180482  0.310441         +  1.311510  2.353870  0.577409  \n",
       "30   0.443058  1.050340         +  1.539700  1.502350  0.305428  \n",
       "31        NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "32        NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "33        NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "34   2.476900  0.007940         +  0.056618  0.635398  0.928997  \n",
       "35   0.204549  0.403901         +  1.405200  2.211060  0.525081  \n",
       "36   0.068987  0.080693         - -1.081520  3.807300  0.776361  \n",
       "37        NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "38   0.467005  2.526200         +  2.325800  1.463320  0.111970  \n",
       "39   0.227302  1.844440         +  2.848600  2.097480  0.174431  \n",
       "40        NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "41   1.358250  2.304190         - -1.302480  0.858046  0.129025  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Drop duplicate rows\n",
    "SV_AMPPDexonic = SV_AMPPDexonic.drop_duplicates()\n",
    "display(SV_AMPPDexonic)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "SV_AMPPDexonic.to_csv(f'{WORK_DIR}/GBA_AMPPD/GBA_AMPPD.burdenTESTSV.exonic.SingleScore.assoc.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -mu terra-ddf4a6f6 cp -r /home/jupyter/CTSB_AMPPD/GBA_AMPPD.burdenTESTSV.exonic.SingleScore.assoc.csv /home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTESTSV.exonic.SingleScore.assoc.csv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying file:///home/jupyter/CTSB_AMPPD/GBA_AMPPD.burdenTESTSV.exonic.SingleScore.assoc.csv...\n",
      "/ [1/1 files][  3.4 KiB/  3.4 KiB] 100% Done                                    \n",
      "Operation completed over 1 objects/3.4 KiB.                                      \n"
     ]
    }
   ],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r /home/jupyter/CTSB_AMPPD/GBA_AMPPD.burdenTESTSV.exonic.SingleScore.assoc.csv /home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTESTSV.exonic.SingleScore.assoc.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1_updated_formatted.log\r\n",
      "chr1_updated_formatted.pgen\r\n",
      "chr1_updated_formatted.psam\r\n",
      "chr1_updated_formatted.pvar\r\n",
      "chr1_updated.log\r\n",
      "chr1_updated.pgen\r\n",
      "chr1_updated.psam\r\n",
      "chr1_updated.pvar\r\n",
      "COVS.txt\r\n",
      "GBA_AMPPD.burdenTEST.exonic.CMC.assoc\r\n",
      "GBA_AMPPD.burdenTEST.exonic.CMCWald.assoc\r\n",
      "GBA_AMPPD.burdenTEST.exonic.Fp.assoc\r\n",
      "GBA_AMPPD.burdenTEST.exonic.log\r\n",
      "GBA_AMPPD.burdenTEST.exonic.MadsonBrowning.assoc\r\n",
      "GBA_AMPPD.burdenTEST.exonic.Zeggini.assoc\r\n",
      "GBA_AMPPD.burdenTESTSV.exonic.log\r\n",
      "GBA_AMPPD.burdenTESTSV.exonic.SingleScore.assoc\r\n",
      "GBA_AMPPD.burdenTESTSV.exonic.SingleScore.assoc.csv\r\n",
      "GBA.annovar.avinput\r\n",
      "GBA.annovar.hg38_multianno.txt\r\n",
      "GBA.annovar.hg38_multianno.vcf\r\n",
      "GBA.bed\r\n",
      "GBA.bim\r\n",
      "GBA.burden.exonic.log\r\n",
      "GBA.burden.exonic.SkatO.assoc\r\n",
      "GBA.burden.nonsynonymous_lof.log\r\n",
      "GBA.burden.nonsynonymous_lof.SkatO.assoc\r\n",
      "GBA.exonic.log\r\n",
      "GBA.exonic.variantstoKeep.txt\r\n",
      "GBA.exonic.vcf.gz\r\n",
      "GBA.exonic.vcf.gz.tbi\r\n",
      "GBA.fam\r\n",
      "GBA.log\r\n",
      "GBA.nonsynonymous_lof.log\r\n",
      "GBA.nonsynonymous_lof.variantstoKeep.txt\r\n",
      "GBA.nonsynonymous_lof.vcf.gz\r\n",
      "GBA.nonsynonymous_lof.vcf.gz.tbi\r\n",
      "GBA.pgen\r\n",
      "GBA.psam\r\n",
      "GBA.pvar\r\n",
      "GBA.vcf.gz\r\n",
      "GBA.vcf.gz.tbi\r\n"
     ]
    }
   ],
   "source": [
    "! ls /home/jupyter/GBA_AMPPD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -mu terra-ddf4a6f6 cp -r /home/jupyter/GBA_AMPPD/ gs://fc-49394ff4-0f5f-49e6-a30a-cf07b1dba334\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.nonsynonymous_lof.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTEST.nonsynonymous_lof.MadsonBrowning.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTESTSV.exonic.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.exonic.variantstoKeep.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.burden.nonsynonymous_lof.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/chr1_updated_formatted.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/COVS.txt [Content-Type=text/plain]...    \n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.exonic.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTEST.exonic.CMC.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.exonic.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.nonsynonymous_lof.variantstoKeep.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/chr1_updated.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTEST.exonic.CMCWald.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.psam [Content-Type=application/octet-stream]...\n",
      "==> NOTE: You are uploading one or more large file(s), which would run\n",
      "significantly faster if you enable parallel composite uploads. This\n",
      "feature can be enabled by editing the\n",
      "\"parallel_composite_upload_threshold\" value in your .boto\n",
      "configuration file. However, note that if you do this large files will\n",
      "be uploaded as `composite objects\n",
      "<https://cloud.google.com/storage/docs/composite-objects>`_,which\n",
      "means that any user who downloads such objects will need to have a\n",
      "compiled crcmod installed (see \"gsutil help crcmod\"). This is because\n",
      "without a compiled crcmod, computing checksums on composite objects is\n",
      "so slow that gsutil disables downloads of composite objects.\n",
      "\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTEST.nonsynonymous_lof.CMC.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/chr1_updated_formatted.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/chr1_updated.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTEST.nonsynonymous_lof.Fp.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTEST.exonic.Fp.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/chr1_updated_formatted.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.exonic.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTESTSV.exonic.SingleScore.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTEST.nonsynonymous_lof.CMCWald.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/chr1_updated_formatted.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTESTSV.exonic.SingleScore.assoc.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTEST.exonic.MadsonBrowning.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.nonsynonymous_lof.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.nonsynonymous_lof.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.burden.exonic.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTEST.nonsynonymous_lof.Zeggini.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTEST.nonsynonymous_lof.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/chr1_updated.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.burden.exonic.SkatO.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTEST.exonic.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.burden.nonsynonymous_lof.SkatO.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/chr1_updated.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA_AMPPD.burdenTEST.exonic.Zeggini.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/GBA_AMPPD/GBA.vcf.gz [Content-Type=text/vcard]...  \n",
      "/ [48/48 files][ 37.8 GiB/ 37.8 GiB] 100% Done  75.7 MiB/s ETA 00:00:00         \n",
      "Operation completed over 48 objects/37.8 GiB.                                    \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r /home/jupyter/GBA_AMPPD/ {WORKSPACE_BUCKET}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.16"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
